{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import PyPDF2\n",
    "import re\n",
    "import requests"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#extract knowledge from sfold papers\n",
    "def extract_text_from_pdfs(directory: str, max_chars: int = 1000):\n",
    "    pdf_texts = []\n",
    "    \n",
    "    # Get all PDF files in directory\n",
    "    pdf_files = [f for f in os.listdir(directory) if f.endswith(\".pdf\")]\n",
    "    \n",
    "    for pdf_file in pdf_files:\n",
    "        pdf_path = os.path.join(directory, pdf_file)\n",
    "        with open(pdf_path, \"rb\") as file:\n",
    "            reader = PyPDF2.PdfReader(file)\n",
    "            text = \"\\n\".join(page.extract_text() or \"\" for page in reader.pages)\n",
    "            pdf_texts.append(text)\n",
    "    \n",
    "    full_text = \"\\n\".join(pdf_texts)\n",
    "    \n",
    "    # Chunk the text while preserving sentences\n",
    "    chunks = []\n",
    "    sentences = re.split(r'(?<=[.!?])\\s+', full_text)  # Split while keeping sentence structure\n",
    "    current_chunk = \"\"\n",
    "    \n",
    "    for sentence in sentences:\n",
    "        if len(current_chunk) + len(sentence) <= max_chars:\n",
    "            current_chunk += (\" \" if current_chunk else \"\") + sentence\n",
    "        else:\n",
    "            chunks.append(current_chunk)\n",
    "            current_chunk = sentence\n",
    "    \n",
    "    if current_chunk:\n",
    "        chunks.append(current_chunk)\n",
    "    \n",
    "    return chunks\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['BRIEF COMMUNICATION\\nSTarMirDB: A database of microRNA binding sites\\nWilliam Rennie, Shaveta Kanoria, Chaochun Liu, Bibekanand Mallick *, Dang Long **, Adam Wolenc, C. Steven Carmack,\\nJun Lu#, and Ye Ding\\nWadsworth Center, New York State Department of Health, Center for Medical Science, Albany, NY, USA\\nARTICLE HISTORY\\nReceived 14 March 2016\\nAccepted 18 April 2016ABSTRACT\\nmicroRNAs (miRNAs) are an abundant class of s mall endogenous non-coding RNAs (ncRNAs) of »22\\nnucleotides (nts) in length. These small regulatory molecules are involved in diverse developmental,\\nphysiological and pathological processes. miRNAs target mRNAs (mRNAs) for translational repression\\nand/or mRNA degradation. Predictions of miRNA bind ing sites facilitate experimental validation of\\nmiRNA targets. Models developed with data from CLIP studies have been used for predictions of\\nmiRNA binding sites in the whole transcriptomes of human, mouse and worm.', 'The predictionresults have been assembled into STarMirDB, a ne w database of miRNA binding sites available at\\nhttp://sfold.wadsworth.org/starmirDB.php. STarMirDB can be searched by miRNAs or mRNAs separately\\nor in combination. The search results are categorized into seed and seedless sites in 3\\n0UTR, CDS and 50\\nUTR. For each predicted site, STarMirDB provides a comprehensive list of sequence, thermodynamic and\\ntarget structural features that are known to in ﬂuence miRNA: target interaction. A high resolution PDF\\ndiagram of the conformation of the miRNA:target hybrid is also available for visualization and publication. The results of a database search are available through both an interactive viewer and downloadable textﬁles.KEYWORDS\\nBinding site prediction; CLIP;\\nmicroRNA; seed site; seedless\\nsite\\nIntroduction\\nmiRNAs are a class of single-stranded, non-coding RNAs of »22\\nnucleotides in length.', 'They have b een discovered in plants, animals\\nas well as in some viruses.1-3miRNAs play essential roles in cell\\nproliferation, differen tiation, development, and are associated with\\nhuman diseases.2,4A mature miRNA can guide miRNA-induced\\nsilencing complex (miRISC) for target recognition by sequencecomplementarity between the miRNA and sequences typically inthe 3\\n0untranslated regions (30UTRs) of the cognitive mRNAs\\n(mRNAs). Successful target bindin g usually results in translational\\nrepression and/or mRNA degradation.5Each human miRNA is\\npredicted to be able to regulate several hundred different mRNAs.6\\nComputational prediction algorithms have proven to be\\nvaluable in the discovery of new miRNA targets. Most of theexisting algorithms are based on the seed rule, i.e., the targetsite within 3\\n0UTR forms Watson-Crick (WC) pairs with bases\\nat positions 2 through 7/8 of the 50end of the miRNA.7How-\\never, numerous exceptions to the seed rule have been well-documented.', '8-13Other sequence features have been proposed\\nbased on their enhancement of targeting speci ﬁcity. These\\ninclude sequence conservation, strong base-pairing to the 30\\nend of the miRNA, local AU content and location of miRNA\\nbinding sites (near either end of the 30UTR is favorable).14The\\nimportance of target structural accessibility for miRNA targetrecognition has been supported by numerous studies. 15-21In recent years, experimental methods based on cross-link-\\ning immunoprecipitation (CLIP) have been developed. For\\nhuman and mouse studies, these include HITS-CLIP,22PAR-\\nCLIP23and variations of such techniques.24The CLIP approach\\nhas also been successful in worm.25\\nThe CLIP studies have provided high throughput quality\\ndatasets for regions of mRNAs containing miRNA binding sites.These data allowed us to develo p models for improved predic-\\ntions of miRNA binding sites.', '18,26The models are based on a\\ncomprehensive list of sequen ce, thermodynamic and target\\nstructure features that were enriched for miRNA binding sites\\nidenti ﬁed from CLIP data, and were validated by intra-data set,\\ninter-dataset as well as cross-s pecies validations. For human,\\nmouse and worm, we have used these models to carry outtranscriptome-scale predictio ns of both seed and seedless sites\\nin the 3\\n0untranslated region (30UTR), coding sequence (CDS)\\nregion, and 50untranslated region (50UTR) of mRNAs. The\\nresults have been assembled into STarMirDB, a new database\\napplication module of the Sfold RNA package.27,28In this article,\\nwe describe this new resource.', 'The unique tools of STarMirDBshall complement the existing miRNA target resources forcomputational predictions and experimental target data.Examples of these include, but are not limited to, TargetScan,\\n29\\nDiana-microT,30TarBase,31StarBase,32miRecords33and\\nmiRTarBase.34\\nCONTACT Ye Ding sfoldrna@gmail.com\\n*Present Address: RNA Biology and Functional Genomics Laboratory, Department of Life Science, National Institute of Technology, Odisha, India\\n**Present Address: Biotechnology Department, Faculty of Chemistry, Danang University of Science and Technology, Danang, Vietnam\\n#Present Address: Department of Genetics and Yale Stem Cell Center, Yale University, New Haven, CT, USA\\nPublished with license by Taylor & Francis Group, LLC © Wadsworth Center\\nThis is an Open Access article distributed under th e terms of the Creative Commons Attribution-Non- Commercial License (http ://creativecommons.o rg/licenses/by-nc/3.0/), which permits\\nunrestricted non-commercial use, distribution, and reproduction in any medium, provided the original work is properly cited.', 'The moral rights of th e named author(s) have been asserted.RNA BIOLOGY\\n2016, VOL. 13, NO. 6, 554 –560\\nhttp://dx.doi.org/10.1080/15476286.2016.1182279\\n\\nGeneration of transcriptome-scale data for STarMirDB\\nThe database currently contains records for 3 species, H. sapi-\\nens(human), M. musculus (mouse) and C. elegans (worm). For\\nhuman and mouse, we used complete mRNA sequences fromNCBI RefSeq build 36.3 and 37.2, respectively. For worm, 3\\n0\\nUTR sequences were obtained from the Wormbase versionWS-190. The current release of STarMirDB includes 38,745transcripts for human, 34,631 for mouse and 22,926 3\\n0UTRs\\nfor worm. miRNA sequences were obtained from miRBase\\nrelease 18.35We collected 1,921 miRNA sequences for human,\\n1,157 for mouse and 368 for worm. Our CLIP based models were used to make transcriptome\\nscale predictions of both seed and seedless binding sites.18For\\neach site, a comprehensive list of sequence, thermodynamicand target structure features are computed ( Table 1 ).', 'A logistic\\nprobability is provided as a measure of con ﬁdence in the pre-\\ndicted site. The number of binding sites is astronomical, so that\\nin the database we only included those with a probability of 0.5or higher. This ﬁlter also helps assure a reasonable response\\ntime for database search queries. In the case of interest in thoselow con ﬁdence sites with probabilities under 0.5, the user can\\nuse the STarMir web server that presents all predicted sites forsingle or multiple miRNAs and a target mRNA. 36The databasecan be searched by one or more miRNAs or targets, separately\\nor in a combination. For worm, we provide a user interface thatallows developmental stage speci ﬁc search of miRNA binding\\nsites within the 3\\n0UTR of transcripts. This interface is activated\\nwhen C. elegans is selected as the species for database search. Additionally, for C. elegans , all the prediction data for miRNA\\nbinding sites within the 30UTR of transcripts are also provided\\nas downloadable ﬁles.', 'Input of database query\\nSTarMirDB presents a collection of predicted miRNA binding\\nsites on mRNAs through a web interface that enables bothsearch and retrieval of data and visualization of the conforma-tion of predicted miRNA:target hybrid for each predicted site.The web interface has four input ﬁelds: species, miRNAs,\\nmRNAs, and logistic probability threshold. The requirements\\nfor each input ﬁeld are described below in detail. To start the database search, the user should ﬁrst select the\\nspecies from a dropdown menu. Currently three species areincluded in the database: human ( Homo sapiens ), mouse ( Mus\\nmusculus ) and worm ( C. elegans ). Next, one or a set of miRNAs\\ncan be selected from the miRNA scroll down list, whichTable 1. Description of site information and features for STarMirDB output.', 'Site ID Predicted sites are sequentially numbered along the target sequence\\nTarget Accession number of the target mRNAGene Gene symbol of the target mRNAmiRNA Name of the microRNA (miRNA)Target_Len Length of the targetSite_Position Start and end position of the target region (site) predicted to be bound by miRNASeed_Position Start and end position of the target sub-region complementary to the miRNA seed (i.e.', 'positions 2 –7/8 of the miRNA)\\nSeed_Type 6mer, offset 6mer, 7mer-A1, 7mer-m8, and 8mer seed sites\\n14, 45\\nSite_Access A measure of structural accessibility as computed by the average probability of a nucleotide being single-stranded (i.e., unpaired) fo r the\\nnucleotides in the predicted binding site18\\nSeed_Access A measure of structural accessibility as computed by the average of single-stranded probabilities of the nucleotides in the target sub- region\\ncomplementary to the miRNA seed18\\nUpstream_Access\\n(# nt)A measure of structural accessibility as computed by the average of single-stranded probabilities for the block of nucleotides upstream of the\\npredicted binding site (# nt: block size of 5, 10, 20, 25 or 30)18\\nDwstream_Access\\n(# nt)A measure of structural accessibility as computed by the average of single-stranded probabilities for the block of nucleotides downstream of\\nthe predicted binding site18\\nUpstream_AU\\n(# nt)Percentage of AU for the block of nucleotides upstream of the binding site\\nDwstream_AU\\n(# nt)Percentage of AU for the block of nucleotides downstream of the binding site\\nSite_Location Relative starting location of the predicted binding site along the length of the sequence(e.g., for 30UTR, 0 indicates the 50end of the UTR,\\nand 1 corresponds to the 30end)14\\n30_bp Presence of contiguous Watson Crick base pairing for miRNA nucleotide positions 12 –17 (sites with 30_bp are also called 30compensatory/\\nsupplementary sites)14\\nSite_Consv Conservation score by the PhastCons program46for the binding site\\nSeed_Consv Conservation score by the PhastCons program for the target sub-region complementary to the miRNA seed\\nOffseed_Consv Conservation score by the PhastCons program for nucleotides within the target site, but outside the seed complementary region\\ndG_hybrid DGhybrid (in kcal/mol): a measure of stability for miRNA:target hybrid as computed by RNAhybrid45\\ndG_nucl DGnucl(in kcal/mol): a measure of the potential of nucleation for miRNA:target hybridization17\\ndG_total DGtotal(in kcal/mol): A measure of the total energy change of the hybridization17\\nLogitProb Logistic probability of the site being an miRNA binding site as predicted by our logistic model18\\nTarget_Mismatch Nucleotides in the target binding site that are not base paired with the miRNA\\nTarget_Match Nucleotides in the target binding site that are base paired with the miRNA\\nMir_Match Nucleotides in the miRNA that are base paired with the target mRNAMir_Mismatch Nucleotides in the miRNA that are not base paired with the target mRNAHybrid Conformation The last 4 ﬁelds above present information for the miRNA:target hybrid conformation predicted by RNAhybrid.', 'In each of the ﬁelds, spaces\\nare included so the ﬁelds can be easily aligned to produce a simple diagram of the hybrid conformation as illustrated below:\\nTarget_Mismatch: U UUUCC U ATarget_Match: GACU AUGUA CUACCUC\\nMir_Match: UUGA UACGU GAUGGAG\\nMir_Mismatch: UGGAU ARNA BIOLOGY 555\\n\\ndisplays all available miRNAs assembled for the selected spe-\\ncies. Additionally, one or more miRNA names can be enteredin the text box. The database follows the naming conventionused by miRBase,\\n35i.e., all the miRNAs can be identi ﬁed bytheir miRNA name/identi ﬁer (e.g., hsa-let7-5p for human,\\nmmu-let7a for mouse, and cel-mir-1018 for worm). Target mRNA information has to be entered into the pro-\\nvided text box. For human and mouse, either Genbank\\nFigure 1. An illustrative example of the STarMirDB input page. Binding sites are searched for miRNAs selected from a pre-stored list as well as manually entered by a user,\\nmultiple targets, and a speci ﬁc logistic probability threshold selected by the user.', 'The option of the most relevant site features is selected for output display.556 W. RENNIE ET AL. accession number or Gene symbol, as assigned by the HUGO\\nGene Nomenclature Committee (HGNC), can be provided. Forworm, Wormbase ID is required. For search result displaythrough an interactive site viewer, a user can choose to displayonly the most relevant site features for each binding site, or thecomplete list of features. The most relevant features are consid-ered by us to be the most informative. They were selected fromthose used in the development of the prediction models. 18,26A\\nuser may choose to input merely miRNAs while leaving the tar-\\nget input box blank. In this case, the database server willretrieve predicted sites for the entire transcriptome assembledfor the species. The user can also choose to input merelymRNA IDs, which will prompt the database server to identifyall miRNAs assembled for the species that have binding siteson those mRNAs.', 'This can be useful, e.g., when the question iswhether an mRNA is targeted by any miRNA. A database\\nsearch is typically instantaneous. However, if the database is\\nqueried with only miRNA(s) without target information, thesearch will take minutes. Finally, the user can use a drop downmenu to ﬁlter out miRNA binding sites with logistic probabili-\\nties below the speci ﬁed threshold. Output of database query\\nRelevant data in the database are retrieved in response to a spe-\\nciﬁc database query and are available through both an interac-\\ntive site viewer and downloadable ﬁles. For the interactive siteviewer, the data is classi ﬁed into three mRNA regions (50UTR,\\nCDS and 30UTR) and seed and seedless sites. To facilitate\\nonline viewing, the number of sites displayed in the interactivearea is limited to top-ranked sites according to the decreasingorder of their logistic probabilities. By default, 100 binding sitesare displayed.', 'Alternatively, the user can choose to display thetop 250, 500 or 1,000 sites. The results of a search are also avail-able for download as text ﬁles, wherein all of the retrieved bind-\\ning sites are listed. The interactive viewer presents the results\\nwith either the most relevant site features or all of the site fea-tures as speci ﬁed by the user in the input page. The download-\\nable text ﬁles provide all site features. In the text ﬁles, features\\nmarked with an asterisk are those used in the model computa-tions of the logistic probabilities. In addition to comprehensivesequence, thermodynamic and target structural features(Table 1 ), a high resolution PDF diagram of the conformation\\nof the miRNA:target hybrid is also provided. The diagram was\\ndeveloped to be high quality so that the user can choose to usethem for publication purposes.', 'When both the miRNA and themRNA were included in the CLIP study for the predictionmodel development,\\n18,26an indicator ﬁeld named “CLIP ”will\\nbe given a value of 1 if the predicted site is supported by theCLIP data, and 0 otherwise. CLIP studies are limited to abun-dant miRNAs and expressed transcripts. When either the\\nmiRNA or the mRNA was absent in the CLIP study, a value of\\n“NA ”is assigned to the CLIP indicator. In the database, less\\nthan 1% of sites have a CLIP indicator value of 0 or 1. Thus,\\nFigure 2. STarMirDB output page for the default display of top 100 sites, with the tab selected for displaying seedless sites in the 30UTR.RNA BIOLOGY 557\\n\\nour prediction data complement the CLIP data. A ﬁle providing\\ndeﬁnitions of site features is available via the link for ‘Feature\\ndeﬁnitions ’under the table listing predicted sites. Illustration of database search\\nFor an illustration of the database search, Fig.', '1 shows the input\\nscreen for a query starting with ‘Human (V-CLIP; NCBI RefSeq\\nBuild 36.3) ’selected in the species dropdown menu. From the\\ndropdown list of miRNAs, hsa-7a-5p, hsa-7b-5p , and hsa-7d-5p\\nwere selected. In addition, 2 miRNAs, hsa-let-7c and hsa-let-7e\\nwere manually entered. For mRNA targets, accession numbersNM_0000024, NM_0000021, NM_0000017, and Gene symbolLin54 were entered. Next, the option of “Show predictions with\\nthe most relevant features in interactive viewer ”was selected. Finally, a logistic probability threshold of 0.6 was selected. The“Search ”button was then clicked for submitting the query\\ninput information for processing by the database server. Upon completion of data retrieval by the database server, the\\nuser is presented with an interactive site viewer ( Fig. 2 ). By default,\\nthe list of the top 100 sites is displayed in decreasing order of logis-tic probabilities. An alternative number of sites can be selectedfrom a dropdown menu.', 'The tab for “3\\n0UTR-seedless ”was\\nselected for presenting seedless sites in the 30UTR of the target. For example, the ﬁrst entry in the site table has a logistic probabil-\\nity of 0.9126, which indicates a high con ﬁdence in this predicted\\nsite. A rather low value of ¡17.3 kcal/mol for DGtotalindicates a\\nhigh structural accessibility at the target site.17In the “Hybrid Con-\\nformation ”column, a link is provided for a high resolution PDF\\ndiagram of the conformation of the miRNA:target hybrid at the\\npredicted site. Clicking this link will open the diagram in a new\\ntab or window, depending on the con ﬁguration of the user ’sw e b\\nbrowser. Multiple windows/tabs facilitate comparison of hybridconformations for multiple binding sites. Fig. 3 shows hybrid dia-\\ngrams for a seed site and a seedless site. Under the interactive site viewer, links are provided for\\ndownloading ﬁles of the query results for the 6 combinations of\\nregions and site types ( Fig. 2 ).', 'For site feature information, the\\ndownloadable ﬁles provide all site features whereas the interac-\\ntive viewer displays either all or the most relevant features asselected by the user in the query i nput page. The user can initiate\\na new search by clicking on the link at the bottom of the page. Conclusions\\nSTarMirDB is a new bioinformatics resource for facilitating\\nmiRNA target studies. The current release of database includes\\n96,302 mRNAs and 3,446 miRNAs for human, mouse andworm. It will be periodically updated and likely extended too t h e rs p e c i e s .I tp r e s e n t sp r e d i c t i o n sf o ra l l3m R N Ar e g i o n sa n dfor both seed and seedless sites. Importantly, it presents a proba-bility for each site as an indicator of con ﬁdence in the prediction. In addition to use for visualizat ion and publication, high quality\\ndiagrams of miRNA:target hybrid s can facilitate design of nucle-\\notide mutations for experimental validation of binding sites.', 'The\\noption for search by developmental stage shall be useful for stud-ies of miRNAs in worm. The unique tools from STarMirDB willcomplement the existing miRNA target resources for computa-tional predictions and experime ntal target data. The database\\ncan retrieve miRNA binding sites for single or multiple miRNAsand/or one or more targets. For example, this capability will beuseful for elucidating miRNA regulation of genes of interest. It\\nwill also be useful in miRNA overexpression and knockout stud-\\nies, wherein differentially expressed genes can be further exam-ined by prediction and validation of miRNA binding sites. We have also developed STarMir, a web server for predic-\\ntion of miRNA binding sites. 36STarMir and STarMirDB are\\ncomplementary tools. While the database allows fast search ofpre-computed results, STarMir makes predictions for anymiRNA:mRNA pair from any species of interest.', 'For example,\\nthe user can use STarMir in making predictions for a new iso-\\nform absent in the current database release. The provision of extensive predictions of seedless sites (i.e.,\\nnon-canonical sites) is a major feature for both the databaseand the web server. The functionality of seedless sites has beendemonstrated by numerous studies based on diverse methods,which include reporter assays, nucleotide mutation analysis,analysis of microarray data, analysis of proteomics data, and\\nphenotypic analysis. 11,37-43However, a study primarily based\\non microarray data failed to ﬁnd support for functional seedless\\nsites.44Further experimental investigations will be helpful for\\nFigure 3. Conformation diagrams of miRNA:target hybrids for a seed site (A) and a\\nseedless site (B).558 W. RENNIE ET AL. addressing this lack of consensus. Our tools will facilitate exper-\\nimental testing of predicted seedless sites, especially those withhigh logistic probabilities.', 'Disclosure of potential con ﬂicts of interest\\nNo potential con ﬂicts of interest were disclosed. Acknowledgments\\nThe Bioinformatics Core at the Wadsworth Center is acknowledged for\\nsupporting computing resources for this work. Funding\\nThis work is supported in part by the National Science Foundation (DBI-0650991 to Y.D.), National Institutes of Health (GM099811 to Y.D. and J. L.). Author contributions\\nY.D. conceived and supervised the study. W.R. performed development,implementation and deployment of the database. C.L. and B.M. contrib-uted to generation of data for the database. S.K. and J.L. performed testing of database interface. D. L. and A.W. wrote the initial software for the computation of several target site featuresused by the database, and C.C. provided hardware and system support. W.R., S.K. and Y.D. wrote the paper with contributions from all authors. All\\nauthors read and approved the ﬁnal manuscript. References\\n1. Li C, Zhang B. MicroRNAs in Control of Plant Development.', 'J Cell\\nPhysiol 2016; 231:301-313; http://dx.doi.org/10.1002/jcp.25125\\n2. Ambros V. The functions of animal microRNAs. Nature 2004;\\n431:350-5; PMID:15372042; http://dx.doi.org/10.1038/nature02871\\n3. Liu DG. MicroRNAs in human virus genomes: helping hands for viral\\ninfection. Microrna 2014; 3:75-85; PMID:25226027; http://dx.doi.org/\\n10.2174/2211536603666140825193447\\n4. Garzon R, Calin GA, Croce CM. MicroRNAs in Cancer. Annu Rev\\nMed 2009; 60:167-79; PMID:19630570; http://dx.doi.org/10.1146/\\nannurev.med.59.053006.104707\\n5. Fabian MR, Sonenberg N. The mechanics of miRNA-mediated gene\\nsilencing: a look under the hood of miRISC. Nat Struct Mol Biol 2012;19:586-93; PMID:22664986; http://dx.doi.org/10.1038/nsmb.2296\\n6. Friedman RC, Farh KK, Burge CB, Bartel DP. Most mammalian\\nmRNAs are conserved targets of microRNAs. Genome Res 2009;19:92-105; PMID:18955434; http://dx.doi.org/10.1101/gr.082701.108\\n7. Lewis BP, Burge CB, Bartel DP.', 'Conserved seed pairing, often ﬂanked\\nby adenosines, indicates that thousands of human genes are micro-RNA targets. Cell 2005; 120:15-20; PMID:15652477; http://dx.doi.org/10.1016/j.cell.2004.12.035\\n8. Didiano D, Hobert O. Perfect seed pairing is not a generally reliable\\npredictor for miRNA-target interactions. Nat Struct Mol Biol 2006;13:849-51; PMID:16921378; http://dx.doi.org/10.1038/nsmb1138\\n9. Tay Y, Zhang J, Thomson AM, Lim B, Rigoutsos I. MicroRNAs to\\nNanog, Oct4 and Sox2 coding regions modulate embryonic stem cell\\ndifferentiation. Nature 2008; 455:1124-8; PMID:18806776; http://dx.doi.org/10.1038/nature07299\\n10. Vella MC, Choi EY, Lin SY, Reinert K, Slack FJ. The C. elegans micro-\\nRNA let-7 binds to imperfect let-7 complementary sites from the lin-41 3\\n0UTR. Genes Dev 2004; 18:132-7; PMID:14729570; http://dx.doi. org/10.1101/gad.1165404\\n11. Lal A, Navarro F, Maher CA, Maliszewski LE, Yan N, O ’Day E,\\nChowdhury D, Dykxhoorn DM, Tsai P, Hofmann O, et al.', 'miR-24Inhibits cell proliferation by targeting E2F2, MYC, and other cell-cyclegenes via binding to “seedless ”3\\n0UTR microRNA recognition ele-\\nments. Mol Cell 2009; 35:610-25; PMID:19748357; http://dx.doi.org/10.1016/j.molcel.2009.08.020\\n12. Ha I, Wightman B, Ruvkun G. A bulged lin-4/lin-14 RNA duplex\\nis suf ﬁcient for Caenorhabditis eleg ans lin-14 temporal gradient\\nformation. Genes Dev 1996; 10:3041-50; PMID:8957004; http://dx.doi.org/10.1101/gad.10.23.3041\\n13. Stern-Ginossar N, Elefant N, Zimmermann A, Wolf DG, Saleh N, Biton\\nM, Horwitz E, Prokocimer Z, Prich ard M, Hahn G, et al. Host immune\\nsystem gene targeting by a viral miRNA. Science 2007; 317:376-81;PMID:17641203; http://dx.doi.org/10.1126/science.1140956\\n14. Bartel DP. MicroRNAs: target recognition and regulatory functions. Cell 2009; 136:215-33; PMID:19167326; http://dx.doi.org/10.1016/j.cell.2009.01.002\\n15. Zhao Y, Samal E, Srivastava D.', 'Serum response factor regulates a mus-\\ncle-speci ﬁc microRNA that targets Hand2 during cardiogenesis. Nature 2005; 436:214-20; PMID:15951802; http://dx.doi.org/10.1038/nature03817\\n16. Long D, Chan CY, Ding Y. Analysis of microRNA-target interactions\\nby a target structure based hybridization model. Pac Symp Biocomput\\n2008; 13:64-74; PMID:18232104; http://dx.doi.org/10.1142/9789812776136_0008\\n17. Long D, Lee R, Williams P, Chan CY, Ambros V, Ding Y. Potent effect\\nof target structure on microRNA fun c t i o n .N a tS t r u c tM o lB i o l2 0 0 7 ;\\n14:287-94; PMID:17401373; http://dx.doi.org/10.1038/nsmb1226\\n18. Liu C, Mallick B, Long D, Rennie WA, Wolenc A, Carmack CS, Ding\\nY. CLIP-based prediction of mammalian microRNA binding sites. Nucleic Acids Res 2013; 41(14):e138; http://dx.doi.org/10.1093/nar/gkt435\\n19. Robins H, Li Y, Padgett RW. Incorporating structure to predict micro-\\nRNA targets.', 'Proc Natl Acad Sci U S A 2005; 102:4006-9;\\nPMID:15738385; http://dx.doi.org/10.1073/pnas.0500775102\\n20. Hammell M, Long D, Zhang L, Lee A, Carmack CS, Han M, Ding\\nY, Ambros V. mirWIP: microRNA target prediction based on\\nmicroRNA-containing ribonucleoprotein-enriched transcripts.Nat Methods 2008; 5:813-9; PMID:19160516; http://dx.doi.org/10.1038/nmeth.1247\\n21. Kertesz M, Iovino N, Unnerstall U, Gaul U, Segal E. The role of site\\naccessibility in microRNA target recognition. Nat Genet 2007;39:1278-84; PMID:17893677; http://dx.doi.org/10.1038/ng2135\\n22. Chi SW, Zang JB, Mele A, Darnell RB. Argonaute HITS-CLIP decodes\\nmicroRNA-mRNA interaction maps. Nature 2009; 460:479-86;\\nPMID:19536157; http;//dx.doi.org/10.1038/nature08170\\n23. Hafner M, Landthaler M, Burger L, Khorshid M, Hausser J, Berninger\\nP, Rothballer A, Ascano M Jr, Jungkamp AC, Munschauer M, et al. Transcriptome-wide identi ﬁcation of RNA-binding protein and\\nmicroRNA target sites by PAR-CLIP.', 'Cell 2010; 141:129-41;PMID:20371350; http://dx.doi.org/10.1016/j.cell.2010.03.009\\n24. Kishore S, Jaskiewicz L, Burger L, Hausser J, Khorshid M, Zavolan M. A quantitative analysis of CLIP methods for identifying binding sitesof RNA-binding proteins. Nat Methods 2011; 8:559-64;PMID:21572407; http://dx.doi.org/10.1038/nmeth.1608\\n25. Zisoulis DG, Lovci MT, Wilbert ML, Hutt KR, Liang TY, Pasquinelli\\nAE, Yeo GW. Comprehensive discovery of endogenous Argonautebinding sites in Caenorhabditis elegans. Nat Struct Mol Biol 2010;17:173-9; http://dx.doi.org/10.1038/nsmb.1745\\n26. Liu C, Rennie WA, Mallick B, Kanoria S, Long D, Wolenc A, Carmack\\nCS, Ding Y. MicroRNA binding sites in C. elegans 3\\n0UTRs. RNA Biol\\n2014; 11:693-701; PMID:24827614; http://dx.doi.org/10.4161/\\nrna.28868\\n27. Ding Y, Lawrence CE. A statistical sampling algorithm for RNA sec-\\nondary structure prediction. Nucleic Acids Res 2003; 31:7280-301;PMID:14654704; http://dx.doi.org/10.1093/nar/gkg938\\n28. Ding Y, Chan CY, Lawrence CE.', 'Sfold web server for statistical folding\\nand rational design of nucleic acids. Nucleic Acids Res 2004; 32:W135-41; PMID:15215366; http://dx.doi.org/10.1093/nar/gkh449\\n29. Grimson A, Farh KK, Johnston WK, Garrett-Engele P, Lim LP, Bartel\\nDP. MicroRNA targeting speci ﬁcity in mammals: determinants\\nbeyond seed pairing. Mol Cell 2007; 27:91-105; PMID:17612493;http://dx.doi.org/10.1016/j.molcel.2007.06.017RNA BIOLOGY 559\\n\\n30. Paraskevopoulou MD, Georgakilas G, Kostoulas N, Vlachos IS, Ver-\\ngoulis T, Reczko M, Filippidis C, Dalamagas T, Hatzigeorgiou AG.DIANA-microT web server v5.0: service integration into miRNAfunctional analysis work ﬂows. Nucleic Acids Res 2013; 41:W169-73;\\nPMID:23680784; http://dx.doi.org/10.1093/nar/gkt393\\n31. Papadopoulos GL, Reczko M, Simossis VA, Sethupathy P, Hatzigeor-\\ngiou AG. The database of experimentally supported targets: a func-\\ntional update of TarBase. Nucleic Acids Res 2009; 37:D155-8;\\nPMID:18957447; http://dx.doi.org/10.1093/nar/gkn809\\n32.', 'Li JH, Liu S, Zhou H, Qu LH, Yang JH. starBase v2.0: decoding\\nmiRNA-ceRNA, miRNA-ncRNA and protein-RNA interaction net-\\nworks from large-scale CLIP-Seq data. Nucleic Acids Res 2014; 42:\\nD92-7; PMID:24297251; http://dx.doi.org/10.1093/nar/gkt1248\\n33. Xiao F, Zuo Z, Cai G, Kang S, Gao X, Li T. miRecords: an integrated\\nresource for microRNA-target interactions. Nucleic Acids Res 2009;\\n37:D105-10; PMID:18996891; http://dx.doi.org/10.1093/nar/gkn851\\n34. Hsu SD, Tseng YT, Shrestha S, Lin YL, Khaleel A, Chou CH, Chu CF,\\nHuang HY, Lin CM, Ho SY, et al. miRTarBase update 2014: an infor-mation resource for experimentally validated miRNA-target interac-\\ntions. Nucleic Acids Res 2014; 42:D78-85; PMID:24304892; http://dx. doi.org/10.1093/nar/gkt1266\\n35. Kozomara A, Grif ﬁths-Jones S. miRBase: annotating high con ﬁdence\\nmicroRNAs using deep sequencing data. Nucleic Acids Res 2014; 42:\\nD68-73; http://dx.doi.org/10.1093/nar/gkt1181\\n36.', 'Rennie W, Liu C, Carmack CS, Wolenc A, Kanoria S, Lu J, Long D,\\nDing Y. STarMir: a web server for prediction of microRNA binding\\nsites. Nucleic Acids Res 2014; 42:W114-8; PMID:24803672; http://dx. doi.org/10.1093/nar/gku376\\n37. Chi SW, Hannon GJ, Darnell RB. An alternative mode of microRNA\\ntarget recognition. Nat Struct Mol Biol 2012; 19:321-7;\\nPMID:22343717; http://dx.doi.org/10.1038/nsmb.2230\\n38. Loeb GB, Khan AA, Canner D, Hiatt JB, Shendure J, Darnell RB,\\nLeslie CS, Rudensky AY. Transcriptome-wide miR-155 binding\\nmap reveals widespread noncanonical microRNA targeting. MolCell 2012; 48:760-70; PMID:23142080; http://dx.doi.org/10.1016/j. molcel.2012.10.002\\n39. Khorshid M, Hausser J, Zavolan M, van Nimwegen E. A biophysical\\nmiRNA-mRNA interaction model infers canonical and noncanonical\\ntargets. Nat Methods 2013; 10:253-5; PMID:23334102; http://dx.doi. org/10.1038/nmeth.2341\\n40. Grosswendt S, Filipchyk A, Manzano M, Klironomos F, Schilling M,\\nHerzog M, Gottwein E, Rajewsky N.', 'Unambiguous identi ﬁcation of\\nmiRNA:target site interactions by different types of ligation reactions.Mol Cell 2014; 54:1042-54; PMID:24857550; http://dx.doi.org/10.1016/j.molcel.2014.03.049\\n41. Tan SM, Kirchner R, Jin J, Hofmann O, McReynolds L, Hide W, Lie-\\nberman J. Sequencing of captive target transcripts identi ﬁes the net-\\nwork of regulated genes and functions of primate-speci ﬁc miR-522. Cell Rep 2014; 8:1225-39; PMID:25131211; http://dx.doi.org/10.1016/\\nj.celrep.2014.07.023\\n42. Helwak A, Kudla G, Dudnakova T, Tollervey D. Mapping the human\\nmiRNA interactome by CLASH reveals frequent noncanonical bind-ing. Cell 2013; 153:654-65; PMID:23622248; http://dx.doi.org/\\n10.1016/j.cell.2013.03.043\\n43. Zhang H, Artiles KL, Fire AZ. Functional relevance of “seed ”and\\n“non-seed ”sequences in microRNA-mediated promotion of C. ele-\\ngans developmental progression. Rna 2015; 21(11):1980-92; http://dx. doi.org/10.1261/rna.053793.115\\n44. Agarwal V, Bell GW, Nam JW, Bartel DP.', 'Predicting effective micro-\\nRNA target sites in mammalian mRNAs. eLife 2015; 4; http://dx.doi. org/10.7554/eLife.05005. 45. Rehmsmeier M, Steffen P, Hochsmann M, Giegerich R. Fast and effec-\\ntive prediction of microRNA/target duplexes. RNA 2004; 10:1507-17;PMID:15383676; http://dx.doi.org/10.1261/rna.5248604\\n46. Siepel A, Bejerano G, Pedersen JS, Hinrichs AS, Hou M, Rosenbloom\\nK, Clawson H, Spieth J, Hillier LW, Richards S, et al. Evolutionarilyconserved elements in vertebrate, insect, worm, and yeast genomes. Genome Res 2005; 15:1034-50; http://dx.doi.org/10.1101/gr.3715005560 W. RENNIE ET AL. Potent effect of target structure on microRNA function\\nDang Long1, Rosalind Lee2, Peter Williams2, Chi Yu Chan1, Victor Ambros2& Ye Ding1\\nMicroRNAs (miRNAs) are small noncoding RNAs that repress protein synthesis by binding to target messenger RNAs. We\\ninvestigated the effect of target secondary structure on the efﬁcacy of repression by miRNAs.', 'Using structures predicted bythe Sfold program, we model the interaction between an miRNA and a target as a two-step hybridization reaction: nucleationat an accessible target site followed by hybrid elongation to disrupt local target secondary structure and form the complete\\nmiRNA-target duplex. This model accurately accounts for the sensitivity to repression by let-7 of various mutant forms of the\\nCaenorhabditis elegans lin-41 3¢untranslated region and for other experimentally tested miRNA-target interactions in C. elegans\\nandDrosophila melanogaster . These ﬁndings indicate a potent effect of target structure on target recognition by miRNAs and\\nestablish a structure-based framework for genome-wide identiﬁcation of animal miRNA targets. miRNAs are a class of small noncoding RNAs found in plants and\\nanimals that regulate gene expression by base-pairing to mRNAtargets, causing either target degradation or translational repression\\n1.', 'Experimental and computational studies have identiﬁed thousands ofmiRNAs encoded in animal genomes\\n1,2. miRNAs have regulatory roles\\nin developmental timing, patterning, embryogenesis, differentiation,organogenesis, stem cell and germline proliferation, growth control\\nand apoptosis; they are also involved in physiological processes,\\nincluding cancer, aging, hematopoiesis and endocrine function\\n1,3–5. However, there is still much to be learned about the biological\\nfunctions of miRNAs and the molecular mechanisms by which they\\nregulate gene expression1,6,7. In plants, miRNAs bind their targets by complete or nearly\\ncomplete complementarity, so target identiﬁcation is straightforward8.', 'By contrast, animal miRNAs are typically only partially complemen-tary to their targets, which poses a challenge for the accurate\\ncomputational identiﬁcation of animal miRNA targets\\n6,9,10.A n\\nimproved understanding of the requirements for functional interac-\\ntions between miRNAs and their targets is essential for preciselyelucidating miRNA targets in animals. T o date, most studies of miRNA-target interactions have focused\\nprimarily on the characteristics of the sequence complementaritybetween the miRNA and putative target sites in the mRNA. Tissue-\\nculture experiments\\n11,12and computational analyses of base-pairing\\nbetween miRNAs and mRNA targets13,14have suggested that perfect\\nWatson-Crick complementarity of seven or eight consecutive bases(typically at positions 2–8) in the 5 ¢ends of miRNAs (the ‘seed’\\nregion) is an important signal for target regulation.', 'A study using an\\nin vivo assay for miRNA repression in transgenic fruitﬂies has further\\nsuggested that strong complementarity in the 3 ¢end of a miRNA can\\ncontribute to miRNA efﬁcacy and speciﬁcity by compensating forweak base-pairing in the 5 ¢end\\n6.Although the base-pairing between a miRNA and target site is\\nimportant, the sequence context surrounding miRNA-binding siteshas been reported to inﬂuence sensitivity to repression by a\\nmiRNA\\n15,16. Sequence context could inﬂuence miRNA efﬁcacy by\\nmediating the binding of hypothetical cofactor proteins or by affecting\\nthe secondary structure of a target site and hence its accessibility tobinding by the miRNA\\n15. Although it is reasonable to postulate that\\ntarget structure could be a factor in miRNA efﬁcacy, the conventionalminimum free energy (MFE) approach for modeling RNA secondarystructure has limitations when it comes to accurately representing the\\nstructure of an mRNA in vivo . An analysis of ﬁve known miRNA-\\ntarget pairs in C.', 'elegans andD. melanogaster , using mfold\\n17to model\\nmRNA structure, has suggested that the presence of three consecutivesingle-stranded nucleotides in the target facilitates pairing with\\nnucleotides in the 5 ¢seed region of the miRNA\\n18. A prediction\\nmethod incorporating this feature results in perhaps fewer false\\npositive predictions, but it also suffers from reduced accuracy. In\\nparticular, this method does not identify the two let-7 binding sites in\\nthelin-41 3¢untranslated region (UTR), because neither site meets the\\nrequirement for a fully Watson-Crick 5 ¢seed region. Although some\\nexisting target search methods such as RNAhybrid19can identify\\nlin-41 as a target of let-7 , they do not account for the observed\\ndifferences in sensitivity to repression by let-7 oflin-41 3¢UTR\\nmutants that differ only in sequences outside of the let-7 –\\ncomplementary sites15.', 'Here, we use Sfold20to model mRNA structures and employ a novel\\ntwo-step model for the hybridization between an miRNA and astructured target. In the two-step model, hybridization nucleates atan accessible target site, and then the hybrid elongates to form thecomplete miRNA-target duplex ( Fig. 1 ). The reliable performance of\\nthe model strongly suggests a potent effect of mRNA secondarystructure on target recognition by miRNAs. The two-step model\\nReceived 19 January; accepted 7 March; published online 1 April 2007; doi:10.1038/nsmb12261Wadsworth Center, New York State Department of Health, 150 New Scotland Avenue, Albany, New York 12208, USA.2Department of Genetics, Dartmouth Medical\\nSchool, Hanover, New Hampshire 03755, USA .Correspondence should be addressed to Y.D. (yding@wadsworth.org) or V.A. (vambros@dartmouth.edu).', 'NATURE STRUCTURAL & MOLECULAR BIOLOGY VOLUME 14 NUMBER 4 APRIL 2007 287ARTICLES\\n©2007 Nature Publishing Group  http://www.nature.com/nsmb\\nprovides a structure-based, mechanistic framework for genome-wide\\nidentiﬁcation of animal miRNA targets. RESULTS\\nPrevious data for lin-41 3¢UTR mutants\\nThe 3 ¢UTR of the C. elegans lin-41 mRNA contains six imperfectly\\ncomplementary sites for let-7 . Of these, site 1 and site 2 are well\\nconserved in the orthologous gene in Caenorhabditis briggsae15. V ella\\nand co-workers15have shown that a sequence from the lin-41 3¢UTR\\nthat contains only sites 1 and 2 (along with an intervening 27-ntspacer sequence) is sufﬁcient, when placed in the context of the unc-54\\n3¢UTR, for repression of a lacZ reporter gene in transgenic worms\\n15. Notably, this study15found that altering the 27-nt spacer abolishes\\nrepression by let-7 .', 'In particular, among ten tested constructs\\n(Table 1 ), pMV9 and pMV19 both contain just sites 1 and 2; however,\\npMV9 contains the wild type 27-nt spacer, whereas pMV19 contains amutated 27-nt spacer. pMV9 was repressed by let-7 ,b u tp M V 1 9w a s\\nnot repressed. Alteration of the relative conﬁguration of sites 1 and 2was also found to impair sensitivity to let-7 . Construct pMV16\\ncontains three copies of site 1 and no other complementary sites,\\nand pMV17 contains only three copies of site 2. Neither of these latter\\nconstructs has normal sensitivity to let-7 in vivo. These results suggest\\nthat speciﬁc features of the conﬁguration of sites 1 and 2 in the lin-41\\n3¢UTR are important for sensitivity to let-7 repression. The authors\\nconsidered altered local RNA secondary structure of the mutant3¢UTR as a possible basis for the impaired functionality of pMV19,\\nbut they could draw no satisfactory interpretation of the results from\\nstructures predicted by mfold\\n15.', 'Below, we analyze the published\\nlin-41 3¢UTR reporter data15using Sfold and the two-step model\\nfor miRNA-target hybridization ( Fig. 1 ). Frequencies of open nucleotides in target sites\\nT o examine the potential importance of contiguous open nucleotides\\nin a miRNA-complementary site, we used Sfold to generate a sample\\nof secondary structures from the mRNA sequence formed by each ofthe mutant 3 ¢UTRs fused to the lacZ coding sequence\\n15.W et h e nu s e d\\nthe sample of structures to identify blocks of open nucleotides of\\nvarious lengths in the let-7 –binding sites of the constructs. A nucleo-\\ntide is considered open if its probability of being unpaired, as\\nestimated from the Sfold structure sample, is greater than or equal\\nto 0.5; a block of nnucleotides (nt) is considered open if the\\nprobability that all nnucleotides in the block are simultaneously\\nunpaired is greater than or equal to 0.5.', 'For putative binding sites ineach construct, the numbers of open nucleotides and the numbers of\\nopen blocks are presented in Supplementary Table 1 online. For most of the constructs, a larger number of open single\\nnucleotides in the complementary sites tends to correlate with greater\\nrepression. However, this trend does not hold for pMV16 and pMV17;\\nmoreover, for pMV9 and pMV19, the numbers of open singlenucleotides do not contrast as sharply as do the repression sensiti-\\nvities. Thus, the number of open single nucleotides is not sufﬁcient to\\nfully account for the data. In contrast, for pMV19 and pMV9, therepression sensitivity is better correlated with the frequency of open\\nblocks of 4 nt. These ﬁndings suggest a potential effect of target\\nstructure in miRNA function; in particular, they suggest that effectivemiRNA-target interaction requires an open block of four or more\\nconsecutive complementary bases on the target.', 'Sfold target accessibility proﬁling\\nT o further examine the structural accessibility of let-7 –complementary\\nsequences in the 3 ¢UTRs of the lacZ reporter mRNAs corresponding\\nto the pMV9 and pMV19 constructs, we used the Sfold samples ofmRNA structures for these constructs to compute target accessibility\\nproﬁles for sites 1 and 2 ( Fig. 2a ). The let-7 –lin-41 hybrid conforma-\\ntions predicted for the two sites in the wild-type 3 ¢UTR\\n15are shown\\ninFigure 2b . For a window length of 4 nt, the accessibility proﬁle\\nshows the probability that the four consecutive nucleotides starting at\\n5′3′ 3′\\n5′5′5′\\n3′3′\\n∆Gtotal = ∆Ghybrid – ∆Gdisruption∆GN = Σ ∆Gstack( j) a\\nbj = 1, 2, 3Figure 1 A two-step model for hybridization between a structured mRNA\\nand a partially complementary miRNA, illustrated for a single structural\\nconformation of the target. ( a) In the ﬁrst step, the miRNA nucleates base-\\npairing at an accessible site of four unpaired nucleotides.', 'The nucleationpotential, DG\\nN, is calculated by summing the stacking energies DGstack( j)\\n(where ( j)i st h e jth base pair stack, 1 rjr3) for the most stable\\n4-base-pair duplex at the complementary site; nucleation is consideredfavorable if ( DG\\nN+DGinitiation )o0k c a lm o l/C01.(b) In the second step, the\\nmiRNA-target hybrid elongates, resulting in local disruption of the targetsecondary structure across the miRNA-complementary region. DG\\ntotal, the\\ntotal energy change for the hybridization, is the key energetic characteristicfor this step (see Methods).', 'Table 1 miRNA-target interaction energy predicts sensitivity of lin-41\\n3¢UTR reporters to repression by let-7\\nPDGtotal(kcal mol–1)a\\nMutant\\nconstructbReported\\nrepression\\nsensitivitybDGinitiation ¼\\n4.09 kcal mol/C01cDGinitiation ¼\\n0.0 kcal mol/C01PDGhybrid\\n(kcal mol/C01)d\\npMV1 + + /C043.3 (+)e/C069.5 (+) /C076.2 (+)f\\npMV8 + + /C043.1 (+) /C052.3 (+) /C057.0 (+)\\npMV9 + + + /C043.4 (+) /C043.4 (+) /C057.0 (+)\\npMV5 + 0.0 ( /C0) /C014.0 (+) /C051.7 (+)\\npMV12 + /C020.3 (+) /C036.6 (+) /C056.6 (+)\\npMV19 /C0/C0 8.3 (/C0) /C040.1 (+) /C057.0 (+)\\npMV6 /C0 0.0 (/C0) 0.0 ( /C0) /C023.0 (+)\\npMV16 /C0/C0 5.7 (/C0) /C052.6 (+) /C083.3 (+)\\npMV7 /C0 0.0 (/C0) 0.0 ( /C0) /C028.6 (+)\\npMV17 /C0/C0 5.6 (/C0) /C055.2 (+) /C087.0 (+)\\naSee Methods. Because DGtotalis computed for a site only if it has ( DGN+DGinitiation )o\\n0.0 kcal mol/C01, the value of DGtotalfor a site, and hencePDGtotalfor multiple sites on the\\nsame target, is dependent on the choice of the initiation energy DGinitiation .bRef. 15.cRef. 24.', 'dDGhybrid is computed from RNAhybrid, and a site is counted for the sum if DGhybridr\\n/C014 kcal mol/C01(an energetic threshold previously considered for miRNA-target duplexes35;\\nevery lin-41 –complementary site meets this threshold). This calculation ignores effects of target\\nstructure and nucleation.eForDGtotal,alet-7–target interaction is predicted to be functional (+)\\nif, for lin-41 –complementary sites15,PDGtotalo/C010 kcal mol/C01; otherwise, the interaction is\\npredicted to be nonfunctional ( /C0).fForDGhybrid,alet-7–target interaction is predicted to be\\nfunctional (+) if, for lin-41 complementary sites,PDGhybridr/C014 kcal mol/C01(that is, at least\\none site meets the threshold); otherwise, the interaction is predicted to be nonfunctional ( /C0).ARTICLES\\n288 VOLUME 14 NUMBER 4 APRIL 2007 NATURE STRUCTURAL & MOLECULAR BIOLOGY\\n©2007 Nature Publishing Group  http://www.nature.com/nsmb\\nthe indicated nucleotide are all single-stranded, as predicted by\\nSfold21,22.', 'A window length of 4 nt was chosen for this study, as use\\nof this length led to a good correlation between our accessibility\\nproﬁle predictions and data from antisense experiments on rabbit\\nb-globin21. Furthermore, a 4-nt accessible block can be sufﬁcient for\\nthe nucleation step of the hybridization between a complementary\\nnucleic acid molecule and its target23. Finally, we observed a better\\ncorrelation with lin-41 data for the block size of 4 nt than for block\\nsizes of 1, 2 or 3 nt ( Supplementary Table 1 ). The accessibility proﬁles of pMV9 and pMV19 ( Fig. 2a )a r ei n\\nsubstantial agreement with the sensitivities of the two constructs to\\nrepression by let-7 . For pMV9, which is repressed by let-7 ,t h e3 ¢end\\nof target site 1 is highly accessible for nucleation, as is the 5 ¢end of\\nsite 2. In contrast, for pMV19, which is inactive for repression, neitherlet-7 site is accessible.', 'Notably, for pMV9, nucleation of hybridization\\nis predicted to occur at the miRNA 5 ¢end (3 ¢end of the target site) for\\nsite 1, but at the miRNA 3 ¢end (5 ¢end of the target site) for site 2. Energetic analysis of miRNA-target hybridizationTo model the potential function of each let-7 –complementary site in\\nthelin-41 mutant 3 ¢UTR reporter constructs, we computed the\\nnucleation potential ( DG\\nN) for each site (see Methods). For\\nsites with favorable nucleation potential (where the nucleation poten-\\ntial overcomes the initiation threshold DGinitiation ,s ot h a t( DGN+\\nDGinitiation )o0 kcal mol/C01, we calculated the total energy change\\n(DGtotal) for the hybridization to each site, thenPDGtotalfor multiple\\nsites (see Methods). A miRNA–3 ¢UTR interaction is predicted to be\\nfunctional if it has a favorablePDGtotal.', 'T o estimate a practical\\nthreshold for favorablePDGtotal, we considered the performance of\\nvarious threshold values in accommodating the experimental results\\nforlin-41 3¢UTR mutants. We found that aPDGtotal of/C010 kcal\\nmol/C01or less seems to separate efﬁcient interactions from inefﬁcient\\ninteractions (data not shown). The energetic predictions for lin-41 3¢UTR reporter constructs are\\nin good agreement with the reported repression sensitivities ( Table 1 ). In general, repression by let-7 was substantial for constructs with\\nfavorable nucleation sites and large negative hybridization energies. Incontrast, constructs with weak or absent repression by let-7 generally\\nhad poor nucleation sites, small hybridization energies or both.', 'For\\nexample, despite a substantial number of open nucleotides in thepMV16 and pMV17 3 ¢UTRs ( Supplementary Table 1 ), their let-7 –\\ncomplementary sites\\n15are nevertheless predicted to be lacking in\\nfavorable nucleation potential, and overall the pMV16 and pMV17\\n3¢UTRs have small hybridization energies. This analysis predicts that\\npMV5 should not interact with let-7 , yet this construct did show weak\\nrepression in vivo , and pMV5 and pMV12 were given the same\\nranking for repression sensitivity15. However, the measured level of\\nrepression for pMV5 is lower than that for pMV12 (ref. 15), which is\\nconsistent with the rankings of these two constructs by our energetic\\nanalysis ( Table 1 ). An accurate accounting for the lin-41 3¢UTR data requires\\napplication of both a nucleation-potential threshold (speciﬁed viaan initiation energy) and a structure-based hybridization energycalculation.', 'If we ignore the effects of target structure and nucleation\\nby using only the energy gain from hybridization, DG\\nhybrid ,i nt h e\\ncalculation, we ﬁnd that all ten constructs have large negative\\nSDGhybrid values, even though ﬁve of them were observed to be\\ninsensitive to let-7 in vivo (Table 1 ). Even ignoring just the initiation\\nenergy, by setting it to 0.0 kcal mol/C01instead of 4.09 kcal mol/C01(the\\nvalue empirically determined in ref. 24), results in rather poorpredictions for pMV19, pMV16 and pMV17 ( Table 1 ). Although\\ninclusion of a nucleation-potential ﬁlter is crucial, the predictions arerobust over a range of initiation energies from 4.0 kcal mol\\n/C01to\\n5.5 kcal mol/C01(Supplementary Fig. 1 online). For example, for\\n5.2 kcal mol/C01(the other published value for initiation energy23),\\nthese predictions for the lin-41 3¢UTR were the same as for\\n4.09 kcal mol/C01.', 'Although these two published values for the initia-\\ntion energy generally performed similarly when applied as thenucleation-potential thresholds, we focus on results for the value of4.09 kcal mol\\n/C01because of its somewhat superior predictive consis-\\ntency for certain other miRNA-target interactions (see below). miRNA-target interactions in C.elegans andDrosophila\\nT o examine the general applicability of the two-step hybridization\\nmodel, we considered other miRNA-target interactions that had been\\npredicted previously for C. elegans orD. melanogaster ,a n df o rw h i c h\\nexperimental validation had been published ( Supplementary Table 2\\nonline). For some of these interactions, the validation experimentsinclude tests of genetic epistasis (where targeting is validated byobserving that loss of function of the target counteracts the effects\\nof loss of function of the miRNA). For other interactions, validation is\\ntypically based on reporter gene expression.', 'For each miRNA-targetpair, we calculatedPDG\\ntotaland compared it with the corresponding\\nvalue computed for ten ‘randomers’ , random control sequences\\ngenerated by dinucleotide shufﬂing of the miRNA sequence with\\nDishufﬂe25. T o statistically test whether the averagePDGtotalfor the\\nauthentic miRNA sequences is lower than that for the randomers, we\\nperformed the one-sided, unequal-variance t-test and the nonpara-\\nmetric Wilcoxon rank-sum test. We found that the degree of correspondence between our calcula-\\ntions and the published validation results depends somewhat on the\\nSite 23′3′ 3′\\n5′\\n5′5′AUlet-7 RNA let-7 RNA\\n27-nt spacer\\nlin-41  mRNA 3 ′ UTR Site 1AU3,350 3,340 3,330 3,320 3,310 3,300 3,290 3,280 3,270\\nNucleotide position00.10.20.30.40.50.60.70.80.91AccessibilitySite 1 Site 2pMV19 pMV9a\\nbFigure 2 Target accessibility proﬁling by S fold.', '( a) Sfold accessibility\\nproﬁles for the region containing the two let-7–binding sites (shaded) in the\\nlin-41 3¢UTR mutant construct pMV9 and in pMV19. For pMV9, which is\\nsensitive to repression by let-7 (ref. 15), the 3 ¢end of target site 1 is\\nhighly accessible for nucleation, as is the 5 ¢end of site 2. For the mutant\\n3¢UTR construct pMV19, which is insensitive to repression, neither site is\\naccessible. ( b) Neither site 1 nor site 2 satisﬁes the criteria for full Watson-\\nCrick base-pairing across the 5 ¢seed region of the miRNA; a bulged A in\\nsite 1 and a wobble G /C15U base pair in site 2 are indicated in red.ARTICLES\\nNATURE STRUCTURAL & MOLECULAR BIOLOGY VOLUME 14 NUMBER 4 APRIL 2007 289\\n©2007 Nature Publishing Group  http://www.nature.com/nsmb\\nnature of the experimental supporting evidence.', 'For the group of 11\\npredicted interactions for which the supporting evidence includesconventional genetic epistasis experiments, we calculated that all 11\\nshould interact effectively, on the basis ofPDG\\ntotal(Supplementary\\nTable 2 online). For 9 of these 11 miRNA-target pairs, a particularly\\nstrong interaction is predicted on the basis of hybridization energies.The meanPDG\\ntotalfor the miRNAs ( /C063.70 kcal mol/C01)i ss t a t i s t i -\\ncally distinct from the mean for the randomers ( /C01.76 kcal mol/C01;\\nFig. 3 ): the t-test returned a highly signiﬁcant P¼3.55/C210/C03,a n d\\nthe corresponding Wilcoxon rank-sum test yielded P¼1.11/C210/C012. The signal-to-noise ratio, deﬁned here as the ratio of the averagePDGtotalfor the miRNAs to that for the randomers, is 36.1. There are two cases of relatively small hybridization energies:\\nnamely, the lin-4 –lin-28 pair and the miR-273 –die-1 pair.', 'It has\\nbeen reported that the repression of lin-28 bylin-4 may occur in\\nconjunction with an additional ‘ lin-4 –independent circuit’26, suggest-\\ning that effective repression of lin-28 bylin-4 requires additional\\nfactor(s). Similarly, the putative miR-273 –die-1 interaction may\\nalso require other unknown factors. For the initiation energy of\\n5.20 kcal mol/C01,let-7 does not have a single favorable nucleation\\nsite on the lin-28 3¢UTR. Thus, we use 4.09 kcal mol/C01in the\\nsubsequent analyses because of its better predictive consistency andhigher inclusiveness of potential sites. Our model predicts a functional interaction for 27 of 39 inter-\\nactions whose in vivo efﬁcacy is supported mainly by reporter gene\\ntests, but for which genetic epistasis evidence is not available.', 'Thus,\\nthe structure-based model performs less precisely in accounting for\\nthe experimental results for interactions tested by nongenetic criteria(69% true positive), compared with interactions validated genetically\\n(100% true positive). Nevertheless, the averagePDG\\ntotalfor this set of\\nmiRNA-target interactions is statistically signiﬁcant compared with\\nthat for randomers ( Fig. 3 ): the averagePDGtotalfor the miRNAs is\\n/C032.51 kcal mol/C01and the average for the randomers is /C01.67 kcal\\nmol/C01, a difference of /C030.84 kcal mol/C01, yielding P¼3.89/C210/C03by\\nthet-test and P¼4.50/C210/C07by the Wilcoxon rank-sum test, and the\\nsignal-to-noise ratio is 19.5. T o examine the success of our model in accounting for negative\\nresults from experimental validation tests, we examined a set of 12putative targets of the C. elegans lsy-6 miRNA that had been predicted\\nbased on conserved seed matches, but for which in vivo tests did notvalidate the predicted interactions\\n27.', 'For only one of these 12 putative\\nlsy-6 targets (8%) was a functional interaction (albeit relatively weak)\\npredicted by our structure-based model ( Table 2 ). For 11 of the 12\\nputative targets (92%), we did not predict a functional interaction\\n(Table 2 ), in complete agreement with the experimental results for\\nthese targets27. In contrast, if we ignore the effects of target structure\\nand nucleation by using justPDGhybrid , functional interactions are\\npredicted for 10 of the 12 negative cases ( Table 2 ). This strongly\\nsuggests that mRNA secondary structure is a major factor behind the\\ninsensitivity of these putative targets to lsy-6 in vivo .N o t et h a t cog-1 ,\\nwhich has previously been conﬁrmed by various functional criteria to\\nbe regulated by lsy-6 (refs. 27,28), is predicted by our model to have an\\neffective interaction with lsy-6 (Table 2 ).', 'A comparison of SDGtotal\\nbetween the authentic miRNAs and randomers for these 12 targetsyielded P¼0.2364 for the t-test and 0.9117 for the Wilcoxon rank-\\nsum test, and the signal-to-noise ratio is merely 2.2 ( Fig. 3 ); for the 11\\nfalse positive predictions by seed matches (see Table 2 ), there is no\\nappreciable difference between the signal and the noise according toour model (for this subset, the P¼0.559 for the t-test and 0.964 for\\nthe Wilcoxon rank-sum test) ( Supplementary Table 2 ). In vivo testing of newly designed lin-41 3¢UTR mutants\\nT o further test for effects of target structure and accessibility on\\nmiRNA efﬁcacy—in this case, on repression of lin-41 bylet-7 in\\nC. elegans —we carried out in vivo experiments using six new lacZ\\nreporter transgenes containing rationally designed lin-41 3¢UTR muta-\\ntions.', 'For four of the new constructs, pVT701, pVT702, pVT704 and\\npVT705, the corresponding lin-41 mutant 3 ¢UTR sequence contains\\none copy of site 1 and one copy of site 2 separated by a 27-nt spacer\\nsequence. For pVT701 and pVT702, the wild-type 27-nt spacer wasmutated to preserve the accessibility of both sites (a target site is\\nconsidered highly accessible if it has both favorable nucleation potential\\nand favorable DG\\ntotal). For pVT704 and pVT705, the spacer was\\ndesigned to reduce the accessibility of the sites. Constructs pVT712\\nand pVT713 contain, respectively, three copies of sites 1 or 2, in\\nsequence contexts designed to permit good accessibility. These new constructs, along with pMV9 and pMV19, were tested in\\ntransgenic worms (see Methods).', 'There was substantial agreementmiRNAs\\nRandomers\\nS/N = 2.2\\nNegative evidence Nongenetic Genetic\\nPositive evidenceS/N = 19.5S/N = 36.1–160\\n–140\\n–120\\n–100\\n–80\\n–60\\n–40–20\\n0Σ∆Gtotal (kcal mol–1)\\nFigure 3 The averagePDGtotalfor miRNAs compared with that calculated\\nfor randomers, for positive miRNA-target interactions supported either by\\ngenetic epistasis evidence or by nongenetic evidence, and for the set of 12putative lsy-6–target pairs predicted by conserved seed matching but having\\nnegative interactions in vivo\\n27(Table 2 ). S/N, signal-to-noise ratio. Error\\nbars represent s.d. for the three sets of interactions. Table 2 miRNA-target interaction energy predicts sensitivity of seed-\\nmatch targets to repression by lsy-6\\nTargetaRepression\\nsensitivitybPDGtotalc\\n(kcal mol/C01)PDGhybridc\\n(kcal mol/C01)\\ncog-1 + /C039.89 (+)d/C075.9 (+)b\\nZK637.13 /C0/C0 13.86 (+) /C017.3 (+)\\nC02B8.4 /C0/C0 4.22 ( /C0) /C062.3 (+)\\nF55G1.1 /C0 0.00 ( /C0)0 .', '0 ( /C0)\\nC48D5.2a /C0/C0 0.04 ( /C0) /C048.4 (+)\\nF59A6.1 /C0/C0 2.72 ( /C0) /C063.2 (+)\\nF40H3.4 /C0 0.00 ( /C0) /C017.0 (+)\\nT05C12.8 /C0 0.00 ( /C0) /C045.0 (+)\\nC27H6.3 /C0/C0 0.06 ( /C0) /C014.8 (+)\\nT23E1.1 /C0 0.00 ( /C0) /C014.5 (+)\\nT14G12.2 /C0 0.00 ( /C0) /C015.0 (+)\\nT20G5.9 /C0 0.00 ( /C0) /C016.5 (+)\\nR07E3.5 /C0 0.00 ( /C0)0 . 0 ( /C0)\\naRef.', '27.bForDGhybrid, an interaction is predicted to be functional (+) ifPDGhybridr\\n/C014 kcal mol/C01(see Table 1 ) and nonfunctional ( /C0) otherwise.cCalculated as in Table 1\\nfor sites identiﬁed from lsy-6–3 ¢UTR alignment (see Methods).dForDGtotal,alsy-6–target\\ninteraction is predicted to be functional (+) ifPDGtotalo/C010 kcal mol/C01and nonfunctional\\n(/C0) otherwise.ARTICLES\\n290 VOLUME 14 NUMBER 4 APRIL 2007 NATURE STRUCTURAL & MOLECULAR BIOLOGY\\n©2007 Nature Publishing Group  http://www.nature.com/nsmb\\nbetween predicted accessibility of the let-7 binding sites and the\\nobserved in vivo temporal repression of the corresponding transgenes\\n(Table 3 ). For pVT701, pVT702, pVT712 and pVT713, the expression\\nof the lacZ reporter constructs is substantially reduced in adult worms,\\nas compared with the expression of pMV19, which has two poorly\\naccessible sites.', 'For pVT704 and pVT705, the expression of the lacZ\\nreporter constructs is substantially increased in adults, as compared\\nwith the expression of pMV9. Furthermore, the correlation between\\nthein vivo repression andPDGtotalfor these constructs is 0.8083, with\\na signiﬁcant P-value of 0.0152. In a linear regression analysis usingPDGtotalof the binding sites to predict the degree of the repression\\n(Fig. 4 ),R2is 0.6534 and the P-value for the predictor is 0.0152. The\\nunderlying assumptions for a linear regression analysis were found tohold well, and an alternative weighted regression analysis yielded very\\nsimilar results (see Supplementary Data online). The R\\n2indicates that\\nour model can account for about two-thirds of the variability in the\\nrepression sensitivity.', 'Other factors are probably responsible for\\nthe remaining one-third of the variability, as well as for the appre-\\nciable difference in activity between pMV19 and pVT705, twoconstructs with comparablePDG\\ntotal values but different empirical\\nrankings ( Table 3 ). Our results indicate that constructs with multiple copies of a single\\nsite (pVT712 and pVT713) can be repressed by let-7 , provided that the\\nsites are structurally accessible. In this regard, the results for pVT712and pVT713 contrast with the activities of similar single-site\\nconstructs, pMV16 and pMV17, reported previously15. A major\\ndifference among these constructs is that adjacent sites in pVT712\\nand pVT713 are separated by the 27-nt spacer that is normally\\nbetween site 1 and site 2 of the wild-type lin-41 3¢UTR, whereas\\npMV16 and pMV17 contain a 4-nt spacer. Notably, pVT712 and\\npVT713 are predicted to contain accessible sites, whereas pMV16 and\\npMV17 are predicted to contain inaccessible sites.', 'We interpret these\\nresults to indicate that what is most essential for let-7 activity is the\\npresence of structurally accessible target sites, not the speciﬁc tandem\\nconﬁguration of site 1 and site 2 or the sequence content of the spacer\\nbetween adjacent sites. Finally,PDGhybrid cannot account for the\\nweak or absent repression of pMV19, pVT704 and pVT705, asPDGhybrid does not take into account the effects of target structure\\nand nucleation ( Table 3 ). DISCUSSION\\nIn this study, we examined the potential effects of the secondary\\nstructure of a target mRNA on the mRNA ’s sensitivity to regulationby miRNAs. We analyzed published data on the in vivo activity of\\nC. elegans reporter genes containing modiﬁed lin-41 3¢UTR\\nsequences, as well as other experimentally studied miRNA-targetinteractions from C. elegans and D.', 'melanogaster .W ef o u n dt h a ta\\ntwo-step model for miRNA-target hybridization that incorporates\\ntarget structure as a governing principle reliably accounts for these\\npublished data. We also tested in vivo a set of new C. elegans lin-41\\n3¢UTR mutants that were speciﬁcally designed to change the\\npredicted accessibility of let-7 –complementary sites without changing\\nthe sequences of the sites themselves. We found that our modelaccounts for the relative sensitivities of these mutants to let-7 repres-\\nsion, further supporting the idea that target structure exerts a potenteffect on miRNA activity. We model the binding of an miRNA to a complementary target site\\nas a two-step process. In the ﬁrst step, the miRNA nucleates base-pairing with a block of four contiguous, single-stranded nucleotidesof the target. In the second step, the miRNA completes base-pairingwith the mRNA, accompanied by the disruption of target secondary\\nstructure in the region of hybridization.', 'Computational implementa-\\ntion of each of these steps requires an accurate representation of thesecondary structure of the mRNA target, followed by the identiﬁcation\\nof open blocks for nucleation, and then accurate accounting for both\\nthe free-energy gain and energy loss in the miRNA-target base-pairingtransaction.', 'We represent the secondary structure of a particular\\nmRNA by a statistical sample of probable secondary structuresTable 3 miRNA-target interaction energy predicts sensitivity of novel mutant lin-41 3¢UTR reporters to repression by let-7\\n3¢UTR constructaFeatures of constructPDGtotalb\\n(kcal mol/C01)PDGhybridb\\n(kcal mol/C01)Observed\\nrepressioncRatio % b-gal\\n(adults/larvae) Worm lines ( n)\\npMV9 Wild-type sites and spacer /C043.44 (+)d/C057.0 (+)d+++ 0.39 ± 0.03 5\\npMV19 Spacer mutation /C08.31 ( /C0) /C057.0 (+) /C0 1.01 ± 0.25 3\\npVT701 Spacer mutation /C043.02 (+) /C057.0 (+) +++ 0.38 ± 0.16 3\\npVT702 Spacer mutation /C043.36 (+) /C057.0 (+) ++ 0.54 ± 0.06 3\\npVT704 Spacer mutation /C06.35 ( /C0) /C057.0 (+) /C0 0.88 ± 0.29 3\\npVT705 Spacer mutation /C07.21 ( /C0) /C057.0 (+) + 0.78 ± 0.15 2\\npVT712 Three copies of site 1 /C054.79 (+) /C083.3 (+) +++ 0.42 ± 0.10 4\\npVT713 Three copies of site 2 /C073.23 (+) /C087.0 (+) ++ 0.53 ± 0.26 2\\napMV9 and pMV19 were previously designed15; pVT701, pVT702, pVT703, pVT704, pVT705, pVT712 and pVT713 are new constructs designed for this study.bCalculated as in Table 1 .', 'cEmpirical ranking of repression is based on the ratio ( b-gal adults)/( b-gal larvae), adapting the published convention15(see Table 1 ), where the level of repression shown by pMV9 is scored as +++,\\nthe lack of repression for pMV19 as /C0and intermediate degrees of repression as + or ++. pVT704 was judged to be similar to pMV19; pVT701 and pVT712 are similar to pMV9.dInteractions are\\nscored as functional (+) or nonfunctional ( /C0)a si n Table 1 . 0 –20 –40 –60 –8000.20.40.60.811.21.4\\nR2 = 0.6534\\nP = 0.01517Ratio of β-gal % (adults/larvae)\\nΣ∆Gtotal (kcal mol–1)\\nFigure 4 Linear regression prediction of in vivo repression sensitivity\\n(measured by b-galactosidase ( b-gal) expression ratios in adult and larval\\nstages) by thePDGtotalfor the lin-41 3¢UTR mutant constructs (see\\nTable 3 ). The correlation between repression sensitivity andPDGtotalis\\n0.8083. Error bars represent s.d.', 'for multiple worm lines (see Table 3 ).ARTICLES\\nNATURE STRUCTURAL & MOLECULAR BIOLOGY VOLUME 14 NUMBER 4 APRIL 2007 291\\n©2007 Nature Publishing Group  http://www.nature.com/nsmb\\ngenerated by Sfold. Compared with the conventional MFE approach,\\nSfold has been shown to make improved structure predictions\\nand to better represent likely structure populations for mRNAs29,30. We also compared the performance of our model implemented\\nwith Sfold to the performance of models based on (i) the MFE\\nstructure, (ii) a heuristic set of suboptimal folds or (iii) a set of thelowest-energy structures. We found that the Sfold implementationperformed best in accounting for the experimentally observed in vivo\\nsensitivities of C. elegans lin-41 3¢UTR mutants ( Supplementary\\nTable 3 online).', 'We found that accurate modeling of miRNA-target interactions\\nrequires incorporation of an initiation-energy penalty (or nucleation-potential threshold) for overcoming the energy threshold required forbimolecular initiation of RNA-RNA base-pairing. When we omitted\\nthe nucleation-potential threshold, we found that differences in the\\nhybridization energies alone were not sufﬁcient to account for thelin-41 3¢UTR reporter data. In contrast, when the nucleation-\\npotential threshold was introduced, the model was able to more\\nconsistently account for the experimentally tested miRNA-target\\ninteractions. Our model seems to generally accommodate values forthe nucleation-potential threshold in the range encompassed by the\\ntwo published empirical values for the initiation energy, 4.09 kcal\\nmol\\n/C01(ref. 24) and 5.20 kcal mol/C01(ref. 23).', 'However, the value of\\n4.09 kcal mol/C01seemed to perform somewhat better than 5.2 kcal\\nmol/C01in our analyses, and so, for the purposes of this study, we\\nconsidered 4.09 kcal mol/C01as the best currently available approxima-\\ntion for the miRNA-target nucleation-potential threshold in vivo . Our approach is limited to the prediction of RNA secondary\\nstructures; it does not incorporate the potential effects of RNA-binding proteins and other associated factors, including componentsof the RNA-induced silencing complex (RISC), that could affect the\\nsecondary structure and accessibility of an mRNA target site or the\\nenergetics of its interaction with miRNAs in vivo . However, although\\nmRNA secondary structure is unlikely to be the only factor inﬂuenc-\\ning target accessibility, our results suggest that it is an important factor\\nin most cases that we examined and therefore that it probablycontributes to accessibility in most miRNA-target interactions.', 'The\\ndegree to which accessibility is also governed by associated proteins\\nmay depend on the particular miRNA-target interaction, the cellularcontext or both. Consistent with our ﬁndings that target structure canaffect miRNA activity, previous studies of factors governing the\\nefﬁciency of short interfering RNAs have suggested the importance\\nof target structure\\n31,32. Some previous studies of miRNA-target interactions have consid-\\nered the role of mRNA secondary structure. It has been proposed thatin vertebrates, miRNAs preferentially target 3 ¢UTRs with less stable\\nstructures\\n33; however, in another study of mammalian miRNAs, an\\neffect of target structure was not detected34. We found that the\\nfrequency of open blocks of 4 nt is better correlated with let-7 –\\nlin-41 data than is the frequency of 3-nt blocks, which was previously\\nconsidered18. The accuracy of those previous analyses of target\\nstructure was probably compromised by a reliance on predictions ofthe MFE structures.', 'A number of existing algorithms for miRNA target searching are\\nbased primarily on the identiﬁcation of conserved continuousWatson-Crick base-pairing in the 5 ¢seed region (nucleotides 2–8) of\\nthe miRNA\\n6,10,35,36. We believe that a strength of our structure-based\\napproach is that it does not speciﬁcally stipulate the positions ofhybridization-nucleating bases in the miRNA or the ﬁnal conﬁgura-tion of base-pairing between the miRNA and the target. Thus, our\\napproach should encompass interactions that would be identiﬁed byalgorithms based on the 5 ¢seed model and also identify interactions\\nthat do not ﬁt the strict 5 ¢seed pairing criteria.', 'The 5 ¢seed approach\\nhas been successful in identifying many bona ﬁde functional miRNA-\\ntarget interactions, probably reﬂecting an important role for seed\\npairing in general; however, the range of permissible seed-target base-\\npairing conﬁgurations is unknown, and evidence indicates that acontinuous Watson-Crick helix for the seed region is not alwaysnecessary or sufﬁcient for effective repression\\n6,15,27,37. Relaxation of\\nthe seed base-pairing requirement would encompass more authentictargets, but this leads to a lower signal-to-noise ratio and thus agreater chance of false positive predictions\\n13,36. Functional interactions\\nmay also include cases where nucleation occurs outside of the 5 ¢seed\\nregion. The results of our accessibility proﬁling analysis suggest thatnucleation can occur in the 3 ¢region of the miRNA, at least for the\\nlet-7 miRNA and site 2 of the lin-41 3¢UTR ( Fig. 2a ). For a set of C. elegans and D.', 'melanogaster miRNA-target inter-\\nactions predicted by seed pairing models, our structure-based modelperformed well, particularly for those interactions that had negative\\nvalidation tests ( Table 2 ). This suggests that sites meeting 5 ¢seed\\ncriteria can be functional if the sites are also structurally accessible,\\nwhereas other 5 ¢seed sites can be nonfunctional owing to structural\\ninaccessibility. Thus, our ﬁndings provide one explanation for theobservation that some targets predicted purely by seed pairing modelstest positively in validation tests, and others test negatively. Indeed,\\nthe lack of repression of certain lin-41 mutant constructs by let-7\\nis explained by our model, but not by other current approaches\\n(Tables 1 and3). Thus, we anticipate that the structure-based model\\ncan substantially improve the speciﬁcity of target predictions based on5¢seed pairing or other algorithms such as RNAhybrid, by reducing\\nthe rate of false positive predictions.', 'It is currently not possible to accurately assess the relative rates of\\nfalse negative predictions by our method compared with previoustarget-prediction approaches. This would require systematic testingof targets not predicted by each method, and such data are not\\nyet available. However, we did ﬁnd that our method conﬁrmed\\n100% of the set of interactions supported by genetic epistasis tests(Fig. 3 andSupplementary Table 2 ). The agreement with experi-\\nmental data was less (69%) for predictions that had been validatedonly by nongenetic criteria ( Fig. 3 and Supplementary Table 2 ),\\nsuggesting a false negative rate of 31%. Nevertheless, we observeda highly signiﬁcant difference (see Results) between miRNAs and\\nrandomers for both the genetic set and the nongenetic set, and the\\ndifference is much greater for the genetic set ( Fig. 3 andSupplemen-\\ntary Table 2 ). There are a number of possible explanations for our\\nfalse negative predictions.', 'First, the interactions that are currentlysupported only by nongenetic tests are generally less extensivelystudied than the other set, and some of these could become reclassiﬁed\\nas negative after further testing. Second, some of the negative predic-\\ntions could be erroneous on account of misannotated mRNAsequences, which could lead to inaccurate structural modeling of the\\ntarget mRNA in such cases. Perhaps the most intriguing explanation\\nwould be the possible existence of a subclass of mRNA targets that areconditionally accessible to miRNA binding—where target accessibilitycould be modulated in vivo by RNA-binding proteins and/or RISC\\ncomponents to counteract otherwise intrinsically closed local targetsecondary structure. Nearly all of the miRNA-target interactions that we examined here\\nhad been predicted based on phylogenetic conservation of the pre-dicted base-pairing.', 'Conservation of miRNA target sites in ortho-logous genes is a powerful criterion for identifying potentially\\nfunctional targets. However, as we do not fully understand whatARTICLES\\n292 VOLUME 14 NUMBER 4 APRIL 2007 NATURE STRUCTURAL & MOLECULAR BIOLOGY\\n©2007 Nature Publishing Group  http://www.nature.com/nsmb\\nfactors exert selective pressure on conserved 3 ¢UTR sequences, it is\\npossible that some conserved sites are authentic, whereas others are\\nnot functional in vivo . Our ﬁndings suggest that inclusion of target-\\naccessibility criteria could reduce the frequency of such false positive\\npredictions from conserved sequences. Conservation-based algorithms\\ncan also be prone to false negative predictions in cases where sites thatare not conserved in sequence alignments are nevertheless demon-strably functional in vivo\\n34or in cases of poorly conserved human or\\nvirally encoded miRNAs2,38.', 'The structure-based model described\\nhere, combined other known features of functional miRNA targetsites, including aspects of the 5 ¢seed pairing model, could provide the\\nbasis for the reliable identiﬁcation of both conserved and non-conserved miRNA regulatory sites in genome-wide searches. METHODS\\nPrediction of mRNA secondary structures. Accurate modeling of mRNA\\nstructure must take into account the principle that a speciﬁc mRNA molecule\\nprobably exists in vivo in dynamic equilibrium among a population of\\nstructures. Sfold is used for this purpose, because its algorithm generates a statis-tically representative sample from the Boltzmann-weighted ensemble of RNAsecondary structures for the RNA\\n22. Samples of 1,000 structures can faithfully\\nand reproducibly characterize structure ensembles of enormous sizes22,30. Therefore, sampling estimates of structural features for a particular RNA arestatistically reproducible from one sample to another.', 'Additional informationon RNA folding can be found in the Supplementary Methods online. Energetic modeling of two-step hybridization. On the basis of studies on\\nnucleic acid hybridization\\n23,39–42, we modeled the miRNA-target hybridization\\nas a two-step process: (i) nucleation (the rate-limiting step), involving fourconsecutive complementary nucleotides in the two RNAs, and (ii) elongationof the hybrid to form a stable intermolecular duplex ( Fig. 1 ). An miRNA itself\\nis unlikely to have stable secondary structure owing to its short length. Therefore, for the identiﬁcation of potential nucleation sites for an miRNA\\non a structured target, the miRNA itself was assumed to be unstructured. Computation of nucleation potential.', 'Nucleation of base-pairing between two\\nnucleic acid molecules requires a gain in free energy from base-pairing at the\\nnucleation site that is greater than the energy cost for the translational androtational entropy loss when two RNA strands are ﬁxed in a conformation by\\nintermolecular base-pairing\\n24. This latter energy cost—that is, the penalty of\\n‘initiation energy’—presents an energy threshold for nucleation. Therefore, we\\ndetermined the nucleation competence of a putative nucleation site by\\ncomputing the free energy gained from base-pairing at that nucleation site\\n(the ‘nucleation potential’ , DGN) and comparing it to a chosen value of the\\ninitiation-energy threshold ( DGinitiation ). There are two published values for\\nthe initiation-energy threshold, 4.09 kcal mol/C01(ref. 24) and 5.2 kcal mol/C01\\n(ref. 23). We considered both values in our analyses.', 'Sites with nucleationpotentials that are favorable compared with the initiation-energy threshold areconsidered to be competent for nucleation. In our model, we assume that four unpaired bases in the target can initiate\\nnucleation\\n23, and we therefore identiﬁed possible miRNA binding sites deﬁned\\nby miRNA-mRNA complementary matches of at least four contiguous nucleo-tide matches anywhere in the miRNA. (For lin-41 mutants, we considered the\\npreviously described complementary sites\\n15as possible let-7–binding sites). We\\nthen computed the nucleation potential of a possible miRNA-binding site byidentifying the particular open (single-stranded) 4-nt block within the site thatwould form the most stable 4-base-pair duplex with the miRNA. T o identify\\nopen blocks of nucleotides in the target site, we used a sample of 1,000\\nstructures for the target RNA predicted by Sfold.', 'We considered each of the1,000 structures in the sample separately and calculated the average nucleationpotential over the whole set of structures in the sample (see Supplementary\\nMethods ). Nucleation is considered favorable if the calculated average nuclea-\\ntion potential can overcome the initiation energy threshold—that is, DG\\nN+\\nDGinitiationo0k c a lm o l/C01. Computation of total hybridization energy. After nucleation of miRNA-\\ntarget binding, a hybrid elongation process follows, involving breakage ofintramolecular base pairs in the target to accommodate the formation of anoptimal intermolecular miRNA-target duplex.', 'Because nucleation is the rate-limiting step for nucleic acid hybridization, we calculated the total energy\\nchange for the hybridization ( DG\\ntotal) only for those sites on the target with\\nfavorable nucleation potential ( DGN+DGinitiationo0k c a lm o l/C01).DGtotal¼\\nDGhybrid/C0DGdisruption ,w h e r e DGdisruption is the energy for the disruption of\\nthe intramolecular base pairs that involve binding-site nucleotides, andDG\\nhybrid is the energy gain owing to the complete intermolecular hybridization\\nof the miRNA with the binding site ( Fig. 1b ). The complete hybrid may contain\\none or more loops ( Fig. 1b ) and is assumed to be in the conformation of the\\nlowest free energy. Therefore, we calculated DGhybrid using RNAhybrid, which\\nﬁnds the most stable miRNA-target hybrid conformation.', 'T o calculate DGdisruption , we adopted the simplifying assumption that the\\nbinding of an miRNA to a relatively much longer mRNA should cause a local\\nstructural alteration at the target site but not longer-range effects on overall\\ntarget secondary structure. Speciﬁcally, we deﬁned local structural alteration as\\nthe breakage of only those target intramolecular base pairs that must be brokento permit formation of the miRNA-target duplex predicted by RNAhybrid. DG\\ndisruption was calculated as the energy difference between DGbefore ,t h ef r e e\\nenergy of the original mRNA structure, and DGafter, the free energy of the new\\nlocally altered structure ( DGdisruption ¼DGbefore/C0DGafter). We calculated\\nDGbefore as the average energy of the original 1,000 structures predicted by\\nSfold and DGafter as the average energy of all of the 1,000 locally altered\\nstructures. Therefore, under the assumption of local disruption, the calcula-\\ntions do not require refolding of the rest of the target sequence.', 'T o model the cooperative effects of multiple binding sites on a single 3 ¢UTR,\\nwe computedPDGtotal for sites with favorable nucleation potentials. This\\nenergetic additivity implicitly assumes independent contributions from multi-ple sites. However, it is possible that occupancy at one site could alter the\\naccessibility for another site. Modiﬁcations of the above energetic calculations\\nwill be required for modeling potential structure-mediated interactions amongmultiple sites. In vivo testing of lin-41 3¢UTR mutants. Mutant constructs were generated\\nand tested as described\\n15(see also Supplementary Methods andSupplemen-\\ntary Tables 4 and5online). Software availability. We have developed STarMir, a new application module\\nfor the Sfold software. ST arMir implements the approach described here to\\nperform energy calculations for miRNA-target hybridization and is available\\nthrough the Sfold web server at http://sfold.wadsworth.org.', 'Note: Supplementary information is available on the Nature Structural &Molecular\\nBiology website. ACKNOWLEDGMENTS\\nWe acknowledge the Computational Molecular Biology and Statistics Core at theWadsworth Center for providing computing resources. This work was supportedin part by US National Science Foundation grants DMS-0200970 and DBI-0650991 and US National Institutes of Health grant GM068726 to Y.D., and by\\nUS National Institutes of Health grants GM34028 and GM066826 to V .A. We\\nthank F. Slack of Y ale University for gifts of plasmids, and A. Lee, G. Ambrosand members of the Ambros lab for technical help and advice. AUTHOR CONTRIBUTIONS\\nD.L. and Y.D. designed the algorithm and performed computational modeling of\\nRNA structure and thermodynamics, R.L. and V .A. analyzed lin-41 reporter genes\\ninC. elegans , P .W. performed computational modeling and C.Y .C. developed the\\nweb interface. COMPETING INTERESTS STATEMENT\\nThe authors declare no competing ﬁnancial interests.', 'Published online at http://www.nature.com/nsmb\\nReprints and permissions information is available online at http://npg.nature.com/reprintsandpermissions\\n1. Ambros, V. The functions of animal microRNAs. Nature 431, 350–355 (2004). 2. Bentwich, I. et al. Identiﬁcation of hundreds of conserved and nonconserved human\\nmicroRNAs. Nat. Genet. 37, 766–770 (2005).ARTICLES\\nNATURE STRUCTURAL & MOLECULAR BIOLOGY VOLUME 14 NUMBER 4 APRIL 2007 293\\n©2007 Nature Publishing Group  http://www.nature.com/nsmb\\n3. Boehm, M. & Slack, F. A developmental timing microRNA and its target regulate life\\nspan in C. elegans .Science 310, 1954–1957 (2005). 4. Calin, G.A. & Croce, C.M. MicroRNA-cancer connection: the beginning of a new tale. Cancer Res. 66, 7390–7394 (2006). 5. Cuellar, T.L. & McManus, M.T. MicroRNAs and endocrine biology. J. Endocrinol. 187,\\n327–332 (2005). 6. Brennecke, J., Stark, A., Russell, R.B. & Cohen, S.M. Principles of microRNA-target\\nrecognition. PLoS Biol. 3, e85 (2005). 7.', 'Sethupathy, P., Corda, B. & Hatzigeorgiou, A.G. TarBase: a comprehensive database of\\nexperimentally supported animal microRNA targets. RNA 12, 192–197 (2006). 8. Jones-Rhoades, M.W. & Bartel, D.P. Computational identiﬁcation of plant microRNAs\\nand their targets, including a stress-induced miRNA. Mol. Cell 14, 787–799 (2004). 9. Lai, E.C. Predicting and validating microRNA targets. Genome Biol. 5, 115 (2004). 10. Lewis, B.P., Burge, C.B. & Bartel, D.P . Conserved seed pairing, often ﬂanked by\\nadenosines, indicates that thousands of human genes are microRNA targets. Cell120,\\n15–20 (2005). 11. Doench, J.G. & Sharp, P.A. Speciﬁcity of microRNA target selection in translational\\nrepression. Genes Dev. 18, 504–511 (2004). 12. Kiriakidou, M. et al. A combined computational-experimental approach predicts\\nhuman microRNA targets. Genes Dev. 18, 1165–1178 (2004). 13. Lewis, B.P., Shih, I.H., Jones-Rhoades, M.W., Bartel, D.P. & Burge, C.B. Prediction of\\nmammalian microRNA targets.', 'Cell115, 787–798 (2003). 14. Rajewsky, N. microRNA target predictions in animals. Nat. Genet. 38Suppl, S8–S13\\n(2006). 15. Vella, M.C., Choi, E.Y., Lin, S.Y., Reinert, K. & Slack, F.J. The C. elegans microRNA\\nlet-7 binds to imperfect let-7 complementary sites from the lin-41 3¢UTR. Genes Dev. 18, 132–137 (2004). 16. Vella, M.C., Reinert, K. & Slack, F.J. Architecture of a validated microRNA:target\\ninteraction. Chem. Biol. 11, 1619–1623 (2004). 17. Zuker, M. Mfold web server for nucleic acid folding and hybridization prediction. Nucleic Acids Res. 31, 3406–3415 (2003). 18. Robins, H., Li, Y. & Padgett, R.W. Incorporating structure to predict microRNA targets. Proc. Natl. Acad. Sci. USA 102, 4006–4009 (2005). 19. Rehmsmeier, M., Steffen, P., Hochsmann, M. & Giegerich, R. Fast and effective\\nprediction of microRNA-target duplexes. RNA 10, 1507–1517 (2004). 20. Ding, Y., Chan, C.Y. & Lawrence, C.E. Sfold web server for statistical folding\\nand rational design of nucleic acids.', 'Nucleic Acids Res. 32, W135–W141 (2004). 21. Ding, Y. & Lawrence, C.E. Statistical prediction of single-stranded regions in RNA\\nsecondary structure and application to predicting effective antisense target sites and\\nbeyond. Nucleic Acids Res. 29, 1034–1046 (2001). 22. Ding, Y. & Lawrence, C.E. A statistical sampling algorithm for RNA secondary structure\\nprediction. Nucleic Acids Res. 31, 7280–7301 (2003). 23. Hargittai, M.R., Gorelick, R.J., Rouzina, I. & Musier-Forsyth, K. Mechanistic insights\\ninto the kinetics of HIV-1 nucleocapsid protein-facilitated tRNA annealing to the\\nprimer binding site. J. Mol. Biol. 337, 951–968 (2004).24. Xia, T. et al. Thermodynamic parameters for an expanded nearest-neighbor model\\nfor formation of RNA duplexes with Watson-Crick base pairs. Biochemistry 37,\\n14719–14735 (1998). 25. Clote, P., Ferre, F., Kranakis, E. & Krizanc, D. Structural RNA has lower folding\\nenergy than random RNA of the same dinucleotide frequency. RNA 11, 578–591\\n(2005).', '2 6 .S e g g e r s o n ,K . ,T a n g ,L .&M o s s ,E . G .T w og e n e t i cc i r c u i t sr e p r e s st h e Caenorhabditis\\nelegans heterochronic gene lin-28 after translation initiation. Dev. Biol. 243, 215–225\\n(2002). 27. Didiano, D. & Hobert, O. Perfect seed pairing is not a generally reliable predictor for\\nmiRNA-target interactions. Nat. Struct. Mol. Biol. 13, 849–851 (2006). 28. Johnston, R.J. & Hobert, O. A microRNA controlling left/right neuronal asymmetry in\\nCaenorhabditis elegans .Nature 426, 845–849 (2003). 29. Ding, Y., Chan, C.Y. & Lawrence, C.E. RNA secondary structure prediction by centroids\\nin a Boltzmann weighted ensemble. RNA 11, 1157–1166 (2005). 30. Ding, Y., Chan, C.Y. & Lawrence, C.E. Clustering of RNA secondary structures with\\napplication to messenger RNAs. J. Mol. Biol. 359, 554–571 (2006). 31. Overhoff, M. et al. Local RNA target structure inﬂuences siRNA efﬁcacy: a systematic\\nglobal analysis. J. Mol. Biol. 348, 871–881 (2005). 32.', 'Schubert, S., Grunweller, A., Erdmann, V.A. & Kurreck, J. Local RNA target structure\\ninﬂuences siRNA efﬁcacy: systematic analysis of intentionally designed bindingregions. J. Mol. Biol. 348, 883–893 (2005). 33. Zhao, Y., Samal, E. & Srivastava, D. Serum response factor regulates a muscle-\\nspeciﬁc microRNA that targets Hand2 during cardiogenesis. Nature 436, 214–220\\n(2005). 34. Farh, K.K. et al. The widespread impact of mammalian MicroRNAs on mRNA\\nrepression and evolution. Science 310, 1817–1821 (2005). 35. Enright, A.J. et al. MicroRNA targets in Drosophila .Genome Biol. 5,R 1( 2 0 0 3 ) . 36. Krek, A. et al. Combinatorial microRNA target predictions. Nat. Genet. 37, 495–500\\n(2005). 37. Miranda, K.C. et al. A pattern-based method for the identiﬁcation of microRNA\\nbinding sites and their corresponding heteroduplexes. Cell 126, 1203–1217\\n(2006). 38. Cullen, B.R. Viruses and microRNAs. Nat. Genet. 38Suppl, S25–S30 (2006). 39. Paillart, J.C., Skripkin, E., Ehresmann, B., Ehresmann, C.', '& Marquet, R. A loop-loop\\n‘‘kissing’’ complex is the essential part of the dimer linkage of genomic HIV-1 RNA. Proc. Natl. Acad. Sci. USA 93, 5572–5577 (1996). 40. Reynaldo, L.P., Vologodskii, A.V., Neri, B.P. & Lyamichev, V.I. The kinetics of\\noligonucleotide replacements. J. Mol. Biol. 297, 511–520 (2000). 41. Milner, N., Mir, K.U. & Southern, E.M. Selecting effective antisense rea-\\ngents on combinatorial oligonucleotide arrays. Nat. Biotechnol. 15,5 3 7 – 5 4 1\\n(1997). 42. Kolb, F.A. et al. Bulged residues promote the progression of a loop-loop interaction\\nto a stable and inhibitory antisense-target RNA complex. Nucleic Acids Res. 29,\\n3145–3153 (2001).ARTICLES\\n294 VOLUME 14 NUMBER 4 APRIL 2007 NATURE STRUCTURAL & MOLECULAR BIOLOGY\\n©2007 Nature Publishing Group  http://www.nature.com/nsmb\\nCLIP-based prediction of mammalian microRNA\\nbinding sites\\nChaochun Liu, Bibekanand Mallick, Dang Long, William A. Rennie, Adam Wolenc,\\nC.', 'Steven Carmack and Ye Ding*\\nWadsworth Center, New York State Department of Health, Center for Medical Science, 150 New Scotland\\nAvenue, Albany, NY 12208, USA\\nReceived February 11, 2013; Revised March 29, 2013; Accepted April 29, 2013\\nABSTRACT\\nPrediction and validation of microRNA (miRNA)\\ntargets are essential for understanding functions ofmiRNAs in gene regulation. Crosslinking immuno-\\nprecipitation (CLIP) allows direct identification of a\\nhuge number of Argonaute-bound target sequencesthat contain miRNA binding sites. By analysing datafrom CLIP studies, we identified a comprehensive listof sequence, thermodynamic and target structurefeatures that are essential for target binding bymiRNAs in the 3\\n0untranslated region (30UTR),\\ncoding sequence (CDS) region and 50untranslated\\nregion (50UTR) of target messenger RNA (mRNA).', 'The total energy of miRNA:target hybridization, ameasure of target structural accessibility, is theonly essential feature common for both seedand seedless sites in all three target regions.Furthermore, evolutionary conservation is an import-ant discriminating feature for both seed and seedlesssites. These features enabled us to develop novel\\nstatistical models for the predictions of both seed\\nsites and broad classes of seedless sites. Throughboth intra-dataset validation and inter-dataset valid-ation, our approach showed major improvementsover established algorithms for predicting seedsites and a class of seedless sites. Furthermore, weobserved good performance from cross-species val-idation, suggesting that our prediction framework\\ncan be valuable for broad application to other mam-\\nmalian species and beyond.', 'Transcriptome-widebinding site predictions enabled by our approachwill greatly complement the available CLIP data,which only cover small fractions of transcriptomes\\nand known miRNAs due to non-detectable levels of\\nexpression. Software and database tools based onthe prediction models have been developed and areavailable through Sfold web server at http://sfold.wadsworth.org. INTRODUCTION\\nMicroRNAs (miRNAs) are an abundant class of small\\nendogenous non-coding RNAs (ncRNAs) of /C2422 nucleo-\\ntides (nt) in length that have been found in plants, animals\\nand viruses. miRNAs are post-transcriptional regulators\\ninvolved in the regulation of diverse developmental\\nprocesses, molecular and cellular pathways and human\\ndiseases (1). A mature miRNA can guide RNA-induced\\nsilencing complex (RISC) for target recognition bysequence complementarity between the miRNA and\\nsequences typically in the 3\\n0untranslated regions\\n(30UTRs) of the cognitive messenger RNAs (mRNAs).', 'Successful target binding usually results in translationalrepression and/or mRNA degradation (2). Each human\\nmiRNA is believed to be able to regulate several\\nhundred different mRNAs (3). On the other hand, an in-\\ndividual mRNA may be simultaneously targeted bymultiple miRNAs. These highlight the versatility and ex-\\ntensiveness of gene regulation by miRNAs. For animal\\nmiRNAs, the complementarity is typically imperfectwith mismatches. This presents a major challenge for\\ntarget identiﬁcation, as straightforward sequence-\\nalignment-based genome-scale target search generates\\ntoo many hits to be useful for efﬁcient experimental\\nvalidation. *To whom correspondence should be addressed.', 'Tel: + 518 486 1719; Fax: + 518 402 4623; Email: yding@wadsworth.org\\nThe authors wish it to be known that, in their opinion, the ﬁrst two authors should be regarded as joint First Authors.Present addresses:\\nBibekanand Mallick, RNA Biology and Functional Genomics Laboratory, Department of Life Science, National Institute of Technology, Rourkela-769008, Odisha, India.Dang Long, Biotechnology Department, Faculty of Chemistry, Danang University of Science and Technology, 54 Nguyen Luong Bang St., Danang,Vietnam.Published online 22 May 2013 Nucleic Acids Research, 2013, Vol. 41, No. 14 e138\\ndoi:10.1093/nar/gkt435\\n/C223The Author(s) 2013. Published by Oxford University Press. This is an Open Access article distributed under the terms of the Creative Commons Attribution License (http://creativecommons.org/licenses/by/3.0/), which\\npermits unrestricted reuse, distribution, and reproduction in any medium, provided the original work is properly cited.', 'at NYS Department of Health on October 15, 2013 http://nar.oxfordjournals.org/ Downloaded from  at NYS Department of Health on October 15, 2013 http://nar.oxfordjournals.org/ Downloaded from  at NYS Department of Health on October 15, 2013 http://nar.oxfordjournals.org/ Downloaded from  at NYS Department of Health on October 15, 2013 http://nar.oxfordjournals.org/ Downloaded from  at NYS Department of Health on October 15, 2013 http://nar.oxfordjournals.org/ Downloaded from  at NYS Department of Health on October 15, 2013 http://nar.oxfordjournals.org/ Downloaded from  at NYS Department of Health on October 15, 2013 http://nar.oxfordjournals.org/ Downloaded from  at NYS Department of Health on October 15, 2013 http://nar.oxfordjournals.org/ Downloaded from  at NYS Department of Health on October 15, 2013 http://nar.oxfordjournals.org/ Downloaded from  at NYS Department of Health on October 15, 2013 http://nar.oxfordjournals.org/ Downloaded from  at NYS Department of Health on October 15, 2013 http://nar.oxfordjournals.org/ Downloaded from  at NYS Department of Health on October 15, 2013 http://nar.oxfordjournals.org/ Downloaded from  at NYS Department of Health on October 15, 2013 http://nar.oxfordjournals.org/ Downloaded from \\nExperimental validation of computational predictions\\nhas been the most popular and simplest approach fortarget identiﬁcation.', 'Most of the existing algorithms are\\nbased on the seed rule, i.e., the target site within 3\\n0UTR\\nforms Watson–Crick (WC) pairs with bases at positions 2through 7 or 8 of the 5\\n0end of the miRNA (3). However,\\nnumerous exceptions to the seed rule have been well docu-\\nmented (4–7). Other sequence features have been proposedbased on their enhancement of targeting speciﬁcity. These\\ninclude sequence conservation, strong base-pairing to the\\n3\\n0end of the miRNA, local AU content and location of\\nmiRNA binding sites (near either end of the 30UTR is\\nfavorable) (8). The importance of target structural acces-\\nsibility for miRNA target recognition has been suggestedby several studies (9–12). In particular, a target structure-based model validated with worm target data (12) was\\nsupported by an independent mammalian study with a\\nhigher true positive rate than the seed-based predictions(13). Findings from analyses of additional miRNA target\\ndata further support the importance of target structure\\n(14–15).', 'While computational prediction algorithms have\\nproven to be valuable in the discovery of new miRNA\\ntargets, current algorithms are known to make highnumbers of false (positive or negative) predictions (16).In recent years, experimental methods based on\\ncrosslinking immunoprecipitation (CLIP) have been de-\\nveloped (7,17–21). CLIP methods involve UV irradiationof tissues, organisms or cells, for covalently crosslinking\\nmiRNA targets to the Argonaute (AGO) proteins, the\\ncatalytic components of the RISC complex. Thecrosslinked RNAs are reduced in size typically to /C2450 nt\\nby partial RNase digestion. The short RNAs are ampliﬁed\\nby RT-PCR and then sequenced for the identiﬁcation ofAGO tags that contain miRNA binding sites on target\\nmRNAs. For mammalian systems, these include HITS-\\nCLIP for mouse brain (18), PAR-CLIP for human celllines (17), and variations referred to here for convenienceas V-CLIP, V-PAR-CLIP and V-PAR-CLIP-MNase for\\nthe same cell lines as PAR-CLIP (20).', 'Unlike previous\\nexperimental techniques of mRNA expression proﬁlingand proteomics (22–24), these CLIP techniques allow\\ndirect identiﬁcation of a huge pool of short target se-\\nquences that contain native miRNA binding sites. CLIPmethods not only provide much higher resolution data\\nwith respect to the precise locations of the binding sites,\\nbut also avoid the problem of potential secondary effectsassociated with mRNA expression proﬁling and prote-omics approaches. In addition, these CLIP datasets are\\npowerful for revealing the presence of ‘seedless’ sites\\n(non-canonical sites) and possible seed or seedless siteslocated in the coding sequences (CDS) or 5\\n0UTRs\\n(4,25,26). The CLIP studies have provided high throughout\\nquality datasets for the identiﬁcation of important\\nmiRNA binding site features as the basis for the develop-\\nment of a novel framework for improved predictions ofmiRNA binding sites.', 'To pursue this objective, we started\\nwith a comprehensive enrichment analysis of target site\\nfeatures for both seed and seedless sites in all threemRNA regions (5\\n0UTR, CDS, 30UTR). EnrichedmiRNA binding site features were used for the develop-\\nment of logistic models for miRNA binding site pre-dictions. Model-based predictions were evaluated byintra-dataset and inter-dataset validation, with the latterincluding cross-species validation. In addition, the predic-tions were compared with available predictions by severalwell-established algorithms. MATERIALS AND METHODS\\nThe pipeline of CLIP data analysis and modeling is\\noutlined in Figure 1, followed by description of proced-ures for each step of the pipeline. Description and processing of CLIP data\\nHITS-CLIP\\n(18). For seed sites, we directly downloaded\\ntarget sites identiﬁed by the authors (http://ago.rockefeller.edu/tag_mm9.php) for deﬁning IP + sites.', 'For\\nseedless sites, the raw AGO-mRNA HITS-CLIP Solexa\\nsequencing data were downloaded from the authors’website (http://ago.rockefeller.edu/rawdata.php) andthen mapped to mouse genome using RMAP (27). Amaximum of two mismatched nucleotides were allowedto ensure >90% identity in sequence alignment. Only\\nreads with unique mapping were considered. Forensuring experimental reproducibility, two differentantibodies, 2A8 with three biological replicates and 7G1-1 with two replicates, were used in the HITS-CLIP study.We identiﬁed the common tags reproduced by at least onereplicate for each of the two antibodies. Two tags for the\\ntwo antibodies are considered a common tag if and only if\\nthey have exactly the same nucleotide sequence andchromosomal location. The locations of the tags withintarget mRNAs were obtained with annotation from theMap Viewer of NCBI Build 37.2 for Mus musculus\\ngenome.', 'Finally, we identiﬁed 6,666 unique AGO tagsdistributed across mRNAs of 2211 genes among 12 389genes in P13 mouse brain used by the HITS-CLIP study(gene list provided by the authors). Thus, these 6666 AGOtags were used for our analysis. Among them, /C2461% tags\\nwere mapped to 3\\n0UTRs, /C2437% to CDS and /C242% to 50\\nUTRs. Each of these AGO tags is considered to harbor at\\nleast one miRNA binding site. The HITS-CLIP study\\nfocused on 20 most abundant miRNA families (114\\nmiRNAs) comprising 88% of all AGO-associatedmiRNAs. We thus focused on these miRNA families inour data analysis and modeling. PAR-CLIP\\n(17). This study identiﬁed 17 319 miRNA\\nsite-containing AGO crosslink-centered regions (CCRs).Among exonic CCRs representing 85% of all AGOCCRs, 4% mapped to 5\\n0UTRs, 50% to CDS and 46%\\nto 30UTRs. The 100 most abundant miRNAs in HEK 293\\ncells accounted for 95% of total miRNA sequence reads. These miRNAs were used in data analysis by the authors,\\nand were the focus of our analysis.', 'The AGO CCRs datawere obtained from the authors. Based on genomic loca-tions according to the annotation of Homo sapiens hg18,\\nthe CCRs are distributed across 5307 mRNAs among 22466 transcripts expressed in human embryonic kidney(HEK) 293 cells (transcript list provided by the authors).e138 Nucleic Acids Research, 2013, Vol. 41, No. 14 P\\nAGE2OF13\\nV-CLIP, V-PAR-CLIP, V-PAR-CLIP-MNase (20). We\\ndownloaded the three CLIP datasets from Gene\\nExpression Omnibus: GSE28865. We selected experimen-\\ntal replicate A for our analysis due to higher reliability as\\nreported (20). For V-CLIP, there are 54 905 CCRsdistributed across 8818 mRNAs among 17 928 transcripts\\nexpressed in human HEK 293 cells. For V-PAR-CLIP, 91\\n362 CCRs are distributed across 9676 of 16 491 expressed\\ntranscripts. V-PAR-CLIP-MNase generated 44 497 CCRs\\nacross 8562 of 20 424 expressed transcripts.', 'For each setof expressed transcripts, by using accession number, we\\ncompiled the list of those that are annotated in NCBI\\nRefSeq database (build 36.3). This resulted in 44 557\\nCCRs across 7507 of 12 440 RefSeq transcripts for\\nV-CLIP, 73 551 CCRs across 8099 of 11 660 transcriptfor V-PAR-CLIP and 37 125 CCRs across 7293 of 13 419\\ntranscripts for V-PAR-CLIP-MNase. For V-CLIP, /C242%\\nof CCRs were mapped to 5\\n0UTRs, /C2431% to CDS and\\n/C2467% to 30UTRs. For V-PAR-CLIP, /C242% of CCRs\\nwere mapped to 50UTRs, /C2435% to CDS and /C2463% to\\n30UTRs. For V-PAR-CLIP-MNase, /C243% of CCRs were\\nmapped to 50UTRs, /C2441% to CDS and /C2456% to 30\\nUTRs. A set of the 10 most abundant miRNA families\\ncontaining 44 mature miRNAs was used by the authors\\nfor analyses of data generated by the 3 CLIP protocols. These miRNAs were used in our study, with sequences\\nprovided by the authors and names updated usingmiRBase Release 18 (28).', 'Identiﬁcation of miRNA binding sites\\nTo identify the potential miRNA binding sites, we used\\nthe RNAhybrid program (29). RNAhybrid was run using\\ntwo options for aligning a miRNA with an mRNA. Forthe ﬁrst option, as in our earlier work for worm (30), we\\nused a threshold of /C015 kcal/mol for /C1G\\nhybrid (a measureof hybrid stability) to identify potential sites with a stable\\nhybrid and to account for potential seedless sites that can\\ninclude rare centered sites (31). The second option is to\\nforce base-pairing for the seed region (miRNA nt 2-7, 2-8,\\n3-8) without enforcing an energy threshold. The latter will\\ncapture all cases of offset 6mer, 6mer, 7mer-A1, 7mer-m8\\nand 8mer seed sites (8). Some of these sites can have weak\\nor no base-pairing beyond the seed region such that they\\nwould be missed by the ﬁrst option due to the enforced\\nenergy threshold. However, the use of the two options can\\nlead to site redundancy, e.g., two heavily overlapped sites\\nof different seed types.', 'To remove redundancy, we used\\nthe following procedure for selecting one out of multiple\\noverlapped sites: 8mer is the most preferred, followed by\\n7mer-m8, 7mer-A1, 6mer, offset 6mer and seedless sites; in\\nthe case of same site types, the one with lower /C1Ghybrid\\n(i.e., greater hybrid stability) is preferred. The procedure\\nwas based on the established seed hierarchy (8) and\\nthermodynamic consideration. Site sequence features\\nTo study the contribution of sequence characteristics on\\ntarget recognition, we considered site sequence features as\\npreviously reported (8).', 'These include seed types of offset\\n6mer, 6mer, 7mer-A1, 7mer-m8 and 8mer, miRNA 30\\ncompensatory/supplementary base-pairing as deﬁned bythe presence of contiguous WC base-pairing for miRNA\\nnucleotide position 12-17, site location as deﬁned by the\\nlocation of the site relative to the length of the region\\n(3\\n0UTR, CDS or 50UTR; e.g., for 30UTR, 0 indicates\\n50end of the 30UTR and 1 corresponds to the 30end.), AU\\ncontent deﬁned as the percentage of the adenosine or\\nuridine bases for a nucleotide block either upstream or\\ndownstream of the site. The sizes of the block used for\\ncomputing are 5, 10, 15, 20, 25 and 30 nt. Thermodynamic and target structure features\\nWe computed /C1Gnucl, a measure of nucleation potential,\\nand /C1Gtotal, a measure of total energy change for\\nmiRNA–target hybridization. These target structure-\\nbased parameters are key characteristics of our two-step\\nmodel for miRNA:target interaction (12).', '/C1Ghybrid,a\\nmeasure of the stability of miRNA–target duplex as\\ncomputed by RNAhybrid (29) was also included in our\\nanalysis. While /C1Gtotalis an energetic measure of struc-\\ntural accessibility at the target site, we also computed\\nseveral probabilistic measures of structural accessibility\\nas follows. For a block of nucleotides, the accessibility\\ncan be measured by the average probability that a nucleo-\\ntide in the block is single stranded, based on the RNA\\nsecondary structure sampling algorithm implemented by\\nSfold (32,33). We considered the entire target site for\\ncomputing site accessibility, the block within the target\\nsite complementary to miRNA seed for seed accessibility,\\nand the block upstream or downstream the target site for\\nupstream accessibility or downstream accessibility.', 'The\\nsizes for upstream or downstream block are 5, 10, 15,\\n20, 25 and 30 nt.mRNA sequences miRNA sequences\\nIdentification of potential sites\\nby RNAhybrid and seed match\\nComputation of sequence, thermodynamic and \\nstructure features for both IP+ and IP- sitesClassification into IP+ or IP- site \\nLogistic modeling for\\ntarget site predictionIdentification of site features\\nenriched in IP+ set\\nIntra-dataset and inter-dataset\\nmodel validationPerformance comparison\\nwith established algorithmswith AGO crosslinked regions\\nFigure 1. Flow chart for the CLIP data analysis and modeling.PAGE3OF13 Nucleic Acids Research, 2013, Vol. 41, No. 14 e138\\nConservation score\\nWe calculated conservation score for each binding site. For seed sites, we also computed the conservation score\\nfor the seed complementary region within the target site,and for the set of target site nucleotides outside the seedcomplementary region (off-seed region).', 'The conservationscore for a set of nucleotides is the average of conservationscores for individual nucleotides. The nucleotide conser-vation score ﬁles were downloaded from UCSC genomebrowser (http://genome.ucsc.edu/). These ﬁles weregenerated by the PhastCons program (35) throughmultiple-sequence alignments of 44 vertebrate genomesto the human genome (hg18) and 30 vertebrate genomesto the mouse genome (mm9). The hg18 and mm9 tracktables in UCSC genome browser were used to map thenucleotide conservation scores from chromosomes tomRNAs. Classiﬁcation of binding sites into IP + set and IP /C0set\\nAmong potential miRNA binding sites (seed or seedless)\\npredicted by RNAhybrid, those reside within an AGO-crosslinked region (tag or CCR) were considered IP +sites. The remaining sites deﬁne the set of IP /C0sites. Enrichment analysis\\nSeed types and miRNA 3\\n0compensatory/supplementary\\nbase-pairing are categorical features, whereas others arecontinuous.', 'The objective is to identify the categories orintervals of the features that are enriched for the IP + set.For conservation score, some sites ( <1% of all sites) were\\nremoved because their conservation scores were not avail-able due to the absence of their mRNAs in the UCSCtrack tables. Given a category (e.g., 8mer) or an interval(e.g., ( /C010 kcal/mol, /C09 kcal/mol) for /C1G\\ntotal, (0.4, 0.5)\\nfor conservation score), we measured the degree of enrich-\\nment by the ratio of the odds of the feature category orinterval occurring in the IP +set to the odds of it occurringin IP/C0set. The odds are P/(1/C0P), with Pestimated by the\\nfrequency (percentage) of the feature category or intervalin the whole IP + or IP /C0set.', 'If the enrichment (odds)\\nratio = 1, the feature category or interval is equallylikely to occur in the IP + set and the IP /C0set; ratio >1\\nindicates more likely occurrence in the IP + set and thusenrichment of the feature; ratio <1 indicates more likely\\noccurrence in the IP /C0set and depletion of the feature. Logistic regression modeling for prediction of miRNA\\nbinding sites\\nFor model-based prediction of miRNA binding sites, we\\nemployed logistic regression (35) to address the problemof predicting binary outcome. For potential target sites,the two possible outcomes are 1 for IP + and 0 for IP /C0. Because the enrichment of a feature in the IP + set indi-cates its predictive value, we included all enriched featuresin the development of multivariate logistic regressionmodel using established procedures (35) and the Rpackage (http://www.r-project.org), an open source statis-tical software package.', 'For categorical features, we usedthe odds ratio of each category to represent their valuesfor sites belonging to this category. We focused on non-\\nlinear logistic modeling, as some of the features may becorrelated, e.g., AU content is correlated to structural\\naccessibility (8). For the vector of enriched features,\\nX={X\\ni}, representing a binding site, the non-linear logis-\\ntic model was deﬁned as log[ P(X)/(1/C0P(X)] =/C11+P/C12iXi\\n+P\\ni6¼jgijXiXj, where XiXjis an interaction (i.e., quad-\\nratic) term which measures potential interaction of a pairof likely correlated features. The logistic model was\\ntrained on each CLIP dataset. For modeling incorporating\\nconservation score, we excluded sites for which conserva-tion scores were not available. The IP /C0set is larger than\\nthe IP + set by /C21/C247-fold for seed sites, and by at least an\\norder of magnitude for seedless sites.', 'To avoid potentialbias in model ﬁtting due to imbalance in the sizes ofdatasets, we randomly selected a subset of the IP /C0set\\nwith its size equal to that of the IP + set. We then\\ntrained a logistic model on the whole IP + set and theselected IP /C0subset. The random selection from the IP /C0\\nsubset and the subsequent model training were performed10 times. For a model-testing site with feature vectorX\\nt={Xt\\ni}, logistic model ﬁtting returns a classiﬁcation\\nprobability for each of the 10 iterations. We used P(Xt),\\nthe average of the probabilities of the 10 iterations as a\\nsingle probabilistic prediction that the site is a binding site.Thus, for a pre-speciﬁed probability threshold q, a deter-\\nministic prediction can be made, i.e., the testing site ispredicted to be a binding site if and only if P(X\\nt)/C21q.', 'Model performance assessment\\nThe true positive rate (TPR = sensitivity), false positive\\nrate (FPR = 1 /C0speciﬁcity) are commonly used\\nmeasures for assessing the performance of a predictionmodel. By changing probability threshold q,we con-\\nstructed receiver operator characteristic (ROC) curve\\nplotting the TPR against the FPR. However, neither\\nTPR nor FPR provides an overall measure of perform-ance. By changing q, sensitivity can be improved at the\\ncost of decreased speciﬁcity and vise versa. For an alter-native quantitative assessment of performance, we usedthe Youden’s J statistic (Youden’s index) (36). TheYouden’s J statistic is an overall measure of performance\\ndeﬁned as (TPR /C0FPR) = sensitivity+speciﬁcity /C01. For applications of the models, the top-ranked predic-\\ntions are of high interest for experimental testing, as they\\nare expected to have a high likelihood of positive valid-\\nation. We thus also computed the TPR for top-rankedpredictions.', 'For a given set of top-ranked sites predictedby a method, the TPR is computed by the number of truepositive sites (according to CLIP data) divided by the totalnumber of the top-ranked sites. Intra-dataset validation\\nFor each CLIP dataset, intra-dataset validation was per-\\nformed with the standard 10-fold cross-validationstrategy, to avoid the potential problem of model over-ﬁtting. The IP + set and the IP /C0set were each randomly\\ndivided into 10 subsets of equal size. At the i-th\\nrepetition ( I=1 , ...,10), the i-th IP + subset and the i-th\\nIP/C0subset were selected for testing of model accuracy, thee138 Nucleic Acids Research, 2013, Vol. 41, No. 14 P\\nAGE4OF13\\nremaining nine IP + subsets were combined into set Ai,a\\nsubset of the entire IP + set, and the remaining nine IP /C0\\nsubsets were combined into set Bi, a subset of the entire\\nIP/C0set. These two datasets were used for model training.', 'The problem of data imbalance between AiandBiis ad-\\ndressed by the strategy described for logistic regressionmodeling above. The ﬁnal modeling performance wasthen computed by averaging over the 10 repetitions. Inter-dataset validation\\nThe availability of multiple datasets generated by similar\\nexperimental protocols presents the opportunity for modelvalidation by independent data (inter-dataset validation).We performed inter-dataset validation using the ﬁve CLIPdatasets by training logistic model on IP + and IP /C0sets\\nfrom one CLIP dataset then testing model performance on\\nthe other four CLIP datasets. For model training, dataimbalance between IP + and IP /C0sets was also addressed. Performance comparison with other algorithms\\nFor an established algorithm to be included in our per-\\nformance comparison, either target site predictions areavailable for mm9 and hg18 or hg19, or the software im-plementing the algorithm can be run locally to generatepredictions.', 'To this end, we identiﬁed four target predic-tion algorithms: the sequence-based TargetScan (37), thestructure-based PITA (11), the composite-feature-based\\nmirSVR (38) and the pattern-based RNA22 (39). TargetScan 6.0 was downloaded from (http://www.targetscan.org/) and run locally with default setting.PITA version 6 was downloaded from (http://genie.weizmann.ac.il/pubs/mir07/mir07_exe.html) and runlocally with the settings: one GU, one mismatch in seedregion. mirSVR database (August 2010 Release) was\\ndownloaded from (http://www.microrna.org). Because\\nhg19 was the newest version for mirSVR predictiondatabase, we further processed mirSVR predictions toensure that all of the mirSVR binding sites can be foundin hg18 for comparison with predictions for hg18. RNA22database (miRNAs from miRBase 16 and genes from\\nENSEMBL 62) was downloaded from (http://cm. jefferson.edu/rna22v1.0/). Same as mirSVR, we alsofurther processed RNA22 prediction for comparisonwith predictions for hg18.', 'For seed sites in 3\\n0UTRs, pre-\\ndictions by the four algorithms were compared with ourCLIP-based logistic models. For seedless sites, TargetScandoes not make predictions, and PITA, mirSVR and\\nRNA22 prefer to predict the class with one GU pair\\nand/or one mismatch in the seed complementary region.We thus considered this class of seedless sites for perform-ance comparison. For construction of a ROC curve by varying the thresh-\\nold for a score, we used context score for TargetScan,mirSVR score for mirSVR, structure-based energy scorefor PITA, and pattern-based score for RNA22. Foroverall performance comparison, we present the highestpossible value of Youden’s J statistic by identifying the\\nthreshold for the score of each of these algorithms that\\nmaximizes the difference between TPR and FPR.', 'We notethat the scores of these algorithms are incompatible withlogistics probability, so that the comparison using\\nYouden’s J statistic is limited to the best-performingscore threshold for each of these algorithms. RESULTS\\nIdentiﬁcation of enriched target site features\\nFor each of the sequence, thermodynamic and structure\\nfeatures, we performed enrichment analysis. For both seedand seedless sites, the results are highly similar for HITS-CLIP, PAR-CLIP, V-CLIP and V-PAR-CLIP, for whichmore enriched features were observed than for V-PAR-CLIP-MNase (Supplementary Tables S1–S5), suggestinga substantially different proﬁle by MNase for the AGOCCRs. The enrichment signals for V-CLIP are appreciably\\nstronger than HITS-CLIP, PAR-CLIP and V-PAR-CLIP. Among the three regions of mRNAs, the enrichmentsignals are strongest for 3\\n0UTRs and weakest for 50\\nUTRs. Below we summarize our ﬁndings on 30UTRs\\nfor V-CLIP data.', 'Among all of the seed sites, only 7% are within the\\nAGO CCRs (i.e., IP + ) from the V-CLIP study (12% for8mer and 7mer sites), indicating that seed alone is a pre-dictor with high FPR. Features enriched for IP +seed sites\\ninclude /C1G\\nnucl,/C1Gtotal (Figure 2A), site accessibility\\n(Figure 2B), seed accessibility (Figure 2C), upstream ac-cessibility (window size of 15 nt, Figure 2D), downstreamaccessibility (10 nt), 8mer and 7mer seed, 3\\n0base-pairing,\\nupstream AU (30 nt), downstream AU (30 nt), sitelocation, seed conservation and site conservation. Forexample, more favorable energy is enriched for /C1G\\ntotal. In particular, /C1Gtotalis the only feature enriched for all\\nthree mRNA regions, for both seed and seedless sites, andfor all ﬁve CLIP datasets. Among enriched 8mer, 7mer-A1\\nand 7mer-m8, the enrichment of 8mer is the strongest. The\\nenrichment of 7mer-A1 and 7mer-m8 are enhanced by 3\\n0\\nbase-pairing (Figure 2E), consistent with the supplemen-tary role of 3\\n0base-pairing (8).', 'However, 30base-pairing\\ndoes not enhance the enrichment for 8mer seed sites.There is a lack of enrichment for 6mer and offset 6merseed sites. There is a far greater conservation for IP +seedsites, with higher conservation for seed complementaryregion than the whole target site (Figure 2F). For the 3\\n0\\nUTR, the gap in conservation score between IP +and IP /C0\\nis larger than that of CDS or 50UTR. These suggest that\\nmiRNA bindings sites in particular the seed complemen-\\ntary regions tend to reside in conserved blocks on target 30\\nUTR. For the threshold of 0.57 (vertical dashed line inFigure 2F) for indicating conservation across all mammals(34), the seed complementary region is conserved for/C2452% in IP + seed sites, but only /C2428% in IP /C0seed\\nsites, further supporting the importance of seed. Among all of the identiﬁed miRNA binding sites\\nresiding within the AGO CCRs, /C2496% are seedless.', \"This together with conservation evidence (Figure 3D)strongly suggests substantial involvement of seedlesssites in gene regulation by miRNAs. For seedless sitesand for features that are not seed related, the results ofenrichment analysis are similar to those for seed sites(Supplementary Tables S1–S5). The enriched featuresP\\nAGE5OF13 Nucleic Acids Research, 2013, Vol. 41, No. 14 e138\\n0.0 0.5 1.0 1.5 2.0 2.5 3.0 3.5 \\n6mer 7mer-A1 7mer-m8 8mer offset \\n6mer Enrichment odds ratio \\nType of miRNA tar get sites Seed without 3' pairing \\nSeed with 3' pairing \\n0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1 \\n1 0.9 0.8 0.7 0.6 0.5 0.4 0.3 0.2 0.1 0 % of conservation over threshold \\nThreshold for conservation score IP+ site \\nIP+ seed \\nIP+ off-seed IP- site \\nIP- seed \\nIP- off-seed AB\\nCD\\nEF\\nFigure 2.\", 'Enrichment of representative seed site features (30UTRs for V-CLIP data): ( A)/C1Gtotal;(B) site accessibility; ( C) seed accessibility; ( D)\\nupstream accessibility (window size of 15 nt); ( E) type of miRNA target sites; ( F) percentage ( Y-axis) of sites/seed/off-seed regions with conservation\\nscores greater than or equal to a pre-speciﬁed threshold ( X-axis), in the IP + set or the IP /C0set (dashed line corresponding to a threshold of 0.57\\npreviously used for deﬁning conservation (34,38).e138 Nucleic Acids Research, 2013, Vol. 41, No. 14 PAGE6OF13\\ninclude /C1Gnucl,/C1Gtotal (Figure 3A), site accessibility\\n(Figure 3B), upstream accessibility (10 nt, Figure 3C),\\ndownstream accessibility (30 nt), 30base-pairing,\\nupstream AU (30 nt), downstream AU (30 nt), site loca-\\ntion and site conservation (Figure 3D). Prediction of seed sites in 30UTRs and performance\\nevaluation\\nFor each CLIP dataset, using enriched features identiﬁed\\nfor the dataset, we ﬁrst performed logistic model training.', 'The model performance was tested by both intra-dataset\\nvalidation and inter-dataset validation. For seed sites in 30\\nUTRs, we compared our predictions with predictions\\navailable from TargetScan (37), mirSVR (38), PITA (11)\\nand RNA22 (39). For a wide range of logistic probability\\nthreshold centering around 0.5, logistic models trained on\\nthe ﬁve CLIP datasets all have substantially higher TPR\\nthan TargetScan and mirSVR, RNA22 with comparable\\nFPR, and lower FPR than PITA, RNA22 with compar-\\nable TPR (Figure 4A, C, E, G and I). Furthermore, there\\nare major overall improvements as measured by the\\nYouden’s J statistic (Figure 4B, D, F, H and J). Among\\nthe four established algorithms, TargetScan is the best\\nperformer, followed by PITA, mirSVR and RNA22.We also computed the TPR of top-ranked (from top\\n1% to 10%) predictions for each of the ﬁve CLIP\\ndatasets (Supplementary Figure S1a, c, e, g and i).', 'Ourlogistic model generally outperforms TargetScan and\\nmirSVR, and is substantially better than PITA and\\nRNA22, especially for more highly ranked predictions(e.g., 1%). For inter-dataset validation, the model trained on\\nV-PAR-CLIP-MNase is the worst performer among the\\nﬁve CLIP-based logistic models, further indicating sub-\\nstantial difference between the data for MNase and thedata for other CLIP protocols. For model testing on the\\nHITS-CLIP dataset from mouse brain (Figure 4A and B),\\nlogistic models trained on HITS-CLIP and PAR-CLIP arethe two best performers. For logistic probability threshold\\nof 0.5, the improvement by either model in Youden’s J\\nstatistic is about 0.13 over TargetScan and PITA, 0.15over mirSVR and 0.28 over RNA22. For each of the re-\\nmaining three models trained on V-CLIP, V-PAR-CLIP,\\nV-PAR-CLIP-MNase datasets, the improvement is about0.1 over TargetScan and PITA, 0.12 over mirSVR and\\n0.25 over RNA22.', 'For cross-species validation, we observed good results\\non mouse CLIP data for models trained on human0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1 \\n1 0.9 0.8 0.7 0.6 0.5 0.4 0.3 0.2 0.1 0 % of conservation over threshold \\nThreshold for conservation score IP+ site \\nIP- site \\nAB\\nCD\\nFigure 3. Enrichment of representative seedless site features (30UTRs for V-CLIP data): ( A)/C1Gtotal;(B) site accessibility; ( C) upstream accessibility\\n(window size of 10 nt); ( D) percentage ( Y-axis) of seedless sites with conservation scores greater than or equal to a pre-speciﬁed threshold ( X-axis), in\\nthe IP + set or the IP /C0set.PAGE7OF13 Nucleic Acids Research, 2013, Vol. 41, No.', \"14 e138\\n0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1 \\n0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1 True positive rate (sensitivity) \\nFalse positive rate (1-specificity) Model 10-fold-CV on HITS-CLIP \\nModel trained on PAR-CLIP \\nModel trained on V-CLIP \\nModel trained on V-PAR-CLIP \\nModel trained on V-PAR-CLIP-MNase \\nPITA \\nRNA22 \\nmirSVR \\nTargetScan \\n-0.05 0.00 0.05 0.10 0.15 0.20 0.25 0.30 0.35 0.40 0.45 0.50 0.55 \\n0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1 Probability threshold for logistic model prediction Model 10-fold-CV on HITS-CLIP \\nModel trained on PAR-CLIP \\nModel trained on V-CLIP \\nModel trained on V-PAR-CLIP \\nModel trained on V-PAR-CLIP-MNase \\nPITA \\nRNA22 \\nmirSVR \\nTargetScan Youden's J statistics (sensitivity+specificity-1) Youden's J statistics (sensitivity+specificity-1) Youden's J statistics (sensitivity+specificity-1)0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1 \\n0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1 True positive rate (sensitivity) \\nFalse positive rate (1-specificity) Model trained on HITS-CLIP \\nModel 10-fold-CV on PAR-CLIP \\nModel trained on V-CLIP \\nModel trained on V-PAR-CLIP \\nModel trained on V-PAR-CLIP-MNase \\nPITA \\nRNA22 \\nmirSVR \\nTargetScan \\n-0.05 0.00 0.05 0.10 0.15 0.20 0.25 0.30 0.35 0.40 0.45 0.50 0.55 \\n0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1 Probability threshold for logistic model prediction Model trained on HITS-CLIP \\nModel 10-fold-CV on PAR-CLIP \\nModel trained on V-CLIP \\nModel trained on V-PAR-CLIP \\nModel trained on V-PAR-CLIP-MNase \\nPITA \\nRNA22 \\nmirSVR \\nTargetScan \\n0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1 \\n0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1 True positive rate (sensitivity) \\nFalse positive rate (1-specificity) Model trained on HITS-CLIP \\nModel trained on PAR-CLIP \\nModel 10-fold-CV on V-CLIP \\nModel trained on V-PAR-CLIP \\nModel trained on V-PAR-CLIP-MNase \\nPITA \\nRNA22 \\nmirSVR \\nTargetScan \\n-0.05 0.00 0.05 0.10 0.15 0.20 0.25 0.30 0.35 0.40 0.45 0.50 0.55 \\n0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1 Probability threshold for logistic model prediction Model trained on HITS-CLIP \\nModel trained on PAR-CLIP \\nModel 10-fold-CV on V-CLIP \\nModel trained on V-PAR-CLIP \\nModel trained on V-PAR-CLIP-MNase \\nPITA \\nRNA22 \\nmirSVR \\nTargetScan AB\\nCD\\nEF\\nFigure 4.\", 'Inter-dataset model validation by using one of the ﬁve CLIP datasets as an independent set for testing performance of each of models\\ndeveloped with other four datasets; intra-dataset validation by 10-fold cross validation (CV); and performance comparison with established algo-rithms for predictions of seed sites in 3\\n0UTRs (dashed diagonal line for random predictions). ROC curve and Youden’s J statistic are shown for\\nmodel testing on HITS-CLIP data ( A,B), on PAR-CLIP data ( C,D), on V-CLIP data ( E,F), on V-PAR-CLIP data ( G,H) and on V-PAR-CLIP-\\nMNase data ( I,J). The color-matched dots on ROC curves correspond to a logistic probability threshold of 0.5. The rectangle, triangle, square and\\ndiamond correspond to the best-performing score threshold (according to Youden’s J statistic) for TargetScan, PITA, mirSVR and RNA22,respectively.e138 Nucleic Acids Research, 2013, Vol. 41, No. 14 PAGE8OF13\\n(continued)\\nCLIP datasets (Figure 4A and B).', 'Furthermore, the\\nHITS-CLIP-based model is well validated on all human\\nCLIP datasets (Figure 4C–J). In particular, for testingperformance on any of the human CLIP datasets, the\\nHITS-CLIP-based model performed as well as the PAR-\\nCLIP-based model. These ﬁndings suggest that ouranalysis captured intrinsically important target site char-\\nacteristics that are species independent. Thus, the model\\ndeveloped for one species using quality CLIP data may be\\ngenerally applicable to target site prediction for other\\nmammalian species and beyond, because the regulatorymachineries are either identical or highly similar. For each of the four models trained on different human\\nCLIP datasets, we computed the average performance of\\nboth intra-dataset and inter-dataset validation using\\nall four human CLIP datasets. This analysis showed\\nthat V-CLIP derived model is the best performer (Supple-\\nmentary Figure S2a and b).', 'The improved predictability\\nby V-CLIP model is an expected result due to stronger\\nenrichment signals for V-CLIP data. The ﬁndings from\\nenrichment analysis and model comparison suggest thatdata from V-CLIP has the best quality, at least from the\\nperspective of data analysis and modeling. Prediction of seedless sites\\nThe seedless site predictions are similar to seed site predic-\\ntions in the trends of ROC curves and Youden’s J statistic,\\nas well as in the levels of improvements over random pre-dictions (Supplementary Figure S3). For seedless sites\\nwith one GU pair and/or one mismatch in the seed com-\\nplementary region, a class of seedless sites predicted by allof PITA, mirSVR and RNA22, we found that levels ofimprovements by logistic modeling are greater than those\\nfor seed site predictions (Figures 4 and 5; Supplementary\\nFigure S4).', 'We note that the TPRs and FPRs for mirSVRand RNA22 are very low, due to the small number of\\npredictions for this class of seedless sites that are available\\nfrom their databases. Again, V-CLIP derived model is thebest performer among four models derived from human\\nCLIP datasets (Supplementary Figure S2c–f).', \"For top-\\nranked predictions, our model has a substantially higher0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1 \\n0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1 True positive rate (sensitivity) \\nFalse positive rate (1-specificity) Model trained on HITS-CLIP \\nModel trained on PAR-CLIP \\nModel trained on V-CLIP \\nModel 10-fold-CV on V-PAR-CLIP \\nModel trained on V-PAR-CLIP-MNase \\nPITA \\nRNA22 \\nmirSVR \\nTargetScan \\n-0.05 0.00 0.05 0.10 0.15 0.20 0.25 0.30 0.35 0.40 0.45 0.50 0.55 \\n0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1 Probability threshold for logistic model prediction Model trained on HITS-CLIP \\nModel trained on PAR-CLIP \\nModel trained on V-CLIP \\nModel 10-fold-CV on V-PAR-CLIP \\nModel trained on V-PAR-CLIP-MNase \\nPITA \\nRNA22 \\nmirSVR \\nTargetScan Youden's J statistic (sensitivity+specificity ‐1)\\n0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1 \\n0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1 True positive rate (sensitivity) \\nFalse positive rate (1-specificity) Model trained on HITS-CLIP \\nModel trained on PAR-CLIP \\nModel trained on V-CLIP \\nModel trained on V-PAR-CLIP \\nModel 10-fold-CV on V-PAR-CLIP-MNase \\nPITA \\nRNA22 \\nmirSVR \\nTargetScan \\n-0.05 0.00 0.05 0.10 0.15 0.20 0.25 0.30 0.35 0.40 0.45 0.50 0.55 \\n0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1 Probability threshold for logistic model prediction Model trained on HITS-CLIP \\nModel trained on PAR-CLIP \\nModel trained on V-CLIP \\nModel trained on V-PAR-CLIP \\nModel 10-fold-CV on V-PAR-CLIP-MNase \\nPITA \\nRNA22 \\nmirSVR \\nTargetScan Youden's J statistic (sensitivity+specificity ‐1)GH\\nIJ\\nFigure 4.\", 'Continued.PAGE9OF13 Nucleic Acids Research, 2013, Vol. 41, No. 14 e138\\nTPR than mirSVR, PITA and RNA22 (Supplementary\\nFigure S1b, d, f, h and j). Prediction for both CDS and 50UTR\\nThe enriched features identiﬁed for CDS and 50UTRs\\nallowed us to develop models speciﬁcally for these tworegions. For both seed and seedless sites, the predictionsare similar to those for 3\\n0UTRs in the trends of ROC\\ncurves and Youden’s J statistic, as well as in thelevels of improvements over random predictions(Supplementary Figure S5). Again, V-CLIP derivedmodel is the best performer among four models forhuman CLIP datasets (Supplementary Figure S6). Software and database availability\\nWe have implemented the HITS-CLIP-based logistics re-\\ngression model for mouse and the V-CLIP-based modelfor human into the STarMir module of the Sfold software(http://sfold.wadsworth.org/cgi-bin/starmir.pl), allowingusers to submit any miRNA and mRNA for predictionof binding sites by our models.', 'We have also developedSTarMirDB (http://sfold.wadsworth.org/starmirDB.php),a web searchable database currently including all pre-dicted binding sites by our models for miRNAs andmRNAs in the HITS-CLIP and V-CLIP studies, with anindicator showing whether a site is supported by the CLIPstudy. The database is being extended to include wholetranscriptome predictions for human and mouse. DISCUSSION\\nThe high throughput datasets from CLIP studies pre-\\nsented a unique opportunity for us to identify target sitecharacteristics important for binding by miRNAs, todevelop novel statistical models for target site prediction,and to test performance of models by not only theconventional intra-dataset validation, but also inter-\\ndataset validation including cross-species validation. The\\nmajor improvement in performance by our approach over\\nestablished algorithms is partly attributable to the high\\nquality of the data from the CLIP studies.', 'From the comprehensive enrichment analyses, we found\\nthat data from the CLIP studies not only support the im-\\nportance of most target site features reported in the litera-\\nture, but also suggest several modiﬁcations and reveal\\npreviously unknown characteristics. Importantly, most\\nof the ﬁndings are common not only for both seed and\\nseedless sites but also for all of three regions of mRNA. Although three types of seed sites, 8mer, 7mer-m8 and\\n7mer-A1, were found to be enriched in IP + set, the levels\\nof enrichment for 7mer-m8 and 7mer-A1 are highly com-\\nparable (Figure 2E). Thus, there is a lack of support for\\ngreater effectiveness by 7mer-m8 over 7mer-A1 as previ-\\nously reported (8). In contrast to 7mer-m8 and 7mer-A1,\\npresence of 3\\n0base-pairing may not enhance enrichment\\nfor 8mers (Figure 2E), perhaps due to adequate stability\\nfor the seed portion of the miRNA:target hybrid for\\n8mers.', 'Our consideration of seedless sites in addition to seed sites\\nwas supported by evolutionary conservation (Figure 3D). Although the CDS is much more conserved than the 30\\nUTR, the signal of enrichment for CDS was weaker than3\\n0UTR, perhaps due to the possibility that greater portion\\nof CLIP hits for CDS are for transient rather than stable\\nassociation between RNAs and AGO. Although it is impos-\\nsible to distinguish transient binding from stable binding,\\nbinding can still be predicted for each region using a\\nregion-speciﬁc model. Furthermore, the seed complemen-\\ntary region is more conserved than the remaining nucleo-\\ntides within the entire seed site (Figure 2F), further\\nsupporting the signiﬁcance of conserved seed pairing (3).', \"There is a lack of enrichment for /C1Ghybrid with the ex-\\nception of data for seed sites from V-PAR-CLIP-MNase0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1 \\n0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1 True positive rate (sensitivity) \\nFalse positive rate (1-specificity) Model trained on HITS-CLIP \\nModel trained on PAR-CLIP \\nModel trained on V-CLIP \\nModel trained on V-PAR-CLIP \\nModel trained on V-PAR-CLIP-MNase \\nPITA \\nRNA22 \\nmirSVR \\n-0.05 0.00 0.05 0.10 0.15 0.20 0.25 0.30 0.35 0.40 0.45 0.50 0.55 \\n0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1 \\nProbability threshold for logistic model prediction Model trained on HITS-CLIP \\nModel trained on PAR-CLIP \\nModel trained on V-CLIP \\nModel trained on V-PAR-CLIP \\nModel trained on V-PAR-CLIP-MNase \\nPITA \\nRNA22 \\nmirSVR Youden's J statistic (sensitivity+specificity ‐1) AB\\nFigure 5.\", 'Performance comparison of logistic models trained on ﬁve CLIP data sets with established algorithms for predictions of seedless sites with\\none GU pair and/or one mismatch within seed complementary region in 30UTRs (dashed diagonal line for random predictions). ROC curve ( A) and\\nYouden’s J statistic ( B) are shown for model testing on V-CLIP.e138 Nucleic Acids Research, 2013, Vol. 41, No. 14 PAGE10OF13\\n(Supplementary Tables S1–S5). This indicates that\\nmiRNA:target hybrid stability is generally not importantfor target recognition, and further suggests the CLIP data\\nproﬁle for MNase is different from those for other studies. We found that /C1G\\ntotalis the only feature enriched for all\\nCLIP datasets, all three target regions, and for both seed\\nand seedless sites. Thus, as one of the important factors\\nfor target binding by miRNAs in general, target structuralaccessibility is likely the most important factor for seedless\\nsites.', 'The ﬁndings lend additional support to the structure-\\nbased hybridization model with /C1G\\ntotal as the key\\nenergetic characteristic (12). For region upstream or\\ndownstream of target site, we observed enrichment of ac-\\ncessibility as measured via structure prediction or by AUcontent. This is consistent with the report that the RISCcomplex can bind non-speciﬁcally to single-stranded\\nregions near target sites to facilitate target recognition\\n(40). A previous analysis of target site features examined 3\\n0\\nUTRs and a subset of abundant miRNAs from HITS-CLIP and PAR-CLIP studies (41). In their study, onlythe single best predicted site on a 3\\n0UTR was considered,\\nand seed sites and seedless sites were not separated. Different methods were used for the computation ofsome of the features and for the analysis and modelingof CLIP data.', 'Despite the differences in scope and meth-\\nodology for analysis and modeling, some of the important\\nfeatures reported in their study (e.g., conservation andaccessibility of target site, upstream or downstream AU\\ncontent) are further supported by our enrichment analysis. In comparison to other established algorithms, our\\nanalysis and modeling framework offers much greater\\nﬂexibility in selection of enriched features for different\\nCLIP datasets, in modeling the contribution of eachenriched feature, and in modeling potential interactions\\nbetween different features. Furthermore, our approach\\nmakes predictions for broad classes seedless sites includingsites with one GU or one mismatch for the seed comple-mentary region, G-bulge sites (42), sites with 3\\n0comple-\\nmentary base-pairing (8), centered sites (31) and beyond. Such capability will facilitate studies on widespread tar-geting by seedless sites (7). We chose the nonlinear option of logistic regression for\\nmodel development.', 'This addresses potential correlationbetween some of the enriched features. For example, AU\\ncontent and the probabilistic measure of structural acces-\\nsibility could be correlated, as AU-rich regions are morelikely to be single stranded and thus structurally access-ible. We also performed linear logistic modeling. We\\nfound that, in general, the nonlinear option outperformed\\nthe linear option (Supplementary Figure S7). Ourapproach presents a probability from logistic regression\\nmodeling for predicting a miRNA binding site. Sites\\nwith higher probabilities will have a greater chance forpositive experimental validation (see Supplementary\\nFigure S1). The similar results by enrichment analysis and the gen-\\nerally good performances from inter-dataset validations\\nindicate that our modeling framework is robust across\\nindependent datasets generated by different experimentalprotocols for different species.', 'The ﬁndings also indicatethat majority of characteristics for miRNA binding sites\\nare shared by different species, because the regulatorymachineries are either identical or highly similar. These\\nsuggest that a model trained on data from a high through-\\nput quality CLIP study for one species may be applicableto binding site predictions for another species. The ﬁndings from our study conﬁrm the power of CLIP\\ndata for the development of improved algorithm for pre-diction of miRNA binding sites. However, the CLIP tech-\\nniques and available CLIP data have a number of\\nlimitations. The identity of the miRNA(s) binding to anAGO crosslinked regions is unknown and needs to be pre-\\ndicted for further analysis. The abundant miRNAs in\\nthese studies only represent a small portion of all knownhuman or mouse miRNAs. The majority of the mRNAs in\\nthe complete human or mouse transcriptome are absent in\\nthe CLIP datasets, due to either lack of expression or alow level of expression.', 'Furthermore, CLIP experiments\\nare difﬁcult to perform, and successful applications are\\nstill limited. For example, successful CLIP applicationhas not been reported for ﬂy. Therefore, our prediction\\nmodels will greatly complement the existing CLIP data by\\nenabling genome-scale miRNA target predictions for anyspecies of interest with higher prediction accuracy than the\\nestablished algorithms. The CLIP methods are binding assays rather than func-\\ntional assays. Therefore, models developed from CLIPdata are limited to prediction of miRNA binding sites. In addition, the best large-scale data on miRNA binding\\nsites are from CLIP studies. For these reasons, we con-sidered model validation by independent CLIP data (i.e.,\\ninter-dataset validation) as the gold standard for valid-\\nation and performance comparison.', 'These models donot make functional predictions that include the\\noutcome of miRNA binding (i.e., target degradation or\\ntranslational repression) and the degree of regulation oneither the mRNA or the protein level. Such predictions\\nwill require computational modeling of the effects of\\nmultiple miRNA binding sites on the same target, aswell as analysis and modeling of high throughput func-\\ntional data for miRNA:target interactions. SUPPLEMENTARY DATA\\nSupplementary Data are available at NAR Online:\\nSupplementary Tables 1–5, Supplementary Figures 1–7and Supplementary Text File. ACKNOWLEDGEMENTS\\nThe Computational Molecular Biology and Statistics Core\\nat the Wadsworth Center is acknowledged for supporting\\ncomputing resources for this work. The authors thank the\\nlabs of Robert Darnel, Thomas Tuschl and MihaelaZavolan for providing additional data, and Christina\\nLeslie for clariﬁcations on mirSVR predictions. Y.D. conceived and supervised the study. B.M.', 'compiled andprocessed data from the HITS-CLIP and PAR-CLIP\\nstudies. C.L. compiled and processed the data from the\\nV-CLIP, V-PAR-CLIP and V-PAR-CLIP-MNase studies.P\\nAGE11OF13 Nucleic Acids Research, 2013, Vol. 41, No. 14 e138\\nC.L. and B.M. performed sequence feature computations\\nand enrichment analyses. C.L. performed model develop-ment and prediction validation. D.L. wrote the initial\\nsoftware for the computation of several target site\\nfeatures and enrichment. W.A.R. and A.W. providedprogramming support and performed software and webserver implementation, and C.S.C. provided hardwareand system support for both cluster computing and the\\nSfold web server. Y.D., C.L. and B.M. wrote the article. FUNDING\\nNational Science Foundation (NSF) [DBI-0650991 to\\nY.D.]; National Institutes of Health (NIH) [GM099811to Y.D.]; Nafosted Fund of Vietnam [102.03-2010.04 to\\nD.L.]. Funding for open access charge: NSF and NIH\\ngrants. Conﬂict of interest statement . None declared. REFERENCES\\n1. Erson,A.E.', 'and Petty,E.M. (2008) MicroRNAs in development\\nand disease. Clin. Genet. ,74, 296–306. 2. Fabian,M.R. and Sonenberg,N. (2012) The mechanics of miRNA-\\nmediated gene silencing: a look under the hood of miRISC. Nat. Struct. Mol. Biol. ,19, 586–593. 3. Lewis,B.P., Burge,C.B. and Bartel,D.P. (2005) Conserved seed\\npairing, often ﬂanked by adenosines, indicates that thousands ofhuman genes are microRNA targets. Cell,120, 15–20. 4. Tay,Y., Zhang,J., Thomson,A.M., Lim,B. and Rigoutsos,I. (2008)\\nMicroRNAs to Nanog, Oct4 and Sox2 coding regions modulate\\nembryonic stem cell differentiation. Nature ,455, 1124–1128. 5. Vella,M.C., Choi,E.Y., Lin,S.Y., Reinert,K. and Slack,F.J. (2004)\\nThe C. elegans microRNA let-7 binds to imperfect let-7complementary sites from the lin-41 3\\n0UTR. Genes Dev. ,18,\\n132–137. 6. Didiano,D. and Hobert,O. (2006) Perfect seed pairing is not a\\ngenerally reliable predictor for miRNA-target interactions. Nat. Struct. Mol. Biol. ,13, 849–851. 7.', 'Loeb,G.B., Khan,A.A., Canner,D., Hiatt,J.B., Shendure,J.,\\nDarnell,R.B., Leslie,C.S. and Rudensky,A.Y. (2012)Transcriptome-wide miR-155 binding map reveals widespread\\nnoncanonical microRNA targeting. Mol. Cell ,48, 760–770. 8. Bartel,D.P. (2009) MicroRNAs: target recognition and regulatory\\nfunctions. Cell,136, 215–233. 9. Zhao,Y., Samal,E. and Srivastava,D. (2005) Serum response\\nfactor regulates a muscle-speciﬁc microRNA that targets Hand2during cardiogenesis. Nature ,436, 214–220. 10. Robins,H., Li,Y. and Padgett,R.W. (2005) Incorporating structure\\nto predict microRNA targets. Proc. Natl Acad. Sci. USA ,102,\\n4006–4009. 11. Kertesz,M., Iovino,N., Unnerstall,U., Gaul,U. and Segal,E. (2007)\\nThe role of site accessibility in microRNA target recognition.Nat. Genet. ,39, 1278–1284. 12. Long,D., Lee,R., Williams,P., Chan,C.Y., Ambros,V. and Ding,Y. (2007) Potent effect of target structure on microRNA function. Nat. Struct. Mol. Biol. ,14, 287–294. 13. Malhas,A., Saunders,N.J. and Vaux,D.J.', '(2010) The nuclear\\nenvelope can control gene expression and cell cycle progressionvia miRNA regulation. Cell Cycle ,9, 531–539. 14. Hausser,J., Landthaler,M., Jaskiewicz,L., Gaidatzis,D. and\\nZavolan,M. (2009) Relative contribution of sequence and\\nstructure features to the mRNA binding of Argonaute/EIF2C-miRNA complexes and the degradation of miRNA\\ntargets. Genome Research ,19, 2009–2020. 15. Long,D., Chan,C.Y. and Ding,Y. (2008) Analysis of microRNA-\\ntarget interactions by a target structure based hybridization\\nmodel. Paciﬁc Symposium on Biocomputing ,13, 64–74.16. Ørom,U.A. and Lund,A.H. (2010) Experimental identiﬁcation of\\nmicroRNA targets. Gene,451, 1–5. 17. Hafner,M., Landthaler,M., Burger,L., Khorshid,M., Hausser,J.,\\nBerninger,P., Rothballer,A., Ascano,M. Jr, Jungkamp,A.C.,\\nMunschauer,M. et al. (2010) Transcriptome-wide identiﬁcation of\\nRNA-binding protein and microRNA target sites by PAR-CLIP.Cell,141, 129–141. 18. Chi,S.W., Zang,J.B., Mele,A. and Darnell,R.B.', '(2009) Argonaute\\nHITS-CLIP decodes microRNA-mRNA interaction maps. Nature ,\\n460, 479–486. 19. Zisoulis,D.G., Lovci,M.T., Wilbert,M.L., Hutt,K.R., Liang,T.Y.,\\nPasquinelli,A.E. and Yeo,G.W. (2010) Comprehensive discoveryof endogenous Argonaute binding sites in Caenorhabditis elegans.Nat. Struct. Mol. Biol. ,17, 173–179. 20. Kishore,S., Jaskiewicz,L., Burger,L., Hausser,J., Khorshid,M. and\\nZavolan,M. (2011) A quantitative analysis of CLIP methods foridentifying binding sites of RNA-binding proteins. Nat. Methods ,\\n8, 559–564. 21. Leung,A.K., Young,A.G., Bhutkar,A., Zheng,G.X., Bosson,A.D.,\\nNielsen,C.B. and Sharp,P.A. (2011) Genome-wide identiﬁcation of\\nAgo2 binding sites from mouse embryonic stem cells with andwithout mature microRNAs. Nat. Struct. Mol. Biol. ,18, 237–244. 22. Lim,L.P., Lau,N.C., Garrett-Engele,P., Grimson,A., Schelter,J.M.,\\nCastle,J., Bartel,D.P., Linsley,P.S. and Johnson,J.M. (2005)Microarray analysis shows that some microRNAs downregulatelarge numbers of target mRNAs.', 'Nature ,433, 769–773. 23. Baek,D., Villen,J., Shin,C., Camargo,F.D., Gygi,S.P. and\\nBartel,D.P. (2008) The impact of microRNAs on protein output.Nature ,455, 64–71. 24. Selbach,M., Schwanhausser,B., Thierfelder,N., Fang,Z., Khanin,R. and Rajewsky,N. (2008) Widespread changes in protein synthesis\\ninduced by microRNAs. Nature ,455, 58–63. 25. Lal,A., Navarro,F., Maher,C.A., Maliszewski,L.E., Yan,N.,\\nO’Day,E., Chowdhury,D., Dykxhoorn,D.M., Tsai,P., Hofmann,O. et al. (2009) miR-24 Inhibits cell proliferation by targeting E2F2,\\nMYC, and other cell-cycle genes via binding to ‘‘seedless’’ 3\\n0UTR\\nmicroRNA recognition elements. Mol. Cell ,35, 610–625. 26. Lytle,J.R., Yario,T.A. and Steitz,J.A. (2007) Target mRNAs\\nare repressed as efﬁciently by microRNA-binding sites in the 50\\nUTR as in the 30UTR. Proc. Natl Acad. Sci. USA ,104,\\n9667–9672. 27. Smith,A.D., Xuan,Z. and Zhang,M.Q. (2008) Using quality scores\\nand longer reads improves accuracy of Solexa read mapping.BMC Bioinform. ,9, 128. 28.', 'Grifﬁths-Jones,S., Saini,H.K., van Dongen,S. and Enright,A.J. (2008) miRBase: tools for microRNA genomics. Nucleic Acids\\nRes.,36, D154–D158. 29. Rehmsmeier,M., Steffen,P., Hochsmann,M. and Giegerich,R. (2004) Fast and effective prediction of microRNA/target duplexes. RNA ,10, 1507–1517. 30. Hammell,M., Long,D., Zhang,L., Lee,A., Carmack,C.S., Han,M.,\\nDing,Y. and Ambros,V. (2008) mirWIP: microRNA targetprediction based on microRNA-containing ribonucleoprotein-enriched transcripts. Nat. Methods ,5, 813–819. 31. Shin,C., Nam,J.W., Farh,K.K., Chiang,H.R., Shkumatava,A. and\\nBartel,D.P. (2010) Expanding the microRNA targeting code:functional sites with centered pairing. Mol. Cell ,38, 789–802. 32. Ding,Y. and Lawrence,C.E. (2001) Statistical prediction of single-\\nstranded regions in RNA secondary structure and application topredicting effective antisense target sites and beyond. Nucleic\\nAcids Res. ,29, 1034–1046. 33. Ding,Y. and Lawrence,C.E.', '(2003) A statistical sampling\\nalgorithm for RNA secondary structure prediction. Nucleic Acids\\nRes.,31, 7280–7301. 34. Siepel,A., Bejerano,G., Pedersen,J.S., Hinrichs,A.S., Hou,M.,\\nRosenbloom,K., Clawson,H., Spieth,J., Hillier,L.W., Richards,S. et al. (2005) Evolutionarily conserved elements in vertebrate,\\ninsect, worm, and yeast genomes. Genome Res. ,15, 1034–1050. 35. Hosmer,D.W. and Lemeshow,S. (1989) Applied Logistic\\nregression . John Wiley & Sons, New York. 36. Youden,W.J. (1950) Index for rating diagnostic tests. Cancer ,3,\\n32–35. 37. Grimson,A., Farh,K.K., Johnston,W.K., Garrett-Engele,P.,\\nLim,L.P. and Bartel,D.P. (2007) MicroRNA targeting speciﬁcitye138 Nucleic Acids Research, 2013, Vol. 41, No. 14 PAGE12OF13\\nin mammals: determinants beyond seed pairing. Mol. Cell ,27,\\n91–105. 38. Betel,D., Koppal,A., Agius,P., Sander,C. and Leslie,C. (2010)\\nComprehensive modeling of microRNA targets predicts\\nfunctional non-conserved and non-canonical sites. Genome Biol. ,\\n11, R90. 39.', 'Miranda,K.C., Huynh,T., Tay,Y., Ang,Y.-S., Tam,W.-L.,\\nThomson,A.M., Lim,B. and Rigoutsos,S. (2006) A pattern-based\\nmethod for the identiﬁcation of microrna binding sites and their\\ncorresponding heteroduplexes. Cell,126, 1203–1217.40. Ameres,S.L., Martinez,J. and Schroeder,R. (2007) Molecular basis\\nfor target RNA recognition and cleavage by human RISC. Cell,\\n130, 101–112. 41. Wen,J., Parker,B.J., Jacobsen,A. and Krogh,A. (2011) MicroRNA\\ntransfection and AGO-bound CLIP-seq data sets reveal distinct\\ndeterminants of miRNA action. RNA ,17, 820–834. 42. Chi,S.W., Hannon,G.J. and Darnell,R.B. (2012) An alternative\\nmode of microRNA target recognition. Nat. Struct. Mol. Biol. ,\\n19, 321–327.PAGE13OF13 Nucleic Acids Research, 2013, Vol. 41, No. 14 e138\\nCLIP-Based Prediction of Mammalian MicroRNA Binding Sites \\nChaochun Liu, Bibekanand Mallick, Dang Long, William A. Rennie,  \\nAdam Wolenc, C.', 'Steven Carmack and Ye Ding  \\n \\n \\nList of Supplementary Tables and Figures \\nSupplementary Table 1 Summary of feature enrichment  analysis for HITS-CLIP data \\nSupplementary Table 2 Summary of feature enrichment  analysis for PAR-CLIP data \\nSupplementary Table 3 Summary of feature enrichment analysis for V-CLIP data \\nSupplementary Table 4 Summary of feature enrichment analysis for V-PAR-CLIP data \\nSupplementary Table 5 Summary of feature enrichment ana lysis for V-PAR-CLIP-MNase data \\nSupplementary Figure 1 True positive rate comparison for top ranked pr edictions of seed sites and seedless sites with  \\none GU pair and/or one mismatch in the seed complementary region in 3 \\x01 UTRs   \\nSupplementary Figure 2 Average performance evaluation fo r each of four models trained on human CLIP data by \\ntesting on all four human CLIP datasets (i.e ., one intra-dataset a nd three inter-dataset \\nvalidations) for seed sites, all seedless sites and seedless sites with one GU pair and/or one \\nmismatch within seed complementary region in 3 \\x01 UTRs \\nSupplementary Figure 3 Intra-dataset and inter-dataset  validation of five CLIP-bas ed models (seedless sites in  \\n3\\x01 UTRs) \\nSupplementary Figure 4 Intra-dataset and inter-dataset validation of five CLIP-based mo dels (seedless sites with one \\nGU pair and/or one mismatch within seed complementary region in 3 \\x01 UTRs) \\nSupplementary Figure 5  Intra-dataset and inter-dataset  validation of five CLIP-based models (seed and seedless sites  \\nin CDS and 5 \\x01 UTRs) \\nSupplementary Figure 6  Average performance ev aluation for each of four models trained on human CLIP data by \\ntesting on all four human CLIP datasets (i.e ., one intra-dataset a nd three inter-dataset \\nvalidations) for seed and seedless sites in CDS and 5 \\x01 UTRs \\nSupplementary Figure 7 Performance comparison between nonlinear logi stic modeling and linear logistic modeling \\n \\n \\nSupplementary Figure Legends \\n \\nSupplementary Figure 1.', 'True positive rate comparison for top ranked (from top 1% to 10%) \\npredictions of seed sites (left panel) and seedless si tes with one GU pair and/or one mismatch in the \\nseed complementary region (right panel) in 3 \\x01 UTRs for testing of models on HITS-CLIP data ( a, b), \\non PAR-CLIP data ( c, d), on V-CLIP data ( e, f), on V-PAR-CLIP data ( g, h), and on V-PAR-CLIP-\\nMNase data ( i, j). Supplementary Figure 2. \\x01Average performance evaluation for each of four models trained on \\nhuman CLIP data by testing on all four human CLIP  datasets (i.e., one intra-dataset and three inter-\\ndataset validations) and average performance compar ison with established algorithms for predictions \\nin 3\\x01 UTRs (dashed diagonal line for random predi ctions). Average ROC curve and Youden’s J \\nstatistic are shown in for seed sites ( a, b), for all seedless sites ( c, d), and for seedless sites with one \\nGU pair and/or one mismatch with in seed complementary region ( e, f). Supplementary Figure 3.', '\\x01Inter-dataset model validation by usin g one of the five CLIP datasets as an \\nindependent set for testing performance of each of m odels developed from other four datasets; intra-\\ndataset validation by ten-fold cross validation (CV) (dashed diagonal line for random predictions) for \\nseedless sites in 3 \\x01 UTRs. For each figure, ROC curve and Y ouden’s J statistic are shown for model \\ntesting on HITS-CLIP data ( a, b), on PAR-CLIP data ( c, d), on V-CLIP data ( e, f), on V-PAR-CLIP \\ndata ( g, h), and on V-PAR-CLIP-MNase data ( i, j). Supplementary Figure 4. Performance comparison of logistic models trained on five CLIP datasets \\nwith PITA, RNA22 and mirSVR for prediction of s eedless sites with one GU pair and/or one mismatch \\nwithin seed complementary region in 3 \\x01UTRs. For each figure, ROC cu rve and Youden’s J statistic are \\nshown for model testing on HITS-CLIP data ( a, b), on PAR-CLIP data ( c, d), on V-PAR-CLIP data ( e, \\nf), and on V-PAR-CLIP-MNase data ( g, h).', 'Supplementary Figure 5. \\x01Inter-dataset model validation by usin g one of the five CLIP datasets as an \\nindependent set for testing performance of each of m odels developed from other four datasets; intra-\\ndataset validation by ten-fold cross validation (CV) (dashed diagonal line for random predictions) for \\nseed sites in CDS ( 5.1-10) , seed sites in 5 \\x01 UTRs ( 5.11-20 ), seedless sites in CDS ( 5.21-30 ) and \\nseedless sites in 5 \\x01 UTRs ( 5.31-40 ). For each figure, ROC curve and Youden’s J statistic are shown \\nfor model testing on HITS-CLIP data ( 5.1-2 , 5.11-12 , 5.21-22 , 5.31-32 ), on PAR-CLIP data ( 5.3-4 , \\n5.13-14 , 5.23-24 , 5.33-34 ), on V-CLIP data ( 5.5-6 , 5.15-16 , 5.25-26 , 5.35-36 ), on V-PAR-CLIP data \\n(5.7-8 , 5.17-18 , 5.27-28 , 5.37-38 ), and on V-PAR-CLIP-MNase data ( 5.9-10 , 5.19-20 , 5.29-20 , 5.39-\\n40). Supplementary Figure 6.', '\\x01Average performance evaluation for each of four models trained on \\nhuman CLIP data by testing on all four human CLIP  datasets (i.e., one intra-dataset and three inter-\\ndataset validations) for seed sites in CDS, 5 \\x01 UTRs and seedless sites in CDS, 5 \\x01 UTRs (dashed \\ndiagonal line for random predictions). Average RO C curve and Youden’s J statistic are shown for \\nseed sites in CDS ( a, b), seed sites in 5 \\x01 UTRs ( c, d), seedless sites in CDS ( e, f), seedless sites in 5 \\x01 \\nUTRs ( g, h). Supplementary Figure 7. Performance comparison between nonli near logistic modeling and linear \\nlogistic modeling. ROC curve and Youden’s J statisti c are shown for model tested on V-CLIP data for \\nseed sites in 3 \\x01 UTRs ( a, b), in CDS ( c, d), tested on HITS-CLIP for seedless sites in 3 \\x01 UTRs ( e, f), in \\n5\\x01UTRs ( g, h).', 'Supplementary Table 1      Summary of feature enrichment Analysis for HITS-CLIP data Seed Seedless Features 3′ UTR CDS 5′ UTR 3′ UTR CDS 5′ UTR ΔGtotal •  •  •  •  •  •  ΔGhybrid         ΔGnucl •  •   •  •  •  Seed type •  •  •     miRNA 3′ base-pairing  •  • • • • • Site accessibility •  •  • • •  • Seed accessibility •  •      Upstream accessibility • 15 nt  • 15 nt  • 15 nt  • 15 nt  • 20 nt  Downstream accessibility • 15 nt  • 15 nt  • 15 nt  • 15 nt  • 20 nt Upstream AU content • 30 nt   • 30 nt  • 20 nt  • 30 nt • 30 nt • 20 nt Downstream AU content  • 30 nt    • 30 nt • 30 nt Site conservation score •   •    •  •  •  Seed conservation score •  •        Site location •    •    •  •  •   1. • indicates enrichment; 2. Number of nts gives the size of the nucleotide block for which the strongest enrichment was observed.', 'Supplementary Table 2      Summary of feature enrichment analysis for PAR-CLIP data Seed Seedless Features 3′ UTR CDS 5′ UTR 3′ UTR CDS 5′ UTR ΔGtotal •  •  •  •  •  •  ΔGhybrid         ΔGnucl •  •       •  Seed type •  •  •     miRNA 3′ base-pairing  •  • • • • • Site accessibility •  •  • • •  • Seed accessibility   •      Upstream accessibility  • 15 nt  • 10 nt  • 10 nt  • 10 nt  Downstream accessibility • 10 nt  • 20 nt • 10 nt • 10 nt  • 10 nt  • 10 nt Upstream AU content • 30 nt   • 20 nt  • 30 nt  • 30 nt • 30 nt • 25 nt Downstream AU content  • 25 nt  • 25 nt • 15 nt • 15 nt • 30 nt Site conservation score •   •    •  •  •  Seed conservation score •  •        Site location •     •  •   •   1. • indicates enrichment; 2. Number of nts gives the size of the nucleotide block for which the strongest enrichment was observed.', 'Supplementary Table 3             Summary of feature enrichment analysis for V-CLIP data Seed Seedless Features 3′ UTR CDS 5′ UTR 3′ UTR CDS 5′ UTR ΔGtotal •  •  •  •  •  •  ΔGhybrid         ΔGnucl •  •   •  •  •  Seed type •  •  •     miRNA 3′ base-pairing  •  • • • • • Site accessibility •  •  • • •  • Seed accessibility •  •      Upstream accessibility • 15 nt  • 30 nt  • 10 nt  • 10 nt  • 10 nt  Downstream accessibility • 10 nt  • 30 nt • 30 nt • 30 nt  • 15 nt  • 10 nt Upstream AU content • 30 nt   • 20 nt  • 30 nt  • 30 nt • 20 nt • 25 nt Downstream AU content • 30 nt  • 30 nt  • 20 nt • 30 nt • 20 nt • 30 nt Site conservation score •   •    •  •  •  Seed conservation score •  •        Site location •     •  •  •  •   1. • indicates enrichment; 2. Number of nts gives the size of the nucleotide block for which the strongest enrichment was observed.', 'Supplementary Table 4    Summary of feature enrichment analysis for V-PAR-CLIP data Seed Seedless Features 3′ UTR CDS 5′ UTR 3′ UTR CDS 5′ UTR ΔGtotal •  •  •  •  •  •  ΔGhybrid         ΔGnucl •  •  •  •  •  •  Seed type •  •  •     miRNA 3′ base-pairing  •  • • • • • Site accessibility •  •  • • •  • Seed accessibility •  •      Upstream accessibility • 10 nt  • 15 nt  • 10 nt  • 15 nt   Downstream accessibility • 30 nt  • 10 nt • 25 nt • 10 nt  • 15 nt  • 15 nt Upstream AU content • 30 nt   • 30 nt  • 25 nt  • 25 nt • 20 nt • 25 nt Downstream AU content • 30 nt  • 25 nt  • 30 nt • 30 nt • 20 nt • 30 nt Site conservation score •   •    •  •  •  Seed conservation score •  •        Site location •     •  •  •  •   1. • indicates enrichment; 2. Number of nts gives the size of the nucleotide block for which the strongest enrichment was observed.', 'Supplementary Table 5        Summary of feature enrichment analysis for V-PAR-CLIP-MNase data Seed Seedless Features 3′ UTR CDS 5′ UTR 3′ UTR CDS 5′ UTR ΔGtotal •  •  •  •  •  •  ΔGhybrid •    •  •     ΔGnucl •  •         Seed type •  •  •     miRNA 3′ base-pairing  •  • • • • • Site accessibility   •    •   Seed accessibility   •      Upstream accessibility  • 15 nt   • 20 nt   Downstream accessibility  • 15 nt   • 15 nt   Upstream AU content • 20 nt   • 20 nt   • 30 nt • 20 nt  Downstream AU content • 20 nt  • 20 nt   • 30 nt • 25 nt  Site conservation score •   •    •  •  •  Seed conservation score •  •        Site location •    •  •  •  •  •   1. • indicates enrichment; 2. Number of nts gives the size of the nucleotide block for which the strongest enrichment was observed. !!!!!!!!!!', 'Percentage for sites with top scoresTrue positive rates for top−scored predicted sites\\n01%2%3%4%5%6%7%8%9%10%00.020.040.060.080.1!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!Model 10−fold−CV on HITS−CLIPModel trained on PAR−CLIPModel trained on V−CLIPModel trained on V−PAR−CLIPModel trained on V−PAR−CLIP−MNasePITARNA22mirSVRTargetScan\\n!!!!!!!!!! Percentage for sites with top scoresTrue positive rates for top−scored predicted sites\\n01%2%3%4%5%6%7%8%9%10%00.010.020.030.040.05\\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!Model trained on HITS−CLIPModel trained on PAR−CLIPModel trained on V−CLIPModel trained on V−PAR−CLIPModel trained on V−PAR−CLIP−MNasePITARNA22mirSVR\\n!!!!!!!!!!', 'Percentage for sites with top scoresTrue positive rates for top−scored predicted sites\\n01%2%3%4%5%6%7%8%9%10%00.020.040.060.080.1!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!Model trained on HITS−CLIPModel 10−fold−CV on PAR−CLIPModel trained on V−CLIPModel trained on V−PAR−CLIPModel trained on V−PAR−CLIP−MNasePITARNA22mirSVRTargetScan\\n!!!!!!!!!! Percentage for sites with top scoresTrue positive rates for top−scored predicted sites\\n01%2%3%4%5%6%7%8%9%10%00.010.020.030.040.05!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!Model trained on HITS−CLIPModel trained on PAR−CLIPModel trained on V−CLIPModel trained on V−PAR−CLIPModel trained on V−PAR−CLIP−MNasePITARNA22mirSVR\\n!!!!!!!!!!', 'Percentage for sites with top scoresTrue positive rates for top−scored predicted sites\\n01%2%3%4%5%6%7%8%9%10%00.070.140.210.280.35!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!Model trained on HITS−CLIPModel trained on PAR−CLIPModel 10−fold−CV on V−CLIPModel trained on V−PAR−CLIPModel trained on V−PAR−CLIP−MNasePITARNA22mirSVRTargetScan!!!!!!!!!! Percentage for sites with top scoresTrue positive rates for top−scored predicted sites\\n01%2%3%4%5%6%7%8%9%10%00.030.060.090.120.15!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!Model trained on HITS−CLIPModel trained on PAR−CLIPModel trained on V−CLIPModel trained on V−PAR−CLIPModel trained on V−PAR−CLIP−MNasePITARNA22mirSVR\\n!!!!!!!!!!', 'Percentage for sites with top scoresTrue positive rates for top−scored predicted sites\\n01%2%3%4%5%6%7%8%9%10%00.090.180.270.360.45!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!Model trained on HITS−CLIPModel trained on PAR−CLIPModel trained on V−CLIPModel 10−fold−CV on V−PAR−CLIPModel trained on V−PAR−CLIP−MNasePITARNA22mirSVRTargetScan!!!!!!!!!! Percentage for sites with top scoresTrue positive rates for top−scored predicted sites\\n01%2%3%4%5%6%7%8%9%10%00.040.080.120.160.2!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!Model trained on HITS−CLIPModel trained on PAR−CLIPModel trained on V−CLIPModel trained on V−PAR−CLIPModel trained on V−PAR−CLIP−MNasePITARNA22mirSVR\\n!!!!!!!!!!', 'Percentage for sites with top scoresTrue positive rates for top−scored predicted sites\\n01%2%3%4%5%6%7%8%9%10%00.080.160.240.320.4!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!Model trained on HITS−CLIPModel trained on PAR−CLIPModel trained on V−CLIPModel trained on V−PAR−CLIPModel 10−fold−CV on V−PAR−CLIP−MNasePITARNA22mirSVRTargetScan!!!!!!!!!! Percentage for sites with top scoresTrue positive rates for top−scored predicted sites\\n01%2%3%4%5%6%7%8%9%10%00.020.040.060.080.1!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!Model trained on HITS−CLIPModel trained on PAR−CLIPModel trained on V−CLIPModel trained on V−PAR−CLIPModel trained on V−PAR−CLIP−MNasePITARNA22mirSVR\\nFalse Positive Rate (1−Specificity)True Positive Rate (Sensitivity)aSupplementary Figure 2\\n! 0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 10 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1!! ! ! ! !', \"!Model trained on PAR−CLIP\\nModel trained on V−CLIPModel trained on V−PAR−CLIPModel trained on V−PAR−CLIP−MNasePITARNA22mirSVRTargetScan\\nThreshold for logistic regressionY ouden's J statistic (Sensitivity+Specificity−1)b0\\n0.05\\n0.1\\n0.15\\n0.2\\n0.25\\n0.3\\n0.35\\n0.4\\n0.45\\n0.5\\n0.55\\n0.6\\n0.65\\n0.7\\n0.75\\n0.8\\n0.85\\n0.9\\n0.95\\n1−0.05 0 0.05 0.1 0.15 0.2 0.25 0.3 0.35Model trained on PAR−CLIP\\nModel trained on V−CLIP\\nModel trained on V−PAR−CLIP\\nModel trained on V−PAR−CLIP−MNasePITA\\nRNA22\\nmirSVR\\nTargetScan\\nFalse Positive Rate (1−Specificity)True Positive Rate (Sensitivity)c\\n! 0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 10 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1!! ! ! ! !\", \"!Model trained on PAR−CLIP\\nModel trained on V−CLIPModel trained on V−PAR−CLIPModel trained on V−PAR−CLIP−MNase\\nThreshold for logistic regressionY ouden's J statistic (Sensitivity+Specificity−1)d0\\n0.05\\n0.1\\n0.15\\n0.2\\n0.25\\n0.3\\n0.35\\n0.4\\n0.45\\n0.5\\n0.55\\n0.6\\n0.65\\n0.7\\n0.75\\n0.8\\n0.85\\n0.9\\n0.95\\n1−0.05 0 0.05 0.1 0.15 0.2 0.25 0.3Model trained on PAR−CLIP\\nModel trained on V−CLIP\\nModel trained on V−PAR−CLIP\\nModel trained on V−PAR−CLIP−MNase\\nFalse Positive Rate (1−Specificity)True Positive Rate (Sensitivity)e\\n! 0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 10 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1!! ! ! ! !\", \"!Model trained on PAR−CLIP\\nModel trained on V−CLIPModel trained on V−PAR−CLIPModel trained on V−PAR−CLIP−MNasePITARNA22mirSVR\\nThreshold for logistic regressionY ouden's J statistic (Sensitivity+Specificity−1)f0\\n0.05\\n0.1\\n0.15\\n0.2\\n0.25\\n0.3\\n0.35\\n0.4\\n0.45\\n0.5\\n0.55\\n0.6\\n0.65\\n0.7\\n0.75\\n0.8\\n0.85\\n0.9\\n0.95\\n1−0.05 0 0.05 0.1 0.15 0.2 0.25 0.3Model trained on PAR−CLIP\\nModel trained on V−CLIP\\nModel trained on V−PAR−CLIP\\nModel trained on V−PAR−CLIP−MNase\\nPITARNA22\\nmirSVR\\nFalse Positive Rate (1−Specificity)True Positive Rate (Sensitivity)aSupplementary Figure 3\\n! 0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 10 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1!!! ! ! ! ! !\", \"!Model 10−fold−CV on HITS−CLIP\\nModel trained on PAR−CLIPModel trained on V−CLIPModel trained on V−PAR−CLIPModel trained on V−PAR−CLIP−MNase\\nThreshold for logistic regressionY ouden's J statistic (Sensitivity+Specificity−1)b0\\n0.05\\n0.1\\n0.15\\n0.2\\n0.25\\n0.3\\n0.35\\n0.4\\n0.45\\n0.5\\n0.55\\n0.6\\n0.65\\n0.7\\n0.75\\n0.8\\n0.85\\n0.9\\n0.95\\n1−0.05 0 0.05 0.1 0.15 0.2 0.25 0.3 0.35 0.4Model 10−fold−CV on HITS−CLIP\\nModel trained on PAR−CLIP\\nModel trained on V−CLIP\\nModel trained on V−PAR−CLIPModel trained on V−PAR−CLIP−MNase\\nFalse Positive Rate (1−Specificity)True Positive Rate (Sensitivity)c\\n! 0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 10 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1!!! ! ! ! ! !\", \"!Model trained on HITS−CLIP\\nModel 10−fold−CV on PAR−CLIPModel trained on V−CLIPModel trained on V−PAR−CLIPModel trained on V−PAR−CLIP−MNase\\nThreshold for logistic regressionY ouden's J statistic (Sensitivity+Specificity−1)d0\\n0.05\\n0.1\\n0.15\\n0.2\\n0.25\\n0.3\\n0.35\\n0.4\\n0.45\\n0.5\\n0.55\\n0.6\\n0.65\\n0.7\\n0.75\\n0.8\\n0.85\\n0.9\\n0.95\\n1−0.05 0 0.05 0.1 0.15 0.2 0.25 0.3 0.35Model trained on HITS−CLIP\\nModel 10−fold−CV on PAR−CLIP\\nModel trained on V−CLIP\\nModel trained on V−PAR−CLIP\\nModel trained on V−PAR−CLIP−MNase\\nFalse Positive Rate (1−Specificity)True Positive Rate (Sensitivity)e\\n! 0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 10 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1!!! ! ! ! ! !\", \"!Model trained on HITS−CLIP\\nModel trained on PAR−CLIPModel 10−fold−CV on V−CLIPModel trained on V−PAR−CLIPModel trained on V−PAR−CLIP−MNase\\nThreshold for logistic regressionY ouden's J statistic (Sensitivity+Specificity−1)f0\\n0.05\\n0.1\\n0.15\\n0.2\\n0.25\\n0.3\\n0.35\\n0.4\\n0.45\\n0.5\\n0.55\\n0.6\\n0.65\\n0.7\\n0.75\\n0.8\\n0.85\\n0.9\\n0.95\\n1−0.05 0 0.05 0.1 0.15 0.2 0.25 0.3 0.35 0.4Model trained on HITS−CLIP\\nModel trained on PAR−CLIP\\nModel 10−fold−CV on V−CLIP\\nModel trained on V−PAR−CLIPModel trained on V−PAR−CLIP−MNase\\nFalse Positive Rate (1−Specificity)True Positive Rate (Sensitivity)g\\n! 0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 10 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1!!! ! ! ! ! !\", \"!Model trained on HITS−CLIP\\nModel trained on PAR−CLIPModel trained on V−CLIPModel 10−fold−CV on V−PAR−CLIPModel trained on V−PAR−CLIP−MNase\\nThreshold for logistic regressionY ouden's J statistic (Sensitivity+Specificity−1)h0\\n0.05\\n0.1\\n0.15\\n0.2\\n0.25\\n0.3\\n0.35\\n0.4\\n0.45\\n0.5\\n0.55\\n0.6\\n0.65\\n0.7\\n0.75\\n0.8\\n0.85\\n0.9\\n0.95\\n1−0.05 0 0.05 0.1 0.15 0.2 0.25 0.3Model trained on HITS−CLIP\\nModel trained on PAR−CLIP\\nModel trained on V−CLIPModel 10−fold−CV on V−PAR−CLIP\\nModel trained on V−PAR−CLIP−MNase\\nFalse Positive Rate (1−Specificity)True Positive Rate (Sensitivity)i\\n! 0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 10 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1!!! ! ! ! ! !\", \"!Model trained on HITS−CLIP\\nModel trained on PAR−CLIPModel trained on V−CLIPModel trained on V−PAR−CLIPModel 10−fold−CV on V−PAR−CLIP−MNase\\nThreshold for logistic regressionY ouden's J statistic (Sensitivity+Specificity−1)j0\\n0.05\\n0.1\\n0.15\\n0.2\\n0.25\\n0.3\\n0.35\\n0.4\\n0.45\\n0.5\\n0.55\\n0.6\\n0.65\\n0.7\\n0.75\\n0.8\\n0.85\\n0.9\\n0.95\\n1−0.05 0 0.05 0.1 0.15 0.2 0.25Model trained on HITS−CLIP\\nModel trained on PAR−CLIP\\nModel trained on V−CLIP\\nModel trained on V−PAR−CLIP\\nModel 10−fold−CV on V−PAR−CLIP−MNase\\nFalse Positive Rate (1−Specificity)True Positive Rate (Sensitivity)aSupplementary Figure 4\\n! 0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 10 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1!!! ! ! ! ! !\", \"!Model trained on HITS−CLIP\\nModel trained on PAR−CLIPModel trained on V−CLIPModel trained on V−PAR−CLIPModel trained on V−PAR−CLIP−MNasePITARNA22mirSVR\\nThreshold for logistic regressionY ouden's J statistic (Sensitivity+Specificity−1)b0\\n0.05\\n0.1\\n0.15\\n0.2\\n0.25\\n0.3\\n0.35\\n0.4\\n0.45\\n0.5\\n0.55\\n0.6\\n0.65\\n0.7\\n0.75\\n0.8\\n0.85\\n0.9\\n0.95\\n1−0.05 0 0.05 0.1 0.15 0.2 0.25 0.3 0.35 0.4Model trained on HITS−CLIP\\nModel trained on PAR−CLIP\\nModel trained on V−CLIP\\nModel trained on V−PAR−CLIP\\nModel trained on V−PAR−CLIP−MNase\\nPITARNA22\\nmirSVR\\nFalse Positive Rate (1−Specificity)True Positive Rate (Sensitivity)c\\n! 0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 10 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1! !! ! ! ! ! !\", \"!Model trained on HITS−CLIP\\nModel trained on PAR−CLIPModel trained on V−CLIPModel trained on V−PAR−CLIPModel trained on V−PAR−CLIP−MNasePITARNA22mirSVR\\nThreshold for logistic regressionY ouden's J statistic (Sensitivity+Specificity−1)d0\\n0.05\\n0.1\\n0.15\\n0.2\\n0.25\\n0.3\\n0.35\\n0.4\\n0.45\\n0.5\\n0.55\\n0.6\\n0.65\\n0.7\\n0.75\\n0.8\\n0.85\\n0.9\\n0.95\\n1−0.05 0 0.05 0.1 0.15 0.2 0.25 0.3 0.35Model trained on HITS−CLIP\\nModel trained on PAR−CLIP\\nModel trained on V−CLIP\\nModel trained on V−PAR−CLIP\\nModel trained on V−PAR−CLIP−MNasePITA\\nRNA22\\nmirSVR\\nFalse Positive Rate (1−Specificity)True Positive Rate (Sensitivity)e\\n! 0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 10 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1!!! ! ! ! ! !\", \"!Model trained on HITS−CLIP\\nModel trained on PAR−CLIPModel trained on V−CLIPModel trained on V−PAR−CLIPModel trained on V−PAR−CLIP−MNasePITARNA22mirSVR\\nThreshold for logistic regressionY ouden's J statistic (Sensitivity+Specificity−1)f0\\n0.05\\n0.1\\n0.15\\n0.2\\n0.25\\n0.3\\n0.35\\n0.4\\n0.45\\n0.5\\n0.55\\n0.6\\n0.65\\n0.7\\n0.75\\n0.8\\n0.85\\n0.9\\n0.95\\n1−0.05 0 0.05 0.1 0.15 0.2 0.25 0.3Model trained on HITS−CLIP\\nModel trained on PAR−CLIPModel trained on V−CLIP\\nModel trained on V−PAR−CLIP\\nModel trained on V−PAR−CLIP−MNase\\nPITA\\nRNA22mirSVR\\nFalse Positive Rate (1−Specificity)True Positive Rate (Sensitivity)g\\n! 0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 10 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1!!! ! ! ! ! !\", \"!Model trained on HITS−CLIP\\nModel trained on PAR−CLIPModel trained on V−CLIPModel trained on V−PAR−CLIPModel trained on V−PAR−CLIP−MNasePITARNA22mirSVR\\nThreshold for logistic regressionY ouden's J statistic (Sensitivity+Specificity−1)h0\\n0.05\\n0.1\\n0.15\\n0.2\\n0.25\\n0.3\\n0.35\\n0.4\\n0.45\\n0.5\\n0.55\\n0.6\\n0.65\\n0.7\\n0.75\\n0.8\\n0.85\\n0.9\\n0.95\\n1−0.05 0 0.05 0.1 0.15 0.2 0.25Model trained on HITS−CLIP\\nModel trained on PAR−CLIP\\nModel trained on V−CLIP\\nModel trained on V−PAR−CLIP\\nModel trained on V−PAR−CLIP−MNasePITA\\nRNA22\\nmirSVR\\nFalse Positive Rate (1−Specificity)True Positive Rate (Sensitivity)1Supplementary Figure 5\\n! 0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 10 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1! !! ! ! ! ! !\", \"!Model 10−fold−CV on HITS−CLIP\\nModel trained on PAR−CLIPModel trained on V−CLIPModel trained on V−PAR−CLIPModel trained on V−PAR−CLIP−MNase\\nThreshold for logistic regressionY ouden's J statistic (Sensitivity+Specificity−1)20\\n0.05\\n0.1\\n0.15\\n0.2\\n0.25\\n0.3\\n0.35\\n0.4\\n0.45\\n0.5\\n0.55\\n0.6\\n0.65\\n0.7\\n0.75\\n0.8\\n0.85\\n0.9\\n0.95\\n1−0.05 0 0.05 0.1 0.15 0.2 0.25 0.3 0.35 0.4Model 10−fold−CV on HITS−CLIP\\nModel trained on PAR−CLIP\\nModel trained on V−CLIP\\nModel trained on V−PAR−CLIPModel trained on V−PAR−CLIP−MNase\\nFalse Positive Rate (1−Specificity)True Positive Rate (Sensitivity)3\\n! 0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 10 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1! !! ! ! ! ! !\", \"!Model trained on HITS−CLIP\\nModel 10−fold−CV on PAR−CLIPModel trained on V−CLIPModel trained on V−PAR−CLIPModel trained on V−PAR−CLIP−MNase\\nThreshold for logistic regressionY ouden's J statistic (Sensitivity+Specificity−1)40\\n0.05\\n0.1\\n0.15\\n0.2\\n0.25\\n0.3\\n0.35\\n0.4\\n0.45\\n0.5\\n0.55\\n0.6\\n0.65\\n0.7\\n0.75\\n0.8\\n0.85\\n0.9\\n0.95\\n1−0.05 0 0.05 0.1 0.15 0.2 0.25 0.3Model trained on HITS−CLIP\\nModel 10−fold−CV on PAR−CLIP\\nModel trained on V−CLIPModel trained on V−PAR−CLIP\\nModel trained on V−PAR−CLIP−MNase\\nFalse Positive Rate (1−Specificity)True Positive Rate (Sensitivity)5\\n! 0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 10 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1! !! ! ! ! ! !\", \"!Model trained on HITS−CLIP\\nModel trained on PAR−CLIPModel 10−fold−CV on V−CLIPModel trained on V−PAR−CLIPModel trained on V−PAR−CLIP−MNase\\nThreshold for logistic regressionY ouden's J statistic (Sensitivity+Specificity−1)60\\n0.05\\n0.1\\n0.15\\n0.2\\n0.25\\n0.3\\n0.35\\n0.4\\n0.45\\n0.5\\n0.55\\n0.6\\n0.65\\n0.7\\n0.75\\n0.8\\n0.85\\n0.9\\n0.95\\n1−0.05 0 0.05 0.1 0.15 0.2 0.25 0.3 0.35 0.4Model trained on HITS−CLIP\\nModel trained on PAR−CLIP\\nModel 10−fold−CV on V−CLIP\\nModel trained on V−PAR−CLIPModel trained on V−PAR−CLIP−MNase\\nFalse Positive Rate (1−Specificity)True Positive Rate (Sensitivity)7\\n! 0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 10 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1! !! ! ! ! ! !\", \"!Model trained on HITS−CLIP\\nModel trained on PAR−CLIPModel trained on V−CLIPModel 10−fold−CV on V−PAR−CLIPModel trained on V−PAR−CLIP−MNase\\nThreshold for logistic regressionY ouden's J statistic (Sensitivity+Specificity−1)80\\n0.05\\n0.1\\n0.15\\n0.2\\n0.25\\n0.3\\n0.35\\n0.4\\n0.45\\n0.5\\n0.55\\n0.6\\n0.65\\n0.7\\n0.75\\n0.8\\n0.85\\n0.9\\n0.95\\n1−0.05 0 0.05 0.1 0.15 0.2 0.25 0.3 0.35Model trained on HITS−CLIP\\nModel trained on PAR−CLIP\\nModel trained on V−CLIP\\nModel 10−fold−CV on V−PAR−CLIP\\nModel trained on V−PAR−CLIP−MNase\\nFalse Positive Rate (1−Specificity)True Positive Rate (Sensitivity)9\\n! 0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 10 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1! !! ! ! ! ! !\", \"!Model trained on HITS−CLIP\\nModel trained on PAR−CLIPModel trained on V−CLIPModel trained on V−PAR−CLIPModel 10−fold−CV on V−PAR−CLIP−MNase\\nThreshold for logistic regressionY ouden's J statistic (Sensitivity+Specificity−1)100\\n0.05\\n0.1\\n0.15\\n0.2\\n0.25\\n0.3\\n0.35\\n0.4\\n0.45\\n0.5\\n0.55\\n0.6\\n0.65\\n0.7\\n0.75\\n0.8\\n0.85\\n0.9\\n0.95\\n1−0.05 0 0.05 0.1 0.15 0.2 0.25 0.3Model trained on HITS−CLIP\\nModel trained on PAR−CLIP\\nModel trained on V−CLIPModel trained on V−PAR−CLIP\\nModel 10−fold−CV on V−PAR−CLIP−MNase\\nFalse Positive Rate (1−Specificity)True Positive Rate (Sensitivity)11\\n! 0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 10 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1! !! ! ! ! ! !\", \"!Model 10−fold−CV on HITS−CLIP\\nModel trained on PAR−CLIPModel trained on V−CLIPModel trained on V−PAR−CLIPModel trained on V−PAR−CLIP−MNase\\nThreshold for logistic regressionY ouden's J statistic (Sensitivity+Specificity−1)120\\n0.05\\n0.1\\n0.15\\n0.2\\n0.25\\n0.3\\n0.35\\n0.4\\n0.45\\n0.5\\n0.55\\n0.6\\n0.65\\n0.7\\n0.75\\n0.8\\n0.85\\n0.9\\n0.95\\n1−0.05 0.05 0.15 0.25 0.35 0.45 0.55Model 10−fold−CV on HITS−CLIP\\nModel trained on PAR−CLIP\\nModel trained on V−CLIPModel trained on V−PAR−CLIP\\nModel trained on V−PAR−CLIP−MNase\\nFalse Positive Rate (1−Specificity)True Positive Rate (Sensitivity)13\\n! 0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 10 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1! !! ! ! ! ! !\", \"!Model trained on HITS−CLIP\\nModel 10−fold−CV on PAR−CLIPModel trained on V−CLIPModel trained on V−PAR−CLIPModel trained on V−PAR−CLIP−MNase\\nThreshold for logistic regressionY ouden's J statistic (Sensitivity+Specificity−1)140\\n0.05\\n0.1\\n0.15\\n0.2\\n0.25\\n0.3\\n0.35\\n0.4\\n0.45\\n0.5\\n0.55\\n0.6\\n0.65\\n0.7\\n0.75\\n0.8\\n0.85\\n0.9\\n0.95\\n1−0.05 0 0.05 0.1 0.15 0.2 0.25 0.3Model trained on HITS−CLIP\\nModel 10−fold−CV on PAR−CLIP\\nModel trained on V−CLIPModel trained on V−PAR−CLIP\\nModel trained on V−PAR−CLIP−MNase\\nFalse Positive Rate (1−Specificity)True Positive Rate (Sensitivity)15\\n! 0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 10 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1!! !! ! ! ! !\", \"!Model trained on HITS−CLIP\\nModel trained on PAR−CLIPModel 10−fold−CV on V−CLIPModel trained on V−PAR−CLIPModel trained on V−PAR−CLIP−MNase\\nThreshold for logistic regressionY ouden's J statistic (Sensitivity+Specificity−1)160\\n0.05\\n0.1\\n0.15\\n0.2\\n0.25\\n0.3\\n0.35\\n0.4\\n0.45\\n0.5\\n0.55\\n0.6\\n0.65\\n0.7\\n0.75\\n0.8\\n0.85\\n0.9\\n0.95\\n1−0.05 0 0.05 0.1 0.15 0.2 0.25 0.3 0.35 0.4Model trained on HITS−CLIP\\nModel trained on PAR−CLIP\\nModel 10−fold−CV on V−CLIP\\nModel trained on V−PAR−CLIPModel trained on V−PAR−CLIP−MNase\\nFalse Positive Rate (1−Specificity)True Positive Rate (Sensitivity)17\\n! 0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 10 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1! !!! ! ! ! !\", \"!Model trained on HITS−CLIP\\nModel trained on PAR−CLIPModel trained on V−CLIPModel 10−fold−CV on V−PAR−CLIPModel trained on V−PAR−CLIP−MNase\\nThreshold for logistic regressionY ouden's J statistic (Sensitivity+Specificity−1)180\\n0.05\\n0.1\\n0.15\\n0.2\\n0.25\\n0.3\\n0.35\\n0.4\\n0.45\\n0.5\\n0.55\\n0.6\\n0.65\\n0.7\\n0.75\\n0.8\\n0.85\\n0.9\\n0.95\\n1−0.05 0 0.05 0.1 0.15 0.2 0.25 0.3 0.35 0.4 0.45Model trained on HITS−CLIP\\nModel trained on PAR−CLIP\\nModel trained on V−CLIPModel 10−fold−CV on V−PAR−CLIP\\nModel trained on V−PAR−CLIP−MNase\\nFalse Positive Rate (1−Specificity)True Positive Rate (Sensitivity)19\\n! 0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 10 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1!!!! ! ! ! !\", \"!Model trained on HITS−CLIP\\nModel trained on PAR−CLIPModel trained on V−CLIPModel trained on V−PAR−CLIPModel 10−fold−CV on V−PAR−CLIP−MNase\\nThreshold for logistic regressionY ouden's J statistic (Sensitivity+Specificity−1)200\\n0.05\\n0.1\\n0.15\\n0.2\\n0.25\\n0.3\\n0.35\\n0.4\\n0.45\\n0.5\\n0.55\\n0.6\\n0.65\\n0.7\\n0.75\\n0.8\\n0.85\\n0.9\\n0.95\\n1−0.05 0 0.05 0.1 0.15 0.2 0.25 0.3 0.35Model trained on HITS−CLIP\\nModel trained on PAR−CLIP\\nModel trained on V−CLIP\\nModel trained on V−PAR−CLIP\\nModel 10−fold−CV on V−PAR−CLIP−MNase\\nFalse Positive Rate (1−Specificity)True Positive Rate (Sensitivity)21\\n! 0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 10 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1! !!! ! ! ! !\", \"!Model 10−fold−CV on HITS−CLIP\\nModel trained on PAR−CLIPModel trained on V−CLIPModel trained on V−PAR−CLIPModel trained on V−PAR−CLIP−MNase\\nThreshold for logistic regressionY ouden's J statistic (Sensitivity+Specificity−1)220\\n0.05\\n0.1\\n0.15\\n0.2\\n0.25\\n0.3\\n0.35\\n0.4\\n0.45\\n0.5\\n0.55\\n0.6\\n0.65\\n0.7\\n0.75\\n0.8\\n0.85\\n0.9\\n0.95\\n1−0.05 0 0.05 0.1 0.15 0.2 0.25 0.3 0.35 0.4Model 10−fold−CV on HITS−CLIP\\nModel trained on PAR−CLIP\\nModel trained on V−CLIP\\nModel trained on V−PAR−CLIPModel trained on V−PAR−CLIP−MNase\\nFalse Positive Rate (1−Specificity)True Positive Rate (Sensitivity)23\\n! 0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 10 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1! !!! ! ! ! !\", \"!Model trained on HITS−CLIP\\nModel 10−fold−CV on PAR−CLIPModel trained on V−CLIPModel trained on V−PAR−CLIPModel trained on V−PAR−CLIP−MNase\\nThreshold for logistic regressionY ouden's J statistic (Sensitivity+Specificity−1)240\\n0.05\\n0.1\\n0.15\\n0.2\\n0.25\\n0.3\\n0.35\\n0.4\\n0.45\\n0.5\\n0.55\\n0.6\\n0.65\\n0.7\\n0.75\\n0.8\\n0.85\\n0.9\\n0.95\\n1−0.05 0 0.05 0.1 0.15 0.2 0.25 0.3Model trained on HITS−CLIP\\nModel 10−fold−CV on PAR−CLIP\\nModel trained on V−CLIPModel trained on V−PAR−CLIP\\nModel trained on V−PAR−CLIP−MNase\\nFalse Positive Rate (1−Specificity)True Positive Rate (Sensitivity)25\\n! 0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 10 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1! !!! ! ! ! !\", \"!Model trained on HITS−CLIP\\nModel trained on PAR−CLIPModel 10−fold−CV on V−CLIPModel trained on V−PAR−CLIPModel trained on V−PAR−CLIP−MNase\\nThreshold for logistic regressionY ouden's J statistic (Sensitivity+Specificity−1)260\\n0.05\\n0.1\\n0.15\\n0.2\\n0.25\\n0.3\\n0.35\\n0.4\\n0.45\\n0.5\\n0.55\\n0.6\\n0.65\\n0.7\\n0.75\\n0.8\\n0.85\\n0.9\\n0.95\\n1−0.05 0 0.05 0.1 0.15 0.2 0.25 0.3 0.35Model trained on HITS−CLIP\\nModel trained on PAR−CLIP\\nModel 10−fold−CV on V−CLIP\\nModel trained on V−PAR−CLIP\\nModel trained on V−PAR−CLIP−MNase\\nFalse Positive Rate (1−Specificity)True Positive Rate (Sensitivity)27\\n! 0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 10 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1! !!! ! ! ! !\", \"!Model trained on HITS−CLIP\\nModel trained on PAR−CLIPModel trained on V−CLIPModel 10−fold−CV on V−PAR−CLIPModel trained on V−PAR−CLIP−MNase\\nThreshold for logistic regressionY ouden's J statistic (Sensitivity+Specificity−1)280\\n0.05\\n0.1\\n0.15\\n0.2\\n0.25\\n0.3\\n0.35\\n0.4\\n0.45\\n0.5\\n0.55\\n0.6\\n0.65\\n0.7\\n0.75\\n0.8\\n0.85\\n0.9\\n0.95\\n1−0.05 0 0.05 0.1 0.15 0.2 0.25 0.3 0.35Model trained on HITS−CLIP\\nModel trained on PAR−CLIP\\nModel trained on V−CLIP\\nModel 10−fold−CV on V−PAR−CLIP\\nModel trained on V−PAR−CLIP−MNase\\nFalse Positive Rate (1−Specificity)True Positive Rate (Sensitivity)29\\n! 0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 10 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1! !!! ! ! ! !\", \"!Model trained on HITS−CLIP\\nModel trained on PAR−CLIPModel trained on V−CLIPModel trained on V−PAR−CLIPModel 10−fold−CV on V−PAR−CLIP−MNase\\nThreshold for logistic regressionY ouden's J statistic (Sensitivity+Specificity−1)300\\n0.05\\n0.1\\n0.15\\n0.2\\n0.25\\n0.3\\n0.35\\n0.4\\n0.45\\n0.5\\n0.55\\n0.6\\n0.65\\n0.7\\n0.75\\n0.8\\n0.85\\n0.9\\n0.95\\n1−0.05 0 0.05 0.1 0.15 0.2Model trained on HITS−CLIP\\nModel trained on PAR−CLIPModel trained on V−CLIP\\nModel trained on V−PAR−CLIP\\nModel 10−fold−CV on V−PAR−CLIP−MNase\\nFalse Positive Rate (1−Specificity)True Positive Rate (Sensitivity)31\\n! 0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 10 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1! !! ! ! ! ! !\", \"!Model 10−fold−CV on HITS−CLIP\\nModel trained on PAR−CLIPModel trained on V−CLIPModel trained on V−PAR−CLIPModel trained on V−PAR−CLIP−MNase\\nThreshold for logistic regressionY ouden's J statistic (Sensitivity+Specificity−1)320\\n0.05\\n0.1\\n0.15\\n0.2\\n0.25\\n0.3\\n0.35\\n0.4\\n0.45\\n0.5\\n0.55\\n0.6\\n0.65\\n0.7\\n0.75\\n0.8\\n0.85\\n0.9\\n0.95\\n1−0.05 0.05 0.15 0.25 0.35 0.45 0.55Model 10−fold−CV on HITS−CLIP\\nModel trained on PAR−CLIP\\nModel trained on V−CLIPModel trained on V−PAR−CLIP\\nModel trained on V−PAR−CLIP−MNase\\nFalse Positive Rate (1−Specificity)True Positive Rate (Sensitivity)33\\n! 0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 10 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1! ! !! ! ! ! !\", \"!Model trained on HITS−CLIP\\nModel 10−fold−CV on PAR−CLIPModel trained on V−CLIPModel trained on V−PAR−CLIPModel trained on V−PAR−CLIP−MNase\\nThreshold for logistic regressionY ouden's J statistic (Sensitivity+Specificity−1)340\\n0.05\\n0.1\\n0.15\\n0.2\\n0.25\\n0.3\\n0.35\\n0.4\\n0.45\\n0.5\\n0.55\\n0.6\\n0.65\\n0.7\\n0.75\\n0.8\\n0.85\\n0.9\\n0.95\\n1−0.05 0 0.05 0.1 0.15 0.2 0.25Model trained on HITS−CLIP\\nModel 10−fold−CV on PAR−CLIP\\nModel trained on V−CLIP\\nModel trained on V−PAR−CLIP\\nModel trained on V−PAR−CLIP−MNase\\nFalse Positive Rate (1−Specificity)True Positive Rate (Sensitivity)35\\n! 0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 10 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1!! !! ! ! ! !\", \"!Model trained on HITS−CLIP\\nModel trained on PAR−CLIPModel 10−fold−CV on V−CLIPModel trained on V−PAR−CLIPModel trained on V−PAR−CLIP−MNase\\nThreshold for logistic regressionY ouden's J statistic (Sensitivity+Specificity−1)360\\n0.05\\n0.1\\n0.15\\n0.2\\n0.25\\n0.3\\n0.35\\n0.4\\n0.45\\n0.5\\n0.55\\n0.6\\n0.65\\n0.7\\n0.75\\n0.8\\n0.85\\n0.9\\n0.95\\n1−0.05 0 0.05 0.1 0.15 0.2 0.25 0.3Model trained on HITS−CLIP\\nModel trained on PAR−CLIP\\nModel 10−fold−CV on V−CLIPModel trained on V−PAR−CLIP\\nModel trained on V−PAR−CLIP−MNase\\nFalse Positive Rate (1−Specificity)True Positive Rate (Sensitivity)37\\n! 0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 10 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1!! !! ! ! ! !\", \"!Model trained on HITS−CLIP\\nModel trained on PAR−CLIPModel trained on V−CLIPModel 10−fold−CV on V−PAR−CLIPModel trained on V−PAR−CLIP−MNase\\nThreshold for logistic regressionY ouden's J statistic (Sensitivity+Specificity−1)380\\n0.05\\n0.1\\n0.15\\n0.2\\n0.25\\n0.3\\n0.35\\n0.4\\n0.45\\n0.5\\n0.55\\n0.6\\n0.65\\n0.7\\n0.75\\n0.8\\n0.85\\n0.9\\n0.95\\n1−0.05 0 0.05 0.1 0.15 0.2 0.25 0.3 0.35Model trained on HITS−CLIP\\nModel trained on PAR−CLIP\\nModel trained on V−CLIP\\nModel 10−fold−CV on V−PAR−CLIP\\nModel trained on V−PAR−CLIP−MNase\\nFalse Positive Rate (1−Specificity)True Positive Rate (Sensitivity)39\\n! 0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 10 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1!! !! ! ! ! !\", \"!Model trained on HITS−CLIP\\nModel trained on PAR−CLIPModel trained on V−CLIPModel trained on V−PAR−CLIPModel 10−fold−CV on V−PAR−CLIP−MNase\\nThreshold for logistic regressionY ouden's J statistic (Sensitivity+Specificity−1)400\\n0.05\\n0.1\\n0.15\\n0.2\\n0.25\\n0.3\\n0.35\\n0.4\\n0.45\\n0.5\\n0.55\\n0.6\\n0.65\\n0.7\\n0.75\\n0.8\\n0.85\\n0.9\\n0.95\\n1−0.05 0 0.05 0.1 0.15Model trained on HITS−CLIP\\nModel trained on PAR−CLIP\\nModel trained on V−CLIP\\nModel trained on V−PAR−CLIPModel 10−fold−CV on V−PAR−CLIP−MNase\\nFalse Positive Rate (1−Specificity)True Positive Rate (Sensitivity)aSupplementary Figure 6\\n! 0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 10 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1!! ! ! ! !\", \"!Model trained on PAR−CLIP\\nModel trained on V−CLIPModel trained on V−PAR−CLIPModel trained on V−PAR−CLIP−MNase\\nThreshold for logistic regressionY ouden's J statistic (Sensitivity+Specificity−1)b0\\n0.05\\n0.1\\n0.15\\n0.2\\n0.25\\n0.3\\n0.35\\n0.4\\n0.45\\n0.5\\n0.55\\n0.6\\n0.65\\n0.7\\n0.75\\n0.8\\n0.85\\n0.9\\n0.95\\n1−0.05 0 0.05 0.1 0.15 0.2 0.25 0.3Model trained on PAR−CLIP\\nModel trained on V−CLIP\\nModel trained on V−PAR−CLIP\\nModel trained on V−PAR−CLIP−MNase\\nFalse Positive Rate (1−Specificity)True Positive Rate (Sensitivity)c\\n! 0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 10 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1!!! ! ! !\", \"!Model trained on PAR−CLIP\\nModel trained on V−CLIPModel trained on V−PAR−CLIPModel trained on V−PAR−CLIP−MNase\\nThreshold for logistic regressionY ouden's J statistic (Sensitivity+Specificity−1)d0\\n0.05\\n0.1\\n0.15\\n0.2\\n0.25\\n0.3\\n0.35\\n0.4\\n0.45\\n0.5\\n0.55\\n0.6\\n0.65\\n0.7\\n0.75\\n0.8\\n0.85\\n0.9\\n0.95\\n1−0.05 0 0.05 0.1 0.15 0.2 0.25 0.3 0.35Model trained on PAR−CLIP\\nModel trained on V−CLIP\\nModel trained on V−PAR−CLIP\\nModel trained on V−PAR−CLIP−MNase\\nFalse Positive Rate (1−Specificity)True Positive Rate (Sensitivity)e\\n! 0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 10 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1!!! ! ! !\", \"!Model trained on PAR−CLIP\\nModel trained on V−CLIPModel trained on V−PAR−CLIPModel trained on V−PAR−CLIP−MNase\\nThreshold for logistic regressionY ouden's J statistic (Sensitivity+Specificity−1)f0\\n0.05\\n0.1\\n0.15\\n0.2\\n0.25\\n0.3\\n0.35\\n0.4\\n0.45\\n0.5\\n0.55\\n0.6\\n0.65\\n0.7\\n0.75\\n0.8\\n0.85\\n0.9\\n0.95\\n1−0.05 0 0.05 0.1 0.15 0.2 0.25 0.3Model trained on PAR−CLIP\\nModel trained on V−CLIP\\nModel trained on V−PAR−CLIP\\nModel trained on V−PAR−CLIP−MNase\\nFalse Positive Rate (1−Specificity)True Positive Rate (Sensitivity)g\\n! 0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 10 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1! !! ! ! !\", \"!Model trained on PAR−CLIP\\nModel trained on V−CLIPModel trained on V−PAR−CLIPModel trained on V−PAR−CLIP−MNase\\nThreshold for logistic regressionY ouden's J statistic (Sensitivity+Specificity−1)h0\\n0.05\\n0.1\\n0.15\\n0.2\\n0.25\\n0.3\\n0.35\\n0.4\\n0.45\\n0.5\\n0.55\\n0.6\\n0.65\\n0.7\\n0.75\\n0.8\\n0.85\\n0.9\\n0.95\\n1−0.05 0 0.05 0.1 0.15 0.2 0.25Model trained on PAR−CLIP\\nModel trained on V−CLIPModel trained on V−PAR−CLIP\\nModel trained on V−PAR−CLIP−MNase\\n          \\nFalse Positive Rate (1−Specificity)True Positive Rate (Sensitivity)! 0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 10 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1!! !! !! ! !! ! ! ! ! ! ! ! ! !\", \"!quadratic−Model trained on HITS−CLIP\\nlinear−Model trained on HITS−CLIPquadratic−Model trained on PAR−CLIPlinear−Model trained on PAR−CLIPquadratic−Model 10−fold−CV on V−CLIPlinear−Model 10−fold−CV on V−CLIPquadratic−Model trained on V−PAR−CLIPlinear−Model trained on V−PAR−CLIPquadratic−Model trained on V−PAR−CLIP−MNaselinear−Model trained on V−PAR−CLIP−MNase\\nThreshold for logistic regressionY ouden's J statistic (Sensitivity+Specificity−1)\\n0\\n0.05\\n0.1\\n0.15\\n0.2\\n0.25\\n0.3\\n0.35\\n0.4\\n0.45\\n0.5\\n0.55\\n0.6\\n0.65\\n0.7\\n0.75\\n0.8\\n0.85\\n0.9\\n0.95\\n1−0.05 0.05 0.15 0.25 0.35quadratic−Model trained on HITS−CLIP\\nlinear−Model trained on HITS−CLIP\\nquadratic−Model trained on PAR−CLIP\\nlinear−Model trained on PAR−CLIP\\nquadratic−Model 10−fold−CV on V−CLIP\\nlinear−Model 10−fold−CV on V−CLIPquadratic−Model trained on V−PAR−CLIP\\nlinear−Model trained on V−PAR−CLIP\\nquadratic−Model trained on V−PAR−CLIP−MNase\\nlinear−Model trained on V−PAR−CLIP−MNase\\n          \\nFalse Positive Rate (1−Specificity)True Positive Rate (Sensitivity)!\", '0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 10 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1!! !! !!! !! ! ! ! ! ! ! ! ! !', \"!quadratic−Model trained on HITS−CLIP\\nlinear−Model trained on HITS−CLIPquadratic−Model trained on PAR−CLIPlinear−Model trained on PAR−CLIPquadratic−Model 10−fold−CV on V−CLIPlinear−Model 10−fold−CV on V−CLIPquadratic−Model trained on V−PAR−CLIPlinear−Model trained on V−PAR−CLIPquadratic−Model trained on V−PAR−CLIP−MNaselinear−Model trained on V−PAR−CLIP−MNase  \\nThreshold for logistic regressionY ouden's J statistic (Sensitivity+Specificity−1)\\n0\\n0.05\\n0.1\\n0.15\\n0.2\\n0.25\\n0.3\\n0.35\\n0.4\\n0.45\\n0.5\\n0.55\\n0.6\\n0.65\\n0.7\\n0.75\\n0.8\\n0.85\\n0.9\\n0.95\\n1−0.05 0.05 0.15 0.25 0.35quadratic−Model trained on HITS−CLIP\\nlinear−Model trained on HITS−CLIP\\nquadratic−Model trained on PAR−CLIP\\nlinear−Model trained on PAR−CLIP\\nquadratic−Model 10−fold−CV on V−CLIP\\nlinear−Model 10−fold−CV on V−CLIPquadratic−Model trained on V−PAR−CLIP\\nlinear−Model trained on V−PAR−CLIP\\nquadratic−Model trained on V−PAR−CLIP−MNase\\nlinear−Model trained on V−PAR−CLIP−MNase\\nFalse Positive Rate (1−Specificity)True Positive Rate (Sensitivity)!\", '0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 10 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1!! !! !! !! ! ! ! ! ! ! ! ! ! !', \"!quadratic−Model 10−fold−CV on HITS−CLIP\\nlinear−Model 10−fold−CV on HITS−CLIPquadratic−Model trained on PAR−CLIPlinear−Model trained on PAR−CLIPquadratic−Model trained on V−CLIPlinear−Model trained on V−CLIPquadratic−Model trained on V−PAR−CLIPlinear−Model trained on V−PAR−CLIPquadratic−Model trained on V−PAR−CLIP−MNaselinear−Model trained on V−PAR−CLIP−MNase\\nThreshold for logistic regressionY ouden's J statistic (Sensitivity+Specificity−1)\\n0\\n0.05\\n0.1\\n0.15\\n0.2\\n0.25\\n0.3\\n0.35\\n0.4\\n0.45\\n0.5\\n0.55\\n0.6\\n0.65\\n0.7\\n0.75\\n0.8\\n0.85\\n0.9\\n0.95\\n1−0.05 0.05 0.15 0.25 0.35quadratic−Model 10−fold−CV on HITS−CLIP\\nlinear−Model 10−fold−CV on HITS−CLIP\\nquadratic−Model trained on PAR−CLIP\\nlinear−Model trained on PAR−CLIP\\nquadratic−Model trained on V−CLIP\\nlinear−Model trained on V−CLIPquadratic−Model trained on V−PAR−CLIP\\nlinear−Model trained on V−PAR−CLIP\\nquadratic−Model trained on V−PAR−CLIP−MNase\\nlinear−Model trained on V−PAR−CLIP−MNase\\n            \\nFalse Positive Rate (1−Specificity)True Positive Rate (Sensitivity)!\", '0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 10 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1!! !!! !! !! ! ! ! ! ! ! ! ! !', \"!quadratic−Model 10−fold−CV on HITS−CLIP\\nlinear−Model 10−fold−CV on HITS−CLIPquadratic−Model trained on PAR−CLIPlinear−Model trained on PAR−CLIPquadratic−Model trained on V−CLIPlinear−Model trained on V−CLIPquadratic−Model trained on V−PAR−CLIPlinear−Model trained on V−PAR−CLIPquadratic−Model trained on V−PAR−CLIP−MNaselinear−Model trained on V−PAR−CLIP−MNase\\nThreshold for logistic regressionY ouden's J statistic (Sensitivity+Specificity−1)\\n0\\n0.05\\n0.1\\n0.15\\n0.2\\n0.25\\n0.3\\n0.35\\n0.4\\n0.45\\n0.5\\n0.55\\n0.6\\n0.65\\n0.7\\n0.75\\n0.8\\n0.85\\n0.9\\n0.95\\n1−0.05 0.05 0.15 0.25 0.35 0.45 0.55quadratic−Model 10−fold−CV on HITS−CLIP\\nlinear−Model 10−fold−CV on HITS−CLIP\\nquadratic−Model trained on PAR−CLIP\\nlinear−Model trained on PAR−CLIP\\nquadratic−Model trained on V−CLIPlinear−Model trained on V−CLIP\\nquadratic−Model trained on V−PAR−CLIP\\nlinear−Model trained on V−PAR−CLIP\\nquadratic−Model trained on V−PAR−CLIP−MNase\\nlinear−Model trained on V−PAR−CLIP−MNase\\nBIOINFORMATICS\\nRNA secondary structure prediction by centroids\\nin a Boltzmann weighted ensemble\\nYE DING,1CHI YU CHAN,1and CHARLES E.\", 'LAWRENCE1,2\\n1Bioinformatics Center, Wadsworth Center, New York State Department of Health, Albany, New York 12208, USA\\n2Center for Computational Molecular Biology and Division of Applied Mathematics, Brown University, Providence,\\nRhode Island 02912, USA\\nABSTRACT\\nPrediction of RNA secondary structure by free energy minimization has been the standard for over two decades. Here we\\ndescribe a novel method that forsakes this paradigm for predictions based on Boltzmann-weighted structure ensemble. We\\nintroduce the notion of a centroid structure as a representative for a set of structures and describe a procedure for its\\nidentification. In comparison with the minimum free energy (MFE) structure using diverse types of structural RNAs, the\\ncentroid of the ensemble makes 30.0% fewer prediction errors as measured by the positive predictive value (PPV) with\\nmarginally improved sensitivity. The Boltzmann ensemble can be separated into a small number (3.2 on average) of clusters.', 'Among the centroids of these clusters, the ‘‘best cluster centroid’’ as determined by comparison to the known structure\\nsimultaneously improves PPV by 46.5% and sensitivity by 21.7%. For 58% of the studied sequences for which the MFE\\nstructure is outside the cluster containing the best centroid, the improvements by the best centroid are 62.5% for PPV and\\n31.4% for sensitivity. These results suggest that the energy well containing the MFE structure under the current incomplete\\nenergy model is often different from the one for the unavailable complete model that presumably contains the unique native\\nstructure. Centroids are available on the Sfold server at http://sfold.wadsworth.org. Keywords: secondary structure prediction; centroid; Boltzmann ensemble\\nINTRODUCTION\\nRNA molecules are key elements in some of the cell’s most\\nfundamental processes, including catalysis, RNA splicing, and\\nregulation of transcription and translation.', 'To a large degree,\\nthe function of a structural RNA molecule is determined by\\nits structure. Computational methods for modeling RNA\\nsecondary structure have proven to be valuable in many\\ncases in which crystal structures are not available. Free energy minimization is a long-established paradigm\\nin computational structural biology that is based on the\\nassumption that, at equilibrium, the solution to the under-\\nlying molecular folding problem is unique, and that the\\nmolecule folds into the lowest energy state. Applications ofthis paradigm include RNA folding (Zuker 1989), protein\\nfolding (Anfinsen 1973; Abagyan 1993), and transmembrane\\nhelix packing (Pappu et al. 1999). The prediction of RNA\\nsecondary structure has been widely applied, with good\\nsuccess. Efficient algorithms for computing the minimum\\nfree energy (MFE) structure and a set of suboptimal struc-\\ntures (Zuker and Stiegler 1981; Mathews et al.', '1999, 2004)\\nare based on free energy parameters that are estimated or\\nextrapolated from chemical melting experiments (Xia et al. 1998; Mathews et al. 1999, 2004). An alternative approach\\ncomputes all suboptimal foldings within an energy incre-\\nment above the MFE (Wuchty et al. 1999). The exponential\\ngrowth in the number of these foldings motivated recent\\ndevelopment of the RNAshapes method for the efficient\\nrepresentation of the near optimal set (Giegerich et al. 2004). In a drastic departure from the MFE perspective,\\nefforts have been made to characterize the ensemble of\\nstructures (McCaskill 1990; Bonhoeffer et al. 1993). Recently, we have presented an algorithm that draws sam-\\nples from the ensemble of secondary structures in propor-\\ntion to their Boltzmann weights (Ding and Lawrence 2003). In other words, our algorithm guarantees the generation\\nof a statistically representative sample of the Boltzmann-\\na25006 Ding et al.', 'BIOINFORMATICS RAReprint request to: Ye Ding, Bioinformatics Center, Wadsworth Cen-\\nter, New York State Department of Health, 150 New Scotland Avenue,\\nAlbany, NY 12208, USA; e-mail: yding@wadsworth.org; fax: (518) 402-\\n4623; or Charles E. Lawrence, Center for Computational Molecular Biol-\\nogy and Division of Applied Mathematics, Brown University, 182 George\\nStreet, Providence, RI 02912, USA; e-mail: lawrence@dam.brown.edu; fax:\\n(401) 863-1355. Article and publication are at http://www.rnajournal.org/cgi/doi/\\n10.1261/rna.2500605. RNA (2005), 11:1157–1166. Published by Cold Spring Harbor Laboratory Press. Copyright ª2005 RNA Society. 1157\\nweighted ensemble of structures, and thus enables the calcu-\\nlation of sampling statistics for structural features (Ding and\\nLawrence 2001, 2003). In this report, we examine the utility\\nof such samples in RNA secondary structure prediction.', 'In applications of the sampling algorithm, we found that\\nthere often exist distinct clusters in the Boltzmann ensem-\\nble (Ding and Lawrence 2003), with each cluster containing\\nsimilar structures. To aid in the characterization of the sampled structural\\nspace, we introduce the centroid structure as an efficient\\nmeans to characterize the central tendency for a set of\\nstructures and present a procedure for its identification. We examine the predictive value of the centroid of the\\nentire sampled ensemble, the centroid of the largest cluster,\\nand the cluster centroid that is closest to the structure\\ndetermined by comparative sequence analysis. RESULTS\\nClustering results\\nFrom online RNA databases, 81 RNA sequences from nine\\nstructural RNA classes were selected (see Materials and Meth-\\nods section for sequence selection process).', 'For each sequence,\\n1000 structures sampled by our algorithm (Ding and Lawrence\\n2003) are clustered, and then the cluster to which the MFE\\nstructure belongs is determined (see Materials and Methods\\nsection for clustering procedure). Although a structural RNA\\nmay have unique structure in solution, we have found that\\nthere exists a small number (3.2 on average) of distinct clusters\\nof similar structures in the Boltzmann ensemble. Although thenumber of conformational states grows exponentially with\\nsequence length, we found no evidence that the number of\\nclusters increases with the length of the sequence (correlation\\ncoefficient = /C00.1180, P-value = 0.294). Since the algorithm\\nsamples structures in accordance with their Boltzmann-\\nweighted probabilities, the probability of a cluster is estimated\\nby the frequency of structures in that cluster, i.e., the number of\\nstructures in the cluster divided by the sample size.', 'In the case\\nof multiple occurrences of the same structure, each occurrence\\nis counted in the calculation. The MFE structure is present in\\nthe largest cluster for 55 of the 81 RNAs (68%). For 36 of these\\n55 sequences, the largest cluster dominates the structure space,\\nwith a probability of 0.7 or higher. Thus, the MFE structure is\\nin a dominant cluster for only 44% (36/81) of the RNAs. The\\nclustering results in Table 1 for 12 sequences exemplify possible\\nscenarios for the cluster of the MFE structure. The cluster of the\\nMFE structure can be the largest cluster with either a dominant\\nprobability or a moderate probability. The MFE cluster can\\nalso be in a cluster that is secondary in size, or in some cases a\\ncluster of only small or negligible probability. The 23S rRNA\\nsequence for Chlamydomonas reinhardtii (accession number\\nX15727) presents an extreme case for which the MFE structure\\nis not similar to any structure in the sampled ensemble.', 'These\\nfindings suggest that the MFE structure does not always\\nrepresent well the Boltzmann-weighted ensemble, thus moti-\\nvating our search for more reliable representatives. Centroid structures as representatives\\nAs an alternative to the MFE structure, we propose the\\ncentroid structure. The centroid for a given set of struc-\\nTABLE 1. Clusters for sampled structures and MFE structure\\nRNA type OrganismGenBank\\naccession no.', 'Length (nt)Number of\\nclusters Cluster probabilitiesa\\nSSU (16S) rRNA Bordetella bronchiseptica U04948 1532 2 0.930* 0.070\\ntRNA Crossostoma lacustre M91245 70 2 0.906* 0.094\\nGroup I intron Acanthamoeba griffini U02540 556 2 0.950* 0.050\\nLSU (23S) rRNA Thermus thermophilus X12612 2915 3 0.339* 0.335 0.326\\nGroup I intron Acanthamoeba griffini S81337 526 5 0.464*\\n0.0480.400\\n0.0360.052\\nGroup II intron Saccharomyces cerevisiae AJ011856 2520 4 0.557*\\n0.0070.432\\n0.004\\n5S rRNA Agrobacterium tumefaciens X02627 120 2 0.591\\n0.409*\\ntmRNA Dehalococcoides ethenogenes\\nstrain 195GSPb352 2 0.578\\n0.422*\\nRNase P Tarsius syrichta L08801 286 3 0.552\\n0.446* 0.002\\nLSU (23S) rRNA Chlamydomonas reinhardtii X15727 2902 3 0.907 0.093 0.000*\\nRNase P Dermocarpa sp. X97396 359 2 0.803 0.197*\\nRNase P Leptospirillum ferrooxidans AF296042 327 2 0.804 0.196*\\naAsterisk indicates cluster of MFE structure. bhttp://tigrblast.tigr.org/ufmg/index.cgi?database=d_ethenogenes|seq. 1158 RNA, Vol. 11, No. 8Ding et al.', 'tures is the structure in the entire structure ensemble that\\nhas the minimum total base-pair distance to the structures\\nin the set. Thus, the centroid structure can be considered as\\nthe single structure that best represents the central ten-\\ndency of the set. A centroid is refered to as the ensemble\\ncentroid when the set is the entire collection of structures\\nsampled from the ensemble. A centroid of a cluster of\\nsimilar structures is referred to as a cluster centroid .T h e\\nmathematical definition of centroid structure and the deri-\\nvation for its identification are presented in the Materials\\nand Methods section. Ever since the emergence of mfold\\n(multiple folds) it has been a common practice to report a\\nnumber of suboptimal folds for predicitive purposes (Zuker\\n1989, 2003). Both the optimal fold and the best from a long\\nlist of suboptimal folds are of interest for performance\\nevaluation (Mathews et al. 1999).', 'Here we employ a similar\\nevaluation strategy and report on the predictive performance\\nof the ensemble centroid and that of the best of a short list of\\ncluster centroids, i.e., the best centroid. In other words,\\nwhen a reference structure is available as the standard, the\\nbest cluster centroid is defined as the cluster centroid that\\nhas the shortest base-pair distance to this known structure. Of course, just as with the best suboptimal, the identity of\\nthis best centroid cannot be determined when a reference\\nstructure is not available. In keeping with accepted practice\\nin this field, we employed structures determined from com-\\nparative sequence analysis as the standard for comparison\\nand for the identification of the best centroid. Performance measures\\nWe consider three measures for making performance com-\\nparisons between the MFE structure and centroids: base-\\npair distance, sensitivity, and PPV.', 'More specifically, we\\ncompute the base-pair distance between the MFE structure\\nand the structure determined by comparative sequence\\nanalysis and between the ensemble centroid or a cluster\\ncentroid and the structure determined by comparative\\nsequence analysis. The sensitivity for a predicted structure\\nis the percentage of base pairs in the structure determined\\nby comparative sequence analysis that are also present in\\nthe predicted structure. The PPV is the percentage of base\\npairs in the predicted structure that are in the structure\\ndetermined by comparative sequence analysis. These two\\ncomplementary measures have become the standards for\\nmeasuring predictive accuracy (Mathews et al. 1999; Dow-\\nell and Eddy 2004; Mathews 2004).', 'The sensitivity focuses\\non predicting base pairs in the structure determined by\\ncomparative sequence analysis without regard to false posi-\\ntive base pair predictions, while the PPV focuses on accu-\\nracy of the predicted base pairs without regard to false\\nnegative base pairs. A perfect prediction is achieved if\\nboth the sensitivity and the PPV are 100%, in which case\\nthe two structures being compared are identical and have a\\ndistance of zero base pairs.Centroids are closer to the structure determined\\nby comparative sequence analysis than is the\\nMFE structure\\nThe ensemble centroid, the centroid of the largest cluster,\\nand the best centroid are closer in base-pair distance to the\\nstructure determined by comparative sequence analysis than\\nis the MFE structure for 66 (81.5%), 60 (74.1%), and 74\\n(91.4%) sequences, respectively.', 'Furthermore, these centroids\\nare either closer to the structure determined by comparative\\nsequence analysis than is the MFE structure or are as close to\\nthe structure determined by comparative sequence analysis as\\nis the MFE structure for 73 (90.1%), 71 (87.7%), and 80\\n(98.8%) of the 81 sequences, respectively. For each sequence, the percentage of distance improve-\\nment by a centroid over the MFE structure is calculated by\\n[1/C0D(C, P)/ D(M, P)] 3100%, where D(C, P) is the base-\\npair distance between the centroid and the structure deter-\\nmined by comparative sequence analysis and D(M, P) is the\\nbase-pair distance between the MFE structure and the\\nstructure determined by comparative sequence analysis. For each RNA type, the averaged percentage of improve-\\nment is calculated; these values are presented in Table 2. For the best cluster centroid, the average improvement is\\n> 19% for every RNA type.', 'For the ensemble centroid and\\nthe centroid of the largest cluster, substantial improve-\\nments are obtained, except for the SRP RNAs. Centroids yield comparable or improved sensitivities\\nFor each RNA type, the averaged sensitivity by the MFE\\nstructure and the average percentage of improvement in\\nsensitivity by each of the three centroids are presented in\\nTable 3. For the MFE structure, the ensemble centroid, and\\nthe centroid of the largest cluster, the results are comparable\\nwith marginal overall improvements by the centroids. Furthermore, the ensemble centroid and the largest cluster\\ncentroid show equal or improved sensitivity for > 60% of the\\nsequences. For the best cluster centroid, there is an average\\nimprovement of 21.74% for all RNA types, and negative\\nimprovement is only observed for group II introns.', 'Centroid predictions yield fewer errors\\nFor each RNA type, the averaged PPV by the MFE structure\\nand the average percentage of improvement in PPV by each of\\nthe three centroids are presented in Table 4. For both the\\nensemble centroid and the best centroid, there is an\\nimprovement over the MFE structure, with an overall average\\nof 30.0% and an overall average of 46.5%, respectively. For the\\nbest centroid, in particular, the PPV is either the same or\\nimproved for 79 of the 81 sequences (97.5%). For the centroid\\nof the largest cluster, there is an improvement for seven of the\\nnine RNA types, with an overall average improvement of\\n17.6%. www.rnajournal.org 1159Secondary structure prediction by centroids\\nMFE predictions break down when the\\nMFE structure is in the wrong cluster\\nThe degree of improvement by the best centroid largely\\ndepends on the location of the MFE structure.', 'For 34\\nsequences (42.0%) for which the MFE structure is in the\\ncluster of the best centroid, the base-pair distance and the\\nPPV are substantially improved, and the sensitivity is\\nimproved appreciably; for the other 47 sequences (58.0%)\\nfor which the MFE structure is outside the cluster of the\\nbest centroid, the improvements by the best centroid are\\n36.5% for base-pair distance, 62.5% for PPV, and 31.4%\\nfor sensitivity (Table 5). The latter case is illustrated by the\\nenergy landscape of the sampled ensemble and representative\\nstructures for Agrobacterium tumefaciens 5S rRNA (Fig. 1).Large standard deviations due to a wide range\\nof improvements\\nThe unusually large standard deviations in Tables 2–5\\nare due to a wide range of improvements, as illustrated\\nby Figure 2 for the best centroid. For base-pair distance,\\nonly one sequence has a negative improvement of\\n/C01.1%, and the improvement is as high as 100% (Fig. 2A).', 'In terms of sensitivity, there are small to moderate\\nnegative improvements for 20 of the 81 sequences\\n(24.7%), with an average of /C09.6%, and the positive\\nimprovement is as high as 245.5% (Fig. 2B). For PPV,\\nthe improvement is as high as 313.6%, with only two\\nsequences having negative improvements of /C08.3% and\\n/C025.3% (Fig. 2C).TABLE 2.', 'Distance improvement by centroids over MFE structure\\nNumber ofAverage percentage of improvement in base-pair distance, with respect to MFE structurea\\nRNA type sequences Ensemble centroid Largest cluster centroid Best cluster centroid\\nSSU (16S) rRNA 10 12.88 64.82 9.11 66.15 24.03 613.28\\nLSU (23S) rRNA 10 19.66 613.37 16.78 613.57 21.52 610.53\\n5S rRNA 10 13.18 619.68 17.75 628.59 27.90 633.08\\nGroup I intron 9 15.98 622.09 6.10 623.46 23.02 67.19\\nGroup II intron 2 21.73 60.01 21.11 61.40 23.03 61.31\\nRNase P 10 8.72 615.35 2.41 614.06 21.46 617.40\\nSRP RNA 10 –9.48 646.06 –14.64 664.23 19.55 616.60\\ntmRNA 10 19.10 619.65 12.47 616.56 25.20 615.72\\ntRNA 10 21.11 617.87 11.77 615.78 56.32 636.35\\nTotal 81 12.83 623.37 8.07 628.56 27.32 622.97\\naDistance improvement by a centroid with respect to the MFE structure = [1 – (base-pair distance between structure determined by comparative\\nsequence analysis and centroid)/(base-pair distance between structure determined by comparative sequence analysis and MFE structure)] 3\\n100%.', 'TABLE 3.', 'Sensitivity of MFE structure and sensitivity improvement by centroids over MFE structure\\nNumber of SensitivityaAverage percentage of improvement in sensitivity with respect to MFE structureb\\nRNA type sequences of MFE structure Ensemble centroid Largest cluster centroid Best cluster centroid\\nSSU (16S) rRNA 10 49.80 614.62 –4.14 69.31 –2.75 68.18 9.10 616.51\\nLSU (23S) rRNA 10 35.35 613.26 0.75 615.41 0.42 615.33 5.46 612.43\\n5S rRNA 10 55.93 624.52 2.41 637.38 15.15 661.20 41.81 684.82\\nGroup I intron 9 45.48 619.97 6.46 621.72 4.60 624.19 29.06 655.63\\nGroup II intron 2 44.48 66.74 0.54 69.46 0.33 68.08 –2.09 64.66\\nRNase P 10 48.47 618.52 –5.60 613.95 –13.37 633.04 4.48 620.35\\nSRP RNA 10 76.20 613.20 –1.93 69.48 –0.76 614.97 4.00 65.73\\ntmRNA 10 36.16 619.06 31.50 681.06 24.06 678.30 42.64 685.71\\ntRNA 10 64.16 617.55 –0.25 63.39 9.76 631.90 42.83 638.90\\nTotal 81 51.34 621.29 3.54 633.73 4.53 639.81 21.74 650.24\\naSensitivity = (number of base pairs in common between structure determined by comparative sequence analysis and predicted structure)/\\n(number of base pairs in structure determined by comparative sequence analysis) 3100%.', 'bSensitivity improvement by a centroid with respect to the MFE structure = [(sensitivity of centroid)/(sensitivity of MFE structure) – 1] 3\\n100%. 1160 RNA, Vol. 11, No. 8Ding et al. Computational costs and availability\\nThe main memory requirement for the clustering proce-\\ndure is the storage of the distance matrix. The computation\\nof the centroid is a linear operation. The CPU times and\\nmemory requirements for our version of partition function\\ncalculation for sampling 1000 structures and for clustering\\nand centroid calculation are given in Table 6 for several\\nsequences of various lengths. Clustering features including\\ncentroids are available through the module Srna of the\\nSfold software for folding and design of nucleic acids. Sfold is available through Web servers at http://sfold. wadsworth.org and http://www .bioinfo.rpi.edu/applica\\ntions/sfold. Sample output for a folded sequence is avail-\\nable at http://sfold.wadsworth.org/demo.', 'DISCUSSION\\nOur main finding is that ensemble centroids yield more specific\\npredictions with average improvements of 30.0% in PPV and\\n3.5% in sensitivity. More strikingly, the best of a small number\\nof cluster centroids improves the PPV by 46.5% while simulta-\\nneously increasing the sensitivity by > 20%. Perhaps our most\\nprovocative finding is that the MFE structure falls outside the\\ncluster containing the best centr oid for over half of the studied\\nsequences. In such cases, > 31% more base pairs are correctly\\nidentified with > 62% fewer predictive errors (Table 5). The consistent finding of improved PPV suggests that the\\nMFE structure may tend to overpredict. It has been argued\\nthat the structure determined by comparative sequence anal-\\nysis is a minimal model for RNA secondary structure, because\\nonly base pairs for which comparative evidence exists are\\nincluded in the structure model (Larsen and Zwieb 1991).', 'This raises the possibility that overprediction by MFE struc-\\nture is in part due to underrepresentation of base pairs in\\nthe structure determined by comparative sequence analysis.However, recent comparisons of structure determined by\\ncomparative sequence analysis with crystal structures indicate\\nthat covariation analysis for 16S and 23S rRNAs identifies\\nnearly all base pairings (Cannone et al. 2002). For 16S and\\n23S rRNAs, the improvements by the centroids are substan-\\ntial (Tables 2–4), and thus cannot be attributed to potential\\nunderrepresentation of base pa irs in the structure determined\\nby comparative sequence analysis. For other types of RNAs,\\ncomprehensive data for comparing crystal structures with\\nstructure determined by compar ative sequence analysis are\\nneeded for making a more general assessment. To a large degree, the ensemble centroid is reflective of\\nthe high-frequency base pairs in the structure sample.', 'Because the base-pair frequencies are sampling estimates\\nof the base-pair probabilities computed by partition func-\\ntions (McCaskill 1990), the finding of improved PPV by\\nthe ensemble centroid is consistent with the recent report\\nthat base pairs in MFE structure that have high probabil-\\nities have a significantly higher PPV than that of base pairs\\nwith lower probabilities (Mathews 2004). TABLE 5. Improvement by the best centroid with respect to the\\nlocation of the MFE structure\\nImprovement by best centroid over MFE\\nstructure (%)\\nPerformance\\nmeasureMFE structure\\nin the cluster\\nof the best centroid\\n(34 sequences)MFE structure\\noutside the cluster\\nof the best centroid\\n(47 sequences)\\nBase-pair distance 14.62 614.28 36.51 623.79\\nSensitivity 7.03 643.20 31.44 652.87\\nPositive predictive\\nvalue (PPV) 23.77 652.49 62.52 666.87TABLE 4.', 'Positive predictive values (PPV) for MFE structure and PPV improvement by centroids over MFE structure\\nNumber of PPVaAverage percentage of improvement in PPV with respect to MFE structureb\\nRNA type sequences of MFE structure Ensemble centroid Largest cluster centroid Best cluster centroid\\nSSU (16S) rRNA 10 48.10 614.37 14.08 610.67 9.54 611.17 28.31 622.97\\nLSU (23S) rRNA 10 33.68 613.80 43.86 652.10 36.84 647.51 46.37 643.28\\n5S rRNA 10 59.78 625.76 26.52 652.76 21.60 657.60 51.42 688.62\\nGroup I intron 9 37.95 620.81 36.37 644.36 20.04 632.99 56.50 657.24\\nGroup II intron 2 29.31 626.14 33.33 64.37 31.55 61.26 32.84 63.09\\nRNase P 10 42.89 616.49 14.16 618.84 –3.86 637.00 27.99 627.41\\nSRP RNA 10 77.59 615.83 1.69 613.23 –0.48 615.64 9.63 612.99\\ntmRNA 10 30.28 619.83 76.78 6116.97 39.57 691.33 95.55 6122.81\\ntRNA 10 55.15 620.34 26.49 636.50 15.29 630.62 60.07 644.22\\nTotal 81 47.84 623.28 30.00 655.19 17.64 646.87 46.51 664.02\\naPositive predictive value (PPV) = (number of common base pairs between structure determined by comparative sequence analysis and predicted\\nstructure)/(number of base pairs in predicted structure) 3100%.', 'bPPV improvement by a centroid with respect to the MFE structure = [(PPV of centroid)/(PPV of MFE structure) /C01]3100%. www.rnajournal.org 1161Secondary structure prediction by centroids\\nAll of the ensemble centroids in our analysis are based on\\nsamples of structures. However, we could also use the base-\\npair probabilities calculated from partition functions\\n(McCaskill 1990) for this purpose. Because sample base-\\npair frequencies used for centroid calculation approach the\\nbase-pair probabilities as the sample size increases, our\\nsample-based-centroid will approach the partition-func-\\ntion-based centroid. However, because base-pair probabil-\\nities give only the marginal probabilities of individual base\\npairs, the identification of clusters of similar structures\\nbased on base-pair probabilities alone is at best difficult.', 'In contrast, because sampled structures are real-\\nizations from the joint high-dimensional distribution of\\nall base pairs (Ding and Lawrence 2003), clustering is\\ngreatly facilitated. Accordingly, a statistical sample enables\\nthe decomposition of the two-dimensional histogram of\\nbase pairs into subhistograms of distinct structural clusters\\n(Ding and Lawrence 2003).Although the best centroids are the best predictors, these\\ncentroids cannot be defined when a reference structure is\\nunavailable. However, it is an appealing feature that the best\\ncentroid predictions are based on only three to four clusters, on\\naverage. The small number of cluster centroid predictions can\\nfacilitate further structura l determination by allowing the\\nincorporation of other types of information, e.g., partial struc-\\nture information from enzymatic or chemical probing.', 'In\\norder that our comparison be as direct and clear-cut as possi-\\nble, all predicted structures in this analysis are based on the\\nsame set of energy rules (Xia et al. 1998; Mathews et al. 1999). We have not compared these approaches using recently revised\\nenergy rules (Mathews et al. 2004). Comparisons incorporat-\\ning constraints (e.g., for forcing modified bases in tRNAs to be\\nunpaired or for the incorporation of other partial structure\\ninformation) and coaxial stacking also await further study. However, we currently see no reason why the advantages of\\nthese sample-based predictions should not extend to other\\nFig 1. live 4/c\\nFIGURE 1. The energy landscape of the sampled ensemble and representative structures for Agrobacterium tumefaciens 5S rRNA (GenBank\\naccession number X02627) of 120 nt.', 'The structure determined by comparative sequence analysis is in the larger (blue color) cluster with a\\nprobability of 0.591 and the MFE structure is in the smaller cluster (purple color) with a probability of 0.409. The coordinates for a structure is\\n(axis 1, axis 2, energy), where the horizontal axes are from multidimensional scaling (MDS; Kruskal and Wish 1977) for presenting high-\\ndimensional objects in typically two dimensions, and the vertical axis is the free energy of a secondary structure. The base-pair distances between\\nstructures (see Materials and Methods section) are used for MDS. The coordinates are (21.50, /C05.73, /C046.80) for the structure determined by\\ncomparative sequence analysis, ( /C027.92, /C00.45, /C050.50) for the MFE structure, (6.55, 3.15, /C036.40) for the ensemble centroid, (20.14, /C02.88,\\n/C045.80) for the larger cluster centroid, and ( /C025.95, /C00.34, /C050.50) for the smaller cluster centroid. 1162 RNA, Vol. 11, No. 8Ding et al. cases.', 'We also expect that the use of experimental constraints\\nmay improve the predictions, as demonstrated for predictions\\nbased on free energy minimizatio n (Mathews et al. 1999, 2004).We have examined the constrained MFE structure, using\\nbase pairs in the ensemble centroid as the constraints. In com-\\nparison with the ensemble centroid, we found that on average\\nFIGURE 2. Distribution of the improvement percentage by the best centroid for base pair distance ( A), sensitivity ( B), and PPV ( C). The best\\ncentroid is the cluster centroid with the shortest base-pair distance to the structure determined by comparative sequence analysis. TABLE 6.', 'Computational costs of clustering and centroid identification in comparison to those for computing partition functions (PFs) and\\nsampling 1000 structures\\nCPU timea(seconds) Memory usage (MB)\\nRNA sequenceb\\n(GenBank accession no.) Length (nt) PFs SamplingClustering and\\ncentroid computation PFs + SamplingClustering and\\ncentroid computation\\ntRNA (M91245) 70 0.06 1.74 9.92 2.65 65.51\\n5S rRNA (X02627) 120 0.33 3.84 9.73 2.89 65.82\\nRNase P RNA (X97396) 359 6.54 11.26 11.81 5.62 67.12\\nGroup I intron (U02540) 556 25.57 22.00 12.66 10.00 68.82\\nSmall subunit rRNA (U04948) 1532 527.06 160.09 21.31 57.34 77.04\\nLarge subunit rRNA (X12612) 2915 4082.19 430.21 37.22 198.75 95.79\\naBenchmarked on an AMD Opteron 1.8-GHz processor under the Linux operating system. bThe name of the organism of the sequence can be found in Table 1.', 'www.rnajournal.org 1163Secondary structure prediction by centroids\\nthe sensitivity is improved by 2.22% for the constrained\\nMFE structure as a result of more predicted base pairs;\\nhowever, this small improvement has a cost of 7.05% average\\ndecrease in PPV. It remains an open question whether the\\ncombination of centroid prediction with other approaches\\ncan further improve structure prediction. The base pair fre-\\nquencies for the entire structure sample as well as for individual\\nclusters might be used as weights by the maximal weighted\\nmatching or the iterated loop matching method (Tabaska\\net al. 1998; Ruan et al. 2004) for calculating representative\\nstructures with pseudoknots.', 'As alternatives to examining clusters and centroids for\\nsampled structures, one might consider clustering the 1000\\nstructures with the lowest energies computed by RNAsub-\\nopt of the Vienna RNA package (Hofacker 2003) or the\\nabstract shape representation of foldings within an energy\\nincrement from the MFE (Giegerich et al. 2004). These two\\nmethods focus on the lowest end of the free energy density\\nof states, whereas structure sampling allows characteriza-\\ntion of Boltzmann-weighted density of states (Ding and\\nLawrence 2003). Thus, the energy landscape is examined\\nfrom two different perspectives. For short sequences such\\nas tRNAs, there is generally a good correspondence be-\\ntween our centroids and the abstract shapes or centroids\\nfor 1000 best structures (data not shown), as the low energy\\nstructures are well represented in a sample.', 'However, the\\ndegree of correspondence and the overlap in the energy\\ncoverage diminishes as sequence length increases, because\\napparently the Boltzmann-weighted density of states be-\\ncomes increasingly dictated by structures at an energy dis-\\ntance from the MFE, and these structures far outnumber\\nthose with energies near the MFE. For example, for the\\nrabbit b-globin mRNA of 589 nt (GenBank accession no. V00879), the 1000 best structures represent a small free\\nenergy range of 1.3 kcal/mol for default parameter settings\\nof RNAsubopt, while a statistical sample presents represen-\\ntative structures from a much wider energy interval of\\n39.80 kcal/mol. In addition, a statistical sample can reveal\\n‘‘entropic clusters’’ (Ding and Lawrence 2003). For an\\nentropic cluster, each member has a probability too small\\nto command individual attention, but collectively the clus-\\nter has an appreciable probability because of the large\\nnumber of cluster members.', 'In computer RNA folding\\napplications, it is a common practice among users to\\nexamine structures from mfold. Because the structure sam-\\nple from mfold is heuristic rather than statistically or low-\\nenergy representative, the method presented here and the\\nRNAshapes approach present two improved and comple-\\nmentary alternatives. The two methods are also comple-\\nmentary because RNAshapes provides an alternative\\nmethod for the identification of structure clusters with\\ncluster members having a common shape. It will be inter-\\nesting to apply the RNAshapes algorithm to a sampled\\nensemble. The larger number of shape representatives\\nthan the number of our clusters suggests that our clusteringprocedure reports major clusters whereas the abstract\\nshape approach may reveal more subtle structural dissim-\\nilarities. As pointed out by Abagyan (1993), two major compo-\\nnents are needed to solve macromolecular folding prob-\\nlems.', 'First, all essential terms of the free energy of a trial\\nconformation must be calculated with sufficient accuracy,\\nand, second, a procedure is needed to find the minimum of\\nthis energy function. For RNAs, the global minimum can\\nbe found for an incomplete energy model, i.e., only the\\nsecondary structure model. Thus, it may not be a surprise\\nthat, for a majority of analyzed sequences (47 of 81\\nsequences), a centroid of an alternative cluster (probably\\nrepresenting an alternative energy well) is about 37% closer\\nto the structure determined by comparative sequence anal-\\nysis than the MFE structure computed with the incomplete\\nenergy function (Table 5). As argued by Abagyan, since\\nonly a small number of such alternative structural classes\\nare needed to find this best alternative, an approach that\\nemploys a post-analysis filter function maybe a productive\\npath for selecting among clusters for improved structure\\nprediction.', 'Since for protein structural models neither of\\nAbagyan’s two components is attainable, our findings argue\\nthat a more comprehensive examination of the energy\\nlandscape of the approximate models for structures of\\nproteins and other macromolecules may also be worthy of\\nfurther investigation. Even in the case of the complete\\nmodel and energy function, the Boltzmann ensemble view\\nis also important, e.g., for the investigation of metastable\\nstates, particularly RNA conformational switches (Barrick\\net al. 2004; Voss et al. 2004). MATERIALS AND METHODS\\nRNA sequences\\nFrom publicly available databases (Larsen and Zwieb 1991; Sprinzl\\net al. 1998; Brown 1999; Cannone et al. 2002; Alm Rosenblad et al. 2003; Zwieb et al. 2003), we took samples of sequences for diverse\\ntypes of structural RNAs with secondary structures determined by\\ncomparative sequence analysis.', 'For tRNAs, RNase P RNAs,\\ntmRNAs, signal recognition particle (SRP) RNAs, small subunit\\n(16S or 16S-like) rRNAs, large subunit (23S or 23S-like) rRNAs,\\nand 5S rRNAs, 10 sequences were randomly selected for each RNA\\ntype. In addition, nine group I introns without undetermined\\nnucleotides and two group II introns that are available in the\\ndatabases were also included in our analysis. The list of the 81\\nsequences is available from the authors upon request and will also\\nbe posted on the Sfold Web server. Clustering procedure\\nFor comparing two secondary structures, we use the base-pair\\ndistance. The discriminatory power of this distance is adequate\\nin our context, because we are interested in comparing multiple\\n1164 RNA, Vol. 11, No. 8Ding et al. structures sampled for a given RNA sequence. In other circum-\\nstances, e.g., when structures of homologous sequences are com-\\npared, alternative metrics (Moulton et al. 2000) may be more\\nappropriate to account for insertions and deletions.', 'For hierarchi-\\ncal clustering of structures, we employ Diana (Kaufman and\\nRousseeuw 1990), a top-down divisive method that has per-\\nformed well in other contexts (Datta and Datta 2003). For determin-\\ning the number of clusters, we use the CH index (Calinski and\\nHarabasz 1974) that was assessed as the best in a comprehensive\\nstudy (Milligan and Cooper 1985) . For a given divisive level on the\\nclustering tree from Diana , the CH index is calculated as\\nCH(k)=[B(k)/(k/C01)]/[W(k)/(ntotal/C0k)], where kis the number of\\nclusters, ntotalis the total number of objects to be clustered, B(k)i st h e\\nbetween-cluster sums of squares, and W(k) is the within-cluster sums\\nof squares. The number of clusters at which the CH index is maximized\\nis the optimal number of clusters. This number is then used to\\ndetermine the structural clusters b y identifying the corresponding\\ndivisive level for the hierarchical clustering tree produced by Diana .', 'As described below, a secondary structure is an object in a high-\\ndimensional Euclidean space that can be expressed by an upper trian-\\ngular matrix. The cluster means required for the calculation of the sum\\nof squares can be computed in two way s: averaging the corresponding\\nEuclidean coordinates for all st ructures in a cluster or using the\\ncentroid of a cluster as its mean and computing the base-pair distance\\nbetween the centroid and any struc ture in the cluster (see below for\\ncentroid calculation). The former is c omputationally intensive for long\\nsequences whereas the later is a linear operation. We did not observe\\nappreciable differences in the clustering results by the two methods. Therefore, we decided to use cluste r centroids in the implementation\\nof the CH index. For every RNA sequence, we first cluster 1000 statistically\\nsampled structures. We compute the MFE structure with mfold\\n3.1 (Zuker 2003) for the same set of Turner thermodynamic\\nparameters (Xia et al.', '1998; Mathews et al. 1999) that are currently\\nimplemented by our sampling algorithm (Ding and Lawrence\\n2003). To decide which cluster the MFE structure belongs to, we\\nfirst identify the cluster whose centroid has the shortest base-pair\\ndistance to the MFE structure. If this distance is less than or equal\\nto the longest base-pair distance between a structure in the cluster\\nand the cluster centroid, the MFE structure belongs to this cluster;\\notherwise, the MFE structure does not belong to any cluster in the\\nsample, i.e., it is in a new cluster by itself. The structure sample\\nsize of 1000 has been shown to be sufficiently large to guarantee\\nstatistical reproducibility in typical sampling statistics including\\nbase-pair frequencies, even when two independent samples do\\nnot share a single structure (Ding and Lawrence 2003). As\\nreported below, base pair frequencies are all that are needed for\\ncentroid identification.', 'In addition, regardless of sequence length,\\na cluster with an appreciable probability is expected to be repre-\\nsented in a sample of 1000 structures. Larger samples would\\nreveal additional clusters that are insignificant at a significance\\nlevel of 0.001. Matrix representation of RNA secondary structure\\nFor an RNA sequence of nnucleotides, a secondary structure Ican\\nbe expressed by an upper triangular matrix of indicators { Iij},\\n1/C20i<j/C20n, where Iijindicates base-pairing status between base\\niand base j. Specifically, Iij= 1 if the ith base is paired with the jthbase or Iij= 0 otherwise. The requirement of at least three unpaired\\nintervening bases between any base pair implies Iij=0 f o r j=i+1 ,\\ni+2 , a n d i+3 , 1 /C20i,i+3/C20n. The indicators are not independent\\nof one another, because they are subject to constraints. The\\nassumption of no pseudoknots implies IijIi0j0= 0 for i0<i<j0<j.', 'A l s o ,w h e nb a s et r i p l e sa r ep r o h i b i t e d , S1/C20i/C20nIij/C201, and\\nS1/C20j/C20nIij/C201. Base-pair distance\\nWhile the base-pair indicators are binary and under constraints,\\nthey are also coordinates in a Euclidean space of dimension\\n(n/C01)n/2. For two structures I1={Iij1} and I2={Iij2}, consider\\nthe following metric D1and the squared Euclidean distance D2:\\nD1(I1,I2)=P\\n1/C20i<j/C20n|Iij1/C0Iij2|a n d D2(I1,I2)=P\\n1/C20i<j/C20n(Iij1/C0Iij2)2. In general, both metrics have sufficient discriminatory power for\\nthe purpose of clustering. In our context, both metrics are equal\\nto the number of different base pairs in I1andI2. In other words,\\nboth of the metrics are in fact the well-known base-pair distance\\nD(I1,I2). This interpretation does not apply to the square root of\\nD2, i.e., the Euclidean distance.', 'Definition and derivation of centroid\\nFor a set of msecondary structures I1,I2,..., Im, with Ik={Iijk},\\n1/C20k/C20m, the centroid for the set is defined as the structure in the\\nentire ensemble of secondary structures that has the shortest total\\nbase-pair distance to the structures in the set. To compute the\\ncentroid structure, we need to find the secondary structure\\nI={Iij} that minimizes the following sum of distances under the\\nconstraints discussed above:\\nS1/C20k/C20mSiSjIijk/C0Iij/C0/C1 2\\n¼S1/C20k/C20mSiSjIijk/C0/C1 2/C02IijIijkþIij/C0/C12hi\\n¼S1/C20k/C20mSiSjIijk/C0/C1 2/C02SiSjS1/C20k/C20mIijk/C0/C1\\nIijþmSiSjIij\\n¼CsþSiSjm/C02Cij/C0/C1\\nIijð1Þ\\nwhere Cs=S1/C20k/C20mSiSj(Iijk)2is a constant for the given structure\\nset, and Cij=S1/C20k/C20mIijkis the total number of occurrences of base\\npair i/C1jin the structure set. Because Iij2/C17Iij, the nonlinear pro-\\ngramming problem is in fact a linear programming problem with\\nnonlinear constraints.', 'For a base pair with a frequency under\\n50%, it cannot be in the centroid because ( m/C02Cij) > 0, and\\nthus Iijmust be 0 for the centroid. A base pair with a frequency\\nof 50% does not influence the double sum in (1), because\\n(m/C02Cij) = 0. For a base pair with a frequency > 50%, because\\n(m/C02Cij) < 0, the inclusion of this base pair (i.e., Iij= 1) decreases\\nthe double sum in (1). Furthermore, any two base pairs with\\nfrequencies > 50% do not form a pseudoknot, because no base\\npairs in the structure set are involved in pseudoknots. Thus, the\\nconsensus structure formed by all base pairs with a frequency\\n> 50% is a centroid. We note that for base pairs with a frequency\\nof 50%, inclusion of any compatible combination into the > 50%\\nconsensus does define another centroid. However, the > 50%\\nconsensus structure is always the unique centroid with the smal-\\nlest number of base pairs and is the one we use for analysis.', 'The centroid is referred to as the ensemble centroid when the\\nstructure set is the statistical sample generated by our sampling\\nalgorithm, typically with m= 1000. In this case, Cijis the observed\\nwww.rnajournal.org 1165Secondary structure prediction by centroids\\ncount for base pair i/C1jin the sample. For a cluster of similar\\nstructures identified from the statistical sample, the centroid is\\nreferred to as a cluster centroid. In this case, Cijis the observed\\ncount for base pair i/C1jin the cluster. ACKNOWLEDGMENTS\\nThe Computational Molecular Biology and Statistics Core at the\\nWadsworth Center is acknowledged for providing computing\\nresources for this work. This work was supported in part by\\nNational Science Foundation grant DMS-0200970 and National\\nInstitutes of Health grant GM068726 to Y.D. and by National\\nInstitutes of Health grant HG01257 to C.E.L.', 'We are grateful to\\nthe suggestions and observations of the anonymous referees that\\nled to drastic improvements in the run times of clustering and\\ncentroid identification as well as the presentation of the article. Received March 7, 2005; accepted May 9, 2005. REFERENCES\\nAbagyan, R.A. 1993. Towards protein folding by global energy opti-\\nmization. FEBS Lett. 325: 17–22. Alm Rosenblad, M., Gorodkin, J., Knudsen, B., Zwieb, C., and\\nSamuelsson, T. 2003. SRPDB (Signal Recognition Particle Data-\\nbase). Nucleic Acids Res. 31:363–364. Anfinsen, C.B. 1973. Principles that govern the folding of protein\\nchains. Science 181: 223–230. Barrick, J.E., Corbino, K.A., Winkler, W.C., Nahvi, A., Mandal, M.,\\nCollins, J., Lee, M., Roth, A., Sudarsan, N., Jona, I., et al. 2004. New RNA motifs suggest an expanded scope for riboswitches in\\nbacterial genetic control. Proc. Natl. Acad. Sci. 101: 6421–6426. Bonhoeffer, S., McCaskill, J.S., Stadler, P.F., and Schuster, P. 1993. RNA multi-structure landscapes.', 'A study based on temperature\\ndependent partition functions. Eur. Biophys. J. 22:13–24. Brown, J.W. 1999. The ribonuclease P database. Nucleic Acids Res. 27:314. Calinski, R.B. and Harabasz, J. 1974. A dendrite method for cluster\\nanalysis. Comm. Stat. 3:1–27. Cannone, J.J., Subramanian, S., Schnare, M.N., Collett, J.R., D’Souza,\\nL.M., Du, Y., Feng, B., Lin, N., Madabusi, L.V., Muller, K.M., et al. 2002. The comparative RNA Web (CRW) site: An online database\\nof comparative sequence and structure information for ribosomal,\\nintron, and other RNAs. BMC Bioinform. 3:2. Datta, S. and Datta, S. 2003. Comparisons and validation of statistical\\nclustering techniques for microarray gene expression data. Bioin-\\nformatics 19:459–466. Ding, Y. and Lawrence, C.E. 2001. Statistical prediction of single-\\nstranded regions in RNA secondary structure and application to\\npredicting effective antisense target sites and beyond. Nucleic Acids\\nRes.29:1034–1046. ———. 2003.', 'A statistical sampling algorithm for RNA secondary\\nstructure prediction. Nucleic Acids Res. 31:7280–7301. Dowell, R.D. and Eddy, S.R. 2004. Evaluation of several lightweight\\nstochastic context-free grammars for RNA secondary structure\\nprediction. BMC Bioinform. 5:71. Giegerich, R., Voss, B., and Rehmsmeier, M. 2004. Abstract shapes of\\nRNA. Nucleic Acids Res. 32:4843–4851.Hofacker, I.L. 2003. Vienna RNA secondary structure server. Nucleic\\nAcids Res. 31:3429–3431. Kaufman, L. and Rousseeuw, P.J. 1990. Finding groups in data: An\\nintroduction to cluster analysis. John Wiley & Sons, New York. Kruskal, J.B. and Wish, M. 1977. Multidimensional scaling. Sage Pub-\\nlications, Beverly Hills, CA. Larsen, N. and Zwieb, C. 1991. SRP-RNA sequence alignment and\\nsecondary structure. Nucleic Acids Res. 19:209–215. Mathews, D.H. 2004. Using an RNA secondary structure partition\\nfunction to determine confidence in base pairs predicted by free\\nenergy minimization. RNA 10:1178–1190.', 'Mathews, D.H., Sabina, J., Zuker, M., and Turner, D.H. 1999. Expanded sequence dependence of thermodynamic parameters\\nimproves prediction of RNA secondary structure. J. Mol. Biol. 288: 911–940. Mathews, D.H., Disney, M.D., Childs, J.L., Schroeder, S.J., Zuker,\\nM., and Turner, D.H. 2004. Incorporating chemical modification\\nconstraints into a dynamic programming algorithm for predic-\\ntion of RNA secondary structure. Proc. Natl. Acad. Sci. 101:\\n7287–7292. McCaskill, J.S. 1990. The equilibrium partition function and base pair\\nbinding probabilities for RNA secondary structure. Biopolymers\\n29:1105–1119. Milligan, G.W. and Cooper, M.C. 1985. An examination of proce-\\ndures for determining the number of clusters in a data set. Psy-\\nchometrika 50:159–179. Moulton, V., Zuker, M., Steel, M., Pointon, R., and Penny, D. 2000. Metrics on RNA secondary structures. J. Comput. Biol. 7:277–292. Pappu, R.V., Marshall, G.R., and Ponder, J.W. 1999.', 'A potential\\nsmoothing algorithm accurately predicts transmembrane helix\\npacking. Nat. Struct. Biol. 6:50–55. Ruan, J., Stormo, G.D., and Zhang, W. 2004. An iterated loop match-\\ning approach to the prediction of RNA secondary structures with\\npseudoknots. Bioinformatics 20:58–66. Sprinzl, M., Horn, C., Brown, M., Ioudovitch, A., and Steinberg, S. 1998. Compilation of tRNA sequences and sequences of tRNA\\ngenes. Nucleic Acids Res. 26:148–153. Tabaska, J.E., Cary, R.B., Gabow, H.N., and Stormo, G.D. 1998. An\\nRNA folding method capable of identifying pseudoknots and base\\ntriples. Bioinformatics 14:691–699. Voss, B., Meyer, C., and Giegerich, R. 2004. Evaluating the predict-\\nability of conformational switching in RNA. Bioinformatics 20:\\n1573–1582. Wuchty, S., Fontana, W., Hofacker, I.L., and Schuster, P. 1999. Complete suboptimal folding of RNA and the stability of second-\\nary structures. Biopolymers 49:145–165.', 'Xia, T., SantaLucia Jr., J., Burkard, M.E., Kierzek, R., Schroeder, S.J.,\\nJiao, X., Cox, C., and Turner, D.H. 1998. Thermodynamic param-\\neters for an expanded nearest-neighbor model for formation of\\nRNA duplexes with Watson–Crick base pairs. Biochemistry 37:\\n14719–14735. Zuker, M. 1989. On finding all suboptimal foldings of an RNA\\nmolecule. Science 244: 48–52. ———. 2003. Mfold Web server for nucleic acid folding and hybri-\\ndization prediction. Nucleic Acids Res. 31:3406–3415. Zuker, M. and Stiegler, P. 1981. Optimal computer folding of large\\nRNA sequences using thermodynamics and auxiliary information. Nucleic Acids Res. 9:133–148. Z w i e b ,C . ,G o r o d k i n ,J . ,K n u d s e n ,B . ,B u r k s ,J . ,a n dW o w e r ,J . 2003. tmRDB (tmRNA database). Nucleic Acids Res. 31:\\n446–447. 1166 RNA, Vol. 11, No. 8Ding et al. BIOINFORMATICS\\nEffect of target secondary structure on RNAi efficiency\\nYU SHAO,1CHI YU CHAN,1ANIL MALIYEKKEL,2,3CHARLES E. LAWRENCE,1,4\\nIGOR B.', 'RONINSON,2and YE DING1\\n1Wadsworth Center, New York State Department of Health, Albany, New York 12208, USA\\n2Cancer Center, Ordway Research Institute, Albany, New York 12208, USA\\n3Department of Molecular Genetics, University of Illinois at Chicago, Chicago, IL 60607, USA\\n4Center for Computational Molecular Biology, and Division of Applied Mathematics, Brown University, Providence, RI 02912, USA\\nABSTRACT\\nRNA interference (RNAi) mediated by small interfering RNAs (siRNAs) or short hairpin RNAs (shRNAs) has become a powerful\\ntool for gene knockdown studies. However, the levels of knockdown vary greatly. Here, we examine the effect of target\\ndisruption energy, a novel measure of target accessibility, along with other parameters that may affect RNAi efficiency. Based\\non target secondary structures predicted by the Sfold program, the target disruption energy represents the free energy cost forlocal alteration of the target structure to allow target binding by the siRNA guide strand.', 'In analyses of 100 siRNAs and 101\\nshRNAs targeted to 103 endogenous human genes, we find that the disruption energy is an important determinant of RNAi\\nactivity and the asymmetry of siRNA duplex asymmetry is important for facilitating the assembly of the RNA-induced silencingcomplex (RISC). We estimate that target accessibility and duplex asymmetry can improve the target knockdown level\\nsignificantly by nearly 40% and 26%, respectively. In the RNAi pathway, RISC assembly precedes target binding by the siRNA\\nguide strand. Thus, our findings suggest that duplex asymmetry has significant upstream effect on RISC assembly and targetaccessibility has strong downstream effect on target recognition. The results of the analyses suggest criteria for improving the\\ndesign of siRNAs and shRNAs.', 'Keywords: target structure; RNA folding; RNAi\\nINTRODUCTION\\nRNA interference (RNAi) is a sequence-specific gene\\nsilencing mechanism that is induced by double-stranded\\nRNA (dsRNA) homologous to the target gene (Fire et al. 1998). RNAi can be mediated either by small interferingRNAs (siRNAs) of about 21 nucleotides (nt) with two-nucleotide 3 9overhang (Elbashir et al. 2001) or by stably\\nexpressed short hairpin RNAs (shRNAs), which are pro-cessed by Dicer into siRNAs (Brummelkamp et al. 2002;Paddison et al. 2002). During activation of the RNA-induced silencing complex (RISC), the guide (antisense)\\nstrand of the siRNA duplex is preferentially assembled into\\nthe RISC when the stem formed by the 5 9end and its\\ncomplement is less stable than the one formed by the 3 9end\\nand its complement (Khvorova et al. 2003; Schwarz et al. 2003); the ‘‘passenger’’ (sense) strand is cleaved by Argo-naute2 (Ago2), the catalytic component of RISC (Matranga\\net al. 2005; Rand et al. 2005).', 'The antisense strand guidesAgo2 to cleave mRNA by base-pairing with the comple-mentary site in the target. Large variation in the efficiency of siRNAs for different\\nsites on the same target is commonly observed (Holen et al. 2002). Usually, only a small proportion of randomly\\nselected siRNAs are potent. Thus, there has been great\\ninterest in determining rules for improvement of RNAidesign. A number of empirical rules on siRNA duplexfeatures have been reported. These include the asymmetryrule for siRNA duplex ends, which requires that the 5 9end\\nof the antisense strand forms a stem with its complementthat is less stable than the stem formed by the 5 9end of the\\nsense strand (Khvorova et al. 2003; Schwarz et al. 2003). The asymmetry rule is strongly related to the requirements\\nof high A/U content at the 5 9end of the antisense strand\\nand high G/C at the 5 9end of the sense strand (Reynolds\\net al. 2004; Ui-Tei et al. 2004).', 'A number of position-specific nucleotide preferences and other siRNA sequencefeatures have been proposed (Reynolds et al. 2004; Patzelet al. 2005). In addition, the importance of target secondary\\nReprint requests to: Ye Ding, Wadsworth Center, New York State\\nDepartment of Health, 150 New Scotland Avenue, Albany, NY 12208,\\nUSA; e-mail: yding@wadsworth.org; fax: (518) 402 4623. Article published online ahead of print. Article and publication date are\\nat http://www.rnajournal.org/cgi/doi/10.1261/rna.546207. RNA (2007), 13:1631–1640. Published by Cold Spring Harbor Laboratory Press. Copyright /C2112007 RNA Society. 1631 on October 9, 2007 www.rnajournal.org Downloaded from \\nstructure and accessibility has been suggested by several\\nstudies based on computational modeling of target struc-\\nture and accessibility (Kretschmer-Kazemi Far and Sczakiel\\n2003; Luo and Chang 2004; Heale et al. 2005; Schubert et al.', '2005) and was supported by compelling evidence based on\\nexperimentally assessed accessibility (Lee et al. 2002; Bohula\\net al. 2003; Vickers et al. 2003; Overhoff et al. 2005;\\nWesterhout et al. 2005). Strikingly, it was observed that\\nHIV can escape RNAi-mediated inhibition by a single point\\nmutation that alters the accessibility of the target site\\n(Westerhout et al. 2005). The significance of target struc-\\nture has long been established for antisense oligonucleo-\\ntides and trans -cleaving ribozymes (Zhao and Lemke 1998;\\nVickers et al. 2000). For RNAi, however, this has been\\ndisputed in several reports, with one based on limited\\ncomputational analysis (Reynolds et al. 2004; Boese et al. 2005). Rules for siRNA duplex features are straightforward to\\nquantify and implement for the purpose of rational RNAi\\ndesign.', 'However, since a messenger RNA (mRNA) is\\nunlikely to have a single stable structure, computational\\nmodeling of the target secondary structure and assessment\\nof the effect of secondary structure on target accessibility\\nare much more challenging. To address this challenge, we\\nintroduce a novel quantitativ e measure of target accessi-\\nbility, target disruption energy, based on structures\\npredicted by the Sfold program, which generates a\\nstatistically representative sample from the Boltzmann\\nweighted ensemble of secondary structures (Ding and\\nLawrence 2003; Ding et al. 2004). We employ this\\napproach with three aims: (1) quantify target structural\\naccessibility, (2) quantitative ly assess the net contribution\\nof the target accessibility to RN Ai efficiency in the context\\nof the RNAi pathway, and (3) establish a general model\\nfor efficient RNAi. We examine target disruption energy\\nalong with a number of other parameters that can affect\\nRNAi efficiency.', 'From an analysis of 100 published\\nsiRNAs for three endogenous human genes, we found\\nthat disruption energy is the most significant parameter\\nfor one target and is second in signifi-\\ncance level to duplex asymmetry for\\nthe other two targets. To quantitatively\\nassess the effects of target accessibility\\nand duplex asymmetry, we utilize an\\nindependent data set of 101 shRNAs\\nfor 100 endogenous human genes. We\\nfound that target accessibility and\\nduplex asymmetry can improve the\\ntarget knockdown level significantly by\\nnearly 40% and 26%, respectively. These findings suggest that, after RISC\\nassembly, target secondary structure\\nplays an important role in target bind-\\ning by the guide siRNA strand. Thus,\\neffective silencing by RNAi favors siR-NAs with sequence features that facilitate RISC activation,\\nas well as accessible target sites that enable intermolecular\\nbase-pairing for target reco gnition. The results of the\\nanalyses suggest criteria for improving the design of\\nsiRNAs and shRNAs.', 'RESULTS\\nStatistical analyses of siRNA data sets\\nWe first performed weighted regression analyses for the\\nsiRNA data sets (see Materials and Methods and also Fig. 1). For lamin A, DGdisruption is the only significant param-\\neter, with a P-value of 1.05E /C08, and is highly predictive of\\nsiRNA activity with a regression R2of 0.7656 (Fig. 2; Table\\n1,). For PTEN and CD54, DGdisruption is the second best\\npredictor, with a P-value of 1.80E /C016 and a R2of 0.6073\\n(Table 2). DGdisruption is the only parameter that is sig-\\nnificantly correlated with siRNA activities for all data sets. DSSE is the best predictor for PTEN and CD54, with a P-\\nvalue of 5.24E /C036 and a high R2of 0.8849 (Table 2). The lack of significance of DSSE for lamin A could be due\\nto the siRNA design constraints (Harborth et al. 2003);\\nthese could have biased the representation of the nucleotide\\ncomposition for the duplex ends of the tested siRNAs.', 'SDis\\nnot correlated with siRNA activity for any of the data sets\\n(Tables 1 and 2). DGhybrid was found to be significant only\\nfor PTEN and CD54 (Table 2). However, its predictive\\nvalue as measured by R2is relatively poor in comparison\\ntoDGdisruption andDSSE . Thus, in further analyses of effects\\nof target structure on RNAi efficiency, we focused on\\nDGdisruption andDSSE . As alternatives to Sfold, we also computed DGdisruption ,\\nusing target structures predicted by other RNA folding\\nprograms, and performed the weighted regression analysis. We found that Sfold is by far the best performer. For any\\nof the other programs, either there is a lack of statistical\\nsignificance or the R2is rather poor in the case of statistical\\nsignificance (Table 3). FIGURE 1. Energetic exchanges for local target disruption by binding of guide siRNA.', 'Target\\ndisruption energy, DGdisruption , a measure of target accessibility, is the free energy cost\\nfor opening the local secondary structure at the target site; DGhybrid is the free energy gain\\ndue to the hybridization between the guide siRNA strand and the target site.Shao et al. 1632 RNA, Vol. 13, No. 10 on October 9, 2007 www.rnajournal.org Downloaded from \\nQuantitative assessment of the effects of target\\nstructure and duplex asymmetry\\nSeparating the effect of target structure from effects\\nof upstream factors\\nWe were most interested in obtaining a quantitative\\nestimate of the net effect of target structure on RNAi\\nefficiency, by taking advantage of the relatively large\\nindependent shRNA data set. Because target structure is\\nonly relevant for the target recognition step of the RNAi\\npathway, factors that can have negative effects on the\\nupstream steps of the RNAi pathway must be considered\\nin such a quantitative analysis.', 'In other words, the overall\\neffects of upstream factors and target structure are\\nconvoluted in the knockdown data. For example, a siRNA\\nor shRNA targeted to an accessible region will not\\nnecessarily be functional if the guide strand could not be\\nsuccessfully assembled into the RISC. To address this issue\\nof convolution, we adopted the approach of using data\\nfilters to control for the upstream effects of nonstructure\\nfactors. For the shRNAs in the cDNA library, a low or high GC\\ncontent and the occurrence of a AAAA, TTTT, GGGG, or\\nCCCC motif were observed to have negative impacts on\\nRNAi activity. The AAAA motif or the TTTT motif has the\\ntendency to cause the premature termination of transcrip-\\ntion of shRNAs from the RNA polymerase III promoter\\n(Geiduschek and Kassavetis 2001). A GC-rich sequence can\\npromote formation of quadruplex structures (Hardin et al. 1992), and GGGG can form tetraplex structures (Laughlan\\net al.', '1994), or it can cause potential nonspecific effects\\nthrough its interaction with heparin-binding proteins(Stein 1999). Thus, to remove these potential adverse\\neffects, we consider two filters: (1) 30% #GC%#70% and\\n(2) absence of AAAA, TTTT, GGGG, and CCCC motifs. In addition, to separate the downstream effect of target\\nstructure from the upstream effect of duplex asymmetry,\\nwe enforce the rule of asymmetry ( DSSE > 0.0 kcal/mol; see\\nMaterials and Methods) for estimating the effect of target\\nstructure. The siRNAs resulting from shRNA cleavage by Dicer\\nare mostly 19 base pairs (bp) or 20 bp in length (with\\nadditional 2-nt 3 9overhang), at comparable yields (Rose et\\nal. 2005). Because the computational results are highly\\nsimilar for both lengths, we focus on reporting the results\\nfor the length of 19 bp (the guide strand sequences are\\ngiven in Supplemental Table 1).', 'Assessing the net effect of target accessibility\\nFrom weighted regression analysis, we found that\\nDGdisruption is the most important parameter, with a P-\\nvalue of 1.52E /C08. To assess the net effect of target\\naccessibility, we make a two-group comparison between\\naccessible sites and inaccessible sites. Because DGdisruption\\nis a quantitative measure of accessibility and is positively\\ncorrelated with RNAi efficiency, we consider a target site\\naccessible if its DGdisruption >Mkcal/mol, and the site\\ninaccessible if DGdisruption <Nkcal/mol, where MandNare\\ntwo threshold values for defining accessibility and inacces-\\nsibility. For the shRNA data set, we found that the average\\nknockdown level for accessible sites is maximized at M=\\n/C010, and that the average knockdown level for inaccessible\\nsites is minimized at N=/C019.', 'Using these two threshold\\nvalues, the difference in average knockdown levels is 39.7%,\\nand this improvement by accessibility is highly significant\\nwith a P-value of 0.0004 by the t-test and 0.0007 by the\\nnonparametric Wilcoxon rank sum test. For several alter-\\nnative pairs of the thresholds, the difference in the\\nknockdown levels is over 30% (Table 4), and the average\\nimprovement for all five pairs of thresholds is 34.62%. When the rule of asymmetry is not enforced and the two\\nfilters are not applied, the degree of improvement is\\nsubstantially reduced, but still significant with an average\\nof 14.18% (see Supplemental Table 2). These results\\nindicate that the net effect of target structure is substantially\\nunderestimated if upstream factors were not taken into\\nconsideration. FIGURE 2. Weighted linear regression analysis of siRNA knockdown\\ndata for the lamin A siRNA data set using disruption energy, with a\\nregression R2= 0.766 and a P-value of 1.05E /C08.', 'Error bars represent\\nstandard deviations from at least three independent experiments. Larger weights are assigned to data points that have smaller standard\\ndeviations. The fitted regression line is thus dictated by data points\\nwith small standard deviations. TABLE 1. Weighted regression results for lamin A siRNA data set\\nParameter Coefficient P-value R2\\nDGdisruption 0.0156 1.05E /C008 0.7656\\nDGhybrid 0.0023 0.0896 0.1202\\nDSSE /C00.0031 0.5370 0.0168\\nSD /C00.0054 0.7067 0.0063Effect of target structure on RNAi\\nwww.rnajournal.org 1633 on October 9, 2007 www.rnajournal.org Downloaded from \\nAssessing the effect of duplex asymmetry\\nFrom a weighted regression analysis of the shRNA data set,\\nDSSE was found to be significant with a P-value of 0.00096. We next compared the average knockdown level for those\\nshRNAs that meet the rule of asymmetry ( DSSE > 0.0 kcal/\\nmol) and the average for those that do not ( DSSE#0.0\\nkcal/mol).', 'We found that the improvement by enforcing\\nthe rule of asymmetry is 10.21% ( P-value of 0.0197 by the\\none-sided t-test and 0.0284 by the one-sided Wilcoxon\\nrank sum test). For 19 shRNAs that pass the two filters andhave accessible target sites ( DG\\ndisruption >/C010 kcal/mol),\\nthe improvement by duplex asymmetry is 25.99% ( P-value\\nof 0.0082 by the t-test, and 0.0086 by the Wilcoxon rank\\nsum test; also see Table 5). Because the two filters appear\\nto be associated with adverse events upstream of the RISC\\nassembly, 25.99% may be a more accurate estimate of the\\neffect of duplex asymmetry for the shRNA data set. Combined effects of target accessibility\\nand duplex asymmetry\\nTo examine the combined effects of target accessibility and\\nduplex asymmetry, we also assessed the improvement byone parameter under the negative condition specified bythe other parameter (Table 5). The two filters were also\\nused in this assessment to minimize effects of upstreamfactors.', 'For shRNAs that failed the asymmetry test, wefound that target accessibility can still improve the knock-down level by 16.03%. For shRNAs targeted to inaccessiblesites (DG\\ndisruption </C019 kcal/mol), however, duplex asym-\\nmetry did not make an appreciable difference. In the RNAi\\npathway, duplex asymmetry is concerned with the\\nupstream step of RISC assembly, while target accessibility\\npresumably governs target recognition. Our results as\\nsummarized in Table 5 show that duplex asymmetry isnot a rate-limiting factor and that target accessibility is themore influential factor. Furthermore, the combination ofboth duplex asymmetry and accessibility was found to yieldthe highest level of improvement. Software availability\\nDG\\ndisruption ,DSSE , and other tools for structure-based\\nrational RNAi design are available through the application\\nmodule Sirna of the Sfold software for the folding anddesign of nucleic acids. Sfold is available through Webserver at http://sfold.wadsworth.org.', 'DISCUSSION\\nIn this work, we have studied the effects of a number of\\nsiRNA duplex features and the effect of predicted targetsecondary structure on the efficiency of RNAi. We haveintroduced a novel measure of target structural accessibilityDG\\ndisruption ; we found this measure to be the most important\\npredictor for RNAi activity. DSSE , an implementation of the\\nasymmetry rule (Khvorova et al. 2003; Schwarz et al. 2003),\\nwas found to be an important duplex sequence feature. Bytaking advantage of a shRNA data set and by controlling forTABLE 2. Weighted regression results for PTEN and CD54 siRNA\\ndata set\\nParameter Coefficient P-value R2\\nDGdisruption 0.0586 1.80E /C016 0.6073\\nDGhybrid 0.0431 9.49E /C009 0.3651\\nDSSE 0.1393 5.24E /C036 0.8849\\nSD /C00.0530 0.3265 0.0132\\nTABLE 3.', 'Comparison of results of weighted regression for predicting silencing efficiency by DGdisruption computed with target structures\\npredicted by various RNA folding programs\\nMFE structure1000\\nsuboptimal\\nstructures by\\nMfoldc1000\\nlowest energy\\nstructures by\\nRNAsuboptd1000\\nsuboptimal\\nstructures by\\nRNAstructuree1000\\nstructures\\nsampled\\nby Sfoldby\\nMfoldby\\nRNAfoldaby\\nRNAstructureb\\nlamin A R20.0748 0.3324 0.0056 0.0793 0.3322 0.0091 0.7656\\nP-value 0.1860 0.0026 0.7230 0.1730 0.0026 0.6500 1.05E /C008\\nPTEN and\\nCD54R20.0231 0.0908 0.0158 0.0021 0.0673 0.0141 0.6073\\nP-value 0.1930 0.0086 0.2819 0.6977 0.0246 0.3094 1.80E /C016\\naImplementation of free energy minimization by the Vienna RNA package (Hofacker 2003). bImplementation of free energy minimization for Windows platform (Mathews et al. 2004).', 'cA set of 1000 suboptimal structures by Mfold (Zuker 2003) was generated using the following settings for folding parameters: percent\\nsuboptimality P=100, window parameter W=0, and maximum number of foldings MAX =1000 (a much smaller set of structures\\nwas generated for default parameter settings). dComplete suboptimal folding implemented by the Vienna RNA package (Wuchty et al. 1999; Hofacker 2003). eA set of 1000 suboptimal structures by RNAstructure (Mathews et al. 2004) was generated using the following settings for folding parameters:\\nmax % energy difference =100, max number of structures =1000, and window size =0 (a much smaller set of structures was generated for\\ndefault parameter settings).Shao et al. 1634 RNA, Vol. 13, No.', '10 on October 9, 2007 www.rnajournal.org Downloaded from \\nfactors that may negatively affect upstream steps in the RNAi\\npathway, we found that target accessibility and duplex\\nasymmetry can improve the target knockdown level signif-\\nicantly by nearly 40% and 26%, respectively. These percen-tages are far greater than the degree of improvement by any\\nof the single sequence features reported in a previous study\\n(Reynolds et al. 2004). Our qualitative findings are consis-tent with a previous report based on alternative calculations(Heale et al. 2005). For efficient gene silencing, a number of studies reported\\nthe significance of siRNA sequence features (Khvorovaet al. 2003; Schwarz et al. 2003; Reynolds et al. 2004),and other studies reported the importance of targetstructure (Bohula et al. 2003; Kretschmer-Kazemi Far and\\nSczakiel 2003; Vickers et al. 2003; Yoshinari et al. 2004;Overhoff et al. 2005; Schubert et al. 2005; Westerhout et al.2005).', 'Based on our findings, we propose a simple modelfor efficient RNAi that combines both perspectives in the\\ncontext of the RNAi pathway (Fig. 3). The asymmetry of\\nsiRNA duplex ends is important for RISC assembly,whereas target accessibility is important for the down-stream step of target recognition in the RNAi pathway. A siRNA designed for an accessible target site will not\\nnecessarily be functional, if it does not have a favorable\\nDSSE for effective assembly of the guide strand into RISC. Likewise, a siRNA with a favorable DSSE will not neces-\\nsarily yield potent silencing when the guide strand cannot\\neffectively bind to the highly structured target site. Thus,the combination of favorable DSSE and target accessibility\\ncan greatly improve the efficiency of RNAi. Because the twofactors operate sequentially in the RNAi pathway, their\\neffects on the efficiency of RNAi are heavily convoluted.', 'Deconvolution is necessary to tease apart the individualeffects, particularly the net effect of target structure. Weexpect the model to be generally valid in many experimen-\\ntal systems. However, exceptions are likely due to system-specific factors. For example, in a viral system, a number ofsiRNAs targeted to experimentally identified accessible siteswere not effective (Das et al. 2004), and some of them do\\nhave favorable duplex asymmetry. We adopted a population approach to modeling of\\nmRNA secondary structure by employing the Sfold\\nTABLE 5.', 'Improvement in knockdown level for 72 of 101 shRNAs in the cDNA librarya\\nCriterion for assessing\\nimprovement Control conditionImprovement in average\\nknockdown % t-test P-valuebWilcoxon rank sum test\\nP-valueb\\nTarget accessibilitycDSSE >0 kcal/mol 39.72 0.0004 0.0007\\nDSSE#0 kcal/mol 16.03 0.0888 0.0836\\nDuplex asymmetrydDGdisruption >/C010 kcal/mol 25.99 0.0082 0.0086\\nDGdisruption </C019 kcal/mol 2.29 0.4197 0.3605\\naSatisfying the two filtering criteria: (1) 30% #GC%#70% and (2) absence of AAAA, TTTT, GGGG, and CCCC motifs. bOne-tailed test for significance of improvement.', 'cFor sites that meet the control condition on DSSE, improvement by target accessibility =(average knockdown % for accessible sites, i.e., sites\\nwithDGdisruption >/C010 kcal/mol) /C0(average knockdown % for inaccessible sites, i.e., sites with DGdisruption </C019 kcal/mol); for DSSE >0 kcal/\\nmol, there are 10 accessible sites and 8 inaccessible sites; for DSSE #0 kcal/mol, there are nine accessible sites and eight inaccessible sites. dFor sites that meet the control condition on DGdisruption , improvement by duplex asymmetry =(average knockdown % for sites passing the\\nasymmetry test, i.e., DSSE >0 kcal/mol) /C0(average knockdown % for sites failing the asymmetry test, i.e., DSSE #0 kcal/mol); for DGdisruption >\\n/C010 kcal/mol, there are 10 sites passing the asymmetry test and 9 sites failing the test; for DGdisruption </C019 kcal/mol, there are eight sites\\npassing the asymmetry test and eight sites failing the test.TABLE 4.', 'Net improvement in knockdown level by target accessibility for various DGdisruption thresholdsa\\nEnergy threshold in kcal/mol\\nfor accessible sites\\n(number of data points)Energy threshold in kcal/mol\\nfor inaccessible sites\\n(number of data points)Net improvement\\nin average knockdown %bt-test\\nP-valuecWilcoxon rank\\nsum test P-valuec\\nDGdisruption >/C09 (5) DGdisruption </C020 (6) 38.72 0.0066 0.0152\\nDGdisruption >/C010 (10) DGdisruption </C019 (8) 39.72 0.0004 0.0007\\nDGdisruption >/C011 (12) DGdisruption </C018 (12) 31.21 0.0003 0.0004\\nDGdisruption >/C012 (14) DGdisruption </C017 (17) 32.11 9.34E /C005 0.0002\\nDGdisruption >/C013 (16) DGdisruption </C016 (19) 31.32 6.77E /C005 0.0002\\naEstimated by using 42 of 101 shRNAs in the cDNA library that meet the rule of duplex asymmetry (DSSE >0 kcal/mol) and satisfy the two\\nfiltering criteria: (1) 30% #GC%#70% and (2) absence of AAAA, TTTT, GGGG, and CCCC motifs.', 'bNet improvement in average knockdown % due to target accessibility =average knockdown % for accessible sites /C0average knockdown %\\nfor inaccessible sites. cOne-tailed test of significance for average knockdown % for accessible sites being greater than that for inaccessible sites.Effect of target structure on RNAi\\nwww.rnajournal.org 1635 on October 9, 2007 www.rnajournal.org Downloaded from \\nprogram. This approach has been found to perform better\\nthan the minimum free energy method for prediction of\\nthe activities of antisense oligonucleotides (Ding and\\nLawrence 2001). In the current application, the sampling\\napproach was found to outperform other established\\nprograms for RNA secondary structure predictions (Table\\n3). In a recent work on predicting microRNA–target\\ninteractions, Sfold was used for extensive structural anal-\\nyses (Long et al. 2007).', 'These analyses were based on\\nprobabilistic accessibility profiling, statistics of open nucle-\\notide blocks, and a two-step hybridization model that\\ninvolves energy calculations specific to microRNA–target\\ninteractions. A potent effect of target structure on micro-\\nRNA function was observed from both data analyses and in\\nvivo experimental testing (Long et al. 2007). The effect of\\ntarget structure appears to be stronger for microRNAs than\\nfor siRNAs/shRNAs. There are two potential reasons for\\nthis observation. First, the rule of duplex asymmetry is not\\nrelevant for mature microRNAs that are single-stranded. Second, this may be due to mechanistic differences between\\ntarget cleavage by RNAi and translation repression by\\nmicroRNAs. For future research, the applicability of the\\ntwo-step hybridization model to modeling siRNA–target\\ninteractions warrants investigation.', 'In the calculation of DGdisruption , we assumed that the\\nbinding of target mRNA by the siRNA guide strand induces\\nonly a local structural alteration at the target site. It is likely\\nthat in some, if not all, cases, nucleotides outside the target\\nsite will also contribute to the energy change owing to\\nsiRNA binding. An alternative to the local disruption\\nmodel is a global disruption model, which assumes that\\nthe rest of the target mRNA molecule can completely refold\\nafter siRNA binding. For this model, DGdisruption can berecalculated by constraining the target site to be single\\nstranded and refolding the rest of the target mRNA. However, the predictability by DGdisruption is rather low\\n(for lamin A, P-value = 0.0223, R2= 0.2069; for PTEN and\\nCD54, P-value = 2.05E /C005,R2= 0.2213). This suggests that\\ntarget cleavage occurs rapidly after target binding by the\\nsiRNA guide strand such that global refolding of the target\\nbefore cleavage is unlikely.', 'While partial refolding is a\\npossibility, it is highly uncertain what region of the target\\nmay be involved in refolding. Thus, it is difficult to con-\\nstruct a computational model that may represent a reason-\\nable compromise between the local model and the global\\nmodel. For assessing improvement in predictions, the\\nperformance of any intermediate model will need to be\\ncompared with that for the local model. It has been reported that the 5 9bases of the siRNA are\\nmore important than the 3 9bases for the strength of target\\nbinding (Haley and Zamore 2004), and that nucleotides\\n2–8 of the 5 9end of microRNAs are important for target\\nrecognition (Lewis et al. 2005). We thus statistically tested\\nthe hypothesis that functional RNAi requires good acces-\\nsibility for the 3 9end of the target site.', 'For each of all target\\nsites for siRNAs in our study, we computed the average\\naccessibility for the first five bases from the 3 9end of the\\ntarget site and also the average accessibility for nucleotide\\npositions 2–8 from the 3 9end of the target site. The\\nprobability that a base is unpaired is computed by the\\nSfold structure sample (Ding and Lawrence 2001). The\\naverage accessibility for a block of nucleotides is computed\\nby the sum of the unpaired probabilities divided by the\\nnumber of bases in the block, i.e., the average unpaired\\nprobability for the block. To facilitate a two-group statis-\\ntical comparison, a block is considered to be accessible if\\nthe average accessibility is $0.5 and inaccessible if the\\naverage accessibility is <0.5. All siRNA target sites were\\npartitioned into two groups: group 1 with the 3 9end of the\\ntarget site being accessible and group 2 with an inaccessible\\n39end.', 'Here we consider the first five bases from the 3 9end\\nand nucleotide positions 2–8 from the 3 9end separately. A one-sided t-test was then performed to determine if the\\nRNAi activity for target sites in group 1 is significantly\\nhigher than that for target sites in group 2. For the lamin A\\ndata set, the P-value for the first five bases from the 3 9end\\nof the target site is 0.3163, and the P-value for nucleotide\\npositions 2–8 is 0.0973; for the PTEN and CD54 data set,\\ntheP-value for the first five bases is 0.1458, and the P-value\\nfor nucleotide positions 2–8 is 0.2024. Thus, we did not\\nfind statistical support for the hypothesis. This suggests that\\nnucleation of siRNA–target hybridization can occur any-\\nwhere within the target site, not necessarily the 3 9end of\\nthe target site. The same conclusion was reached for\\nmicroRNA–target hybridization in a recent study (Long\\net al. 2007). Our study seeks to improve the potency of gene silencing\\nby RNAi.', 'Based on our findings, we recommend selecting\\nFIGURE 3. A proposed simple model for efficient RNAi. RISC\\nassembly is facilitated by asymmetric ends of siRNA duplex; target\\nrecognition via intermolecular base-pairing is aided by structural\\naccessibility at the target site. The combination of the upstream effect\\nof duplex asymmetry and the downstream effect of target accessibility\\nis generally essential for potent gene silencing.Shao et al. 1636 RNA, Vol. 13, No. 10 on October 9, 2007 www.rnajournal.org Downloaded from \\nsiRNAs or shRNAs with DGdisruption >/C010 kcal/mol and\\nDSSE > 0.0 kcal/mol, in addition to applying the filters of\\nbalanced GC content and the absence of nucleotide repeat\\ntracts. The issue of the specificity of RNAi is also important(Jackson et al. 2003; Semizarov et al. 2003; Pei and Tuschl\\n2006), particularly for high-throughput RNAi screening.Recent studies suggest that some off-target effects areassociated with ‘‘seed’’ matches in the 3 9UTRs (Birmingham\\net al.', '2006; Jackson et al. 2006). These findings and thefindings in our study can be useful for improving thedesign of RNAi experiments, by addressing both the issueof potency and the issue of specificity. MATERIALS AND METHODS\\nPrediction of mRNA secondary structure\\nAn mRNA is likely to exist as a population of structures\\n(Christoffersen et al. 1994). This view has been supported by\\nthe experimental elucidation of multiple equilibrium conforma-tions (Altuvia et al. 1989; Betts and Spremulli 1994). Thus, the\\nuse of a single structure, e.g., the minimum free energy (MFE)\\nstructure, is not well suited to structure prediction for mRNAs. An\\nalternative ensemble-based method has been developed (Ding andLawrence 2003). In this approach, a statistically representative\\nsample from the Boltzmann-weighted ensemble of probable RNA\\nsecondary structures is generated, in a manner to faithfully andreproducibly capture the statistical features of the structure\\nensemble of enormous size.', 'In comparison with MFE predictions,\\nthis method has been shown to substantially improve predictionsfor structural RNAs (Ding et al. 2005) and to better represent the\\nlikely population of mRNA structures (Ding et al. 2006). A sample\\nsize of 1000 structures is sufficient to guarantee statistical\\nreproducibility in sampling statistics (Ding and Lawrence 2003;Ding et al. 2006). The structure sampling method has been\\nimplemented in the Sfold software package (Ding et al. 2004)\\nand is applied here to mRNA folding. Parameters for analyses\\nFor the statistical analyses of knockdown data from RNAi experi-\\nments, we consider several empirical rules in the literature. In\\naddition, we introduce a novel measure of target accessibility. Thestability of the target:guide strand duplex is also considered. The\\nparameters included in our analysis are defined below.', 'DSSE: Differential stability of siRNA duplex ends\\nFor the 5 9end of the antisense (guide) siRNA strand, 5 9-antisense\\nstability ( AntiS , in kcal/mol) is computed by the summation of the\\nfree energies for four base-pair stacks (involving five consecutive\\nbase pairs) and the 3 9dangling base (overhang), with a penalty for\\na terminal A-U pair. Similarly, 5 9-sense stability ( SS, in kcal/mol)\\nis the sum for the 5 9end of the sense siRNA strand. The\\ndifferential stability of siRNA duplex ends ( DSSE , in kcal/mol)\\nis the difference between the 5 9-antisense stability and the 5 9-sense\\nstability, i.e., DSSE =AntiS/C0SS. These calculations are based on\\nthe established RNA thermodynamic rules and parameters (Xiaet al. 1998; Mathews et al. 1999). Because DSSE measures the\\ndifference in stability between the two siRNA duplex ends, a\\nsiRNA duplex meets the rule of asymmetry when DSSE > 0.0 kcal/\\nmol.', 'S\\nD: Dharmacon score\\nThis score was proposed by Dharmacon scientists for rational\\nsiRNA design (Reynolds et al. 2004). As the sum of eight\\ncomponent scores for various sequence features of siRNA duplex,\\nthe Dharmacon score ranges between /C02 and 10. DGdisruption : A measure of target site accessibility\\nDGdisruption is the energy cost of disruption of the mRNA structure\\nso that the binding site becomes completely single stranded (Fig.1). Given the small size of the antisense siRNA, we adopt a localdisruption model, i.e., the alteration of target structure due to\\nsiRNA binding is local rather than global. Specifically, we assume\\nthat only the binding site is involved in structural alternation (Fig.1).', 'Under this assumption, DG\\ndisruption is the energy cost for\\nbreaking those target intramolecular base pairs at the binding siteand is given by the energy difference between DG\\nbefore, the free\\nenergy of the original mRNA structure, and DGafter, the free\\nenergy of the new, locally altered structure, i.e., DGdisruption =\\nDGbefore/C0DGafter. For 1000 structures predicted by Sfold, we\\ncalculate DGbefore by the average energy of the original 1000\\nstructures and DGafterby the average energy of all the 1000 locally\\naltered structures. A largely single-stranded (i.e., structurallyaccessible) site does not require substantial structure alterationfor the guide siRNA strand to bind to the target. The disruption\\nenergy DG\\ndisruption is a quantitative measure of the structural\\naccessibility at the target site. DGhybrid : Stability of hybrid formed by siRNA guide\\nstrand and target\\nDGhybrid is the energy gain due to the hybridization at the binding\\nsite (Fig. 1).', 'This parameter measures the stability of the hybrid\\nformed by the siRNA guide strand and the nucleotides at thetarget site. DG\\nhybrid is calculated as the sum of the stacking\\nenergies for the siRNA guide:target duplex, with the penalty of aninitiation energy:\\nDG\\nhybrid =DGinitiation ++DGstacking ;\\nwhere DGinitiation = 4.1 kcal/mol (Mathews et al. 1999), and the\\nsum is over RNA/RNA stacking energies (Xia et al. 1998). Statistical analyses\\nWeighted least-squares regression\\nTo assess the contribution by each of the above parameters to\\nRNAi efficiency, we employ weighted least-squares regression for\\nprediction of target knockdown level by each of the parameters. For each of the siRNA data sets as described below, there exists\\na large variation in the standard deviations of the measuredknockdown levels.', 'In a statistical analysis, a data point with\\nsmaller standard deviation should carry more weight than oneEffect of target structure on RNAi\\nwww.rnajournal.org 1637 on October 9, 2007 www.rnajournal.org Downloaded from \\nwith larger standard deviation. This consideration can be\\naddressed by the use of weights in the least-squares regression\\n(Weisberg 2005). In other words, the square term in the sum ofsquares for a data point is multiplied by a weight. When the\\nstandard deviation of the knockdown level from multiple mea-\\nsurements is available for every data point, 1/(standard devia-tion)\\n2can be used as the weight. The P-value and R2of the\\nregression analysis for a parameter are, respectively, the measuresof the statistical significance of the parameter and the degree ofvariability in silencing activity that is attributed to the parameter. Statistical tests for two-group comparison\\nThe unpaired t-test was used for comparing data for two indepen-\\ndent groups.', 'The corresponding nonparametric test, the Wilcoxon\\nrank sum test (also known as the Mann–Whitney U-test or the\\nWilcoxon–Mann–Whitney test), was also used to confirm theresults by the t-test, which relies on the assumption of the nor-\\nmality of the data. All of the statistical analyses in this study wereperformed with the statistical package R (http://www.r-project.org). Selection of siRNAs data sets\\nFor selection of RNAi data sets from the literature for analysis, we\\nemployed two criteria: (1) at least 10 siRNAs for the same target\\nmust have been tested and (2) target sites on the same mRNAmust not have substantial overlap. Due to the high costs of\\nsynthetic siRNAs, usually only a few siRNAs are tested for one\\ntarget. Experimental variation between different RNAi knock-down experiments is difficult to account for in statistical analysis. Thus, we focus on data sets that have a sufficient number of\\nsiRNAs for the same target or for multiple targets tested by thesame experimental system.', 'Heavy overlap of target sites canintroduce an autocorrelation bias that is difficult to assess. It is\\nconceivable that a local region of the target could be highly\\nsusceptible to RNAi, e.g., due to high steric accessibility. Criterion2 aims to avoid such likely region bias. We identified published\\nsiRNA data sets for lamin A (Harborth et al. 2003), PTEN, and\\nCD54 (Vickers et al. 2003), and included 25, 36, and 39 siRNAs,respectively, in our analysis. The GenBank accession number and\\nsequence length of the target are NM_170707 and 3181 nt for\\nlamin A, U92436 and 3160 nt for PTEN, and J03132 and 2986 ntfor CD54. For lamin A, another 19 siRNAs were tested for a short\\ndistance walk through single-base shift in the target site. In the\\nlight of criterion 2, however, these 19 siRNAs were not included\\nhere.', 'Inclusion of these 19 siRNAs would enhance support for ourcomputational approach, because these siRNAs are highly func-\\ntional, and their target region is highly accessible by our pre-\\ndiction. For lamin A, siRNA activity was measured at the protein\\nlevel using Western blot. For PTEN and CD54, the siRNA activity\\nwas measured at the mRNA level using RT-PCR. For these data\\nsets, every siRNA was tested at least twice, in triplicate, withstandard error available for measured activity. Description of shRNA data set\\nWe have also analyzed a data set of shRNA activities obtained\\nfrom the analysis of a library of shRNA sequences generated fromrandomly fragmented cDNA of normalized (reduced-redundance)\\ncDNA of all of the genes expressed in the MCF-7 human breastcarcinoma cells. The generation and testing of the library will be\\ndescribed in detail elsewhere (A. Maliyekkel, Y. Shao, N. Warholic,\\nK. Cole, Y. Ding, and I.B. Roninson, in prep.).', 'Briefly, DNAseI-generated fragments of normalized cDNA were converted into\\nshRNA templates by the procedure of Shirane et al. (2004), with\\nsome modifications. The shRNA templates were cloned intolentiviral vector LLCEP TU6LX (Maliyekkel et al. 2006) that\\nexpresses shRNA from RNA polymerase III promoter, which is\\npositively regulated by tetracycline/doxycycline via the tTR-KRABrepressor. cDNA was cut by MmeI to produce 19–21-bp cDNA\\nfragments. These fragments were then ligated with a hairpin\\nadaptor to produce a hairpin–stem with a length of 27–29 bp. Thehairpin–stems are very efficiently processed by Dicer to generateeither 19- or 20-bp siRNA, with the adaptor sequences removed. The positions of the 19-nt and 20-nt Dicer cleavage sites are\\nknown precisely. Individual sequenced shRNAs were matched with the corre-\\nsponding human genes, and randomly selected shRNA sequenceswere transduced into MCF-7 cells expressing tTR-KRAB.', 'shRNAactivity was determined by measuring the levels of each target\\nmRNA by real-time reverse-transcription PCR, in triplicate. Percent knockdown was calculated from the ratio of mRNA levelswith and without doxycycline. The data for 101 shRNA sequencestargeting 100 different genes (i.e., two shRNAs for only one gene)\\nwere used for the analysis. The list of these genes and their lengths\\nare given in Supplemental Table 1. Strategy of data analyses\\nThe siRNA data sets were first analyzed to identify parameters that\\nare important for RNAi efficiency. These parameters were then\\nfurther examined with the independent shRNA data set, and their\\ncontributions to RNAi knockdown levels were quantitatively\\nassessed. Such a statistical assessment was made possible by the\\nrelatively large size of the shRNA data set. SUPPLEMENTAL DATA\\nSupplemental Tables 1 and 2 are available at http://sfold. wadsworth.org/Shao_RNA07_supp.pdf.', 'ACKNOWLEDGMENTS\\nThe Computational Molecular Biology and Statistics Core at theWadsworth Center is acknowledged for providing computingresources for this work. This work was supported in part by\\nNational Science Foundation grant DMS-0200970 and National\\nInstitutes of Health grant R01 GM068726 to Y.D. and National\\nInstitutes of Health grants R33 CA95996, R01 CA62099, and R01\\nAG17921 to I.B.R. Received March 7, 2007; accepted July 10, 2007. REFERENCES\\nAltuvia, S., Kornitzer, D., Teff, D., and Oppenheim, A.B. 1989. Alternative mRNA structures of the cIII gene of bacteriophage\\nldetermine the rate of its translation initiation. J. Mol. Biol. 210:\\n265–280.Shao et al. 1638 RNA, Vol. 13, No. 10 on October 9, 2007 www.rnajournal.org Downloaded from \\nBetts, L. and Spremulli, L.L. 1994. Analysis of the role of the Shine–\\nDalgarno sequence and mRNA secondary structure on the\\nefficiency of translational initiation in the Euglena gracilis chloro-\\nplast atpH mRNA. J. Biol. Chem. 269: 26456–26463.', 'Birmingham, A., Anderson, E.M., Reynolds, A., Ilsley-Tyree, D.,\\nLeake, D., Fedorov, Y., Baskerville, S., Maksimova, E.,\\nRobinson, K., Karpilow, J., et al. 2006. 3 9UTR seed matches,\\nbut not overall identity, are associated with RNAi off-targets. Nat. Methods 3:199–204. Boese, Q., Leake, D., Reynolds, A., Read, S., Scaringe, S.A.,\\nMarshall, W.S., and Khvorova, A. 2005. Mechanistic insights aid\\ncomputational short interfering RNA design. Methods Enzymol. 392: 73–96. Bohula, E.A., Salisbury, A.J., Sohail, M., Playford, M.P.,\\nRiedemann, J., Southern, E.M., and Macaulay, V.M. 2003.The efficacy of small interfering RNAs targeted to the type 1insulin-like growth factor receptor (IGF1R) is influenced bysecondary structure in the IGF1R transcript. J. Biol. Chem. 278:\\n15991–15997. Brummelkamp, T.R., Bernards, R., and Agami, R. 2002. A system for\\nstable expression of short interfering RNAs in mammalian cells.Science 296: 550–553. Christoffersen, R.E., McSwiggen, J.A., and Konings, D. 1994.', 'Appli-\\ncation of computational technologies to ribozyme biotechnologyproducts. J. Mol. Struct. THEOCHEM 311: 273–284. Das, A.T., Brummelkamp, T.R., Westerhout, E.M., Vink, M.,\\nMadiredjo, M., Bernards, R., and Berkhout, B. 2004. Humanimmunodeficiency virus type 1 escapes from RNA interference-mediated inhibition. J. Virol. 78:2601–2605. Ding, Y. and Lawrence, C.E. 2001. Statistical prediction of single-\\nstranded regions in RNA secondary structure and application to\\npredicting effective antisense target sites and beyond. Nucleic Acids\\nRes. 29:1034–1046. Ding, Y. and Lawrence, C.E. 2003. A statistical sampling algorithm for\\nRNA secondary structure prediction. Nucleic Acids Res. 31:7280–\\n7301. doi: 10.1093/nar/gkg938. Ding, Y., Chan, C.Y., and Lawrence, C.E. 2004. Sfold web server for\\nstatistical folding and rational design of nucleic acids. Nucleic Acids\\nRes. 32:W135–W141. doi: 10.1093/nar/gkh449. Ding, Y., Chan, C.Y., and Lawrence, C.E. 2005.', 'RNA secondary\\nstructure prediction by centroids in a Boltzmann weightedensemble. RNA 11:1157–1166. Ding, Y., Chan, C.Y., and Lawrence, C.E. 2006. Clustering of RNA\\nsecondary structures with application to messenger RNAs. J. Mol. Biol. 359: 554–571. Elbashir, S.M., Harborth, J., Lendeckel, W., Yalcin, A., Weber, K., and\\nTuschl, T. 2001. Duplexes of 21-nucleotide RNAs mediate RNAinterference in cultured mammalian cells. Nature 411: 494–498. Fire, A., Xu, S., Montgomery, M.K., Kostas, S.A., Driver, S.E., and\\nMello, C.C. 1998. Potent and specific genetic interference bydouble-stranded RNA in Caenorhabditis elegans .Nature 391:\\n806–811. Geiduschek, E.P. and Kassavetis, G.A. 2001. The RNA polymerase III\\ntranscription apparatus. J. Mol. Biol. 310: 1–26. Haley, B. and Zamore, P.D. 2004. Kinetic analysis of the RNAi enzyme\\ncomplex. Nat. Struct. Mol. Biol. 11:599–606. Harborth, J., Elbashir, S.M., Vandenburgh, K., Manninga, H.,\\nScaringe, S.A., Weber, K., and Tuschl, T. 2003.', 'Sequence, chem-\\nical, and structural variation of small interfering RNAs and shorthairpin RNAs and the effect on mammalian gene silencing.Antisense Nucleic Acid Drug Dev. 13:83–105. Hardin, C.C., Watson, T., Corregan, M., and Bailey, C. 1992. Cation-\\ndependent transition between the quadruplex and Watson–Crickhairpin forms of d(CGCG3GCG). Biochemistry 31:833–841. Heale, B.S., Soifer, H.S., Bowers, C., and Rossi, J.J. 2005. siRNA target\\nsite secondary structure predictions using local stable substruc-\\ntures. Nucleic Acids Res. 33:e30. doi: 10.1093/nar/gni026. Hofacker, I.L. 2003. Vienna RNA secondary structure server. Nucleic\\nAcids Res. 31:3429–3431.Holen, T., Amarzguioui, M., Wiiger, M.T., Babaie, E., and Prydz, H. 2002. Positional effects of short interfering RNAs targeting the\\nhuman coagulation trigger Tissue Factor. Nucleic Acids Res. 30:\\n1757–1766. Jackson, A.L., Bartz, S.R., Schelter, J., Kobayashi, S.V., Burchard, J.,\\nMao, M., Li, B., Cavet, G., and Linsley, P.S. 2003.', 'Expressionprofiling reveals off-target gene regulation by RNAi. Nat. Bio-\\ntechnol. 21:635–637. Jackson, A.L., Burchard, J., Schelter, J., Chau, B.N., Cleary, M.,\\nLim, L., and Linsley, P.S. 2006. Widespread siRNA ‘‘off-target’’\\ntranscript silencing mediated by seed region sequence comple-mentarity. RNA 12:1179–1187. Khvorova, A., Reynolds, A., and Jayasena, S.D. 2003. Functional\\nsiRNAs and miRNAs exhibit strand bias. Cell 115: 209–216. Kretschmer-Kazemi Far, R. and Sczakiel, G. 2003. The activity of\\nsiRNA in mammalian cells is related to structural target accessi-bility: A comparison with antisense oligonucleotides. Nucleic Acids\\nRes. 31:4417–4424. Laughlan, G., Murchie, A.I., Norman, D.G., Moore, M.H.,\\nMoody, P.C., Lilley, D.M., and Luisi, B. 1994. The high-resolutioncrystal structure of a parallel-stranded guanine tetraplex. Science\\n265: 520–524. Lee, N.S., Dohjima, T., Bauer, G., Li, H., Li, M.J., Ehsani, A.,\\nSalvaterra, P., and Rossi, J. 2002.', 'Expression of small interferingRNAs targeted against HIV-1 rev transcripts in human cells. Nat. Biotechnol. 20:500–505. Lewis, B.P., Burge, C.B., and Bartel, D.P. 2005. Conserved seed\\npairing, often flanked by adenosines, indicates that thousands ofhuman genes are microRNA targets. Cell 120: 15–20. Long, D., Lee, R., Williams, P., Chan, C.Y., Ambros, V., and Ding, Y. 2007. Potent effect of target structure on microRNA function. Nat. Struct. Mol. Biol. 14:287–294. Luo, K.Q. and Chang, D.C. 2004. The gene-silencing efficiency of\\nsiRNA is strongly dependent on the local structure of mRNA at thetargeted region. Biochem. Biophys. Res. Commun. 318: 303–310. Maliyekkel, A., Davis, B.M., and Roninson, I.B. 2006. Cell cycle arrest\\ndrastically extends the duration of gene silencing after transient\\nexpression of short hairpin RNA. Cell Cycle 5:2390–2395. Mathews, D.H., Sabina, J., Zuker, M., and Turner, D.H. 1999.', 'Expanded sequence dependence of thermodynamic parametersimproves prediction of RNA secondary structure. J. Mol. Biol. 288:\\n911–940. Mathews, D.H., Disney, M.D., Childs, J.L., Schroeder, S.J., Zuker, M.,\\nand Turner, D.H. 2004. Incorporating chemical modificationconstraints into a dynamic programming algorithm for predictionof RNA secondary structure. Proc. Natl. Acad. Sci. 101: 7287–7292. Matranga, C., Tomari, Y., Shin, C., Bartel, D.P., and Zamore, P.D. 2005. Passenger-strand cleavage facilitates assembly of siRNA intoAgo2-containing RNAi enzyme complexes. Cell 123: 607–620. Overhoff, M., Alken, M., Far, R.K., Lemaitre, M., Lebleu, B.,\\nSczakiel, G., and Robbins, I. 2005. Local RNA target structureinfluences siRNA efficacy: A systematic global analysis. J. Mol. Biol. 348: 871–881. Paddison, P.J., Caudy, A.A., Bernstein, E., Hannon, G.J., and\\nConklin, D.S. 2002. Short hairpin RNAs (shRNAs) inducesequence-specific silencing in mammalian cells. Genes&Dev. 16:\\n948–958.', 'Patzel, V., Rutz, S., Dietrich, I., Koberle, C., Scheffold, A., and\\nKaufmann, S.H. 2005. Design of siRNAs producing unstructured\\nguide-RNAs results in improved RNA interference efficiency. Nat. Biotechnol. 23:\\n1440–1444. Pei, Y. and Tuschl, T. 2006. On the art of identifying effective and\\nspecific siRNAs. Nat. Methods 3:670–676. Rand, T.A., Petersen, S., Du, F., and Wang, X. 2005. Argonaute2\\ncleaves the anti-guide strand of siRNA during RISC activation. Cell\\n123: 621–629. Reynolds, A., Leake, D., Boese, Q., Scaringe, S., Marshall, W.S., and\\nKhvorova, A. 2004. Rational siRNA design for RNA interference. Nat. Biotechnol. 22:326–330.Effect of target structure on RNAi\\nwww.rnajournal.org 1639 on October 9, 2007 www.rnajournal.org Downloaded from \\nRose, S.D., Kim, D.H., Amarzguioui, M., Heidel, J.D.,\\nCollingwood, M.A., Davis, M.E., Rossi, J.J., and Behlke, M.A. 2005. Functional polarity is introduced by Dicer processing ofshort substrate RNAs. Nucleic Acids Res. 33:4140–4156.', 'doi:\\n10.1093/nar/gki732. Schubert, S., Grunweller, A., Erdmann, V.A., and Kurreck, J. 2005. Local RNA target structure influences siRNA efficacy: Systematicanalysis of intentionally designed binding regions. J. Mol. Biol. 348:\\n883–893. Schwarz, D.S., Hutvagner, G., Du, T., Xu, Z., Aronin, N., and\\nZamore, P.D. 2003. Asymmetry in the assembly of the RNAienzyme complex. Cell 115: 199–208. Semizarov, D., Frost, L., Sarthy, A., Kroeger, P., Halbert, D.N., and\\nFesik, S.W. 2003. Specificity of short interfering RNA determinedthrough gene expression signatures. Proc. Natl. Acad. Sci. 100:\\n6347–6352. Shirane, D., Sugao, K., Namiki, S., Tanabe, M., Iino, M., and\\nHirose, K. 2004. Enzymatic production of RNAi libraries from\\ncDNAs. Nat. Genet. 36:190–196. Stein, C.A. 1999. Two problems in antisense biotechnology: In vitro\\ndelivery and the design of antisense experiments. Biochim. Biophys. Acta 1489: 45–52.', 'Ui-Tei, K., Naito, Y., Takahashi, F., Haraguchi, T., Ohki-\\nHamazaki, H., Juni, A., Ueda, R., and Saigo, K. 2004. Guidelinesfor the selection of highly effective siRNA sequences for mamma-lian and chick RNA interference. Nucleic Acids Res. 32:936–948. doi: 10.1093/nar/gkh247.Vickers, T.A., Wyatt, J.R., and Freier, S.M. 2000. Effects of RNA\\nsecondary structure on cellular antisense activity. Nucleic Acids\\nRes. 28:1340–1347. Vickers, T.A., Koo, S., Bennett, C.F., Crooke, S.T., Dean, N.M., and\\nBaker, B.F. 2003. Efficient reduction of target RNAs by small\\ninterfering RNA and RNase H-dependent antisense agents.A comparative analysis. J. Biol. Chem. 278: 7108–7118. Weisberg, S. 2005. Applied linear regression , 3rd ed. John Wiley and\\nSons, New York. W e s t e r h o u t ,E . M . ,O o m s ,M . ,V i n k ,M . ,D a s ,A . T . ,a n d\\nBerkhout, B. 2005. HIV-1 can escape from RNA interferenceby evolving an alternative structure in its RNA genome. Nucleic\\nAcids Res. 33:796–804. doi: 10.1093/nar/gki220.', 'Wuchty, S., Fontana, W., Hofacker, I.L., and Schuster, P. 1999. Complete suboptimal folding of RNA and the stability of second-ary structures. Biopolymers 49:145–165. Xia, T., SantaLucia Jr., J., Burkard, M.E., Kierzek, R., Schroeder, S.J.,\\nJ i a o ,X . ,C o x ,C . ,a n dT u r n e r ,D . H .1 9 9 8 .T h e r m o d y n a m i cp a r a m e t e r s\\nfor an expanded nearest-neighbor model for formation of RNA\\nduplexes with Watson–Crick base pairs. Biochemistry 37:14719–14735. Yoshinari, K., Miyagishi, M., and Taira, K. 2004. Effects on RNAi of\\nthe tight structure, sequence, and position of the targeted region. Nucleic Acids Res. 32:691–699. doi: 10.1093/nar/gkh221. Zhao, J.J. and Lemke, G. 1998. Rules for ribozymes. Mol. Cell. Neurosci. 11:92–97. Zuker, M. 2003. Mfold web server for nucleic acid folding and\\nhybridization prediction. Nucleic Acids Res. 31:3406–3415.Shao et al. 1640 RNA, Vol. 13, No.', '10 on October 9, 2007 www.rnajournal.org Downloaded from \\n    \\n 1 Supplemental Materials  \\n \\nSupplemental Table 1.', 'Sequence and target information for the 101 shRNAs in the MCF-7 cDNA library  \\nshRNA  \\nnumber  Target gene \\nGenBank \\naccession \\nnumber Sequence  \\nlength (nt)  Target \\nposition \\n(for 19-bp \\nDicer \\ncleavage) Guide strand sequence including 2-\\nnt 3\\x02 overhang in lower case (5 \\x02\\x013\\x02) Target mRNA \\nknockdown % \\n1 NM_002799 1012 888-906 GGGCCTCAATGCTCACCACct -5.86 \\n2 NM_003618 4141 418-436 ATATGCAATTTGCAGTTCTtt -0.48 \\n3 NM_004359 1462 354-372 AAGTAGCCGCCCTCGTAGTag 39.85 \\n4 NM_003129 2989 2216-2234 CTTTAGCAGTTTTCTCCATtt 30.93 \\n5 NM_014671 5160 1273-1291 AATAATTTTCGCCAACAGTtt -21.32 \\n6 NM_015231 5383 3549-3567 TTCCTCTTAGGGGATGCTCtt 45.44 \\n7 NM_003600 2346 1886-1904 CTTTCCCCACAGCCAGGCTtt 47.69 \\n8 NM_004526 3453 1085-1103 GGTTTCACCTCCTGGTTCTtt 45.56 \\n9 NM_031946 3152 2056-2074 GCAGCTCAGGCGGCCAGTCtt 23.47 \\n10 NM_005745 1324 914-932 GCACGTGCCAGGTGGAAGCtt -6.44 \\n11 NM_004707 2359 476-494 GGATGGTTCGTGTTCGCTCtt 68.07 \\n12 NM_014330 2349 941-959 CTTCTTTCTGTTCTTTTATtt 4.12 \\n13 NM_052844 1818 1352-1370 GCCCGTCAGTCCCAGCGCTcg -30.24 \\n14 NM_006513 1942 1787-1805 TGGGTCCCTATGCCCATGCtt 45.36 \\n15 NM_004860 2895 2457-2475 AGAGAGGCTCCCCTTACCCtt 12.86 \\n16 NM_152787 3798 2483-2501 CCTGTACTTTGGAGGTCACtt 19.08 \\n17 NM_004336 3486 996-1014 GGGATGTCTCCACCACCTGat -5.54 \\n18 NM_004716 3497 798-816 ACCCTCAGGGCTATAGTTGgg 8.43 \\n19 NM_005415 3297 449-467 ACTGGTAATCAGCGTTGCCat 35.75 \\n20 NM_005463 3514 1147-1165 AAGCACAAATCCAAATCCTtt 44.63 \\n21 NM_004356 1497 1026-1044 ACCGGAAACGTTATATACAcg 15.37 \\n22 NM_000098 3090 1815-1833 TTTCATGGTGGCATCAAACtt 33.12 \\n23 NM_001015 641 556-574 CGGGCCGATGTCCAGCCTCtt 13.03 \\n24 NM_021212 2952 1452-1470 AATAGTCACATGTTATCATtt 14.34 \\n25 NM_001537 1919 319-337 TAAATTATTAGCAACCTTCtt 69.4 \\n26 NM_006640 3929 1533-1551 TGAGTGTGTCCGCCTTGGCtt -1.36 \\n27 NM_005733 2972 1658-1676 TCAATATCATCATCAAGGCtt 70.39 \\n28 NM_002791 1035 248-266 GTGTGACAATTACTGCACAgt 18.59 \\n29 NM_017811 4457 798-816 ATTTTGGTCAAGAATCTGAag -14.38 \\n30 NM_152441 2164 606-624 GTTAAGGCTCTTGAGGCGCtt 10.23 \\n31 NM_015213 4965 2706-2724 ATGTAACAGGTGGGACCATtt 27.49 \\n32 NM_006908 2355 246-264 ACCACACACTTGATGGCCTgc 48.74 \\n33 NM_012432 4420 3208-3226 TGTCTCCCTCATCCTTGATtt 36.68 \\n34 NM_002018 4176 3106-3124 GCTGCTGCGTCATGCGTACtt 52.12 \\n35 NM_004939 2706 942-960 GTTTTTCATATGTGGTGGTtt 40.14 \\n    \\n 2 36 NM_001419 6075 2061-2079 GAAGGGATGCGAGAAATACtt -17.62 \\n37 NM_004941 4201 1548-1566 ATCTCAGCTTCCCGCTGGGcc -17.49 \\n38 NM_001001520 2344 1485-1503 GGAGGGCTCTTTCTTCTTCtg -4.75 \\n39 NM_014267 1637 564-582 TTCTTCAGCAGATTCTTCTtt 59.13 \\n40 NM_001067 5698 829-847 TGACCATTAGTGCAACAATtt 17.2 \\n41 NM_020948 4972 1699-1717 TTTTCATCAAAATCATTTTtt -4.48 \\n42 NM_182931 7139 3858-3876 GCCTCTAGTATCATCCTGCat 1.14 \\n43 NM_000122 2751 1751-1769 AATCTTGTCATTCCTCCTTtc 31.13 \\n44 NM_021259 2561 2356-2374 TCGTTCTTGCAGATCTGATtt 51.8 \\n45 NM_004421 2941 2793-2811 ACGTGGGGGCCAGAGAAGCtt -1.67 \\n46 NM_006098 1125 81-99 TCATGGCGGCGGCGAGAGCtt -0.39 \\n47 NM_015326 6305 2544-2562 ATAAATATCGGCTACATGAcg 29.42 \\n48 NM_001386 4567 2573-2591 TATTAATATGGAATTAAATtt 10.66 \\n49 NM_014891 2383 550-568 CTGTTTCCGGATGATGGCCtt 46.74 \\n50 NM_001029835 2140 770-788 TGGGCACCCTGAGGATGATtt 2.24 \\n51 NM_021259 2561 2223-2241 AGTAGTTGTCGCTAGTCATtt 69.2 \\n52 NM_001001438 2658 1873-1891 AGCAGGAAGTCACAGGCCCtt 26.7 \\n53 NM_002032 1245 739-757 GTGCTTGTCAAAGAGATATtt 51.97 \\n54 NM_001802 2702 571-589 AGTTCCCTTGCTGTGACGTtt -17.74 \\n55 NM_003903 2196 1159-1177 TTTTCTCAAGTGTTGTGGCtt 44.78 \\n56 NM_002623 1383 143-161 GCCTTCTTCATGGTGGCCGcg -26.56 \\n57 NM_172020 6014 4754-4772 GCTTGAAACATACAATCCTtt 44.62 \\n58 NM_004788 6138 1310-1328 GGAGAGAGCTGGAGTAAGTtt -0.93 \\n59 NM_005973 2142 1766-1784 GCAATCAGTTCCAGAGCCCta -5.81 \\n60 NM_004504 4878 569-587 AAAAGAGATTTCAGTGGTTtg 8.35 \\n61 NM_012099 3286 1433-1451 GGCTTCCTCCAGTGGCTCCac -0.47 \\n62 NM_020753 5005 1515-1533 TGGCTCTCGTGGATGTGGCtt -21.95 \\n63 NM_173823 5633 2966-2984 CAGCTAAGCTTTTCCTTATtt -28.84 \\n64 NM_002264 6887 828-846 TCACTGACAAACAGCAACCtt 40.29 \\n65 NM_139280 2109 1221-1239 GGAACCAAGCCATCTACACtc 37.07 \\n66 NM_012479 3747 785-803 GGCGTCGTCGAACGCGGTCtt 25.6 \\n67 NM_015954 1561 1454-1472 AAAATGAAAGATGAGTTTTtt 0.04 \\n68 NM_006284 762 686-704 AAGTACATTTAGGTTGGGTtt 37.22 \\n69 NM_001943 3516 3386-3404 GCTAATTAAACTCTGGGTCag 22.67 \\n70 NM_001122 2010 372-390 TCTGGATGATGGGCAGAGCtt -15.94 \\n71 NM_020198 3294 863-881 CATTCAGTAAGTCTTGTCTtt 14.25 \\n72 NM_006135 2758 415-433 CATAAACAGTACAGAAGCCtt -4.6 \\n73 NM_003318 2984 148-166 TCTTCATTTTTAAACTTATtt 14.54 \\n74 NM_017978 2149 1884-1902 AAAATAGTCTTCATGTCCTtt -6.52 \\n75 NM_014077 1465 1217-1235 ACTGCAGCCTTGTCTGTTTtt 25.42 \\n76 NM_003674 1913 613-631 TGGATAATGAAGTTCCTGTtt 23.62 \\n77 NM_032139 4501 3596-3614 GAATGCGGTGGTGAAACCTtt 30.62 \\n78 NM_021188 2207 1921-1939 AACAGTCAGAATGAAGAACtt 15.24 \\n79 NM_002627 2628 1891-1909 TCTCCGTCAGGTGCTCCACtt 8.72 \\n80 NM_006471 1243 393-411 TTCTTTTGCTCGACATGGTtt 31.99 \\n    \\n 3 81 NM_005312 6121 3228-3246 ATAATGTCGTCGTTCCTCCgc 58.25 \\n82 NM_004652 9018 5232-5250 TAAAAGATTCTTCACATTCtt 2.03 \\n83 NM_001430 5186 1931-1949 GTCTTCAGGGCTATTGGGCgt 52.19 \\n84 NM_006601 1490 1293-1311 TGACAGTGCAAATACAAATtt 26.24 \\n85 NM_003168 1499 767-785 GAGAATGGAAGAGGGAGGCtt -14.24 \\n86 NM_005720 1520 277-295 GTCACAATACGGTTACTCTcg -16.01 \\n87 NM_007126 3198 1017-1035 TGATCTCAGGACCATTGATtt 41.46 \\n88 NM_015906 8339 2106-2124 CAGCGATGGAAGGTTTTCTtt 14.15 \\n89 NM_002067 1540 921-939 GCTCTCCTCCATCCGGTTCtg -8.96 \\n90 NM_013433 4901 864-882 GTCAATCCGCACTTCCAGAag 6.16 \\n91 NM_024811 3524 2969-2987 ACCCAGAAGGGTCTCTAGTtt -22.86 \\n92 NM_004423 5062 3200-3218 CTGGACATCCTTGGGAAGCtt -1.93 \\n93 NM_001024 418 272-290 GAGACGATGCCATCGGCCTtt -6.45 \\n94 NM_178148 2057 1806-1824 CCGCTTGTGCTTTTCTCACtt -6.59 \\n95 NM_006342 2788 2666-2684 AGAATCAGACAGGACAGACtt 46.39 \\n96 NM_005628 2856 2596-2614 CAACTACAGCCGCCAAAATtt -10.15 \\n97 NM_014400 1698 930-948 GCCTCGTGTTCTACTCCCTgt 2.6 \\n98 NM_002806 1590 36-54 AGTCCTGAAGCGCCTTATCtc 25.42 \\n99 NM_003025 2460 1062-1080 AGCTCCCCGTCGTTCTCGGgc 32.63 \\n100 NM_006803 3472 2973-2991 ATCTGGTTGTGTACAGTACac 24.74 \\n 101 NM_021953 3527 630-648 CTGGTTTGGGTTTGAGGCCtt 38.67 \\n \\n    \\n 4  \\n \\n \\nSupplemental Table 2.', 'Improvement in knockdown level by target accessibility for various \\x01Gdisruption  \\nthresholds, as estimated by using all of 101 shRNAs in the cDNA library \\nEnergy threshold in \\nkcal/mol for accessible sites  \\n(number of data points) Energy threshold in \\nkcal/mol for inaccessible \\nsites (number of data points)  Improvement in  \\naverage \\nknockdown % a t-test \\np-value b Wilcoxon \\nrank sum test  \\np-value b \\n\\x01Gdisruption  > \\x029 (20) \\x01Gdisruption  < \\x0220 (18) 14.13 0.0437 0.0384 \\n\\x01Gdisruption  > \\x0210 (29) \\x01Gdisruption  < \\x0219 (23) 17.19 0.0072 0.0067 \\n\\x01Gdisruption  > \\x0211 (31) \\x01Gdisruption  < \\x0218 (30) 15.62 0.0057 0.0066 \\n\\x01Gdisruption  > \\x0212 (37) \\x01Gdisruption  < \\x0217 (41) 12.41 0.0117 0.0151 \\n\\x01Gdisruption  > \\x0213 (41) \\x01Gdisruption  < \\x0216 (46) 11.57 0.0133 0.0198 \\na Improvement in average knockdown % = average knockdown % for accessible sites – average \\nknockdown % for inaccessible sites \\nb One-tailed test of significance for average knockdown % for accessible sites being greater than that for \\ninaccessible sites \\n  \\n \\n \\n \\n©2014 Landes Bioscience.', 'Do not distribute Resea Rch Pa PeR\\nwww.landesbioscience.com  RNa Biology  693RNa Biology 11:6, 693–701; June 2014; © 2014 Landes BioscienceResea Rch Pa PeR\\nIntroduction\\nIn animals and plants, microRNAs (miRNAs) are an abun-\\ndant class of small endogenous non-coding RNAs (ncRNAs) of \\n~22 nucleotides (nts) in length. Since the discovery of the first miRNA lin-4 in Caenorhabditis elegans  by the Ambros lab,\\n1 tens \\nof thousands of miRNAs have been identified and annotated in miRBase. 2 miRNAs are post-transcriptional regulators of diverse \\ndevelopmental and physiological processes. For target recogni -\\ntion and regulation in animals, miRNAs guide the RNA-induced silencing complex (RISC) by binding to partially complementary sequences typically in the 3 ′ untranslated regions (3 ′ UTRs) of \\nthe mRNAs, leading to translational repression and/or mRNA destabilization. 3\\nIdentification of the targets for miRNAs is essential for under -\\nstanding their regulatory functions.', 'Genetic studies have iden -\\ntified numerous targets for some worm miRNAs; however, the regulatory functions for the majority of worm miRNAs are not yet understood. Computational predictions can be helpful for target elucidation. Most of the prediction algorithms have incor -\\nporated the seed rule, i.e., the target site within 3′ UTR forms Watson-Crick (WC) pairs with bases at positions 2 through 7 or 8 of the 5 ′ end of the miRNA. 4 However, exceptions to the seed rule have been reported by both C. elegans  and mammalian stud -\\nies.5-12 Other proposed sequence features for enhancing targeting \\nspecificity include sequence conservation, strong base-pairing to the 3′ end of the miRNA, local AU content, and location of miRNA binding sites (near either end of the 3 ′ UTR is favor -\\nable). 13 Furthermore, the importance of target structural accessi -\\nbility for miRNA target recognition has also been demonstrated by several independent studies.', '14-18\\nIn recent years, experimental target identification methods \\nbased on crosslinking immunoprecipitation (CLIP) and high-throughput sequencing have been reported for mammalian sys -\\ntems and C. elegans . 19-21 CLIP methods involve UV irradiation for \\ncovalently crosslinking miRNA targets to the Argonaute (AGO) proteins, the catalytic components of the RISC complex. The crosslinked RNAs are treated by partial RNase digestion. The shortened RNAs are amplified by RT-PCR and then sequenced for the identification of AGO crosslinked sequences that con -\\ntain miRNA binding sites. For C. elegans , CLIP-derived clusters \\n(CDCs) of ~100 nts were identified for target binding sites of ALG-1, the AGO protein mainly responsible for miRNA func -\\ntion in worm. Importantly, the CLIP experiment was also per -\\nformed for alg-1  genetic mutants so that the background noise \\nin CDCs for wild-type worms could be removed.', \"The CLIP \\n*Correspondence to: Ye Ding; Email: yding@wadsworth.org\\nSubmitted: 03/12/2014; Revised: 04/07/2014; Accepted: 04/12/2014; Published Online: 04/25/2014http://dx.doi.org/10.4161/rna.28868MicroRNA binding sites in C. elegans  3' UTRs\\nchaochun Liu1, William a Ren nie1, Bibekanand Mallick1,†, shaveta Kanoria1, Dang Long1,‡, adam Wolenc1, c steven carm ack1, \\nand Ye Ding1,*\\n1Wadsworth cent er; New York stat e Department of heal th; cent er for Medical scie nce; alba ny, NY U sa; curr ent affiliations: †RNa Biology and Functional Genomics \\nLa\\nboratory; Department of Life scie\\nnce; National Institute of Technology; Odisha, India; ‡Biotechnology Department; Faculty of chem istry; Danang University of scie nce and \\nTechnology; Danang, Vietnam\\nKeywords:  microRNA, target binding site, prediction, GO analysis, developmental stage\\nMicroRN as (mi RNas) ar e post-transcriptional regulators of gene expression.\", 'sinc e the discovery of lin-4 , the founding \\nmember of the miRN a f\\namily, over 360 miRN as ha\\nve been identified for Caenorhabditis elegans  (C. elegans ). Prediction \\nand validation of targets are essential for elucidation of regulatory functions of these miRN as. Fo\\nr C. elegans , crosslinking \\nimmunoprecipitation ( cLIP) h\\nas been successfully performed for the identification of target mRN a se\\nquences bound \\nby argo\\nnaute protein aLG\\n-1. In addition, reliable annotation of the 3 ′ untranslated regions (3 ′ UTRs) as well as develop -\\nmental stage-specific expression profiles for both miRN as an\\nd 3′ UTR isoforms are available. By utilizing these data, we \\ndeveloped statistical models and bioinformatics tools for both transcriptome-scale and developmental stage-specific \\npredictions of miRN a b\\ninding sites in C. elegan s 3′ UTRs.', 'In performance evaluation via cross validation on the aLG\\n-1 cLIP \\nda\\nta, the models were found to offer major improvements over established algorithms for predicting both seed sites and \\nseedless sites. In particular, our top-ranked predictions have a substantially higher true positive rate, suggesting a much higher likelihood of positive experimental validation. a ge\\nne ontology analysis of stage-specific predictions suggests \\nthat miRN as are i\\nnvolved in dynamic regulation of biological functions during C. elegans development. In particular, miR -\\nNas pre\\nferentially target genes related to development, cell cycle, trafficking, and cell signaling processes. a da\\ntabase \\nfor both transcriptome-scale and stage-specific predictions and software for implementing the prediction models are available through the \\nsfold\\n web server at http://sfold.wadsworth.org. ©2014 Landes Bioscience.', 'Do not distribute694 RNa Biology  Volume 11 Issue 6technique not only provides high-resolution data with respect to \\nthe precise locations of the binding sites, but also is powerful for revealing the presence of “seedless” sites (non-canonical sites). In addition to data from ALG-1 CLIP, improved annotation has been established for C. elegans  3′ UTR isoforms expressed dur -\\ning different developmental stages, i.e., embryonic, L1, L2, L3, L4, adult hermaphrodite, and male. 22,23 Moreover, developmental \\nstage-specific expression profile of worm miRNAs has become available. 24\\nIn this work, we performed a comprehensive enrichment anal-\\nysis of target site features for both seed and seedless sites iden -\\ntified from ALG-1 CDCs. We used enriched miRNA binding site features for the development of logistic models for prediction of miRNA binding sites. We assessed accuracy of predictions by cross validation and compared the performance with established algorithms.', 'We used the models to make transcriptome-scale and developmental stage-specific predictions of miRNA binding sites in C. elegans . We performed a gene ontology (GO) analy -\\nsis for stage-specific predictions to examine biological functions that are regulated by miRNAs during C. elegans  development. For dissemination of the results, we have developed both data -\\nbase and software tools that are freely available to the scientific community. Results\\nIdentification of enriched target site featuresFor each of the sequence, thermodynamic, and target struc -\\nture features ( Table 1 ), we performed enrichment analysis to \\nidentify features enriched in ALG-1 CDCs. Among the seed sites, 14\\n 355 (\\n11%) are within CDCs and referred to as the IP+ \\nseed sites, the other 112  589 (\\n89%) are referred to as the IP- seed \\nsites, indicating that seed alone is a poor predictor with high false-positive rate. Features enriched for IP+ seed sites include site accessibility ( Fig.', '1A)\\n, upstream accessibility (window size of 10 \\nnt, Fig. 1B), 6\\nmer and 8mer seed ( Fig. 1C), s\\nite conservation \\nand seed conservation ( Fig. 1D), s\\need accessibility, Δ Gnucl, and ΔGhybrid. Among the seedless sites, 461  798 a re in the IP+ set \\n(within CDCs) and 3  820 416 a\\nre in the IP- set (outside CDCs). The enriched features for IP+ seedless sites include site accessibil -\\nity (Fig. 1E), s\\nite conservation ( Fig. 1F), u\\npstream accessibility \\n(10 nt), downstream accessibility (10 nt), 3 ′ base-pairing, Δ Gnucl, \\nand ΔGhybrid. These enriched features were used for the develop -\\nment of our logistic prediction models.', 'miRNA binding site prediction and performance evaluationFor performance evaluation, we constructed a receiver opera -\\ntor characteristic (ROC) curve for plotting the true positive rate (TPR = sensitivity) against the false positive rate (FPR = 1-specificity) by varying the threshold of a prediction score, e.g., logistic probability of our model, context score of TargetScan,\\n27 \\nenergy score of miRanda,28 or PITA.16 The Youden’s J statistic29 \\ncomputed by (TPR −FPR) was used as the overall measure of \\nperformance. For seed sites in the 3 ′ UTRs, we compared our predictions \\nwith TargetScan, miRanda, and PITA. At a comparable FPR level, our logistic model has a substantially higher TPR than TargetScan, miRanda, and PITA ( Fig. 2A)\\n. For a logistic probability threshold \\nof 0.5, the improvement by the logistic model on Youden’s J sta -\\ntistic is about 0.15 over PITA, 0.19 over miRanda, and 0.20 over TargetScan ( Fig. 2B)\\n.', 'Among the three established algorithms, \\nTargetScan is the worst performer. This is because TargetScan is primarily based on the predictions of 7mer and 8mer seed sites, and only 8mer is marginally enriched in the IP+ set ( Fig. 1C). Fo\\nr seedless site prediction, we only compared our model with \\nPITA and miRanda, as TargetScan does not predict seedless sites. The predictions are similar to seed site predictions in the trends of receiver operating characteristic (ROC) curves and the Youden’s J statistic, with a higher degree of improvement ( Fig. 2C an\\nd D). In particular, for the class of seedless sites with one G•U pair or one mismatch in the seed complementary region, our logistic model was also found to have major improvement over PITA and miRanda (Fig. 2E a\\nnd F). Since top-ranked predictions are of high interest for experi -\\nmental validation, we compared the true positive rates of top-ranked predictions (top 1% to 50%) by our logistic models, Table\\xa01.', 'Features computed for each potential miRN a bi nding site (seed or seedless)*\\nType and name of feature Description\\nSequence\\n(1) seed; (2) aU cont\\nent; (3) \\nsite loca\\ntion; (4) miRN a 3′ base \\npair\\ning(1) Offset 6mer, 6mer, 7mer- a1, 7mer-m8 and 8mer seed sit\\nes;13 (2) percentage of aU for the block (of 5 nt , 10 nt,…, \\n30 nt) upstream or downstream of the binding site; (3) proximity to the ends of 3′  UTR; (4) presence of contiguous \\nWc base pairing f\\nor miRN a nt positions 12–1713,27\\nThermodynamic and target structure(1) ΔG\\nhybrid; (2) ΔGnucl; (3) ΔGtotal;\\n(4) seed accessibilit\\ny; (5) site \\nac\\ncessibility;\\n(6) Upstream accessibility; (7) Downstream accessibility(1) ΔG\\nhybrid is the measure of stability for miRN a:target hybrid as computed by RN ahybrid;26 (2) ΔGnucl measures \\nthe potential of nucleation for miRN a-target hybr\\nidization;18,36 (3) ΔGtotal measures the total energy change of \\nthe hybridization;17 (4–7) structural accessibility is evaluated by a probabilistic measure of single-strandedness \\nfor a block of nucleotides and is computed by using structures predicted by sfold39,40 for the binding site, \\ncomplementary seed region within the binding site, and for the block upstream or downstream of the binding site\\nConservation(1) \\nsite c\\nonservation score;\\n(2) seed/off-seed conserv\\nation \\nscoreThe conservation score by the Phast cons progr\\nam25 through multiple-sequence alignments of five other \\nnematode genomes to the C.', 'elegans genome (ce6) was used to measure conservation for a binding site, and the complementary seed and off-seed region within the binding site\\n*The details of feature computation can be found in reference \\n34. ©2014 Landes Bioscience. Do not distributewww.landesbioscience.com  RNa Biology  695TargetScan, PITA, and miRanda. For a given set of predicted \\ntop-ranked sites, the true positive rate is computed by the num -\\nber of true positive sites (residing in ALG-1 CDCs) divided by the total number of the top-ranked sites. For seed sites in the 3′ UTRs, our logistic model was found to have a substantially \\nhigher true positive rate than PITA, miRanda, and TargetScan, especially for highly ranked predictions ( Fig. 3A)\\n. E.g., for top \\n2% predictions, the true positive rate of the logistic model is about 7% higher than PITA, 9% higher than miRanda, and 11% higher than TargetScan.', 'For either all seedless sites or the class of seedless sites with one G•U pair or one mismatch in the seed complementary region, we also observed substantial improvements by our logistic model over PITA and miRanda  \\n(Fig. 3B a\\nnd C ). Transcriptome-scale and stage-specific predictionsWe applied our logistic models to transcriptome-scale predic -\\ntions of miRNA binding sites in C. elegans . This included 368 \\nmiRNAs in miRBase Release 19\\n2 and 24  503 3′ U TR isoforms.22 \\nWe predicted 429  072 s\\need sites and 14  921 597 s\\needless sites. Each of these sites either resides within a CDC or has a logistic probability above 0.5. The miRNAs and the 3′ UTR isoforms are expressed in different developmental stages, i.e., embryonic, L1, L2, L3, L4, adult hermaphrodite, and male.', 'Accordingly, we further processed transcriptome-scale predictions for stage-specific predictions by collecting predicted sites for co-expressed miRNA:3 ′ UTR pairs in each of the seven stages.Stage-specific patterns of miRNA:target interactions during \\ndevelopment\\nWe next investigated the stage-specific patterns of \\nmiRNA:target interactions during C. elegans  development. We first defined a positive miRNA:target interaction if the miRNA:target pair has at least one seed site with a logistic prob -\\nability above 0.5, or one seedless site with a probability above 0.6. From all of positive miRNA:target interactions for each develop -\\nmental stage, we assembled the set of the targeted genes and the set of their regulating miRNAs. For a more stringent definition of a positive interaction, we also used a probability of above 0.6 for a seed site and above 0.7 for a seedless site. For either defini -\\ntion, every abundant miRNA for each of the seven stages has at least one target.', 'Among all of the 119 miRNAs expressed in at least one stage, 84 (~70.6%) are shared by all the seven stages, which is consistent with the previous observation that most miR -\\nNAs are present at steady-state levels during C. elegans  develop -\\nment. 30 For probabilities of 0.5 and 0.6, we identified 4583, 3895, \\n4540, 4143, 4330, 2086, and 3741 target genes for embryonic, L1, L2, L3, L4, adult hermaphrodite, and male stages, respec -\\ntively. Among 7745 genes targeted by miRNAs in at least one stage, 1062 (~13.7%) are shared by all seven stages. For prob -\\nabilities of 0.6 and 0.7, we identified 2736, 2379, 2869, 2577, 2625, 1205, and 2133 target genes for embryonic, L1, L2, L3, L4, adult hermaphrodite, and male stages, respectively. Among 4633 genes targeted by miRNAs in at least one stage, 644 (~13.9%) are Figure\\xa01.', 'enri chment of representative site features: ( A) site accessibility for seed sites; ( B) upstream accessibility (window size of 10 nt) for seed sites; ( C) \\ntype of miRN a ta\\nrget seed sites; (D ) percentage (Y-axis) of sites/seed/off-seed regions with conservation scores greater than or equal to a pre-specified \\nthreshold (X-axis), in the IP+ seed set or the IP- seed set (dashed line corresponding to a threshold of 0.57 previously used for defining conservation25,41); \\n(E) site accessibility for seedless sites; ( F) percentage (Y-axis) of seedless sites with conservation scores greater than or equal to a pre-specified threshold \\n(X-axis), in the IP+ set or the IP- set. ©2014 Landes Bioscience. Do not distribute696 RNa Biology  Volume 11 Issue 6shared by all seven stages. The high number of common targets \\nis consistent with the previous observation that many miRNA targets seem to be stably and continuously regulated during  \\nC. elegans  development.', '31\\nFor each stage, using the common targets, we identified the \\nset of remaining targets that were not shared by all seven stages. To explore functional themes among the common target genes and the remaining target gene sets, we searched for enriched GO annotations, focusing on GO terms with P  value (by hyper-\\ngeometric distribution) under 0.01 and percentage (number of genes associated with the GO term divided by total number of genes of interest) above 0.01 ( Tables S1 and S2 ).Figure\\xa02. Performance comparison of logistic models with three established algorithms for site predictions in 3 ′ UTRs (dashed diagonal line for random \\npredictions). RO c c\\nurve and Youden’s J statistic are shown for the predictions of seed sites ( A and B ), seedless sites ( C and D ), and seedless site with \\none G•U pair or one mismatch within seed complementary region ( E and F ). The color-matched dots on RO c cu\\nrves correspond to a logistic probability \\nthreshold of 0.5.', 'The rectangle, triangle and square correspond to the best-performing score threshold (according to Youden’s J statistic) for Target scan\\n, \\nPITa and m\\niRanda, respectively. ©2014 Landes Bioscience. Do not distributewww.landesbioscience.com  RNa Biology  697Analysis of targets common to all seven stages\\nFor the two definitions of positive interactions, the enriched \\nGO terms for the common target gene set are highly overlapped so that the key findings are the same as summarized below. For biological processes, the most enriched GO terms are related to development, cell cycle, or trafficking, e.g., nematode lar -\\nval development, embryonic development ending in birth or egg hatching, growth, positive regulation of growth rate, body morphogenesis, locomotion, and positive regulation of locomo -\\ntion.', 'For molecular functions, the most enriched GO terms are related to protein–nucleic acid and protein–protein interactions in cell signaling processes, e.g., structural constituent of ribo -\\nsome, nucleotide binding, GTPase activity, and ATP binding. This is consistent with a previous conclusion that miRNAs preferentially target genes involved in signaling processes dur -\\ning C. elegans  development. 31 For cellular components, the most \\nenriched GO terms are related to cytopoiesis and protein syn -\\nthesis, e.g., cytoplasm, intracellular, ribosome, and ribonucleo -\\nprotein complex.Analysis of remaining targets for each of seven stagesFor the two definitions of positive interactions, the enriched \\nGO terms for the remaining target set for each of the seven stages are largely overlapped such that we have the same conclusions below.', 'For the remaining target gene sets, temporal pattern tran -\\nsition of miRNA:target interactions is evident from the enriched GO terms of each stage ( Tables S1 and S2 ), further supporting \\nthat the miRNA-mediated regulation network is highly dynamic during C. elegans  development. 31 Interestingly, the L1 and male \\nstages have substantially fewer enriched GO terms than other stages, and a large portion of these terms are not enriched in the preceding or following stage. This suggests that the embryo to L1, the L1 to L2, and the L4 to male transitions involve substan -\\ntial changes in miRNA-mediated regulation of gene expression. This conclusion for the embryo to L1 and the L1 to L2 transition is consistent with a previous insight suggested by stage-specific miRNA expression profiles. 31 Although a majority of enriched \\nGO terms are preserved over the L2 to L3, the L3 to L4, and the L4 to adult hermaphrodite transition, a minority of enriched Figure\\xa03.', 'True positive rate comparison for top-ranked (from top 1% to top 50%) predictions of seed sites ( A), seedless sites ( B), seedless sites with one \\nG•U pair or one mismatch in the seed complementary region ( C) in 3′ UTRs. ©2014 Landes Bioscience. Do not distribute698 RNa Biology  Volume 11 Issue 6GO terms vary in these transitions, thereby indicating dynamic \\ntemporal patterns of miRNA:target interactions during develop -\\nment. The contrast between substantial changes in the L4 to male transition and the relatively minor changes in the L4 to adult her -\\nmaphrodite transition indicates that genes related to hermaph -\\nrodite genitalia development and sex differentiation are under strong miRNA regulation, e.g., those genes associated with the enriched GO term of negative regulation of vulval development (Table S1 ).', 'Moreover, for the remaining target gene set for each \\nof the seven stages, enriched GO terms are mainly for biologi -\\ncal processes and rarely for cellular components ( Tables S1 and \\nS2). The GO terms for biological processes are largely related \\nto development, cell cycle, or trafficking, and the GO terms for molecular function are largely related to protein–nucleic acid and protein–protein interactions in cell signaling processes. These observations are the same as the common target gene set.Figure\\xa04. (A) sTar MirDB search result for binding sites of cel-miR-796 on ncs-2  3′ UTR isoforms, with “3 ′ UTR-seedless” option selected for output display; \\n(B) hybrid diagram of a seed site; ( C) hybrid diagram of a seedless site. ©2014 Landes Bioscience.', 'Do not distributewww.landesbioscience.com  RNa Biology  699Database and software tools\\nFor dissemination of the results, transcriptome-scale and stage-\\nspecific predictions are freely available from STarMirDB (http://sfold.wadsworth.org/starmirDB.php), a web searchable database, with an indicator showing whether a site is supported by the ALG-1 CLIP study. We have also implemented the prediction models into the STarMir module of the Sfold web applications\\n37 \\n(http://sfold.wadsworth.org/cgi-bin/starmirWeb.pl), allowing users to submit any miRNA and mRNA sequences for prediction of miRNA binding sites by the models. Information for predicted sites includes site features, a logistic probability as a measure of confidence, and a high-resolution diagram of hybrid conforma -\\ntion.', 'As an illustration of the output from a database search, using cel-miR-796 and ncs-2  (or WBGene00003564) as search \\nkeywords returns a list of seed sites and seedless sites for miR-796 on different ncs-2  3′ UTR isoforms for various developmental \\nstages ( Fig. 4A)\\n. For example, for isoform WBGene00003564_\\nF10G8.5_ncs-2.a_chrI_10034020_10033271, one 7mer-A1 seed site, and one seedless site have high conservation scores of 0.9658 and 0.9756, and logistic probabilities of 0.67 and 0.76, respec -\\ntively. The hybrid diagrams for two example sites are shown in Figure\\n 4B a\\nnd C , where the seed region (nt 2–8) of the miRNA \\nis shown in red color. Discussion\\nAmong various types of seed sites, mammalian miRNA \\ntargeting studies13,27,32,33 established that 8mer seed is the most \\neffective, followed by 7mer-m8, 7mer-A1, 6mer, and offset 6mer. Our analysis of five mammalian CLIP data sets also supported the effectiveness of 8mer and 7mer seed sites.', '34 In contrast, \\nfrom our enrichment analysis of the worm ALG-1 CLIP data, 6mer was found to be the only substantially enriched seed type (Fig. 1C). T\\nhis is a somewhat surprising finding, however, it is \\nconsistent with two observations on miRNA:target interactions in C. elegans . For numerous genetically verified targets, e.g., lin -\\n41, RAS, and pha -4 targeted by let-7 ,\\n6,9,10 and lin-14  targeted by \\nlin-4 ,35 a perfect seed match is absent at the functional sites of \\ninteraction. In a study on the interaction between lsy-6  and its \\ntarget cog-1 , it was shown by single nucleotide mutations that \\nG•U base-pairing in the seed region could be well tolerated. The miRNA machinery in C. elegans  may be more tolerant of G•U \\nbase-pairing and mismatches in the seed region. Furthermore, the number of enriched features and the enrichment signals are generally less than those observed in our mammalian study. These necessitate development of prediction models specific for C. elegan s.', 'Our prediction approach computes a logistic probability as a \\nmeasure of confidence for predictions. The predicted sites with higher probabilities are expected to have a greater chance for posi -\\ntive experimental validation. In particular, the top-ranked pre -\\ndictions by our approach have higher likelihood to be positive than the top-ranked predictions of the established algorithms (Fig. 3). O\\nur approach is a successful application of our previous \\nmethodology34 to C. elegans  ALG-1 CLIP data, and was shown to provide new models specifically for improved predictions of miRNA binding sites in C. elegans . The ALG-1 CLIP data are mainly for the L4 stage. However, \\nbecause the miRNA machinery is expected to be the same throughout the C. elegans  development, rules on miRNA target -\\ning learned from the L4 stage can be generalized to other stages for binding site predictions. Stage-specific CLIP data would be ideal, however, such data are not available.', 'Our GO analysis revealed a dynamic stage-dependent pattern \\nof miRNA:target interactions. In addition to significant changes at the embryo to L1 and the L1 to L2 transitions that confirm a previous observation,\\n31 we observed a significant change at the \\nL4 to male transition and relatively minor changes at the L2 to L3, the L3 to L4, and the L4 to adult hermaphrodite transitions. These further support the previous conclusion that miRNA-mediated regulation is stable and continuous via coordinately targeting or avoiding genes involved in certain biological func -\\ntions during C. elegans  development. 31 Furthermore, we observed \\nthat genes under strong miRNA regulation during development include those related to biological processes of development, cell cycle, trafficking, and sex differentiation, and those related to molecular functions of protein–nucleic acid and protein–protein interactions in cell signaling processes.', 'The observation on cell signaling processes further supports a previous conclusion. 31\\nFor C. elegans, currently the only available CLIP data set is \\nfrom the ALG-1 study.19 Therefore, model testing by other inde -\\npendent C. elegans  CLIP data are not yet possible. Nevertheless, \\nthe cross validation strategy enables an assessment of the gen -\\neralizability of our model to other independent data sets. The CLIP technique provides information on miRNA binding sites, but not functional outcomes (e.g., readout from a reporter) due to miRNA binding. Accordingly, our CLIP-derived models are limited to predictions of miRNA binding sites. Extension of the models for predicting functional outcomes will require analysis and modeling of high-throughput functional data. In conclusion, we have performed a comprehensive enrich -\\nment analysis of data from the ALG-1 CLIP study. The findings allowed us to develop new models for the prediction of worm miRNA binding sites.', 'The advantage of our approach is the utili -\\nzation of recent experimental data for ALG-1 CLIP, 3 ′ UTR anno -\\ntation, and stage-specific expression of miRNAs and mRNAs. In performance evaluation via cross validation on the ALG-1 CLIP data, the models were found to offer major improvements over established algorithms. For future studies, independent CLIP data (preferably for multiple C. elegans  developmental stages) \\nwould allow inter-data set validation. Extensive experimental testing of model predictions would be needed to further assess predictive improvements. Useful information from a CLIP study is limited to abundantly expressed miRNAs and transcripts in the experimental system. Our analysis of ALG-1 CLIP data included 113 abundant miRNAs and 3093 genes in worm. The prediction models enabled transcriptome-scale predictions for 368 miRNAs and 24\\n 503 3′\\n UTR isoforms, as well as stage-specific predictions \\nfor seven C. elegans  developmental stages.', 'The stage-specific GO \\nanalysis revealed that miRNAs regulate diverse biological func -\\ntions during C. elegans  development. The software and database \\n©2014 Landes Bioscience. Do not distribute700 RNa Biology  Volume 11 Issue 6tools will greatly complement the ALG-1 CLIP data for studies \\nof miRNA regulation in C. elegans . Materials and Methods\\nDescription and processing of C. elegans  CLIP data\\nThe C. elegans  CLIP study identified about 4800 CDCs that \\nare unique to the wild-type (WT) worms at the fourth larval stage (L4 stage), in comparison to clusters from ALG-1 genetic mutant worms. 19 These CDCs represent 3093 genes, about 20% \\nof protein-coding genes expressed in L4 stage. Based on the anno -\\ntation from a previous study38 and the NCBI map viewer for ce6 \\ngenome, we compiled 13  356 3′\\n UTR isoform sequences (length \\n≥ 30 nts) for all expressed transcripts in the CLIP experiment.', 'The 113 most abundant miRNAs (sequencing read number ≥ 10) for the WT worms were used in our analysis. Overview of framework for CLIP data analysis, prediction \\nmodel training, and testing\\nFor the development of new prediction models specific for \\nworm using the C. elegans  ALG-1 CLIP data, we adopted the \\ncore methodology that was found to be successful in our recent mammalian study. 34 First, the potential miRNA binding sites \\nwere predicted by RNAhybrid program.26 Those sites residing \\nwithin an ALG-1 CDC were classified as the IP+ sites, while the remaining sites were categorized as the IP- sites. Second, the site features described in Table 1  were computed for each seed or \\nseedless site. For each feature, the degree of enrichment was mea -\\nsured by the ratio of the odds of occurrence in the IP+ set to that of IP- set. A ratio above one indicates enrichment of the feature, whereas a ratio below one indicates depletion.', 'Enriched features were considered important for miRNA targeting in worm, and were used for the development of prediction models, one for seed sites and one for seedless sites. Third, nonlinear logistic regression was used for model development. The logistic model outputs a probability as a measure of confidence for a predicted site. Finally, the standard 10-fold cross validation strategy was used to test the performance of our prediction models on the ALG-1 CLIP data. According to this strategy, the data set was randomly divided into 10 subsets of equal size. During each of the 10 iterations, one subset was set aside for performance testing of the model trained on the collection of the other nine subsets. This model-training-testing process was repeated 10 times, and the mean prediction accuracy is averaged over the 10 iterations. Although currently only one CLIP data set is available for C.', 'elegans , the cross vali -\\ndation strategy enables an assessment of the generalizability of our model to other independent data sets. TargetScan,\\n27 PITA,16 \\nand miRanda,28 which can be run locally with available software, \\nwere used for comparison. The computational procedures here are essentially the same as previously used. 34\\nTranscriptome predictionWe used our models for transcriptome-scale predictions \\nof worm miRNA binding sites. To this end, we included all 368 worm miRNAs in miRBase Release 19,\\n2 and considered the recently annotated 3 ′ UTRs.22 We compiled 24  503 3′  \\nUTR isoforms (sequence length ≥  30 nts) for ~13  830 g\\nenes in \\nWormBase WS190. These 3 ′ UTR isoforms are frequently and \\ndifferentially expressed during different C. elegans  developmen -\\ntal stages.', 'Stage-specific prediction and GO analysisFrom the 24\\n 503 3′ U\\nTR isoforms with stage-specific \\nexpression information, we assembled 8806 isoforms for the embryonic stage, 6568 for the L1 stage, 7341 for the L2 stage, 6495 for the L3 stage, 7049 for the L4 stage, 3213 for the adult hermaphrodite stage, and 6738 for the male stage. Among the 368 worm miRNAs, we assembled 97 abundant miRNAs for the embryonic stage, 97 for the L1 stage, 101 for the L2 stage, 99 for the L3 stage, 100 for the L4 stage, 101 for the adult hermaphrodite stage, and 109 for the male stage. A miRNA is considered abundant if its read number is at least 10 from the sequencing analysis. 24 For each developmental stage, we \\nassembled stage-specific predictions for co-expressing miRNAs and mRNA isoforms, and performed GO analysis. The GO association file was downloaded from WormBase FTP with ver -\\nsion WS190. Disclosure of Potential Conflicts of Interest\\nNo potential conflicts of interest were disclosed.', 'Acknowledgments\\nThe Computational Molecular Biology and Statistics Core at \\nthe Wadsworth Center is acknowledged for supporting comput -\\ning resources for this work. The authors thank Marco Mangone and the lab of Gene W Yeo for providing additional data. This work was supported in part by National Science Foundation (DBI-0650991 to Ding Y), National Institutes of Health grant (GM099811 to Ding Y), and Nafosted Fund of Vietnam (102.03-2010.04 to Long D). Author Contributions\\nDing Y conceived and supervised the study. Liu C and \\nMallick B complied and processed the data from C. elegans  \\nstudies on ALG-1 CLIP, 3 ′ UTR annotation and stage-specific \\nexpression of miRNAs and 3 ′ UTR isoforms, and performed \\nfeature enrichment analysis. Liu C developed prediction mod -\\nels, performed model validation, performance comparison, and GO analysis. Rennie W performed both database and soft -\\nware implementation. Kanoria S performed testing of database interface.', 'Long D and Wolenc A wrote the initial software for the computation of several target site features, and Carmack C provided hardware and system support for both cluster com -\\nputing and the Sfold web server. Liu C and Ding Y wrote the paper with contributions from all authors. All authors read and approved the final manuscript. Supplemental Materials\\nSupplemental materials may be found here: www.landesbioscience.com/journals/rnabiology/article/28868/\\n©2014 Landes Bioscience. Do not distributewww.landesbioscience.com  RNa Biology  701References\\n1. Lee RC, Feinbaum RL, Ambros V. The C. elegans \\nhe\\nterochronic gene lin-4 encodes small RNAs \\nwith antisense complementarity to lin-14. Cell 1993; 75:843-54;  PMID:8252621 ; http://dx.doi. org/10.1016/0092-8674(93)90529-Y\\n2. Koz\\nomara A, Griffiths-Jones S. miRBase: inte -\\ngrating microRNA annotation and deep-sequenc -\\ning data. Nucleic Acids Res 2011; 39:D152-7; PMID:21037258 ; http://dx.doi.org/10.1093/nar/\\ngkq1027\\n3. Fab\\nian MR, Sonenberg N.', 'The mechanics of \\nmiRNA-mediated gene silencing: a look under the hood of miRISC. Nat Struct Mol Biol 2012; 19:586-93; PMID:22664986 ; http://dx.doi.org/10.1038/\\nnsmb.2296\\n4. Lew\\nis BP, Burge CB, Bartel DP. Conserved seed pair -\\ning, often flanked by adenosines, indicates that thou -\\nsands of human genes are microRNA targets. Cell 2005; 120:15-20;  PMID:15652477 ; http://dx.doi. org/10.1016/j.cell.2004.12.035\\n5. Tay Y\\n, Zhang J, Thomson AM, Lim B, Rigoutsos I. MicroRNAs to Nanog, Oct4 and Sox2 coding regions modulate embryonic stem cell differentiation. Nature 2008; 455:1124-8;  PMID:18806776 ; http://dx.doi. org/10.1038/nature07299\\n6. Vel\\nla MC, Choi EY, Lin SY, Reinert K, Slack FJ. The \\nC. elegans microRNA let-7 binds to imperfect let-7 complementary sites from the lin-41 3’UTR. Genes Dev 2004; 18:132-7; PMID:14729570 ; http://\\ndx.doi.org/10.1101/gad.1165404\\n7. Did\\niano D, Hobert O. Perfect seed pairing is not \\na generally reliable predictor for miRNA-target interactions.', 'Nat Struct Mol Biol 2006; 13:849-51; PMID:16921378 ; http://dx.doi.org/10.1038/\\nnsmb1138\\n8. Loe\\nb GB, Khan AA, Canner D, Hiatt JB, Shendure J, \\nDarnell RB, Leslie CS, Rudensky AY. Transcriptome-wide miR-155 binding map reveals widespread noncanonical microRNA targeting. Mol Cell 2012; 48:760-70;  PMID:23142080 ; http://dx.doi. org/10.1016/j.molcel.2012.10.002\\n9. Gro\\nsshans H, Johnson T, Reinert KL, Gerstein M, \\nSlack FJ. The temporal patterning microRNA let-7 regulates several transcription factors at the larval to adult transition in C. elegans. Dev Cell 2005; 8:321-30; PMID :15737928 ; http://dx.doi.org/10.1016/j. devcel.2004.12.019\\n10. Joh\\nnson SM, Grosshans H, Shingara J, Byrom M, \\nJarvis R, Cheng A, Labourier E, Reinert KL, Brown D, Slack FJ. RAS is regulated by the let-7 microRNA family. Cell 2005; 120:635-47; PMID :15766527 ; \\nhttp://dx.doi.org/10.1016/j.cell.2005.01.014\\n11.', 'Lal A\\n, Navarro F, Maher CA, Maliszewski LE, Yan \\nN, O’Day E, Chowdhury D, Dykxhoorn DM, Tsai P, Hofmann O, et al. miR-24 Inhibits cell prolifera -\\ntion by targeting E2F2, MYC, and other cell-cycle genes via binding to “seedless” 3’UTR microRNA recognition elements. Mol Cell 2009; 35:610-25; PMID:19748357 ; http://dx.doi.org/10.1016/j. molcel.2009.08.020\\n12. Kho\\nrshid M, Hausser J, Zavolan M, van Nimwegen \\nE. A biophysical miRNA-mRNA interaction model infers canonical and noncanonical targets. Nat Methods 2013; 10:253-5; PMID:23334102 ; http://\\ndx.doi.org/10.1038/nmeth.2341\\n13. Bar\\ntel DP. MicroRNAs: target recognition and \\nregulatory functions. Cell 2009; 136:215-33; PMID:19167326 ; http://dx.doi.org/10.1016/j. cell.2009.01.00214. Zha\\no Y, Samal E, Srivastava D. Serum response fac -\\ntor regulates a muscle-specific microRNA that targets Hand2 during cardiogenesis. Nature 2005; 436:214-20; PMID :15951802 ; http://dx.doi.org/10.1038/\\nnature03817\\n15. Rob\\nins H, Li Y, Padgett RW.', 'Incorporating structure \\nto predict microRNA targets. Proc Natl Acad Sci U S A 2005; 102:4006-9; PMID:15738385 ; http://\\ndx.doi.org/10.1073/pnas.0500775102\\n16. Ker\\ntesz M, Iovino N, Unnerstall U, Gaul U, Segal \\nE. The role of site accessibility in microRNA tar -\\nget recognition. Nat Genet 2007; 39:1278-84; PMID:17893677 ; http://dx.doi.org/10.1038/ng2135\\n17. Lon\\ng D, Lee R, Williams P, Chan CY, Ambros V, \\nDing Y. Potent effect of target structure on microRNA function. Nat Struct Mol Biol 2007; 14:287-94; PMID:17401373 ; http://dx.doi.org/10.1038/\\nnsmb1226\\n18. Zis\\noulis DG, Lovci MT, Wilbert ML, Hutt KR, \\nLiang TY, Pasquinelli AE, Yeo GW. Comprehensive discovery of endogenous Argonaute binding sites in Caenorhabditis elegans. Nat Struct Mol Biol 2010; 17:173-9;  PMID:20062054 ; http://dx.doi. org/10.1038/nsmb.1745\\n19. Chi S\\nW, Zang JB, Mele A, Darnell RB. Argonaute \\nHITS-CLIP decodes microRNA-mRNA interaction maps. Nature 2009; 460:479-86; PMID :19536157\\n20.', 'Haf\\nner M, Landthaler M, Burger L, Khorshid M, \\nHausser J, Berninger P, Rothballer A, Ascano M Jr., Jungkamp AC, Munschauer M, et al. Transcriptome-wide identification of RNA-binding protein and microRNA target sites by PAR-CLIP. Cell 2010; 141:129-41;  PMID:20371350 ; http://dx.doi. org/10.1016/j.cell.2010.03.009\\n21. Man\\ngone M, Manoharan AP, Thierry-Mieg D, \\nThierry-Mieg J, Han T, Mackowiak SD, Mis E, Zegar C, Gutwein MR, Khivansara V, et al. The landscape of C. elegans 3’UTRs. Science 2010; 329:432-5; PMID:20522740 ; http://dx.doi.org/10.1126/\\nscience.1191244\\n22. Jan C\\nH, Friedman RC, Ruby JG, Bartel \\nDP. Formation, regulation and evolution of Caenorhabditis elegans 3’UTRs. Nature 2011; 469:97-101;  PMID:21085120 ; http://dx.doi. org/10.1038/nature09616\\n23. Kat\\no M, de Lencastre A, Pincus Z, Slack FJ. Dynamic \\nexpression of small non-coding RNAs, including novel microRNAs and piRNAs/21U-RNAs, dur -\\ning Caenorhabditis elegans development.', 'Genome Biol 2009; 10:R54; PMID:19460142 ; http://dx.doi. org/10.1186/gb-2009-10-5-r54\\n24. Sie\\npel A, Bejerano G, Pedersen JS, Hinrichs AS, Hou \\nM, Rosenbloom K, Clawson H, Spieth J, Hillier LW, Richards S, et al. Evolutionarily conserved ele -\\nments in vertebrate, insect, worm, and yeast genomes. Genome Res 2005; 15:1034-50; PMID:16024819 ; \\nhttp://dx.doi.org/10.1101/gr.3715005\\n25. Reh\\nmsmeier M, Steffen P, Hochsmann M, Giegerich \\nR. Fast and effective prediction of microRNA/target duplexes. RNA 2004; 10:1507-17; PMID :15383676 ; \\nhttp://dx.doi.org/10.1261/rna.5248604\\n26. Gri\\nmson A, Farh KK, Johnston WK, Garrett-Engele \\nP, Lim LP, Bartel DP. MicroRNA targeting specific -\\nity in mammals: determinants beyond seed pairing. Mol Cell 2007; 27:91-105; PMID:17612493 ; http://\\ndx.doi.org/10.1016/j.molcel.2007.06.017\\n27. Enr\\night AJ, John B, Gaul U, Tuschl T, Sander \\nC, Marks DS. MicroRNA targets in Drosophila. Genome Biol 2003; 5:R1; PMID:14709173 ; http://\\ndx.doi.org/10.1186/gb-2003-5-1-r128.', 'You\\nden WJ. Index for rating diagnostic tests. Cancer 1950; 3:32-5; PMID:15405679 ; http://\\ndx.doi.org/10.1002/1097-0142(1950)3:1<32::AID-CNCR2820030106>3.0.CO;2-3\\n29. Lim L\\nP, Lau NC, Weinstein EG, Abdelhakim A, \\nYekta S, Rhoades MW, Burge CB, Bartel DP. The microRNAs of Caenorhabditis elegans. Genes Dev 2003; 17:991-1008;  PMID:12672692 ; http://dx.doi. org/10.1101/gad.1074403\\n30. Zha\\nng L, Hammell M, Kudlow BA, Ambros V, \\nHan M. Systematic analysis of dynamic miRNA-target interactions during C. elegans development. Development 2009; 136:3043-55; PMID:19675127 ; \\nhttp://dx.doi.org/10.1242/dev.039008\\n31. Bae\\nk D, Villén J, Shin C, Camargo FD, Gygi SP, \\nBartel DP. The impact of microRNAs on protein output. Nature 2008; 455:64-71; PMID:18668037 ; \\nhttp://dx.doi.org/10.1038/nature07242\\n32. Sel\\nbach M, Schwanhäusser B, Thierfelder N, Fang \\nZ, Khanin R, Rajewsky N. Widespread changes in protein synthesis induced by microRNAs. Nature 2008; 455:58-63;  PMID:18668040 ; http://dx.doi.', 'org/10.1038/nature07228\\n33. Liu C\\n, Mallick B, Long D, Rennie WA, Wolenc A, \\nCarmack CS, Ding Y. CLIP-based prediction of mammalian microRNA binding sites. Nucleic Acids Res 2013; 41:e138; http://dx.doi.org/10.1093/nar/gkt435 ; PMID:23703212\\n34. Ha I, \\nWightman B, Ruvkun G. A bulged lin-4/lin-\\n14 RNA duplex is sufficient for Caenorhabditis ele -\\ngans lin-14 temporal gradient formation. Genes Dev 1996; 10:3041-50;  PMID:8957004 ; http://dx.doi. org/10.1101/gad.10.23.3041\\n35. Lon\\ng D, Chan CY, Ding Y. Analysis of microRNA-\\ntarget interactions by a target structure based hybrid -\\nization model. Pacific Symposium on Biocomputing 2008; 13:64-74. 36. Ham\\nmell M, Long D, Zhang L, Lee A, Carmack CS, \\nHan M, Ding Y, Ambros V. mirWIP: microRNA target prediction based on microRNA-containing ribonucleoprotein-enriched transcripts. Nat Methods 2008; 5:813-9;  PMID:19160516 ; http://dx.doi. org/10.1038/nmeth.1247\\n37. Ren\\nnie WA, Liu C, Carmack CS, Wolenc A, Kanoria \\nS, Lu J, Long D, Ding Y.', 'STarMir: a web server for prediction of microRNA binding sites. Nucleic Acids Res 2014; http://dx.doi.org/10.1093/nar/gku376\\n38. Hil\\nlier LW, Reinke V, Green P, Hirst M, Marra MA, \\nWaterston RH. Massively parallel sequencing of the polyadenylated transcriptome of C. elegans. Genome Res 2009; 19:657-66; PMID:19181841 ; http://\\ndx.doi.org/10.1101/gr.088112.108\\n39. Din\\ng Y, Lawrence CE. Statistical prediction of single-\\nstranded regions in RNA secondary structure and application to predicting effective antisense target sites and beyond. Nucleic Acids Res 2001; 29:1034-46; PMID:11222752 ; http://dx.doi.org/10.1093/\\nnar/29.5.1034\\n40. Din\\ng Y, Lawrence CE. A statistical sampling \\nalgorithm for RNA secondary structure predic -\\ntion. Nucleic Acids Res 2003; 31:7280-301; PMID:14654704 ; http://dx.doi.org/10.1093/nar/\\ngkg938\\n41. Bet\\nel D, Koppal A, Agius P, Sander C, Leslie C. Comprehensive modeling of microRNA targets pre -\\ndicts functional non-conserved and non-canonical sites.', 'Genome Biol 2010; 11:R90; PMID:20799968 ; \\nhttp://dx.doi.org/10.1186/gb-2010-11-8-r90\\nW114–W118 NucleicAcidsResearch,2014,Vol.42,WebServerissue Publishedonline06May2014\\ndoi:10.1093/nar/gku376\\nSTarMir: a web server for prediction of microRNA\\nbinding sites\\nWilliam Rennie1, Chaochun Liu1, C. Steven Carmack1, Adam Wolenc1, Shaveta Kanoria1,\\nJun Lu2, Dang Long1and Ye Ding1,*\\n1Wadsworth Center, New York State Department of Health, Center for Medical Science, 150 New Scotland Avenue,\\nAlbany, NY 12208, USA and2Department of Genetics and Yale Stem Cell Center, Yale University, New Haven, CT\\n06520, USA\\nReceived January 24, 2014; Revised April 04, 2014; Accepted April 18, 2014\\nABSTRACT\\nSTarMir web server predicts microRNA (miRNA)\\nbinding sites on a target ribonucleic acid (RNA). STarMir is an implementation of logistic predic-tion models developed with miRNA binding data\\nfrom crosslinking immunoprecipitation (CLIP) stud-\\nies (Liu,C., Mallick, B., Long, D., Rennie, W.A., Wo-lenc, A., Carmack, C.S.', 'and Ding, Y. (2013). CLIP-\\nbased prediction of mammalian microRNA binding\\nsites. Nucleic Acids Res. , 41(14), e138). In both intra-\\ndataset and inter-dataset validations, the models\\nshowed major improvements over established algo-\\nrithms in predictions of both seed and seedless sites.General applicability of the models was indicated by\\ngood performance in cross-species validations. The\\ninput data for STarMir is processed by the web serverto perform prediction of miRNA binding sites, com-\\npute comprehensive sequence, thermodynamic and\\ntarget structure features and a logistic probability as\\na measure of conﬁdence for each predicted site. For\\neach of seed and seedless sites and for all three re-gions of a mRNA (3\\n/primeUTR, CDS and 5/primeUTR), STarMir\\noutput includes the computed binding site features,\\nthe logistic probability and a publication-quality dia-gram of the predicted miRNA:target hybrid.', 'The pre-\\ndiction results are available through both an interac-\\ntive viewer and downloadable text ﬁles. As an ap-plication module of the Sfold RNA package ( http:\\n//sfold.wadsworth.org ), STarMir is freely available to\\nall at http://sfold.wadsworth.org/starmir.html . INTRODUCTION\\nMicroRNAs (miRNAs) are a class of small endogenous\\nnon-coding RNAs (ncRNAs) of ∼22 nucleotides (nts) inlength that have been found in plants, animals and viruses. miRNAs are involved in the post-transcriptional regu-\\nlation of gene expression by binding to target messen-\\nger RNAs (mRNAs), leading to translational repression\\nand/or mRNA destabilization ( 1). They regulate diverse\\ndevelopmental processes, molecular and cellular pathways\\nand are associated with cancer and other human diseases. Prediction and validation of miRNA targets are essentialforunderstandingfunctionsofmiRNAsingeneregulation.', 'Algorithms for the prediction of miRNA:target binding\\nsitesaretypicallybasedontheseedrule,i.e.thetargetformsWatson–Crick (WC) pairs with bases two through seven or\\neight, at the 5\\n/primeend of the miRNA ( 2). However, numerous\\nexceptionstotheseedrulehavebeendocumented( 3–5).In\\nrecent years, miRNA binding site data from crosslinking\\nimmunoprecipitation(CLIP)studieshavebecomeavailable. TheseincludeHITS-CLIPformousebrain( 6),PAR-CLIP\\nfor human cell lines ( 7) and several variants of PAR-CLIP\\nfor the same human cell lines ( 8).', 'CLIP studies generate\\nshort target fragments that contain miRNA binding sites.Although the CLIP technique only covers abundantly ex-\\npressed miRNAs and transcripts in the experimental sys-\\ntem, we successfully utilized the high throughput data fordeveloping logistic models to improve binding site predic-\\ntionsforanymiRNA:targetpair( 9).Themodelsadoptthe\\nfeatures essential for miRNA binding from a list of com-prehensive sequence, thermodynamic and target structure\\nfeatures ( 10). For model validations, we used five indepen-\\ndentCLIPdatasetsforbothinter-datasetvalidationaswellas cross-species validation. Each CLIP dataset yielded at\\nleast 10 million binding sites, of which ∼5–15% are posi-\\ntive. For performance evaluation, we calculated true posi-tiverate(TPR =sensitivity),falsepositiverate(FPR =1−\\nspecificity) and an overall performance measure: Youden’s\\nJ-statistic (sensitivity + specificity −1) (11). The models\\nwere found to substantially outperform established algo-\\nrithms.', 'Furthermore, the good performance by the modelsin cross-species validation suggests that the models can be\\n*Towhomcorrespondenceshouldbeaddressed.Tel:+15184861719;Fax:+15184024623;Email:yding@wadsworth.org\\nPresentaddress:DangLong,BiotechnologyDepartment,FacultyofChemistry,DanangUniversityofScienceandTechnology,54NguyenLuongBangSt., Danang,\\nVietnam. C/circlecopyrtTheAuthor(s) 2014. Published by Oxford University Presson behalf of NucleicAcidsResearch. This is an Open Access article distributed under the terms of the Creative Commons Attribution License (http: //creativecommons.org /licenses /by/3.0/), which\\npermits unrestrictedreuse,distribution, and reproduction in anymedium, providedthe original work isproperlycited.Downloaded from https://academic.oup.com/nar/article/42/W1/W114/2436353 by guest on 15 March 2025\\n\\nNucleicAcidsResearch,2014,Vol.42,WebServerissue W115\\ngenerally applicable to microRNA binding site prediction\\nforanymammalian speciesand beyond.', 'We have implemented the models into the STarMir ap-\\nplication module of the Sfold RNA package ( 12). STarMir\\nweb server allows users to submit miRNA and mRNA se-\\nquencesforpredictionofbindingsitesbythemodels.Fora\\ngivenpairofmiRNA:targetmRNA,STarMirfirstpredictstargetsecondarystructures( 13).PotentialmiRNAbinding\\nsites are then predicted by the RNAhybrid program for ei-\\ntherseedmatchesorseedlesssiteswithahybridstabilityof−15kcal /molorlower( 14).Foreachsite,acomprehensive\\nlist of sequence, thermodynamic and structure-based fea-\\ntures are computed as previously described ( 9). These fea-\\nturesareusedbyourlogisticmodelwithparametersspecific\\nfor the site type (seed or seedless) and the target region (5\\n/prime\\nUTR,CDSor3/primeUTR)tocomputealogisticprobabilityas\\na measure of confidence in the predicted site. In general, a\\nprobabilityof0.5indicatesafairlygoodchanceofmiRNA\\nbinding. A high likelihood of miRNA binding is predicted\\nby a high probability, e.g. 0.75 or higher.', 'In addition, for\\neachsite,STarMiralsooutputsallofthesitefeaturesalongwithadiagram ofmiRNA:target hybrid conformation. STarMir can be accessed either from the Sfold main\\npage(http://sfold.wadsworth.org ),ordirectlyat http://sfold. wadsworth.org/starmir.html . The STarMir web service is\\nfreelyavailabletoallwithoutaregistrationorloginrequire-\\nment. In this article, we highlight the main features of theweb service. Users are encouraged to consult the online\\nmanual by clicking the ‘MANUAL’ button in the STarMir\\nmenu line and to examine sample output by clicking the‘DEMOOUTPUT’ button. INPUT\\nThe user needs to input sequence information for one or\\nmoremiRNAsandasingletargetmRNAforjobprocessing\\nby the web server. A link is provided for the user to checkontheprogressofthejobandtoaccesstheresults.Detailed\\ndescriptionsof theinputs are givenbelow. Model for prediction\\nThe user first selects a prediction model.', 'Currently avail-\\nable models are: a model trained on V-CLIP data for hu-\\nman (Homo sapiens )(8), a model trained on HITS-CLIP\\ndata for mouse ( Musmusculus )(6) and a recent model for\\nCaenorhabditis elegans based on analysis and modeling of\\nworm ALG-1 CLIP data ( 15,16). Speciesfor prediction\\nThe user next selects the species for prediction. This in-\\nformation will be used if the RefSeq ID is entered for the\\ntarget mRNA sequence so that the server can utilize pre-\\nstored evolutionary conservation information in the mod-elingcomputation.Thechoiceofspecieshasnoeffectifthe\\nmRNAsequenceinformationisenteredmanually.Further-\\nmore,if‘other’isselected,conservationinformationcannot\\nbe used in model prediction. We note that applications of\\nSTarMir are not limited to the species with available CLIPdata and prediction models, as good performance in cross-\\nspecies validation suggests that the models can also be ap-\\nplied toother species( 9).', 'miRNA sequences\\nIn the default option, the user can enter one or more\\nmiRNAIDs,e.g.hsa-let-7a-3p,mmu-mir-128-1,cel-mir-90. Forthisoption,thesequencesareretrievedfromaninternaldatabasebuiltfromrelease20ofthemiRBase( 17).Alterna-\\ntively,oneormoremiRNAsequencescanbepastedintothe\\ninputboxinFASTAformat,oruploadedfromaFASTAfile\\n(Figure1). There is no limit on the number of miRNA se-\\nquencesthatcanbeentered.miRNAsequencesmustbelessthan 55 nts in length. Any character in the miRNA other\\nthan A, T, C, G and U willbe removed. mRNA sequence\\nThere are three methods that can be used to input the\\nsequence information for the target mRNA. The default\\nmethod is to enter the RefSeq ID in the input box pro-vided (Figure 1). The sequence will then be retrieved from\\nourdatabaseofmRNAsequences.Wehavestored ∼19000\\nsequences from National Center for Biotechnology Infor-mation (NCBI) RefSeq Build 36.3 for human and ∼12 000\\nsequences from Build 37.2 for mouse.', 'If the sequence is\\nspecified using the RefSeq ID and is present in our mRNAdatabase, the models will utilize evolutionary conservation\\ninformation ( 9,18) for improved predictions. The sequence\\ndatacanalsobeprovidedbyselectingthe‘Manualsequenceentry’optiontoeitherinputthesequenceinraworFASTA\\nformat or upload from a FASTA file (Figure 1). When the\\nsequence is loaded from a FASTA format file, the file must\\ncontain onlyone sequence. Any character in the mRNA sequences other than A, T,\\nC, G and U will be removed. The current web server limit\\non the length of the mRNA sequence is 5000 nts. Longer\\nsequences will be truncated to 5000 nts starting from the 5\\n/prime\\nend\\nThe mRNA region information needs to be provided to\\nthe server through the region dropdown box directly abovethe sequence input box (Figure 1).', 'The user needs to indi-\\ncate that the sequence entered represents an entire mRNA\\norasingleregion(3\\n/primeUTR,CDSor5/primeUTR).Ifthesequence\\nrepresents the entire mRNA, the nucleotide positions for\\nthe start and end of the coding region must be specified in\\nthe boxes provided belowthe inputwindow. If the sequence is entered via RefSeq ID, the bound-\\nariesofthecodingregionwillberetrievedfromourmRNA\\ndatabase and binding sites will be reported for all three re-gions.ThenameofthesequenceforoutputwillbetheRef-\\nSeq ID. For a manually entered sequence, the user has the\\noptiontonamethesequence.Provisionofanemailaddressis optional. If an email address is entered, the user will re-\\nceiveanotification whenthejob is complete. OUTPUT\\nThe output results are presented to the user through both\\nan interactiveviewer and downloadable files.Downloaded from https://academic.oup.com/nar/article/42/W1/W114/2436353 by guest on 15 March 2025\\n\\nW116NucleicAcidsResearch,2014,Vol.42,WebServerissue\\nFigure1.', \"STarMirinputpagewithmanualentryoptionselectedforbothmiRNAandtargetsequences.Downloaded from https://academic.oup.com/nar/article/42/W1/W114/2436353 by guest on 15 March 2025\\n\\nNucleicAcidsResearch,2014,Vol.42,WebServerissue W117\\nFigure2. STarMiroutputpageshowingtheinteractivesiteviewer(with‘3/primeUTR-seedless’tabselectedfordisplay)andthedownloadlinksfortextfiles. (a)\\n(b)||●●●●●● ||●● ● ●●●●●●●●UUGAUAUG UUGGAU\\nGAUGGAG U\\nUGCUAUAU\\nU\\nU\\nUCAUGAC UU\\nU\\nU\\nC\\nC\\nAUGUAUCUACC UCAC3'\\n−−−5'5'\\n−−−3'\\n1753 1788hsa−let−7a−5pG= −23.2 kcal/mol\\nNM_000024\\n||● ● ● ●●●●● | ●●●UUGAUA\\nU\\nGUU\\nGGA UGAU\\nGGAGU\\nAGAUU\\nU\\nUCC\\nUUCCUAC\\nACCCUUG3'\\n−−−5'5'\\n−−−3'\\n1842 1862hsa−let−7a−5pG= −16.5 kcal/mol\\nNM_000024\\nFigure 3. (a) Hybrid diagram for a seed site (miRNA seed region (nt 2–8)\\ninred);(b)hybriddiagramforaseedless(non-canonical)site.\", 'Interactiveviewer\\nTheresultsappearinasix-tabbedpane(Figure 2).Forease\\nof use, the results are divided into seed and seedless sites\\nfor each of the three target regions (3/primeUTR, CDS and 5/prime\\nUTR).Withineachtab,theresultsforalloroneindividualmiRNAselectedfromthedropdownmenuintheinteractive\\nviewer are presented in a table in descending order of theirlogisticprobabilities.Foreachbindingsite,informationfor\\ncomprehensive sequence, thermodynamic and target struc-\\nturefeaturesisprovided,alongwithalogisticprobabilityasa measure of confidence for the site. In addition, a link to\\na graphic representation of the hybrid conformation and a\\nPDFofthediagramisavailableforvisualizationanddown-load (Figure 3). A file for definitions of the features with\\nreferences is available by clicking the link for ‘Feature defi-\\nnitions’ under theresulttable. Downloadable files\\nThe results can also be downloaded as tab-delimited text\\nfiles.', 'The text files include all site features calculated by\\nSTarMir with the features used in the prediction model\\nmarkedwithanasterisk(*).ThepredictionmodelsexcludefeaturesthatwerenotenrichedinCLIPdataanalysis( 9).A\\ntextfileisavailableforeachofthesixcategoriesrepresented\\nbythetabs.Allresultscanbedownloadedasacompressedarchive. In addition, there is a file representing a simplified\\ntext version of the hybrid conformations for each site and\\na file presenting the probability that each nucleotide in thesite is unpaired (i.e.single-stranded). Additional information\\nSTarMir uses output from the Srna module of the Sfold\\npackageinthecomputationoftargetaccessibilitymeasures.Downloaded from https://academic.oup.com/nar/article/42/W1/W114/2436353 by guest on 15 March 2025\\n\\nW118NucleicAcidsResearch,2014,Vol.42,WebServerissue\\nTheresultpagecontainsadropdownmenuprovidingaccess\\nto predictions generated by Srna as well as other modules\\nof theSfold( 12).', 'CONCLUSION\\nThe STarMir web server enables predictions of miRNA\\nbindingsitesforanyspeciesofinterest.Theserverprovidescomprehensive site features for both seed and seedless sites\\nto facilitate both experimental and computational investi-\\ngations. STarMir performs computations including targetsecondary structure predictions that are time-consuming\\nespecially for long target sequences. Thus, it cannot re-\\nturn results instantly as does a database search. A sepa-rate database has been under our development for the dis-\\ntribution of pre-computed transcriptome-scale results of\\nselect species. STarMir and this database will be comple-mentary tools. While STarMir can make predictions for\\nany miRNA:mRNA pair from any species of interest, the\\ndatabase will allow fast search of pre-computed results formultiplemiRNAs and mRNAs.', 'CITING THE STARMIR WEB SERVER\\nIn research publications, the users of STarMir should cite\\nthis article as well as the papers describing the prediction\\nmodels formiRNA:target interactions ( 9,10). SUPPLEMENTARY DATA\\nSupplementary Data are available at NAR Online. ACKNOWLEDGMENTS\\nThe Computational Molecular Biology and Statistics Core\\nat the Wadsworth Center is acknowledged for supporting\\ncomputingresourcesfor thiswork. FUNDING\\nNational Science Foundation [DBI-0650991 to Y.D.]; Na-\\ntional Institutes of Health [GM099811 to Y.D. and J.L.,\\nR01CA149109toJ.L.];NafostedFundofVietnam[102.03-2010.04toD.L.].Fundingforopenaccesscharge:National\\nInstitutesof Health. Conflictofintereststatement. None declared.REFERENCES\\n1. Fabian,M.R.andSonenberg,N.(2012)Themechanicsof\\nmiRNA-mediatedgenesilencing:a lookunderthehoodofmiRISC. Nat.Struct.Mol.Biol. ,19,586–593. 2. Lewis,B.P.,Burge,C.B.andBartel,D.P.(2005)Conservedseed\\npairing,oftenflankedby adenosines,indicatesthatthousandsofhumangenesare microRNAtargets.', 'Cell,120,15–20. 3. Tay,Y.,Zhang,J.,Thomson,A.M.,Lim,B.andRigoutsos,I.(2008)\\nMicroRNAsto Nanog,Oct4andSox2codingregionsmodulateembryonicstemcelldifferentiation. Nature,455,1124–1128. 4. Vella,M.C.,Choi,E.Y.,Lin,S.Y.,Reinert,K.andSlack,F.J.(2004)The\\nC.elegansmicroRNAlet-7bindstoimperfectlet-7complementarysitesfromthelin-413’UTR. GenesDev. ,18,132–137. 5. Didiano,D.andHobert,O.(2006)Perfectseedpairingisnota\\ngenerallyreliablepredictorformiRNA-targetinteractions. Nat. Struct.Mol.Biol. ,13,849–851. 6. Chi,S.W.,Zang,J.B.,Mele,A.andDarnell,R.B.(2009)Argonaute\\nHITS-CLIPdecodesmicroRNA-mRNAinteractionmaps. Nature,\\n460,479–486. 7. Hafner,M.,Landthaler,M.,Burger,L.,Khorshid,M.,Hausser,J.,\\nBerninger,P.,Rothballer,A.,Ascano,M.Jr,Jungkamp,A.C.,Munschauer,M. etal.(2010)Transcriptome-wideidentificationof\\nRNA-bindingproteinandmicroRNAtargetsitesbyPAR-CLIP. Cell,141,129–141. 8.', 'Kishore,S.,Jaskiewicz,L.,Burger,L.,Hausser,J.,Khorshid,M.and\\nZavolan,M.(2011)AquantitativeanalysisofCLIPmethodsforidentifyingbindingsitesofRNA-bindingproteins. Nat.Methods ,8,\\n559–564. 9. Liu,C.,Mallick,B.,Long,D.,Rennie,W.A.,Wolenc,A.,Carmack,C.S. andDing,Y.(2013)CLIP-basedpredictionofmammalianmicroRNA\\nbindingsites. NucleicAcidsRes. ,41,e138. 10. Long,D.,Lee,R.,Williams,P.,Chan,C.Y.,Ambros,V.andDing,Y. (2007)PotenteffectoftargetstructureonmicroRNAfunction. Nat. Struct.Mol.Biol. ,14,287–294. 11. Youden,W.J.(1950)Indexforratingdiagnostictests. Cancer,3,32–35. 12. Ding,Y.,Chan,C.Y.andLawrence,C.E.(2004)Sfoldwebserverfor\\nstatisticalfoldingandrationaldesignofnucleicacids. NucleicAcids\\nRes.,32,W135–W141. 13. Ding,Y.andLawrence,C.E.(2003)Astatisticalsamplingalgorithm\\nforRNAsecondarystructureprediction. NucleicAcidsRes. ,31,\\n7280–7301. 14. Rehmsmeier,M.,Steffen,P.,Hochsmann,M.andGiegerich,R.(2004)\\nFastandeffectivepredictionofmicroRNA /targetduplexes. RNA,10,\\n1507–1517. 15.', 'Zisoulis,D.G.,Lovci,M.T.,Wilbert,M.L.,Hutt,K.R.,Liang,T.Y.,\\nPasquinelli,A.E.andYeo,G.W.(2010)Comprehensivediscoveryof\\nendogenousargonautebindingsitesincaenorhabditiselegans. Nat. Struct.Mol.Biol. ,\\n17,173–179. 16. Liu,C.,Rennie,W.A.,Mallick,B.,Kanoria,S.,Long,D.,Wolenc,A.,\\nCarmack,C.SandDing,Y.(2014)MicroRNABindingSitesinC. elegans3’UTRs. RNABiology ,11,doi:10.4161 /rna.28868\\n17. Kozomara,A.andGriffiths-Jones,S.(2014)miRBase:annotating\\nhighconfidencemicroRNAsusingdeepsequencingdata. Nucleic\\nAcidsRes. ,42,D68–D73. 18. Siepel,A.,Bejerano,G.,Pedersen,J.S.,Hinrichs,A.S.,Hou,M.,\\nRosenbloom,K.,Clawson,H.,Spieth,J.,Hillier,L.W.,Richards,S.etal.(2005)Evolutionarilyconservedelementsinvertebrate,insect,\\nworm,andyeastgenomes. GenomeRes. ,15,1034–1050.Downloaded from https://academic.oup.com/nar/article/42/W1/W114/2436353 by guest on 15 March 2025\\n\\nSfold web server for statistical folding and rational\\ndesign of nucleic acids\\nYe Ding*, Chi Yu Chan and Charles E.', 'Lawrence\\nBioinformatics Center, Wadsworth Center, New York State Department of Health, 150 New Scotland Avenue, Albany,\\nNY 12208, USA\\nReceived February 12, 2004; Revised and Accepted April 21, 2004\\nABSTRACT\\nTheSfold web server provides user-friendly access to\\nSfold, a recently developed nucleic acid folding soft-\\nware package, via the World Wide Web (WWW). Thesoftware is based on a new statistical sampling para-digm for the prediction of RNA secondary structure.One of the main objectives of this software is to offer\\ncomputational tools for the rational design of RNA-\\ntargetingnucleicacids,whichincludesmallinterferingRNAs (siRNAs), antisense oligonucleotides andtrans -cleaving ribozymes for gene knock-down stu-\\ndies. The methodology for siRNA design is based ona combination of RNA target accessibility prediction,siRNA duplex thermodynamic properties and empiri-cal design rules.', 'Our approach to target accessibility\\nevaluation is an original extension of the underlying\\nRNA folding algorithm to account for the likely exis-tence of a population of structures for the targetmRNA. In addition to the application modules Sirna,\\nSoligo and Sribo for siRNAs, antisense oligos and\\nribozymes, respectively, the module Srna offers com-\\nprehensive features for statistical representation ofsampled structures. Detailed output in both graphical\\nand text formats is available for all modules. The Sfold\\nserver is available at http://sfold.wadsworth.org andhttp://www.bioinfo.rpi.edu/applications/sfold. INTRODUCTION\\nThe prediction of RNA secondary structure for a single\\nsequence is a classic problem in computational biology. Free energy minimization has been one of the most popular\\nmethods for addressing this problem. The established algo-rithm computes the optimal folding and a set of suboptimal\\nfoldings (1,2).', 'A more recent algorithm computes all subopti-\\nmal foldings within any speciﬁed increment of the minimumfree energy (3). However, neither method computes astatistically valid set of foldings. Toward a better character-\\nization of the ensemble of probable RNA secondary structures,\\nMcCaskill developed an algorithm for the calculation of equi-\\nlibrium partition functions and base-pairing probabilities (4).Although these algorithms are computationally efﬁcient, each\\nnevertheless has inherent limitations (5). Perhaps the most\\nimportant limitation stems from the fact that these methodswere developed primarily for structural RNAs that may have\\nunique structures. Messenger RNAs (mRNAs), on the other\\nhand, may exist as a population of structures (6).', 'We recently developed an algorithm to sample rigorously\\nand exactly from the Boltzmann ensemble of secondary struc-\\ntures (5), based on Turner thermodynamic parameters (7,8).This statistical sampling algorithm guarantees the generationof a statistically representative sample of structures. In addi-\\ntion, this algorithm enables the development of unique tools\\nfor a number of important applications that include the rationaldesign of RNA-targeting nucleic acids. Single-stranded regions in RNA secondary structure are\\nlikely to be accessible for RNA-targeting nucleic acidsthrough base-pairing interactions. Target accessibility has\\nlong been established as an important factor for the potency\\nof antisense oligonucleotides (oligos) and trans -cleaving ribo-\\nzymes.', 'Recently, the importance of target structure and acces-sibility for the function of siRNAs has been demonstrated\\nusing a number of experimental approaches that include\\noligo library (9), oligo array (10), antisense evaluation ofaccessibility (11), and targeting the same sequence in both\\nstructured and unstructured sites (12). Based on the RNA structure-sampling algorithm, we also\\ndeveloped a probability proﬁling method for the prediction of\\ntarget accessibility (5,13). A stochastic approach to accessi-\\nbility evaluation may be essential to account for the likelyexistence of a population of structures for mRNAs (6). The\\nprobability proﬁling approach reveals target sites that are com-\\nmonly accessible in a high proportion of statistically repre-sentative structures for the target RNA. Through assignment ofstatistical conﬁdence in predictions, this approach bypasses\\nthe long-standing difﬁculty in accessibility evaluation due\\nto limited representation of probable structures.', '*To whom correspondence should be addressed. Tel: +1 518 486 1719; Fax: +1 518 402 4623; Email: yding@wadsworth.org\\nThe online version of this article has been published under an open access model. Users are entitled to use, reproduce, disseminate, or display the ope n access\\nversion of this article provided that: the original authorship is properly and fully attributed; the Journal and Oxford University Press are attribu ted as the original\\nplace of publication with the correct citation details given; if an article is subsequently reproduced or disseminated not in its entirety but only in part or as a derivative\\nwork this must be clearly indicated. ª2004, the authors\\nNucleic Acids Research, Vol. 32, Web Server issue ªOxford University Press 2004; all rights reservedNucleic Acids Research, 2004, Vol.', '32, Web Server issue W135–W141\\nDOI: 10.1093/nar/gkh449\\nBased on these novel algorithms (5,13), the ﬁrst version of\\nSfold, a nucleic acid folding and design software package, was\\ncompleted during spring 2002. The package currently consists\\nof four application modules. Modules Sirna, Soligo, and Sribo\\nprovide computational tools for target accessibility predictionand for the rational design of siRNAs, antisense oligos and\\ntrans -cleaving ribozymes, respectively. General statistical\\nfolding features and output are available from the fourth mod-ule,Srna. Since April 2003, the software has been available\\nto the scientiﬁc community through web servers at http://\\nsfold.wadsworth.org and http://www. bioinfo.rpi.edu/applications/\\nsfold. A manual and frequently asked questions (FAQs) for\\nthe software are provided on the websites. In this article, we highlight the main features of the software\\nand the web service.', 'Users are encouraged to consult theonline manual for more information on the software, and to\\nexamine sample output located at http://sfold.wadsworth.org/\\ndemo. INPUT\\nA user can submit jobs to the web server by ﬁrst clicking on the\\napplication module of interest on the server’s front page and\\nthen ﬁlling out the job submission form. A job can run in either\\ninteractive mode or batch mode. Current limits are 200 basesfor an interactive job and 5000 bases for a batch job. For abatch job, a correct email address is required for notiﬁcation of\\njob completion. Strategies for sequences >5000 bases in length\\nare described in the online manual. Sequences in raw format,in FASTA format, or in GenBank format are accepted. A\\nsequence can be entered by copying and pasting into the\\nsequence input window; alternatively, a sequence ﬁle in theuser’s ﬁle directory (folder) can be selected for uploading. Any\\ncharacter other than A, C, G, T or U will be edited out.', 'An\\noption is provided for cases in which the RNA sequence to befolded is the reverse complement of the input sequence. For\\nSoligo, the user has the option to set the length of the oligos. The default length is 20 nt. For Sribo, the user can specify a\\nspeciﬁc NUH cleavage triplet for hammerhead ribozymes. Thedefault triplet is GUC. OUTPUT\\nThe output page for each module includes both graphical\\nrepresentations and relevant text ﬁles. Many images are pre-\\nsented through interactive graphics applications via the user’sweb browser. The user can download a colored or black-and-\\nwhite plot in PDF or PostScript format. For monochrome\\nprinters, the black-and-white version is recommended. Allof the graphs, with the exception of RNA structure diagrams,are ﬁrst generated by the public domain software Gnuplot and\\nthen post-processed by Perl scripts to provide additional fea-\\ntures and to improve visualization. Links to output from othermodules with default input settings are provided.', 'Output can\\nbe accessed on the server for up to 72 h after job completion. With the exception of siRNA internal stability proﬁles, all ofthe output in Zip or compressed tar (tar.gz) format is available\\nfor download. After the compressed ﬁle has been uncom-\\npressed by the user, a directory with the job ID as its nameis created under the user’s current directory. Under the job IDdirectory, there are seven subdirectories and a ﬁle readme.txt\\nto describe the ﬁles in the subdirectories. Below, we highlightthe main output features for the four modules. Sirna module\\nThe Sirna module provides tools for the rational design of\\nsiRNAs. The design methodology is based on target accessi-bility evaluation (5,13), typical design rules and published\\nempirical rules, and stability rules for siRNA duplex ends\\nand for the region of the cleavage site (14,15).', 'Recently, ithas been reported that functional siRNA duplexes tend to have\\nlower stability on the 5\\n0-antisense end ( /C244 bp) than on the 50-\\nsense end (14,15). It is proposed that both the absolute andrelative stabilities of the siRNA duplex ends determine the\\ndegree to which each strand participates in the RNAi pathway\\n(14). Furthermore, functional siRNAs tend to have relativeinstability at the cleavage site, which may facilitate productrelease and multiple turnovers (15). However, these rules on\\nsiRNA duplex stabilities do not guarantee siRNA function\\n(15), perhaps because they do not address the structure ofthe target mRNA. By integrating rules on target accessibility,\\nempirical rules and thermodynamic properties for siRNA\\nduplexes, Sirna provides a unique combination of tools for\\nsiRNA design. Probability profiling for target accessibility prediction.', 'For\\nprediction of target accessibility, a complete probability pro-ﬁle of single-stranded regions is generated for the entire target\\nRNA. Sites with high probabilities of being single-stranded are\\npredicted to be accessible. At nucleotide position i, the proﬁle\\nshows the probability that nucleotides i,i+1,i+2,...,\\ni+W/C01 are all unpaired. In other words, the proﬁle is for\\nconsecutive fragments with a width W. Although the proﬁle\\ncan be generated for any value of W, the default width W=4\\nhas been found to be particularly useful (13), perhaps because\\na minimum of four unpaired bases is thought to be required for\\nthe initiation of antisense binding (16). The proﬁle probabil-ities are computed from a representative statistical sample of\\ntarget secondary structures, with a default sample size of 1000\\nstructures. A regional probability proﬁle allows the user toexamine any region of the complete proﬁle.', 'In the interactive\\ngraphic window for regional proﬁling, a new region of 200\\nbases can be selected either by specifying the starting positionor by clicking /C28or/C29for the upstream or the downstream\\nregion. The text ﬁle sstrand.out contains output for probability\\nproﬁling. As an example, the regional proﬁle for the nt 201–400 region of rabbit b-globin mRNA (GenBank accession no. V00879) is illustrated in Figure 1. In addition, loop-speciﬁc\\nproﬁles for W=1 (5) are also available. However, we do not\\nknow whether a certain type of loop is more favorable thanother types for binding by complementary nucleic acids. Empirical rules.', 'The following basic design rules are widely\\nused: (i) siRNA duplexes should be composed of 21 nt senseand 21 nt antisense strands, paired so as to each have a 2 nt 3\\n0\\ndTdT overhang; (ii) the siRNA sequence should have low tomoderate GC content; (iii) sequences with more than three Gsor three Cs in a row should be avoided, because polyG and\\npolyC sequences can form quartets (17) and may interfere in\\nthe siRNA silencing mechanism; (iv) AAAA or TTTT shouldalso be avoided for RNA polymerase III mediated promotersW136 Nucleic Acids Research, 2004, Vol. 32, Web Server issue\\nbecause transcription tends to terminate at these sequences. It\\nis of note that there is a lack of support in the literature for the\\nsigniﬁcance of target patterns such as AA(N19) or NA(N19). Recently, empirical rules on sequence features of functionalsiRNAs have been reported by several groups (18–20). Basedon the largest published siRNA data set, the rules by Reynolds\\nand colleagues (18) are the most comprehensive.', 'However,\\nthese rules alone do not guarantee siRNA function. A numberof potent siRNAs in studies on effect of target structure and\\naccessibility (9–12) do not meet key empirical rules. However,\\ntheir function is explained by accessibility. siRNA duplex thermodynamics. S irna computes a number of\\nthermodynamics indexes for the implementation of rules onsiRNA duplex stabilities, based on recent RNA thermody-namics parameters (7,8). 5\\n0-antisense stability (AntiS, in\\nkcal/mol) is computed as the sum of free energies for 4 bp\\nstacks and the 30dangling T and a penalty for terminal A–U for\\nthe 50end of the antisense siRNA strand; 50-sense stability (SS,\\nin kcal/mol) is the sum for the 50end of the sense siRNA\\nstrand. Differential stability of siRNA duplex ends (DSSE,in kcal/mol) is the difference between the 5\\n0-antisense stability\\nand the 50-sense stability, i.e. DSSE =AntiS/C0SS.', 'For each of\\npositions 2–18 of the antisense strand, the internal stability isthe sum of 4 bp stacks, starting at this position in the 5\\n0!30\\norientation. For position 1, the internal stability is the 50-anti-\\nsense stability. For position 19, the internal stability is the 50-\\nsense stability. The internal stabilities are used for constructingan internal stability proﬁle for each siRNA duplex. For posi-\\ntions 16–19 on the 3\\n0end of the antisense strand, Khvorova and\\ncolleagues (15) extended the target sequence with the targetRNA for the purpose of calculation. This treatment can lead to\\ninaccurate information. For example, the proﬁle is not guaran-\\nteed to be symmetric when the bases for the ends of the duplexare symmetric. For a correct comparison of the stabilities for\\nthe two siRNA duplex ends, we simply reverse the orientation\\nto 3\\n0!50in the calculation for positions 16–19.', 'Average\\ninternal stability at the cleavage site (AIS, in kcal/mol) isthe average of internal stability values for positions 9–14 of\\nthe antisense strand (15). Filters and scores for siRNA duplexes. Based on available\\nrules that we consider to be important, Sirna currently uses\\nthe following ﬁlters for screening siRNA candidates:\\n(i) antisense siRNA binding energy </C010 kcal/mol (target\\naccessibility rule);\\n(ii) duplex feature score of 6 or higher;\\n(iii) DSSE >0 kcal/mol (asymmetry rule);\\n(iv) AIS >/C08.6 kcal/mol (cleavage site instability rule);\\n(v) 30% <GC%<60%;\\n(vi) exclusion of target sequence that has at least one occur-\\nrence of AAAA, CCCC, GGGG, or UUUU. The antisense siRNA binding energy is a weighted sum of the\\nRNA/RNA stacking energies (7) for the hybrid formed by the\\nantisense siRNA and the targeted sequence. For a base-pairstack, the weight for the sum is calculated by the probability of\\nthe unpaired dinucleotide in the target sequence that is\\ninvolved in the stack.', 'In addition, an A–U terminal penaltyfor the hybrid is included and is weighted by the probability of\\nthe unpaired terminal base. This weighting scheme accounts\\nfor the structural variation at the target site. For example, ansiRNA with an antisense binding energy of /C015 kcal/mol is\\npredicted to be more effective than an siRNA with a binding\\nenergy of /C010 kcal/mol. The target accessibility rule is imple-\\nmented by requiring the siRNA binding energy to be belowa threshold value. The current default of the threshold is\\n/C010 kcal/mol. The siRNA duplex feature score is computed with the algo-\\nrithm by Reynolds and colleagues (18) and has a minimum\\nof/C02pointsandamaximumof10points.The‘asymmetryrule’,\\ni.e. that the 5\\n0-antisense end is less stable than the 50-sense end,\\nis enforced by DSSE >0 kcal/mol. The rule of relative instabil-\\nity at the cleavage site is enforced by AIS >/C08.6 kcal/mol, the\\nmidpoint between the minimum ( /C03.6 kcal/mol) and the max-\\nimum ( /C013.6 kcal/mol).', 'In addition, based on antisense\\nFigure 1. Regional profile (window width W=4) for the region between nt 201 and 400 of the rabbit b-globin mRNA (GenBank accession no. V00879). Sites with\\nhigh probabilities are predicted to be accessible.Nucleic Acids Research, 2004, Vol. 32, Web Server issue W137\\nsiRNA binding energy, a target accessibility score is computed\\nand it ranges between 0 and 8 points. A duplex thermo-dynamics score is computed and it can have value 0, 1 or 2,\\nwith 1 point contributed by DSSE >0 (kcal/mol), and\\nanother point by AIS >/C08.6 (kcal/mol). The total siRNA\\nscore is the sum of accessibility score, duplex feature score\\nand duplex thermodynamics score. The maximum total score\\nis 20 points. The output ﬁle ﬁltered.out presents siRNAs that meet all\\nﬁlter criteria. The ﬁle sirna_s.out provides output information\\nfor siRNAs with total score greater or equal to a preset thresh-old. The current threshold is 12 points.', 'The ﬁle sirna.out con-\\ntains output information for all siRNAs. Results for DSSE,\\nAIS and other siRNA duplex stability indices are provided inthe output ﬁle stability.out. siRNA internal stability profiling. On the output page for\\nSirna, the internal stability proﬁle for every possible siRNA\\nduplex is available through the interactive graphic window for\\nsiRNA ends and internal stability proﬁling. On the proﬁle, theuser can use the stabilities for positions 1 and 19 to make a\\ncomparison of the stabilities of siRNA duplex ends. These ﬁles\\nare not included in the Zip or tar.gz ﬁle for Sfold output,\\nbecause of the large number of proﬁles for long mRNAs. However, from the interactive proﬁling window, the user\\ncan download the proﬁles for the selected target sites. Theinternal stability proﬁle for target positions nt 44–64 forthe rabbit b-globin mRNA is illustrated in Figure 2.', 'The 5\\n0-\\nantisense end is less stable than the 50-sense end (DSSE =\\n1.6 kcal/mol), and the siRNA is relatively unstable at thecleavage site, with AIS =/C07.8 kcal/mol. Soligo module\\nThe Soligo module provides tools for the rational design of\\nantisense oligos by combining prediction of target secondarystructure and accessibility with empirical design rules. Prob-\\nability proﬁling for the prediction of accessible regions is\\nthe same as described above for the Sirna module. The ﬁleoligo_f.out gives ﬁltered output for antisense oligos of user-\\nspeciﬁed length. Soligo currently uses the following ﬁlters:\\n(i) antisense oligo binding energy </C08 kcal/mol;\\n(ii) 40% <GC %<60%;\\n(iii) No GGGG in the target sequence. The antisense oligo binding energy is a weighted sum of the\\nDNA/RNA stacking energies (21) for the hybrid formed by the\\nantisense oligo and the targeted sequence.', 'For a base-pairstack, the weight for the sum is calculated by the probabilityof the unpaired dinucleotide in the target sequence that is\\ninvolved in the stack. This weighting scheme accounts for\\nthe structural variation at the target site. Sribo module\\nThis module currently offers tools for the design of hammer-\\nhead ribozymes based on target accessibility prediction. Forevery site of the selected cleavage triplet (e.g. GUC) on thetarget RNA, the probability proﬁle for individual bases (i.e. window width W=1 for ribozyme applications) is produced\\nfor the region that includes the triplet and the two ﬂankingsequences of 15 bases each. The length of 15 bases is sufﬁcient\\nto cover the normal range of lengths for the binding arms of the\\nhammerhead ribozyme. We recommend selection of cleavagesites for which both ﬂanking sequences are at least partially\\naccessible.', 'This is because antisense hybridization is believed\\nto start with nucleation at a site of several unpaired bases, andelongation then occurs by ‘unzipping’ the adjacent helix on the\\ntarget (22). As an example, the GUC triplet at positions 87–89\\nfor the rabbit b-globin mRNA meets this selection criterion\\n(Figure 3). We note that proﬁling of the target RNA onlyaddresses the accessibility of the target. However, it is also\\nimportant to assess the folding of a designed ribozyme whose\\nbinding arms are determined by the cleavage triplet and itsﬂanking sequences. To address the issue of ribozyme folding,\\nthe user can run module Srna to fold the ribozyme. Diagrams of\\nrepresentative structures and associated sampling frequencies\\nFigure 2. Internal stability profile (15) for the siRNA targeting positions 44–64 of the rabbit b-globin mRNA.W138 Nucleic Acids Research, 2004, Vol.', '32, Web Server issue\\nfrom this module are helpful in providing a conﬁdence assess-\\nment about the degree of correct ribozyme folding. Srna module\\nThis module provides tools and sampling statistics to statis-\\ntically characterize the Boltzmann ensemble of RNA second-\\nary structures. A two-dimensional histogram (2Dhist) displays\\nbase pair probabilities computed from a statistical sample witha default size of 1000 structures. In 2Dhist, base pair prob-abilities are shown by solid squares in the upper left triangle,\\nwith the nucleotide positions on both axes. The areas of the\\nsolid squares are proportional to the frequencies of the basepairs in the sampled structures. 2Dhist has an option for the\\ndisplay of base pair probabilities. When this option is selected,\\nthe probability, and the nucleotide positions of the base pair towhich a solid square corresponds, can be shown through\\nmouse pointing (Figure 4).', 'For long RNA sequences, the\\nbase pair probabilities take some time to load after the2Dhist is displayed. We thus have set ‘no base pair probabil-\\nities’ as the default display. Anad hoc representation of the structure sample is given in\\na table format. First, the minimum free energy of structures inthe sample (SMFE) and the largest free energy (LFE) in the\\nsample are computed. The free energy range covering all\\nstructures in the sample (i.e. [SMFE, LFE]) is then dividedinto 10 equally spaced intervals. For each free energy interval,\\nthe structure with the lowest free energy is selected as the\\nrepresentative. For the representative, the table presents itsassociated free energy interval, the frequency with which\\nstructures in the sample fall into the energy interval (the fre-\\nquencies for the 10 intervals sum to 1), the free energy of thestructure and the secondary structural diagram (Figure 5).', 'Wenote that this is a rather crude representation of the structure\\nsample and the Boltzmann ensemble, mainly because struc-\\ntures in a common free energy interval can possess structuralfeatures substantially different from one another. Structure\\ndiagrams are generated using a modiﬁed version of the naview\\nprogram (24). In our modiﬁed version, a number of bugs havebeen ﬁxed to reduce the chance of overlapping betweendiagram components. A Perl script was written to convert\\nthe device-independent .plt2 output from naview into a struc-\\ntural diagram in PostScript format. The structural diagram is\\nalso available in PNG and PDF formats. The PNG format hascapabilities for enlargement and local display. The GCG con-\\nnect ﬁle is also provided. A text ﬁle of base pair frequencies for generating 2Dhist and\\na ﬁle of free energies of sampled structures for plotting free\\nenergy distributions (5) are available. These ﬁles are described\\nin more detail in the online manual. Figure 3.', 'Probability profile (window width W=1 for ribozyme applications) for the region of the cleavage triplet G87UC89and its two flanking sequences of 15 each\\nfor the rabbit b-globin mRNA. Figure 4. Two-dimensional histograms (2Dhist) for a sample of 1000 structures\\ngenerated for Leptomonas collosoma spliced leader RNA of 56 nt in length (23). As an example of the option of ‘with base pair probabilities’, the sampling\\nestimate of the probability for the base pair between nt 25 and 29 is shown to be0.681 when the mouse is pointed at the solid square positioned at (25,29).Nucleic Acids Research, 2004, Vol. 32, Web Server issue W139\\nFUTURE PLANS\\nWe have ongoing collaborations with several experimental\\nlabs to validate and potentially improve the design methods\\nfor RNA-targeting nucleic aids. Additional design tools for\\nmodules Sirna, Soligo and Sribo may be developed from these\\nstudies.', 'For Srna, an appealing method for an efﬁcient statis-\\ntical representation of the Boltzmann ensemble of RNA sec-\\nondary structures is to cluster the structures in the samplegenerated by our algorithm (5). We have made substantial\\nprogress toward algorithmic determination of distinct clusters. We expect to include cluster representation in the output fortheSrna module in the foreseeable future. The ability to allow\\nfolding constraints is an important feature for future develop-\\nment. DNA folding prediction using DNA parameters (25,26)will enable the development of modules for other applicationssuch as rational design of PCR primers and amplicons. Furthermore, oligo self-interactions (27) may be considered\\nin the development of additional screening ﬁlters for Soligo. Alternative programs for RNA structure display will be\\nexplored for improvement in visualization, colored annotationbased on sampling statistics, and the potential of user inter-\\naction through Java.', 'The feasibility of including a standardizedRNA structure format to facilitate data exchange, e.g. the\\nRNAML format (28), is under consideration. The Wadsworth\\nCenter server is hosted on a web cluster currently consisting ofthree 1U servers each with dual Athlon MP 2800 +processors. The Grid Engine software is used as the job management\\nsystem in this web cluster. The master node has 2 GB ofmemory while each of the two execution nodes has 4 GB of\\nmemory. The job load is closely monitored, and the server\\nhardware will be upgraded when this becomes necessary. CITING THE SFOLD WEB SERVER\\nIn research publications, users of Sfold should cite this article\\nand the papers describing the algorithms (5,13), in additionalto including the URL for the main server web site: http://\\nsfold.wadsworth.org. Future relevant articles for citation\\nwill be listed on the server’s front page.', 'ACKNOWLEDGEMENTS\\nThe Computational Molecular Biology and Statistics Core at\\nthe Wadsworth Center is acknowledged for providing comput-ing resources for this work. The Computer Systems Support\\ngroup at the Wadsworth Center is acknowledged for setting up\\nthe hardware for the Wadsworth Center web server. The long-term development of the Sfold software and the maintenance of\\nthe web server are supported by National Science Foundation\\ngrant DMS-0200970 and National Institutes of Health grantGM068726 to Y.D. This work was also supported, in part, by\\nNational Institutes of Health grant HG01257 to C.E.L. The\\nserver at Rensselaer Polytechnic Institute (RPI) runs on aLinux cluster acquired through an IBM SUR grant awardedto RPI/Wadsworth Bioinformatics Center (29). REFERENCES\\n1. Zuker,M. and Stiegler,P. (1981). Optimal computer folding of large RNA\\nsequences using thermodynamics and auxiliary information. Nucleic\\nAcids Res. ,9, 133–148. 2. Zuker,M.', '(1989) On finding all suboptimal foldings of an RNA molecule. Science ,244, 48–52. 3. Wuchty,S., Fontana,W., Hofacker,I.L. and Schuster,P. (1999) Complete\\nsuboptimal folding of RNA and the stability of secondary structures.Biopolymers ,49, 145–165. 4. McCaskill,J.S. (1990) The equilibrium partition function and base pair\\nbinding probabilities for RNA secondary structure. Biopolymers ,29,\\n1105–1119. 5. Ding,Y. and Lawrence,C.E. (2003) A statistical sampling algorithm for\\nRNA secondary structure prediction. Nucleic Acids Res. ,31, 7280–7301. 6. Christoffersen,R.E., McSwiggen,J.A. and Konings,D. (1994)\\nApplication of computational technologies to ribozyme biotechnology\\nproducts. J. Mol. Structure (Theochem) ,311, 273–284. 7. Xia,T., SantaLucia,J.Jr, Burkard,M.E., Kierzek,R., Schroeder,S.J.,\\nJiao,X., Cox,C. and Turner,D.H. (1998) Thermodynamic parameters foran expanded nearest-neighbor model for formation of RNA duplexes withWatson–Crick base pairs. Biochemistry ,37, 14719–14735. 8.', 'Mathews,D.H., Sabina,J., Zuker,M. and Turner,D.H. (1999) Expanded\\nsequence dependence of thermodynamic parameters improves prediction\\nof RNA secondary structure. J. Mol. Biol. ,288, 911–940. 9. Lee,N.S., Dohjima,T., Bauer,G., Li,H., Li,M.J., Ehsani,A., Salvaterra,P. and Rossi,J.(2002) Expression of small interferingRNAs targeted againstHIV-1 rev transcripts in human cells. Nature Biotechnol. ,20, 500–505. Figure 5. Secondary structure diagram generated by a modified version of the\\nnaview program (24) for a predicted structure of Leptomonas collosoma spliced\\nleader RNA.W140 Nucleic Acids Research, 2004, Vol. 32, Web Server issue\\n10. Bohula,E.A., Salisbury,A.J., Sohail,M., Playford,M.P., Riedemann,J.,\\nSouthern,E.M. and Macaulay,V.M. (2003) The efficacy of smallinterfering RNAs targeted to the type 1 Insulin- like growth factorreceptor (IGF1R) is influenced by secondary structure in the IGF1Rtranscript. J. Biol. Chem. ,278, 15991–15997. 11. Far,R.K. and Sczakiel,G.', '(2003) The activity of siRNA in mammalian\\ncells is related to structural target accessibility: a comparison withantisense oligonucleotides. Nucleic Acids Res. ,31, 4417–4424\\n12. Vickers,T.A., Koo,S., Bennett,C.F., Crooke,S.T., Dean,N.M. and\\nBaker,B. (2003) Efficient reduction of target RNAs by small interferingRNA and RNase H-dependent antisense agents. A comparative analysis. J. Biol. Chem. ,278, 7108–7118. 13. Ding,Y. and Lawrence,C.E. (2001) Statistical prediction of single-\\nstranded regions in RNA secondary structure and application topredicting effective antisense target sites and beyond. Nucleic Acids Res. ,\\n29, 1034–1046. 14. Schwarz,D.S., Hutvagner,G., Du,T., Xu,Z., Aronin,N. and Zamore,P.D. (2003) Asymmetry in the assembly of the RNAi enzyme complex. Cell,\\n115, 199–208. 15. Khvorova,A., Reynolds,A. and Jayasena,S.D. (2003) Functional siRNAs\\nand miRNAs exhibit strand bias. Cell,115, 209–216. 16. Zhao,J.J. and Lemke,G. (1998) Rules for Ribozymes. Mol. Cell\\nNeurosci. ,11, 92–97.', '17. Hardin,C.C., Watson,T., Corregan,M. and Bailey,C. (1992) Cation-\\ndependent transition between the quadruplex and Watson–Crick hairpin\\nforms of d(CGCG3GCG). Biochemistry ,31, 833–841. 18. Reynolds,A., Leake,D., Boese,Q., Scaringe,S., Marshall,W.S. and\\nKhvorova,A. (2004) Rational siRNA design for RNA interference.Nature Biotechnol. ,22, 326–330. 19. Ui-Tei,K., Naito,Y., Takahashi,F., Haraguchi,T., Ohki-Hamazaki,H.,\\nJuni,A., Ueda,R. and Saigo,K. (2004) Guidelines for the selection of\\nhighly effective siRNA sequences for mammalian and chick RNAinterference. Nucleic Acids Res. ,32, 936–94820. Amarzguioui,M. and Prydz,H. (2004) An algorithm for selection of\\nfunctional siRNA sequences. Biochem. Biophys. Res. Commun. 316,\\n1050–1058. 21. Sugimoto,N., Nakano,S., Katoh,M., Matsumura,A., Nakamuta,H.,\\nOhmichi,T., Yoneyama,M. and Sasaki,M. (1995) Thermodynamic\\nparameters to predict stability of RNA/DNA hybrid duplexes. Biochemistry ,34, 11211–11216. 22. Milner,N., Mir,K.U. and Southern,E.M.', '(1997). Selecting effective\\nantisense reagents on combinatorial oligonucleotide arrays. Nature\\nBiotechnol. ,15, 537–541. 23. LeCuyer,K.A. and Crothers,D.M. (1993) The Leptomonas collosoma\\nspliced leader RNA can switch between two alternate structural forms. Biochemistry ,32, 5301–5311. 24. Bruccoleri,R.E. and Heinrich, G. (1988) An improved algorithm for\\nnucleic acid secondary structure display. Comput. Appl. Biosci. ,4,\\n167–173. 25. SantaLucia,J.Jr (1998) A unified view of polymer, dumbbell, and\\noligonucleotide DNA nearest-neighbor thermodynamics. Proc. Natl\\nAcad. Sci. USA. ,95, 1460–1465. 26. Owczarzy,R., Vallone,P.M., Goldstein,R.F. and Benight,A.S. (1999)\\nStudies of DNA dumbbells VII: evaluation of the next-nearest-neighborsequence-dependent interactions in duplex DNA. Biopolymers ,52,\\n29–56. 27. Matveeva,O.V., Mathews,D.H., Tsodikov,A.D., Shabalina,S.A.,\\nGesteland,R.F., Atkins,J.F. and Freier,S.M. (2003) Thermodynamic\\ncriteria for high hit rate antisense oligonucleotide design.', 'Nucleic Acids\\nRes.,31, 4989–4994. 28. Waugh,A., Gendron,P., Altman,R., Brown,J.W., Case, D., Gautheret, D.,\\nHarvey, S.C., Leontis,N., Westbrook,J., Westhof,E., Zuker,M. andMajor,F. (2002) RNAML: a standard syntax for exchanging RNA\\ninformation. RNA,8, 707–717. 29. Zuker,M. (2003) Mfold web server for nucleic acid folding and\\nhybridization prediction. Nucleic Acids Res. ,31, 3406–3415.Nucleic Acids Research, 2004, Vol.', '32, Web Server issue W141\\nBioMed Central\\nPage 1 of 12\\n(page number not for citation purposes)BMC Bioinformatics\\nOpen Access Research article\\nA structural analysis of in vitro catalytic activities of hammerhead \\nribozymes\\nYu Shao†, Susan Wu†, Chi Yu Chan, Jessie R Klapper, Erasmus Schneider and \\nYe Ding*\\nAddress: Wadsworth Center, New York State Department of Health, 150 New Scotland Avenue, Albany, NY 12208, USA\\nEmail: Yu Shao - yushao04@yahoo.com; Susan Wu - swu@ wadsworth.org; Chi Yu Chan - chanc@wadsworth.org; \\nJessie R Klapper - jklapper@brand eis.edu; Erasmus Schneider - schneid@wadsworth.org; Ye Ding* - yding@wadsworth.org\\n* Corresponding author    †Equal contributors\\nAbstract\\nBackground: Ribozymes are small catalyti c RNAs that possess the du al functions of sequence-\\nspecific RNA recognition an d site-specific cleavage.', 'Trans -cleaving ribozymes can inhibit translation\\nof genes at the messenger RNA (mRNA) level in both eukaryotic and prokaryotic systems and are\\nthus useful tools for studies of gene function. Howe ver, identification of ta rget sites for efficient\\ncleavage poses a challenge. Here , we have considered a number of structural and thermodynamic\\nparameters that can affect the effi ciency of target cleavage, in an attempt to identify rules for the\\nselection of functional ribozymes. Results: We employed the Sfold program for RNA seco ndary structure prediction, to account for\\nthe likely population of target structures that co -exist in dynamic equilibrium for a specific mRNA\\nmolecule. We designed and prepared 15 hammerhea d ribozymes to target GUC cleavage sites in\\nthe mRNA of the breast cancer resistance protei n (BCRP). These ribozymes were tested, and their\\ncatalytic activities  were measured in vitro .', 'We found that target disruption energy owing to the\\nalteration of the local target st ructure necessary for ribozyme bind ing, and the total energy change\\nof the ribozyme-target hybridization, are two si gnificant parameters for prediction of ribozyme\\nactivity. Importantly, target di sruption energy is the major cont ributor to the predictability of\\nribozyme activity by the total energy change. Furthermore, for a target-site specific ribozyme,\\nincorrect folding of the catalytic core, or interactions involving th e two binding arms and the end\\nsequences of the catalytic co re, can have detrimental ef fects on ribozyme activity. Conclusion: The findings from this study suggest rule s for structure-based rational design of trans-\\ncleaving hammerhead ribozymes in gene knockdow n studies. Tools implementing these rules are\\navailable from the Sribo module and the Srna mo dule of the Sfold program available through Web\\nserver at http://sfold.wadsworth.org .', \"Background\\nRibozymes are short catalytic RNAs that possess the dual\\nfunctions of sequence-specific RNA recognition and site-specific cleavage. For the self-cleaving ( cis-acting) ham-\\nmerhead ribozyme discovered by Haseloff and Gerlach[1], the binding arms at the 5' and 3' ends of the ribozymePublished: 30 November 2007\\nBMC Bioinformatics  2007, 8:469 doi:10.1186/1471-2105-8-469Received: 18 July 2007\\nAccepted: 30 November 2007\\nThis article is available from: http ://www.biomedcentral.com/1471-2105/8/469\\n© 2007 Shao et al; licensee BioMed Central Ltd. This is an Open Access article distributed under the terms of the Creative Commons Attribution License (http://creativecommons. org/licenses/by/2.0 ), \\nwhich permits unrestricted use, distribution, and reproduction in any medium, provided the orig inal work is properly cited.\", \"BMC Bioinformatics  2007, 8:469 http://www.biomedcen tral.com/1471-2105/8/469\\nPage 2 of 12\\n(page number not for citation purposes)form two helices, termed helix I and helix III, with the\\nsubstrate. The catalytic core of the ribozyme contains helix\\nII and largely conserved nucleotides. These structure andsequence features are illustrated in Figure 1A for the ham-merhead conformation of a specific mRNA-targetingribozyme studied here. For inhibition of the expression ofa gene through targeting of the gene's mRNA, trans -cleav-\\ning hammerhead ribozymes can be engineered with bind-\\ning arms whose sequences are complementary to thetarget mRNA sequences flanking a cleavage triplet NUH,where N is any nucleotide and H is any nucleotide exceptG. Among all possible NUH combinations, cleavage at\\nGUC (see Figure 1A) has been reported to be the most\\neffective [2]. In recent years, gene silencing by RNA interference (RNAi)\\n[3] has emerged as a powerful tool for gene knockdownstudies.\", \"However, applications of RNAi are limited to\\neukaryotes, whereas ribozymes can be applied in both\\neukaryotic and prokaryotic systems [4,5]. Furthermore,off-target effects by RNAi have been well noted [6-10].Ribozymes, in contrast, generally possess high target spe-\\n(A) Hammerhead ribozyme conformation arising from hybrid ization between hammerhead ribo zyme GUC19 and the target (BCRP) mRNAFigure 1\\n(A) Hammerhead ribozyme conformation arising from hybridiz ation between hammerhead ribozyme GUC19 and the target \\n(BCRP) mRNA. The GUC triplet in the target site and the conserve d nucleotides in the ribozyme ca talytic core are in boldface. The site of cleavage, i.e., 3' of the C of the GUC triplet, is indicated by an arrow. ( B) Thermodynamic cycle of free energy \\nexchanges.\", 'ΔG\\ndisruption is the target disruption energy, wh ich represents the free energy cost to alter the local secondary struc-\\nture at the target site  for ribozyme binding; ΔGswitch is the change in free energy from a predicted ribozyme conformation to \\nthe \"active\" binding conformation in which the catalytic core is corre ctly folded and both binding arms are single-stranded; \\nΔGhybrid is the free energy gain from the ribozyme-target hybr idization (see Methods for de scription of calculations).ΔGdisruptionΔGswitch\\nΔGhybridStructured hammerhead ribozymeRibozyme binding conformation\\nStructured target mRNA Local structure disruptionRibozyme-target hybrid\\n3\\n 5\\n3\\n 5\\n5\\n 5\\n 3\\n 3\\n5\\n5\\n 3\\n3GUG\\nG AGA\\nCCUCGGAAA U\\nAGUCUGAAAGA UC ACA GUG UCAU CUU . . . . . . . . . . . .', 'Helix III Helix I\\nHelix IITarget mRNA\\nHammerhead ribozymeC AU GUUA CA G\\nAAAUUUAU\\nGC\\n 3\\n3\\n 5\\n5A\\nB\\nBMC Bioinformatics  2007, 8:469 http://www.biomedcen tral.com/1471-2105/8/469\\nPage 3 of 12\\n(page number not for citation purposes)cificity, owing to the dual requirements for cleavage, i.e.,\\ncomplementarity for the binding arms and a cleavage tri-\\nplet. On the other hand, the knockdown effects by RNAiare stronger than those for ribozymes, suggesting that thetwo technologies can play complementary roles in func-tional genomics [11]. Ribozyme libraries constructed withrandomized binding arms have been employed for suc-\\ncessful identification of novel functional genes in mam-\\nmalian cells [12-15]. The activities of trans -cleaving ribozymes can vary greatly\\nfor different sites on the same target mRNA. Such variabil-ity is considered to be largely due to differences in the\\naccessibility of the target sites [11,16].', 'There is compelling\\nevidence that, to a large extent, the secondary structure ofan mRNA molecule determines the accessibility of themRNA for numerous gene regulatory mechanisms thatrequire complementary base-pairing for target recogni-tion, including translational inhibition by antisense oli-gonucleotides [17], target cleavage by ribozymes [18] and\\nsiRNAs [19-23], and, more recently, repression of transla-\\ntion by microRNAs [24,25]. Experimental approaches to the identification of accessi-\\nble target sites are tedious and time-consuming. Thedesign of effective ribozymes presents a challenge and has\\nlargely been based on trial and error. Several computa-\\ntional methods [26-28] make accessibility predictionsthrough structures predicted by the free energy minimiza-tion approach [29]. However, this method is not wellsuited to characterization of the likely population of struc-tures that can exist in dynamic equilibrium in vivo for a\\nspecific mRNA molecule [30,31].', 'In recent years, an alter-\\nnative sampling approach to RNA secondary structure pre-diction has been developed [32] and has beenimplemented as the Sfold program [33]. The structure\\nsampling algorithm generates a statistically representative\\nsample from the Boltzmann-weighted ensemble of RNAsecondary structures for the RNA. In comparison to theminimum free energy method, this approach has beenshown to better represent the likely population of mRNAstructures [34], and to make improved predictions for\\nstructural RNAs [35]. Not surprisingly, predictions by\\nSfold significantly correlate with experimental results ingene down-regulation studies using antisense oligonucle-otides [36,37], RNAi [38], or microRNAs [25]; in contrast,a lack of significant correlation was found in these appli-cations for predictions based on free energy minimiza-\\ntion. Here, we explore the potential value of using Sfold to\\npredicting activities of hammerhead ribozymes.', 'In this study, a set of hammerhead ribozymes targeted to\\nthe transcript of the human ABCG2 gene encoding thebreast cancer resistance protein (BCRP) were designedand analyzed in vitro (Table 1). We considered a number\\nof structural and thermodynamic parameters that can\\naffect the activity of a hammerhead ribozyme. Several ofthese parameters were computed with structures predictedby Sfold for the target mRNA and for the hammerheadribozyme. One of the parameters measures the targetaccessibility, and is termed the target disruption energy; it\\nrepresents the energy cost expended in altering the local\\ntarget structure so as to allow ribozyme binding (Figure1). We found that the target disruption energy and thetotal energy change of the ribozyme-target hybridizationare two significant parameters for prediction of ribozymeactivity. The target disruption energy is the major contrib-\\nutor to the predictability by the total energy change.', 'In\\naddition, for the ribozyme itself, incorrect folding of thecatalytic core or interactions involving the two binding\\nTable 1: Hammerhead ribozymes targeted to 15 GUC cleavage sites in the BCRP mRNA\\nRibozymea Position of target site b Activity [(1-Su3600) ± SD] ΔGdisruption (kcal/mol) ΔGtotal (kcal/mol)\\nGUC1 -174 0.843 ± 0.047 -15.958 -5.423\\nGUC2 -131 0.890 ± 0.045 -11.898 -13.288\\nGUC3 +3 0.811 ± 0.116 -2.466 -17.571\\nGUC4 +16 0.803 ± 0.062 -11.090 -4.677GUC5 +36 0.876 ± 0.030 -10.879 -9.883GUC6 +164 0.802 ± 0.053 -17.339 -2.715GUC7 +221 0.934 ± 0.025 -11.197 -16.294GUC8 +274 0.844 ± 0.027 -7.312 -13.835\\nGUC9 +490 0.803 ± 0.030 -17.625 -6.215\\nGUC10 +506 0.829 ± 0.028 -15.457 -8.610GUC11 +558 0.664 ± 0.067 -10.178 -26.245GUC14 +1109 0.788 ± 0.066 -13.385 -5.469GUC17 +1201 0.914 ± 0.067 -8.426 -11.931GUC18 +1207 0.920 ± 0.042 -5.473 -17.366\\nGUC19 +1398 0.940 ± 0.047 -7.868 -9.736\\na Ribozymes GUC12, GUC13, GUC1 5 and GUC16 were not tested.', 'b Position of G of the GUC cleavage site in th e ABCG2 cDNA relative to the A of the AT G start codon (GenBank accession no. NM_004 827). BMC Bioinformatics  2007, 8:469 http://www.biomedcen tral.com/1471-2105/8/469\\nPage 4 of 12\\n(page number not for citation purposes)arms and the end sequences of the catalytic core can have\\ndetrimental effects on ribozyme activity. These findings\\nsuggest rules for a structure-based rational design of trans -\\ncleaving hammerhead ribozymes. Results\\nMeasurement of ribozyme activity\\nTraditionally, ribozyme activity is determined through in\\nvitro cleavage followed by gel electrophoresis; the latter\\nmost often uses a radiolabeled substrate RNA combinedwith autoradiography [39], although non-radioactivedetection by ethidium bromide staining has also beenemployed [40]. For both methods, quantification then\\nrequires densitometry of the cleavage products on the gel.', 'Here, we wished to evaluate whether ribozyme cleavageactivity can also be measured via quantitative RT-PCR.Accordingly, ribozyme GUC7 was incubated for varyinglengths of time with the appropriate substrate RNA, andthe remaining substrate was analyzed either by agarose gelelectrophoresis followed by densitometry, or else by real\\ntime RT-PCR on the LightCycler. As shown by Figure 2,\\nthe results from these two methods are in good agree-ment. We concluded that quantitative RT-PCR is a validmethod by which to determine ribozyme activity in vitro ;\\nthus, all activity measurements for our study were there-fore made by this method.Statistical analyses\\nFor the 15 ribozymes tested in vitro , we performed both\\ncorrelation and regression analyses, using the ribozymecatalytic activity measured by (1-Su3600), and each of thecomputational parameters (see Methods). First, weobserved that there were two outliers, namely, ribozymesGUC3 and GUC11, which behaved differently from the\\nother 13 ribozymes.', 'We thus initially focused on the anal-\\nyses for the 13 \"normally behaving\" ribozymes, and theninvestigated possible explanations for the two outliers. For the 13 well behaved ribozymes, we found, among the\\nstructural and thermodynamic parameters, that ΔG\\ndisrup-\\ntion and ΔGtotal are significantly correlated with ribozyme\\nactivity (Table 2). The correlation coefficient for ΔGdisrup-\\ntion is 0.6839 with a P-value of 0.0099, and the correlation\\ncoefficient for ΔGtotal is -0.7901 with a P-value of 0.0013. Ribozyme activity, however, was not significantly corre-lated with ΔG\\nswitch, or ΔGhybrid . Because ΔGtotal is com-\\nputed from ΔGdisruption , ΔGswitch, and ΔGhybrid (see\\nMethods), ΔGtotal and ΔGdisruption are significantly corre-\\nlated (correlation coefficient = -0.6349, and P-value =\\n0.0110). Thus, the significance of the correlation with theribozyme activity for ΔG\\ntotal is mainly due to ΔGdisruption .', 'From linear regression analysis, either ΔGdisruption or ΔGtotal\\nis significantly predictive of the ribozyme activity (Table\\n2, Figure 3). Furthermore, in a comparison of the R2 values\\nfor ΔGdisruption and ΔGtotal, a relative improvement of\\nabout 33.5% is observed for ΔGtotal. This suggests that,\\nComparison of methods for quant ification of ribozyme cleavage Figure 2\\nComparison of methods for quant ification of ribozyme cleavage. Ribozyme GUC7  was incubated for various lengths of time \\nfrom 0 to 60 min, as indicated, and subs trate cleavage activity was an alyzed by agarose gel electrophoresis and real-time RT-\\nPCR as described in Methods. After electrophoresis (right panel), the gel wa s stained with ethidium bromide, and the bands \\nwere quantified by densitometry. Rela tive band intensity was then graphe d against time (left panel). Target ( /H17004), remaining sub-\\nstrate; product 1 (  ) and 2 (  ), relative amounts of each of the two cleavage  products.', 'Separately, the substrate was quantified \\nby real-time RT-PCR, and the relative  amount of remaining substrate (  ) was graphed against time. 020406080100\\n02 0 4 0 6 0\\nTime (min)% Product or substrateTarget\\nProduct 1\\nProduct 2\\nSubstrate0 5 10 15 30 60 MW nt\\n6583\\n4981\\n3638\\n2604\\n1908\\n1383\\n955\\n623\\n281\\nBMC Bioinformatics  2007, 8:469 http://www.biomedcen tral.com/1471-2105/8/469\\nPage 5 of 12\\n(page number not for citation purposes)although ΔGswitch and ΔGhybrid are insignificant as individ-\\nual predictors, they do contribute to the improved predict-ability by ΔG\\ntotal. The R2 for ΔGtotal indicates that over 60%\\nof the variability in the ribozyme cleavage activity can beattributed to ΔG\\ntotal. To understand the behaviors of the two outliers,\\nribozymes GUC3 and GUC11, we examined the structurespredicted by Sfold for each of the ribozymes.', 'In the caseof ribozyme GUC3, we found that, for 79.1% of the struc-tures, there are at least four base pairs formed by nucle-otides in the two binding arms and the ends of the\\ncatalytic core sequence (Figure 4). In the \"active\"\\nribozyme binding conformation (Figure 1B), all of notonly the binding arms but also the end sequences of thecatalytic core are single-stranded. Thus, substantialintramolecular structure involving these regions canhinder target binding by the ribozyme, despite the correct\\nformation of helix II (Fig 1A). For GUC11, we found that\\n33.5% of sampled structures have the catalytic core mis-folded so that it lacks a correctly formed helix II (Figure 5).This could explain the observation that GUC11 was theleast effective for target cleavage (Table 1, Figure 3),despite a moderately accessible target site as indicated by\\nΔG\\ndisruption (Table 1).', 'In contrast, for each of the other 14\\nribozymes in our study, the percentage of the sampledstructures having a misfolded core is less than 1%. Discussion and conclusion\\nIn this work, we have considered a number of structuraland thermodynamic parameters and have assessed their\\neffects on the in vitro activities of trans -cleaving hammer-\\nhead ribozymes. We found that ΔG\\ndisruption , a measure of\\naccessibility at the target site, and ΔGtotal, a measure of the\\ntotal energy change for the ribozyme-target hybridizationprocess, are two significant parameters for predictingribozyme activity, and that ΔG\\ndisruption is the major con-\\ntributor to the predictability by ΔGtotal. In this analysis, the\\nstability of the ribozyme-target hybrid as measured byΔG\\nhybrid had no impact on ribozyme cleavage activity.', 'Fur-\\nthermore, we found that incorrect folding of the ribozymecatalytic core or interactions involving the two binding\\narms and the end sequences of the catalytic core can havedetrimental effects on ribozyme activity. For the selection of functional ribozymes for gene knock-\\ndown studies, it is advisable to choose highly accessibletarget sites (i.e., sites with favorable ΔG\\ndisruption ) and favo-\\nrable ΔGtotal. In addition, a ribozyme with either a mis-\\nfolded core or structures that are substantially differentfrom the \"active\" binding conformation (Figure 1B)\\nshould be avoided. Our analyses are limited to modeling of RNA secondary\\nstructures.', 'The R\\n2 of 62.42% for ΔGtotal indicates that theLinear regression for prediction  of the ribozyme activity (as measured by (1-Su3600) for the amount of substrate cleaved at 1 hr) for 13 \"normally behaving\" ribozymesFigure 3\\nLinear regression for prediction  of the ribozyme activity (as \\nmeasured by (1-Su3600) for the amount of substrate cleaved at 1 hr) for 13 \"normally behaving\" ribozymes. ( A) For ΔG\\ndis-\\nruption as the predictor, the R2 for the regression is 0.4677, \\nand the P-value is 0.0099. ( B) For ΔGtotal as the predictor, the \\nR2 for the regression is 0.6242, and the P-value is 0.0013. Also plotted are the two outl iers (GUC11 and GUC3) that \\nwere not included in the regres sion analysis (see Results, Fig-\\nures 4 and 5 for explanations of the outliers).', '0.65 0.70 0.75 0.80 0.85 0.90 0.95GUC1GUC2\\nGUC3GUC4GUC5\\nGUC6 GUC7\\nGUC8 \\nGUC9 GUC10\\nGUC11GUC14GUC17 \\nGUC18 GUC19 \\nP = 0.0099 = 0.4677R2\\nΔGdisruption (kcal mol   )−1−20 −15 −10 −5\\n−1Su3600\\n0.65 0.70 0.75 0.80 0.85 0.90 0.95GUC1GUC2\\nGUC3\\nGUC4GUC5\\nGUC6 GUC7\\nGUC8 \\nGUC9 GUC10\\nGUC11GUC14GUC17 \\nGUC18 GUC19 \\n−1Su3600\\nΔGtotal(kcal mol   )−1P = 0.0013 = 0.6242R2\\n−25 −20 −15 −10 −50A\\nBTable 2: Linear regression and correlation analyses a\\nParameter Linear Regression Correlation \\ncoefficient\\nCoefficient P-value R2\\nΔGdisruption 0.0095 0.0099 0.4677 0.6839\\nΔGswitch 0.0024 0.8616 0.0029 0.0537\\nΔGhybrid -0.0048 0.3498 0.0798 -0.2825\\nΔGtotal -0.0093 0.0013 0.6242 -0.7901\\na Two outliers, GUC3 an d GUC11 (see Figure 3) were excluded in \\nthese analyses\\nBMC Bioinformatics  2007, 8:469 http://www.biomedcen tral.com/1471-2105/8/469\\nPage 6 of 12\\n(page number not for citation purposes)remaining 37.58% of the variability in the ribozyme\\ncleavage activity must be attributed to other factors thatlikely include RNA tertiary interactions [41].', 'We adopted a population approach to modeling of mRNA\\nsecondary structure, by employing the Sfold program. Inan antisense application, predictions by Sfold were signif-icantly correlated with activity of antisense oligonucle-otides, whereas there was a lack of correlation for the\\nminimum free energy (MFE) predictions [36]. In an RNAiapplication, the Sfold predictions were significantly pre-dictive of RNAi efficacy [38]. In contrast, the predictive\\nsignificance was either lacking or poor in terms of R\\n2 for\\npredictions by MFE structures, by heuristic suboptimalfoldings [42], or by complete suboptimal foldings [43].Furthermore, for all of these RNA folding programs, onlyStructural analysis of GUC3, one of  the two outliers in Figure 3 Figure 4\\nStructural analysis of GUC3, one of the tw o outliers in Figure 3. Unfavorable inte ractions involving the two binding arms and \\nthe end sequences of the ribozyme  catalytic core are present in structures predicted for GUC3.', '( A) The representative struc-\\nture (i.e., the centroid of a structural cluster [35]) for 44.9% of structures predic ted by Sfold for the ribozyme sequence. ( B) \\nThe representative structure for 29 .8% of the predicted structures. ( C) The representative structure for 20.9% of the pre-\\ndicted structures. ( D) The representative structure for the remaining 4.4%  of the predicted structures.', \"The sequence for the \\nribozyme 5' binding arm ends at A\\n9, and the sequence for the ribozy me 3' binding arm starts at A32.UUACUGGAACUGAUGAGUCCGU G\\nA\\nG\\nG\\nA\\nC\\nG\\nA\\nA\\nA\\nC\\nA\\nU\\nC\\nU\\nG\\nG\\nA\\nG\\nA\\nΔG= −12.90 kcal/mol3’ 5’32\\n9\\nUUACUGGAACUGAUGAGUCCGU G\\nA\\nG\\nG\\nA\\nC\\nG\\nA\\nA\\nA\\nC\\nA\\nU\\nC\\nU\\nG\\nG\\nA\\nG\\nA32\\n9\\n5’ 3’\\nΔG= −12.60 kcal/mol\\nUUACUGGAACUGAUGAGUCCGU G\\nA\\nG\\nG\\nA\\nC\\nG\\nA\\nA\\nA\\nC\\nA\\nU\\nC\\nU\\nG\\nG\\nA\\nG\\nA\\n3’ 5’32\\n9\\nΔG= −12.20 kcal/molU UACUGGAACU GAUGAGUCCGU G\\nA\\nG\\nG\\nA\\nC\\nG\\nA\\nA\\nAC\\nA\\nU\\nC\\nU\\nG\\nG\\nA\\nG\\nA32\\n9\\n5’3’\\nΔG= −11.00 kcal/molAB\\nCD\\nBMC Bioinformatics  2007, 8:469 http://www.biomedcen tral.com/1471-2105/8/469\\nPage 7 of 12\\n(page number not for citation purposes)Sfold predictions were found to fully explain sensitivity of\\nlin-41 mutants to microRNA repression by let-7 [25].\", 'In\\nthe current application, if we employ the minimum freeenergy predictions by mfold [29] in the calculation of\\nΔG\\ndisruption and ΔGswitch, we observe an insignificant corre-\\nlation with the ribozyme activity for ΔGdisruption (P-value =\\n0.0730, R2 = 0.2632, and correlation coefficient =\\n0.5130), and substantially reduced significance and pre-dictability for ΔGtotal (P-value = 0.0277, R2 = 0.3687, and\\ncorrelation coefficient = -0.6072). This finding further val-idates the sampling approach to characterization of thelikely population of mRNA structures. In the calculation of ΔG\\ndisruption , we assumed that the\\nbinding of target mRNA by a ribozyme induces only alocal structural alteration at the target site. It is likely thatStructural analysis of GUC11, the other outlier in Figure 3Figure 5\\nStructural analysis of GUC11, the other outl ier in Figure 3. For a substantial portion of the structure sample generated by Sfo ld \\nfor GUC11, the predicted structure has a misfolded catalytic core.', '( A) Ribozyme in \"active\" bind ing conformation with cor-\\nrectly folded core (i.e., helix II and sing le stranded end sequences of the core in cluding 9 conserved nucleotides, as shown in  \\nFigure 1A), representing 66.5% of the predicted structures. ( B) A conformation with a misfolded core, representing 28% of the \\npredicted structures. ( C) Another conformation with a misfolded core, repr esenting 5.5% of the predicted structures.', \"The \\nsequence for the ribozyme 5'  binding arm ends at A\\n9, and the sequence for the ribozyme 3' binding arm starts at A32.UCUCCUCCACUGAUGA GUCCGU G\\nA\\nG\\nG\\nA\\nCG\\nA\\nA\\nA\\nC\\nA\\nC\\nA\\nC\\nC\\nA\\nC\\nG\\nG32\\n9\\n5’3’\\nΔG= −9.40 kcal/mol\\nUCUCCUC CAC UGA UGA G\\nU\\nC C\\nG\\nU\\nG\\nA\\nG\\nG\\nA C\\nG\\nAA AC\\nA\\nC\\nA\\nC\\nC\\nA CGG932\\n5’\\n3’\\nΔG= −8.70 kcal/mol\\nCUC\\nCUC\\nCACUGAUGAGUCCGUGAG\\nGAC G\\nA\\nA\\nA\\nC AC\\nACCACGG\\n932\\n5’3’\\nΔG= −8.20 kcal/molA\\nB\\nC\\nBMC Bioinformatics  2007, 8:469 http://www.biomedcen tral.com/1471-2105/8/469\\nPage 8 of 12\\n(page number not for citation purposes)in some, if not all, cases, nucleotides outside the target site\\nwill also contribute to the energy change due to ribozyme\\nbinding. An alternative to the local disruption model is aglobal disruption model, which assumes that the rest ofthe target mRNA molecule can refold after ribozyme bind-ing. For this model, ΔG\\ndisruption can be re-calculated by\\nconstraining the target site to be single-stranded and\\nrefolding the rest of the target mRNA.\", 'Surprisingly, we\\nobserved insignificant correlation and poor predictabilityeither for ΔG\\ndisruption (P-value = 0.7206, R2 = 0.0121, and\\ncorrelation coefficient = 0.1100), or for ΔGtotal (P-value =\\n0.3039, R2 = 0.0956, and correlation coefficient = -\\n0.3093). These results suggest that target cleavage occurs\\nrapidly after the completion of ribozyme-target hybridiza-\\ntion such that global refolding of the target before cleav-age is unlikely. While partial refolding is a possibility, it ishighly uncertain what region of the target will be involvedin refolding. Thus, it is difficult to construct a computa-tional model that can represent a reasonable compromisebetween the local model and the global model. When the standard deviations for the measured activities\\nare available, an alternative to the standard linear regres-sion analysis is the weighted regression analysis.', 'In aweighted least-squares regression, the square term in thesum of squares for a data point is multiplied by a weight\\n[44]. In general, a larger weight is assigned to a data point\\nwith a higher precision as indicated by a smaller standarddeviation. Specifically, with the standard deviation ofcleavage activity available for every ribozyme (Table 1), 1/(standard deviation)\\n2, that is, 1/variance, can be used as\\nthe weight [44]. The weighted regression yielded results\\nthat are highly similar to those from the un-weighted\\nregression analysis. For example, for ΔGtotal, the R2 is\\n0.6119, with a P-value of 0.0016. While we focused on GUC cleavage sites in the present\\nstudy, we have also tested a small set of non-GUC NUH\\nsites. We observed a similar level of correlation between\\nribozyme activity and ΔGdisruption (data not shown), sug-\\ngesting that the critical parameters identified here are gen-erally relevant for the prediction of the activity ofhammerhead ribozymes.', 'However, the levels of activitiesfor non-GUC sites were generally lower than those for theGUC sites, consistent with a previous report that GUC is\\nthe most effective target site [2]. To generalize our findings\\nfor GUC target sites, further in vitro testing using other tar-\\ngets, as well as in vivo testing in cultured cells, will be\\nrequired. The latter is currently in progress for theribozymes described here.Methods\\nPreparation of double-stranded DNA oligonucleotide \\nribozyme templates\\nFor creation of the ribozymes in vitro , two complementary\\noligonucleotides containing the hammerhead ribozymecore sequence flanked by the sequences for the two bind-ing arms and a T7 RNA polymerase promoter sequence\\nwere annealed into duplex DNA in 10 mM Tris, pH 8.0,\\nand 50 mM NaCl, by incubation at 94°C for 5 min, fol-lowed by slow cooling to room temperature. All oligonu-cleotides were obtained from Integrated DNA Technology(IDT, Coralville, IA).', \"In vitro transcription of ribozyme and substrate target \\nRNA\\nIn vitro transcription of the substrate and ribozyme RNA\\nwas performed using the MEGAscript and MEGAshort-script kits (Ambion-ABI, Austin, TX), respectively, follow-ing the manufacturer's instructions. Either 2.5 μg of\\nlinearized plasmid (pTRIamp19, Ambion) containing the\\ntarget ABCG2 cDNA sequence, or 1.5 μg of ribozyme DNA\\nwere used as template. After transcription, the DNA tem-plates were digested with RQ1 RNase-free DNase. Unin-corporated nucleotides were removed from the RNAtranscripts by size-exclusion chromatography with aProbeQuant G-50 Micro Column (GE-Healthcare, Piscat-\\naway, NJ) or by phenol/chloroform extraction, both of\\nwhich were followed by an ethanol precipitation.\", 'Thepurified in vitro -transcribed RNAs and ribozymes were\\nthen quantitated spectrophotometrically, and their qual-ity verified by gel electrophoresis (see Additional file 1).Two separate substrate RNAs were made, one from nucle-\\notides -225 to +1011, and one from nucleotides +586 to\\n+1708, relative to the A of the ATG start codon of the fulllength ABCG2 cDNA (GenBank accession no.NM_004827\\n). Individual ribozymes are numbered con-\\nsecutively in the order of occurrence of the GUC cleavagesites to which they bind, starting from nucleotide -285. Thus, for example, GUC1 refers to the first GUC triplet\\nafter nucleotide -285. A total of 15 hammerheadribozymes targeted to GUC sites were designed and pre-pared (Table 1). These ribozymes were constructed withthe same ribozyme core sequence, but with differentsequences for binding arms that were complementary tothe target sequences at the binding site (Figure 1A; also see\\nAdditional file 2).', \"For each of these ribozymes, the 3'\\nbinding arm had 11 nucleotides, and the 5' binding armhad nine nucleotides. In vitro cleavage of target sequen ce and identification of \\ncleavage products\\nThe target RNA (10 pmol) and ribozyme (50 pmol) under\\nstudy were mixed in 50 mM Tris, pH 8.0, and the in vitro\\ncleavage reaction was initiated by the addition of 20 mMMgCl\\n2. One μl of RNaseGuard was also added, and the\\nBMC Bioinformatics  2007, 8:469 http://www.biomedcen tral.com/1471-2105/8/469\\nPage 9 of 12\\n(page number not for citation purposes)mixture was incubated at 37°C. 10- μl aliquots were\\nremoved after 0, 5, 10, 15, 30, and 60 min, and the reac-\\ntion was terminated by the addition of 50 mM EDTA. Thecleavage products were then analyzed by electrophoresisin a 2% (v/v) formaldehyde/2.0% (w/v) agarose gel for3–4 hr at 70 V. The separated products were stained withSYBR Green or ethidium bromide and photographed\\nunder UV light [40].\", \"Quantification of residual substrate by real-time RT-PCR\\nSince a ribozyme irreversibly cleaves its substrate, we rea-\\nsoned that the cleavage reaction could be quantifiedthrough measurement of the amount of substrate remain-ing by real-time RT-PCR, using primer pairs that span thecleavage site. Accordingly, an aliquot of the cleavage reac-tion containing both the remaining, uncleaved substrate\\nand the cleavage products was added to a one-step real-\\ntime RT-PCR reaction mix containing SYBR Green (Sigma,St. Louis, MO) according to the manufacturer's instruc-tions, and amplification was carried out for 35–45 cyclesin a LightCycler\\n® (Roche, Indianapolis, IN), under condi-\\ntions appropriate for each primer pair (see Additional file3). Primers flanking each cleavage site were chosen suchthat the PCR products were between 600 and 400 bp long.\", 'The amount of uncleaved substrate present was deter-\\nmined from the crossing point values (C\\nT) calculated by\\nthe Lightcycler software from the amplification curve. The\\nrelative amount of template remaining at each time point\\n(Su(t)) was then calculated by , where CT(t) is\\nthe CT value at time t, and CT(0) is the CT value at time 0. Each time point was assayed in duplicate, and each cleav-\\nage reaction was repeated at least four times independ-ently with different batches of substrate RNA. Selected\\nribozymes were also analyzed with differing ribozyme\\npreparations. No significant activity differences wereobserved between separate ribozyme and/or substratepreparations. For the subsequent calculations, the relativeamount of substrate cleaved at 3600 sec (1-Su3600) wasused as the measure of ribozyme activity. In preliminaryexperiments, we determined that the RT-PCR reaction waslinear with the amount of substrate present (data not\\nshown).', 'Prediction of mRNA secondary structure\\nThe determination of mRNA secondary structure presents\\nboth theoretical and experimental challenges. One majorimpediment to the accurate prediction of mRNA struc-tures stems from the likelihood that a specific mRNA mol-ecule does not adopt a single structure in solution, butinstead likely exists in thermodynamic equilibriumamong a population of structures [30,31,45]. Thus, thecomputational prediction of secondary structure based on\\nfree energy minimization is not well suited to the task of\\nproviding a realistic representation of mRNA structures in\\nvivo. An alternative to free energy minimization for characteri-\\nzation of the ensemble of probable structures for a given\\nRNA molecule has been developed [32]. In this approach,\\na statistically representative sample is drawn from theBoltzmann-weighted ensemble of RNA secondary struc-tures for the RNA.', 'Such samples of even moderate size canfaithfully and reproducibly characterize structure ensem-bles of enormous size, so that sampling estimates of struc-\\ntural features are statistically reproducible from one\\nsample to another. In particular, in comparison to free-energy minimization, this method has been shown tomake better structural predictions [35] and to better rep-resent the likely population of mRNA structures [34], andto yield a significant correlation between predictions andantisense inhibition data [36,37]. A sample size of 1,000\\nstructures has been shown to be sufficient to guarantee\\nstatistical reproducibility in typical sampling statistics andstructure clustering features [32,34]. In applications tomodeling RNA target binding by a (partially) comple-mentary nucleic acids, because a single-stranded block offour or five nucleotides is essential for the nucleation step\\nof the hybridization [25,46,47], the probability that such\\nblock is single-stranded must be high.', 'Thus, in the currentand other related applications, we consider the samplesize of 1,000 to be sufficient. In the case that a structuralfeature of small probability is of interest, a much largersample would be required. The structure sampling\\nmethod has been implemented in the Sfold software pro-\\ngram for RNA folding and applications [33] and is usedhere for mRNA folding. Prediction of ribozyme secondary structure\\nThe core of the ribozyme is considered to exist in a mix-\\nture of conformations in solution that can interchange\\nrapidly [48-51]. In accordance with this establisheddynamic view of the hammerhead structure, we alsoemployed Boltzmann structure samples generated bySfold for the prediction of ribozyme secondary structure.Again, a sample size of 1,000 was used for characterizingprobable ribozyme structures at equilibrium. Structural and thermodynamic parameters\\nThe catalytic activity of a trans -cleaving ribozyme can be\\naffected by many factors.', \"Here, we have focused on anumber of structural and thermodynamic parameters.These parameters take into account the secondary struc-\\nture of the target, the secondary structure of the ribozyme,\\nand the stability of the ribozyme-target duplex. Below, wedefine these terms in the current context and compute the20 −−()() ( )CCTt T\\nBMC Bioinformatics  2007, 8:469 http://www.biomedcen tral.com/1471-2105/8/469\\nPage 10 of 12\\n(page number not for citation purposes)total free energy change for modeling the hybridization\\nprocess. ΔGdisruption is the free energy cost for disruption of the sec-\\nondary structure at the ribozyme binding site on the targetmRNA (Figure 1B), and thus is a measure of accessibilityat the target site. For the 15 designed ribozymes, each with\\nnine nucleotides for the 5' binding arm and 11 nucle-\\notides for the 3' binding arm, the binding site involves 20nucleotides, excluding the unpaired C of the GUC triplet(Figure 1A).\", 'To calculate ΔG\\ndisruption , we adopted the sim-\\nplifying assumption that the binding of a ribozyme to arelatively much longer mRNA should induce a local struc-\\ntural alteration at the target site, but no longer-range\\neffects on overall target secondary structure. In otherwords, we defined local structural alteration as the break-age of the intramolecular base pairs involving the targetsite to permit formation of the ribozyme-target duplex(Figure 1). Specifically, ΔG\\ndisruption was calculated as the\\nenergy difference between ΔGbefore, the free energy of the\\noriginal mRNA structure, and ΔGafter, the free energy of the\\nnew, locally altered structure ( ΔGdisruption = ΔGbefore-\\nΔGafter). We calculated ΔGbefore from the average energy of\\nthe original 1,000 structures predicted by Sfold, andΔG\\nafter from the average energy of all of the 1,000 locally\\naltered structures.', 'Therefore, under the local disruption\\nassumption, the calculations did not require refolding of\\nthe rest of the target sequence. ΔGswitch is the free energy cost for the ribozyme to switch\\nfrom one conformation to the conformation that is mostfavorable for target binding and subsequent cleavage. Here, the starting conformation is any conformation pre-\\ndicted by Sfold, and the binding conformation is the onefor which the ribozyme core is correctly folded and bothbinding arms are single-stranded (Figure 1B). Thus, ΔG\\ns-\\nwitch = ΔGs - ΔGb , where ΔGs is the free energy of the start-\\ning conformation, and ΔGb is the free energy of the\\nbinding conformation. In the case that the starting confor-\\nmation is the binding conformation, ΔGswitch = 0.0 kcal/\\nmol. We calculated ΔGs by the average free energy of the\\n1000 structures predicted by Sfold for the ribozyme.', 'ΔGb\\nis the same for different starting conformations of a given\\nribozyme sequence, so there is no need to average over astructure sample. ΔG\\nhybrid is the energy gain due to the complete intermo-\\nlecular hybridization between the ribozyme binding armsand the nucleotide sequence of the target binding site. Itis calculated by the sum of base-pair stack energies for thetwo ribozyme arm-target duplexes, an energetic penalty\\n(\"initiation energy\") for the initialization of bimolecular\\ninteraction [52], and other penalties or energies associ-ated with the multi-branched loop formed by the threeadjacent helices.', 'Specifically, ΔG\\nhybrid = ΔGinitiation +∑1≤i≤10ΔGH3_stacking(i) + ∑1≤j≤8ΔGH1_stacking ( j) + ΔGmulti-loop +\\nΔGH3_terminal + ΔGH1_terminal + ΔGdangle , where the initiation\\nenergy ΔGinitiation = 4.1 kcal/mol [52]; ΔGH3_stacking ( i) (1 ≤ i\\n≤ 10) is the stacking energy for the i-th base-pair stack for\\nhelix III (Figure 1A); ΔGH1_stacking ( j) (1 ≤ j ≤ 8) is the stack-\\ning energy for the j-th base-pair stack for helix I; ΔGmulti-\\nloop is a linear penalty for the multibranched loop formed\\nby the three helices; ΔGH3_terminal is a penalty of 0.5 kcal/\\nmole for the terminal A-U pair for helix III, whileΔG\\nH1_terminal applies the same penalty for a terminal A-U\\nor G-U pair [53] for helix I (e.g., A-U for ribozymeGUC19, Figure 1A); and ΔG\\ndangle is a sum of free energies\\nfor dangling ends (i.e., single base stacks)[52].', 'More spe-\\ncifically, for the linear multibranched loop penalty,\\nΔGmulti-loop = a + b(number of unpaired bases)+ c(number\\nof helices), where a, b, and c are respectively the offset, the\\nfree base penalty and the helix penalty, and a = 3.4 kcal/\\nmol, b = 0.0 kcal/mol, and c = 0.4 kcal/mol [53]. In our\\npresent context, there are 11 unpaired bases and three hel-ices in the loop, so ΔG\\nmulti-loop = 5.2 kcal/mol, a constant\\nfor all ribozymes studied here.', \"For a terminal base-pair N-\\nN' (A-U for ribozyme GUC19, as shown in Figure 1A) forhelix I, ΔG\\ndangle = min [ΔG3(U-A,C), ΔG5(N-N',C)] + ΔG5(A-U,A)\\n+ ΔG3(C-G,G) + ΔG5(G-C,A) + ΔG3(N'-N,C), where the free ener-\\ngies for both 5' and 3' dangling ends [53] are used, andmin [ΔG\\n3(U-A,C), ΔG5(N-N',C) ] is the minimum of the two\\ndangling energies, to take into account two possibilities of\\nsingle-base stacking for the C of the GUC cleavage triplet.It is assumed that a single unpaired nucleotide betweentwo adjacent helices for a multi-branched loop stacksonto the terminal base pair of the helix possessing themore favorable dangling energy. Finally, we computed ΔG\\ntotal, the total energy change for\\nthe ribozyme-target hybridization. ΔGtotal can be calcu-\\nlated through consideration of the energy gain due to thecomplete intermolecular hybridization and the energycosts owing to structure alterations for both the target and\\nthe ribozyme.\", \"With use of the parameters introduced\\nabove, ΔG\\ntotal = ΔGhybrid - ΔGswitch - ΔGdisruption . Statistical analyses\\nThe standard univariate linear regression was used for pre-\\ndicting ribozyme activity by each of the parameters listedabove. The P-value measures the statistical significance of\\nthe parameter, and the R\\n2 of the regression indicates the\\ndegree of variability in ribozyme activity that is attributedto the parameter. The Pearson's correlation coefficientbetween a parameter and the ribozyme activity was alsocomputed. We note that the P-value of the correlation is\\nthe same as the P-value of the parameter from the stand-\\nard univariate regression analysis. The software package R\\n[54] was used for the statistical analyses.\", \"BMC Bioinformatics  2007, 8:469 http://www.biomedcen tral.com/1471-2105/8/469\\nPage 11 of 12\\n(page number not for citation purposes)Availability and requirements\\nThe energetic calculations in this study have been incor-\\nporated into the Sribo module of the Sfold program. Sribois freely available for academic applications through aWeb server at http://sfold.wadsworth.org/sribo.pl\\n. Struc-\\ntural predictions and features are also available from theSrna module of Sfold http://sfold.wadsworth.org/srna.pl\\n. For commercial usage, a license is required http://\\nwww.wadsworth.org/resnres/bioinfo/sfold/\\nLicense_info.html . Competing interests\\nThe author(s) declare that they have no competing inter-ests. Authors' contributions\\nYD and ES conceived the study. YS and CYC performed\\ncomputational analyses. SW and JRK carried out the in\\nvitro experiments. YD supervised the computational work. ES supervised the experimental work. YS, YD and ES wrote\\nthe manuscript.\", \"All authors read and approved the final\\nmanuscript. Additional material\\nAcknowledgements\\nThe Computational Molecular Biology and Statistics Core at the \\nWadsworth Center is acknowledged for providing computing resources \\nfor this work. This work was supported in part by National Science Foun-dation grant DMS-0200970 and National  Institutes of Health grant R01 \\nGM068726 to Y.D.References\\n1. Haseloff J, Gerlach WL: Simple RNA enzymes with new and\\nhighly specific endoribonuclease activities. Nature  1988,\\n334(6183): 585-591. 2. Tanner NK: Ribozymes: the characteri stics and properties of\\ncatalytic RNAs. FEMS microbiology reviews  1999, 23(3): 257-275. 3. Fire A, Xu S, Montgomery MK, Kostas SA, Driver SE, Mello CC:\\nPotent and specific genetic in terference by double-stranded\\nRNA in Caenorhabditis elegans. Nature  1998,\\n391(6669): 806-811. 4. Junn E, Kang C: Detection of hammerhead ribozyme-medi-\\nated cleavage and reduced expression of LacZ' mRNA in E. coli. Genet Anal  1996, 13(1): 1-7. 5.\", \"Tatout C, Gauthier E, Pinon H: Rapid evaluation in Escherichia\\ncoli of antisense RNAs and ribozymes. Letters in applied microbi-\\nology 1998, 27(5): 297-301. 6. Birmingham A, Anderson EM, Reyn olds A, Ilsley-Tyree D, Leake D,\\nFedorov Y, Baskerville S, Maksimova E, Robinson K, Karpilow J, Mar-\\nshall WS, Khvorova A: 3' UTR seed matches, but not overall\\nidentity, are associated with RNAi off-targets. Nature methods\\n2006, 3(3): 199-204. 7. Semizarov D, Frost L, Sarthy A, Kroeger P, Halbert DN, Fesik SW:\\nSpecificity of short interfer ing RNA determined through\\ngene expression signatures. Proc Natl Acad Sci USA  2003,\\n100(11): 6347-6352. 8. Jackson AL, Bartz SR, Schelter J, Kobayashi SV, Burchard J, Mao M, Li\\nB, Cavet G, Linsley PS: Expression profiling reveals off-target\\ngene regulation by RNAi. Nat Biotechnol  2003, 21(6): 635-637. 9.\", 'Jackson AL, Burchard J, Schelter J, Chau BN, Cleary M, Lim L, Linsley\\nPS: Widespread siRNA \"off-target\"  transcript silencing medi-\\nated by seed region sequence complementarity. RNA 2006,\\n12(7): 1179-1187. 10. Pei Y, Tuschl T: On the art of identifyin g effective and specific\\nsiRNAs. Nature methods  2006, 3(9): 670-676. 11. Akashi H, Matsumoto S, Taira K: Gene discovery by ribozyme\\nand siRNA libraries. Nature reviews  2005, 6(5): 413-422. 12. Li QX, Robbins JM, Welch PJ, Wong-Staal F, Barber JR: A novel\\nfunctional genomics approach identifies mTERT as a sup-pressor of fibroblast transformation. Nucleic Acids Res  2000,\\n28(13): 2605-2612. 13. Kashani-Sabet M, Liu Y, Fong S, Desprez PY, Liu S, Tu G, Nosrati M,\\nHandumrongkul C, Liggitt D, Thor AD, Debs RJ: Identification of\\ngene function and functional pathways by systemic plasmid-\\nbased ribozyme targeting in adult mice. Proc Natl Acad Sci USA\\n2002, 99(6): 3878-3883. 14.', 'Beger C, Pierce LN, Kruger M, Marcusson EG, Robbins JM, Welcsh P,\\nWelch PJ, Welte K, King MC, Barber JR, Wong-Staal F: Identifica-\\ntion of Id4 as a regulator of  BRCA1 expression by using a\\nribozyme-library-based inverse genomics approach. Proc Natl\\nAcad Sci USA  2001, 98(1): 130-135. 15. Kawasaki H, Taira K: A functional gene discovery in the Fas-\\nmediated pathway to apoptosis by analysis of transiently\\nexpressed randomized hybrid-ribozyme libraries. Nucleic\\nAcids Res  2002, 30(16): 3609-3614. 16. Bramlage B, Luzi E, Eckstein F: Designing ribozymes for the inhi-\\nbition of gene expression. Trends in biotechnology  1998,\\n16(10): 434-438. 17. Vickers TA, Wyatt JR, Freier SM: Effects of RNA secondary struc-\\nture on cellular antisense activity. Nucleic Acids Res  2000,\\n28(6): 1340-1347. 18. Zhao JJ, Lemke G: Rules for ribozymes. Mol Cell Neurosci  1998,\\n11(1–2): 92-97. 19.', 'Bohula EA, Salisbury AJ, Sohail M, Playford MP, Riedemann J, Southern\\nEM, Macaulay VM: The efficacy of small interfering RNAs tar-\\ngeted to the type 1 insulin- like growth fa ctor receptor\\n(IGF1R) is influenced by secondary structure in the IGF1Rtranscript. J Biol Chem  2003, 278(18): 15991-15997. 20. Kretschmer-Kazemi Far R, Sczakiel G: The activity of siRNA in\\nmammalian cells is related to st ructural target accessibility:\\na comparison with anti sense oligonucleotides. Nucleic Acids\\nRes 2003, 31(15): 4417-4424. 21. Overhoff M, Alken M, Far RK, Lemait re M, Lebleu B, Sczakiel G, Rob-\\nbins I: Local RNA target structure influences siRNA efficacy:\\na systematic global analysis. J Mol Biol  2005, 348(4): 871-881. 22. Schubert S, Grunweller A, Erdmann VA, Kurreck J: Local RNA tar-\\nget structure influences siRN A efficacy: systematic analysisAdditional File 1\\nA representative gel. In-vitro transcribed and purified ribozymes were \\nanalyzed by denaturing gel electrophoresis.', 'The figure shows a represent-\\native gel with four different ribozymes.Click here for file\\n[http://www.biomedcentral.com /content/supplementary/1471-\\n2105-8-469-S1.pdf]\\nAdditional File 2\\nSequences of hammerhead ribozymes. The table lists sequences of 15 ham-merhead ribozymes targeted to the BCRP mRNA. Click here for file\\n[http://www.biomedcentral.com /content/supplementary/1471-\\n2105-8-469-S2.doc]\\nAdditional File 3\\nPrimer sequences for RT-PCR amplif ication.', 'The table lists primer \\nsequences for RT-PCR amplification of the ribozymes for quantifying \\ncleavage reaction.Click here for file\\n[http://www.biomedcentral.com /content/supplementary/1471-\\n2105-8-469-S3.pdf]\\nPublish with BioMed Central   and  every \\nscientist can read your work free of charge\\n\"BioMed Central will be the most significant development for \\ndisseminating the results of biomedical research in our lifetime.\"\\nSir Paul Nurse, Cancer Research UK\\nYour research papers will be:\\navailable free of charge to the entire biomedical community\\npeer reviewed and published immediately upon acceptance\\ncited in PubMed and archived on PubMed Central \\nyours — you keep the copyright\\nSubmit your manuscript here:\\nhttp://www.biomedcentral.com/info/publishing_adv.aspBioMed centralBMC Bioinformatics  2007, 8:469 http://www.biomedcen tral.com/1471-2105/8/469\\nPage 12 of 12\\n(page number not for citation purposes)of intentionally designed binding regions. J Mol Biol  2005,\\n348(4): 883-893. 23.', 'Yoshinari K, Miyagishi M, Taira K: Effects on RNAi of the tight\\nstructure, sequence and positi on of the targeted region. Nucleic Acids Res  2004, 32(2): 691-699. 24. Zhao Y, Samal E, Srivastava D: Serum response fa ctor regulates\\na muscle-specific microRNA th at targets Hand2 during car-\\ndiogenesis. Nature  2005, 436(7048): 214-220. 25. Long D, Lee R, Williams P, Chan CY, Ambros V, Ding Y: Potent\\neffect of target structure on microRNA function. Nat Struct\\nMol Biol  2007, 14:287-294. 26. Denman RB: Using RNAFOLD to predict the activity of small\\ncatalytic RNAs. BioTechniques  1993, 15(6): 1090-1095. 27. Sczakiel G, Tabler M: Computer-aided calculation of the local\\nfolding potential of target RNA and its use for ribozyme\\ndesign. Methods Mol Biol  1997, 74:11-15. 28. James W, Cowe E: Computational approach es to the identifi-\\ncation of ribozyme target sites. Methods Mol Biol  1997, 74:17-26. 29. Zuker M: Mfold web server for nucleic acid folding and hybrid-\\nization prediction.', 'Nucleic Acids Res  2003, 31(13): 3406-3415. 30. Altuvia S, Kornitzer D, Teff D, Oppenheim AB: Alternative mRNA\\nstructures of the cIII gene of bacteriophage lambda deter-\\nmine the rate of its translation initiation. J Mol Biol  1989,\\n210(2): 265-280. 31. Betts L, Spremulli LL: Analysis of the role of the Shine-Dalgarno\\nsequence and mRNA secondary structure on the efficiencyof translational initiation in the Euglena gracilis chloroplast\\natpH mRNA. J Biol Chem  1994, 269(42): 26456-26463. 32. Ding Y, Lawrence CE: A statistical sampling algorithm for RNA\\nsecondary structure prediction. Nucleic Acids Res  2003,\\n31(24): 7280-7301. 33. Ding Y, Chan CY, Lawrence CE: Sfold web server for statistical\\nfolding and rational de sign of nucleic acids. Nucleic Acids Res\\n2004:W135-141. 34. Ding Y, Chan CY, Lawrence CE: Clustering of RNA secondary\\nstructures with application to messenger RNAs. J Mol Biol\\n2006, 359(3): 554-571. 35.', 'Ding Y, Chan CY, Lawrence CE: RNA secondary structure pre-\\ndiction by centroids in a Bo ltzmann weighted ensemble. RNA\\n2005, 11(8): 1157-1166. 36. Ding Y, Lawrence CE: Statistical prediction  of single-stranded\\nregions in RNA secondary stru cture and application to pre-\\ndicting effective antisense target sites and beyond. Nucleic\\nAcids Res  2001, 29(5): 1034-1046. 37. Shao Y, Wu Y, Chan CY, McDonough K, Ding Y: Rational design\\nand rapid screening of antisense oligonucleotides for\\nprokaryotic gene modulation. Nucleic Acids Res  2006,\\n34(19): 5660-5669. 38. Shao Y, Chan CY, Mali yekkel A, Lawrence CE, Roninson IB, Ding Y:\\nEffect of target secondary structure on RNAi efficiency. RNA\\n2007, 13(10): 1631-1640. 39. Kowalski P, Wichert A, Ho lm PS, Dietel M, Lage H: Selection and\\ncharacterization of a high-act ivity ribozyme directed against\\nthe antineoplastic drug resi stance-associated ABC trans-\\nporter BCRP/MXR/ABCG2. Cancer gene therapy  2001,\\n8(3): 185-192. 40.', 'Liu R, Rohe B, Carson DD, Farach-Carson MC: A rapid and simple\\nnonradioactive method for in vi tro testing of ribozyme activ-\\nity. Antisense & nucleic acid drug development  2002, 12(4): 283-288. 41. Martick M, Scott WG: Tertiary contacts distant from the active\\nsite prime a ribozyme for catalysis. Cell 2006, 126(2): 309-320. 42. Zuker M: On finding all suboptimal foldings of an RNA mole-\\ncule. Science  1989, 244(4900): 48-52. 43. Wuchty S, Fontana W, Hofacker IL, Schuster P: Complete subop-\\ntimal folding of RNA and the stability of secondary struc-\\ntures. Biopolymers  1999, 49(2): 145-165. 44. Weisberg S: Applied Linear Regression  3rd edition. John Wiley & Sons,\\nNew York; 2005. 45. Christoffersen RE, McSwiggen JA, Konings D: Application of com-\\nputational technologies to ribozyme biotechnology prod-\\nucts. J Mol Structure (Theochem)  1994, 311:273-284. 46. Milner N, Mir KU, Southern EM: Selecting effective antisense\\nreagents on combinatoria l oligonucleotide arrays.', 'Nat Bio-\\ntechnol  1997, 15(6): 537-541. 47. Hargittai MR, Gorelick RJ, Ro uzina I, Musier-Forsyth K: Mechanistic\\ninsights into the kinetics of HIV-1 nucleocapsid protein-facil-itated tRNA annealing to the primer binding site. J Mol Biol\\n2004, 337(4): 951-968. 48. Nelson JA, Uhlenbeck OC: When to believe what you see. Mol\\nCell 2006, 23(4): 447-450. 49. Peracchi A, Karpeisky A, Malone y L, Beigelman L, Herschlag D: A\\ncore folding model for ca talysis by the hammerhead\\nribozyme accounts for its extr aordinary sensitivity to abasic\\nmutations. Biochemistry  1998, 37(42): 14765-14775. 50. Simorre JP, Legault P, Han gar AB, Michiels P, Pardi A: A conforma-\\ntional change in the cataly tic core of the hammerhead\\nribozyme upon cleavage of an RNA substrate. Biochemistry\\n1997, 36(3): 518-525. 51. Westhof E: A tale in molecular reco gnition: the hammerhead\\nribozyme. J Mol Recognit  2007, 20(1): 1-3. 52.', 'Xia T, SantaLucia J Jr, Burkard ME, Kierzek R, Schroeder SJ, Jiao X,\\nCox C, Turner DH: Thermodynamic parameters for an\\nexpanded nearest-neighbor model for formation of RNA\\nduplexes with Watson-Crick base pairs. Biochemistry  1998,\\n37(42): 14719-14735. 53. Mathews DH, Sabina J, Zuker M, Turner DH: Expanded sequence\\ndependence of thermodynamic parameters improves pre-\\ndiction of RNA secondary structure. J Mol Biol  1999,\\n288(5): 911-940. 54. Team RDC: R: A language and environment for statistical\\ncomputing (ISBN 3-900051-07-0). .']\n"
     ]
    }
   ],
   "source": [
    "# Example usage\n",
    "chunks = extract_text_from_pdfs(\"./pdfs\", max_chars=1000)\n",
    "print(chunks)\n",
    "print(f\"There are {len(chunks)} chunks\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "497\n"
     ]
    }
   ],
   "source": [
    "#upload to xtrace\n",
    "INDEX_NAME='near-hackathon'\n",
    "KNOWLEDGE_BASE='sagent'\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def upload_to_xtrace(chunks):\n",
    "    url = \"https://beta0-api.xtrace.ai/data\"\n",
    "    index_name = \"near-hackathon\"\n",
    "    knowledge_base = \"sagent\"\n",
    "    headers = {\"x-api-key\": \"pR4EPkE9AV5YlLVUlBqax5rN1jWMAPDbaO6Jysxp\"}  # Add your API key here\n",
    "    \n",
    "    for chunk in chunks:\n",
    "        payload = {\n",
    "            \"index_name\": index_name,\n",
    "            \"context\": chunk,\n",
    "            \"knowledge_base\": knowledge_base\n",
    "        }\n",
    "        response = requests.post(url, json=payload, headers=headers)\n",
    "        print(f\"Uploaded chunk, response: {response.status_code}, {response.text}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n",
      "Uploaded chunk, response: 403, {\"message\":\"Forbidden\"}\n"
     ]
    }
   ],
   "source": [
    "upload_to_xtrace(chunks)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "near",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
